// @angular-three/core/esm2015/helpers/lib/spot-light-helper/spot-light-helper.directive.js
// GENERATED
import { ThreeHelper, ThreeObject3d, OBJECT_3D_CONTROLLER_PROVIDER, } from '@angular-three/core';
import { Directive, Input } from '@angular/core';
import { SpotLightHelper } from 'three';
import * as i0 from "@angular/core";
export class SpotLightHelperDirective extends ThreeHelper {
    constructor() {
        super(...arguments);
        this.helperType = SpotLightHelper;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
SpotLightHelperDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: SpotLightHelperDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
SpotLightHelperDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: SpotLightHelperDirective, selector: "ngt-spot-light-helper", inputs: { args: "args" }, providers: [
        {
            provide: ThreeObject3d,
            useExisting: SpotLightHelperDirective,
        },
        OBJECT_3D_CONTROLLER_PROVIDER,
    ], exportAs: ["ngtSpotLightHelper"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: SpotLightHelperDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-spot-light-helper',
                    exportAs: 'ngtSpotLightHelper',
                    providers: [
                        {
                            provide: ThreeObject3d,
                            useExisting: SpotLightHelperDirective,
                        },
                        OBJECT_3D_CONTROLLER_PROVIDER,
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });
