// @angular-three/core/bundles/angular-three-core-stats.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('@angular/common'), require('three/examples/jsm/libs/stats.module'), require('@angular-three/core')) :
    typeof define === 'function' && define.amd ? define('@angular-three/core/stats', ['exports', '@angular/core', '@angular/common', 'three/examples/jsm/libs/stats.module', '@angular-three/core'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory((global["angular-three"] = global["angular-three"] || {}, global["angular-three"].core = global["angular-three"].core || {}, global["angular-three"].core.stats = {}), global.ng.core, global.ng.common, global["three/examples/jsm/libs/stats"].module, global["angular-three"].core));
})(this, (function (exports, i0, common, Stats, i1) { 'use strict';

    function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

    function _interopNamespace(e) {
        if (e && e.__esModule) return e;
        var n = Object.create(null);
        if (e) {
            Object.keys(e).forEach(function (k) {
                if (k !== 'default') {
                    var d = Object.getOwnPropertyDescriptor(e, k);
                    Object.defineProperty(n, k, d.get ? d : {
                        enumerable: true,
                        get: function () { return e[k]; }
                    });
                }
            });
        }
        n["default"] = e;
        return Object.freeze(n);
    }

    var i0__namespace = /*#__PURE__*/_interopNamespace(i0);
    var Stats__default = /*#__PURE__*/_interopDefaultLegacy(Stats);
    var i1__namespace = /*#__PURE__*/_interopNamespace(i1);

    var StatsDirective = /** @class */ (function () {
        function StatsDirective(animationStore, document) {
            this.animationStore = animationStore;
            this.document = document;
        }
        StatsDirective.prototype.ngOnInit = function () {
            var _this = this;
            this.stats = Stats__default["default"]();
            this.document.body.appendChild(this.stats.dom);
            this.animationStore.registerAnimation(function () {
                _this.stats.update();
            });
        };
        StatsDirective.prototype.ngOnDestroy = function () {
            this.stats.end();
        };
        return StatsDirective;
    }());
    StatsDirective.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: StatsDirective, deps: [{ token: i1__namespace.AnimationStore }, { token: common.DOCUMENT }], target: i0__namespace.ɵɵFactoryTarget.Directive });
    StatsDirective.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: StatsDirective, selector: "ngt-stats", exportAs: ["ngtStats"], ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: StatsDirective, decorators: [{
                type: i0.Directive,
                args: [{
                        selector: 'ngt-stats',
                        exportAs: 'ngtStats',
                    }]
            }], ctorParameters: function () {
            return [{ type: i1__namespace.AnimationStore }, { type: Document, decorators: [{
                            type: i0.Inject,
                            args: [common.DOCUMENT]
                        }] }];
        } });

    var ThreeStatsModule = /** @class */ (function () {
        function ThreeStatsModule() {
        }
        return ThreeStatsModule;
    }());
    ThreeStatsModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeStatsModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    ThreeStatsModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeStatsModule, declarations: [StatsDirective], exports: [StatsDirective] });
    ThreeStatsModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeStatsModule });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeStatsModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        declarations: [StatsDirective],
                        exports: [StatsDirective],
                    }]
            }] });

    /**
     * Generated bundle index. Do not edit.
     */

    exports.StatsDirective = StatsDirective;
    exports.ThreeStatsModule = ThreeStatsModule;

    Object.defineProperty(exports, '__esModule', { value: true });

}));
