// @angular-three/core/bundles/angular-three-core-materials.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular-three/core'), require('@angular/core'), require('three')) :
    typeof define === 'function' && define.amd ? define('@angular-three/core/materials', ['exports', '@angular-three/core', '@angular/core', 'three'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory((global["angular-three"] = global["angular-three"] || {}, global["angular-three"].core = global["angular-three"].core || {}, global["angular-three"].core.materials = {}), global["angular-three"].core, global.ng.core, global.three));
})(this, (function (exports, core, i0, three) { 'use strict';

    function _interopNamespace(e) {
        if (e && e.__esModule) return e;
        var n = Object.create(null);
        if (e) {
            Object.keys(e).forEach(function (k) {
                if (k !== 'default') {
                    var d = Object.getOwnPropertyDescriptor(e, k);
                    Object.defineProperty(n, k, d.get ? d : {
                        enumerable: true,
                        get: function () { return e[k]; }
                    });
                }
            });
        }
        n["default"] = e;
        return Object.freeze(n);
    }

    var i0__namespace = /*#__PURE__*/_interopNamespace(i0);

    /*! *****************************************************************************
    Copyright (c) Microsoft Corporation.

    Permission to use, copy, modify, and/or distribute this software for any
    purpose with or without fee is hereby granted.

    THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES WITH
    REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY
    AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT,
    INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM
    LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR
    OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
    PERFORMANCE OF THIS SOFTWARE.
    ***************************************************************************** */
    /* global Reflect, Promise */
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b)
                if (Object.prototype.hasOwnProperty.call(b, p))
                    d[p] = b[p]; };
        return extendStatics(d, b);
    };
    function __extends(d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    }
    var __assign = function () {
        __assign = Object.assign || function __assign(t) {
            for (var s, i = 1, n = arguments.length; i < n; i++) {
                s = arguments[i];
                for (var p in s)
                    if (Object.prototype.hasOwnProperty.call(s, p))
                        t[p] = s[p];
            }
            return t;
        };
        return __assign.apply(this, arguments);
    };
    function __rest(s, e) {
        var t = {};
        for (var p in s)
            if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
                t[p] = s[p];
        if (s != null && typeof Object.getOwnPropertySymbols === "function")
            for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
                if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                    t[p[i]] = s[p[i]];
            }
        return t;
    }
    function __decorate(decorators, target, key, desc) {
        var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
            r = Reflect.decorate(decorators, target, key, desc);
        else
            for (var i = decorators.length - 1; i >= 0; i--)
                if (d = decorators[i])
                    r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    }
    function __param(paramIndex, decorator) {
        return function (target, key) { decorator(target, key, paramIndex); };
    }
    function __metadata(metadataKey, metadataValue) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function")
            return Reflect.metadata(metadataKey, metadataValue);
    }
    function __awaiter(thisArg, _arguments, P, generator) {
        function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
        return new (P || (P = Promise))(function (resolve, reject) {
            function fulfilled(value) { try {
                step(generator.next(value));
            }
            catch (e) {
                reject(e);
            } }
            function rejected(value) { try {
                step(generator["throw"](value));
            }
            catch (e) {
                reject(e);
            } }
            function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
            step((generator = generator.apply(thisArg, _arguments || [])).next());
        });
    }
    function __generator(thisArg, body) {
        var _ = { label: 0, sent: function () { if (t[0] & 1)
                throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
        return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function () { return this; }), g;
        function verb(n) { return function (v) { return step([n, v]); }; }
        function step(op) {
            if (f)
                throw new TypeError("Generator is already executing.");
            while (_)
                try {
                    if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done)
                        return t;
                    if (y = 0, t)
                        op = [op[0] & 2, t.value];
                    switch (op[0]) {
                        case 0:
                        case 1:
                            t = op;
                            break;
                        case 4:
                            _.label++;
                            return { value: op[1], done: false };
                        case 5:
                            _.label++;
                            y = op[1];
                            op = [0];
                            continue;
                        case 7:
                            op = _.ops.pop();
                            _.trys.pop();
                            continue;
                        default:
                            if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) {
                                _ = 0;
                                continue;
                            }
                            if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) {
                                _.label = op[1];
                                break;
                            }
                            if (op[0] === 6 && _.label < t[1]) {
                                _.label = t[1];
                                t = op;
                                break;
                            }
                            if (t && _.label < t[2]) {
                                _.label = t[2];
                                _.ops.push(op);
                                break;
                            }
                            if (t[2])
                                _.ops.pop();
                            _.trys.pop();
                            continue;
                    }
                    op = body.call(thisArg, _);
                }
                catch (e) {
                    op = [6, e];
                    y = 0;
                }
                finally {
                    f = t = 0;
                }
            if (op[0] & 5)
                throw op[1];
            return { value: op[0] ? op[1] : void 0, done: true };
        }
    }
    var __createBinding = Object.create ? (function (o, m, k, k2) {
        if (k2 === undefined)
            k2 = k;
        Object.defineProperty(o, k2, { enumerable: true, get: function () { return m[k]; } });
    }) : (function (o, m, k, k2) {
        if (k2 === undefined)
            k2 = k;
        o[k2] = m[k];
    });
    function __exportStar(m, o) {
        for (var p in m)
            if (p !== "default" && !Object.prototype.hasOwnProperty.call(o, p))
                __createBinding(o, m, p);
    }
    function __values(o) {
        var s = typeof Symbol === "function" && Symbol.iterator, m = s && o[s], i = 0;
        if (m)
            return m.call(o);
        if (o && typeof o.length === "number")
            return {
                next: function () {
                    if (o && i >= o.length)
                        o = void 0;
                    return { value: o && o[i++], done: !o };
                }
            };
        throw new TypeError(s ? "Object is not iterable." : "Symbol.iterator is not defined.");
    }
    function __read(o, n) {
        var m = typeof Symbol === "function" && o[Symbol.iterator];
        if (!m)
            return o;
        var i = m.call(o), r, ar = [], e;
        try {
            while ((n === void 0 || n-- > 0) && !(r = i.next()).done)
                ar.push(r.value);
        }
        catch (error) {
            e = { error: error };
        }
        finally {
            try {
                if (r && !r.done && (m = i["return"]))
                    m.call(i);
            }
            finally {
                if (e)
                    throw e.error;
            }
        }
        return ar;
    }
    /** @deprecated */
    function __spread() {
        for (var ar = [], i = 0; i < arguments.length; i++)
            ar = ar.concat(__read(arguments[i]));
        return ar;
    }
    /** @deprecated */
    function __spreadArrays() {
        for (var s = 0, i = 0, il = arguments.length; i < il; i++)
            s += arguments[i].length;
        for (var r = Array(s), k = 0, i = 0; i < il; i++)
            for (var a = arguments[i], j = 0, jl = a.length; j < jl; j++, k++)
                r[k] = a[j];
        return r;
    }
    function __spreadArray(to, from, pack) {
        if (pack || arguments.length === 2)
            for (var i = 0, l = from.length, ar; i < l; i++) {
                if (ar || !(i in from)) {
                    if (!ar)
                        ar = Array.prototype.slice.call(from, 0, i);
                    ar[i] = from[i];
                }
            }
        return to.concat(ar || Array.prototype.slice.call(from));
    }
    function __await(v) {
        return this instanceof __await ? (this.v = v, this) : new __await(v);
    }
    function __asyncGenerator(thisArg, _arguments, generator) {
        if (!Symbol.asyncIterator)
            throw new TypeError("Symbol.asyncIterator is not defined.");
        var g = generator.apply(thisArg, _arguments || []), i, q = [];
        return i = {}, verb("next"), verb("throw"), verb("return"), i[Symbol.asyncIterator] = function () { return this; }, i;
        function verb(n) { if (g[n])
            i[n] = function (v) { return new Promise(function (a, b) { q.push([n, v, a, b]) > 1 || resume(n, v); }); }; }
        function resume(n, v) { try {
            step(g[n](v));
        }
        catch (e) {
            settle(q[0][3], e);
        } }
        function step(r) { r.value instanceof __await ? Promise.resolve(r.value.v).then(fulfill, reject) : settle(q[0][2], r); }
        function fulfill(value) { resume("next", value); }
        function reject(value) { resume("throw", value); }
        function settle(f, v) { if (f(v), q.shift(), q.length)
            resume(q[0][0], q[0][1]); }
    }
    function __asyncDelegator(o) {
        var i, p;
        return i = {}, verb("next"), verb("throw", function (e) { throw e; }), verb("return"), i[Symbol.iterator] = function () { return this; }, i;
        function verb(n, f) { i[n] = o[n] ? function (v) { return (p = !p) ? { value: __await(o[n](v)), done: n === "return" } : f ? f(v) : v; } : f; }
    }
    function __asyncValues(o) {
        if (!Symbol.asyncIterator)
            throw new TypeError("Symbol.asyncIterator is not defined.");
        var m = o[Symbol.asyncIterator], i;
        return m ? m.call(o) : (o = typeof __values === "function" ? __values(o) : o[Symbol.iterator](), i = {}, verb("next"), verb("throw"), verb("return"), i[Symbol.asyncIterator] = function () { return this; }, i);
        function verb(n) { i[n] = o[n] && function (v) { return new Promise(function (resolve, reject) { v = o[n](v), settle(resolve, reject, v.done, v.value); }); }; }
        function settle(resolve, reject, d, v) { Promise.resolve(v).then(function (v) { resolve({ value: v, done: d }); }, reject); }
    }
    function __makeTemplateObject(cooked, raw) {
        if (Object.defineProperty) {
            Object.defineProperty(cooked, "raw", { value: raw });
        }
        else {
            cooked.raw = raw;
        }
        return cooked;
    }
    ;
    var __setModuleDefault = Object.create ? (function (o, v) {
        Object.defineProperty(o, "default", { enumerable: true, value: v });
    }) : function (o, v) {
        o["default"] = v;
    };
    function __importStar(mod) {
        if (mod && mod.__esModule)
            return mod;
        var result = {};
        if (mod != null)
            for (var k in mod)
                if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k))
                    __createBinding(result, mod, k);
        __setModuleDefault(result, mod);
        return result;
    }
    function __importDefault(mod) {
        return (mod && mod.__esModule) ? mod : { default: mod };
    }
    function __classPrivateFieldGet(receiver, state, kind, f) {
        if (kind === "a" && !f)
            throw new TypeError("Private accessor was defined without a getter");
        if (typeof state === "function" ? receiver !== state || !f : !state.has(receiver))
            throw new TypeError("Cannot read private member from an object whose class did not declare it");
        return kind === "m" ? f : kind === "a" ? f.call(receiver) : f ? f.value : state.get(receiver);
    }
    function __classPrivateFieldSet(receiver, state, value, kind, f) {
        if (kind === "m")
            throw new TypeError("Private method is not writable");
        if (kind === "a" && !f)
            throw new TypeError("Private accessor was defined without a setter");
        if (typeof state === "function" ? receiver !== state || !f : !state.has(receiver))
            throw new TypeError("Cannot write private member to an object whose class did not declare it");
        return (kind === "a" ? f.call(receiver, value) : f ? f.value = value : state.set(receiver, value)), value;
    }

    var ShadowMaterialDirective = /** @class */ (function (_super) {
        __extends(ShadowMaterialDirective, _super);
        function ShadowMaterialDirective() {
            var _this = _super.apply(this, __spreadArray([], __read(arguments))) || this;
            _this.materialType = three.ShadowMaterial;
            return _this;
        }
        return ShadowMaterialDirective;
    }(core.ThreeMaterial));
    ShadowMaterialDirective.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ShadowMaterialDirective, deps: null, target: i0__namespace.ɵɵFactoryTarget.Directive });
    ShadowMaterialDirective.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: ShadowMaterialDirective, selector: "ngt-shadow-material", providers: [{ provide: core.ThreeMaterial, useExisting: ShadowMaterialDirective }], exportAs: ["ngtShadowMaterial"], usesInheritance: true, ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ShadowMaterialDirective, decorators: [{
                type: i0.Directive,
                args: [{
                        selector: 'ngt-shadow-material',
                        exportAs: 'ngtShadowMaterial',
                        providers: [{ provide: core.ThreeMaterial, useExisting: ShadowMaterialDirective }],
                    }]
            }] });

    // GENERATED
    var ThreeShadowMaterialModule = /** @class */ (function () {
        function ThreeShadowMaterialModule() {
        }
        return ThreeShadowMaterialModule;
    }());
    ThreeShadowMaterialModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeShadowMaterialModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    ThreeShadowMaterialModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeShadowMaterialModule, declarations: [ShadowMaterialDirective], exports: [ShadowMaterialDirective] });
    ThreeShadowMaterialModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeShadowMaterialModule });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeShadowMaterialModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        declarations: [ShadowMaterialDirective],
                        exports: [ShadowMaterialDirective],
                    }]
            }] });

    // GENERATED

    var SpriteMaterialDirective = /** @class */ (function (_super) {
        __extends(SpriteMaterialDirective, _super);
        function SpriteMaterialDirective() {
            var _this = _super.apply(this, __spreadArray([], __read(arguments))) || this;
            _this.materialType = three.SpriteMaterial;
            return _this;
        }
        return SpriteMaterialDirective;
    }(core.ThreeMaterial));
    SpriteMaterialDirective.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: SpriteMaterialDirective, deps: null, target: i0__namespace.ɵɵFactoryTarget.Directive });
    SpriteMaterialDirective.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: SpriteMaterialDirective, selector: "ngt-sprite-material", providers: [{ provide: core.ThreeMaterial, useExisting: SpriteMaterialDirective }], exportAs: ["ngtSpriteMaterial"], usesInheritance: true, ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: SpriteMaterialDirective, decorators: [{
                type: i0.Directive,
                args: [{
                        selector: 'ngt-sprite-material',
                        exportAs: 'ngtSpriteMaterial',
                        providers: [{ provide: core.ThreeMaterial, useExisting: SpriteMaterialDirective }],
                    }]
            }] });

    // GENERATED
    var ThreeSpriteMaterialModule = /** @class */ (function () {
        function ThreeSpriteMaterialModule() {
        }
        return ThreeSpriteMaterialModule;
    }());
    ThreeSpriteMaterialModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeSpriteMaterialModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    ThreeSpriteMaterialModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeSpriteMaterialModule, declarations: [SpriteMaterialDirective], exports: [SpriteMaterialDirective] });
    ThreeSpriteMaterialModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeSpriteMaterialModule });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeSpriteMaterialModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        declarations: [SpriteMaterialDirective],
                        exports: [SpriteMaterialDirective],
                    }]
            }] });

    // GENERATED

    var RawShaderMaterialDirective = /** @class */ (function (_super) {
        __extends(RawShaderMaterialDirective, _super);
        function RawShaderMaterialDirective() {
            var _this = _super.apply(this, __spreadArray([], __read(arguments))) || this;
            _this.materialType = three.RawShaderMaterial;
            return _this;
        }
        return RawShaderMaterialDirective;
    }(core.ThreeMaterial));
    RawShaderMaterialDirective.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: RawShaderMaterialDirective, deps: null, target: i0__namespace.ɵɵFactoryTarget.Directive });
    RawShaderMaterialDirective.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: RawShaderMaterialDirective, selector: "ngt-raw-shader-material", providers: [
            { provide: core.ThreeMaterial, useExisting: RawShaderMaterialDirective },
        ], exportAs: ["ngtRawShaderMaterial"], usesInheritance: true, ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: RawShaderMaterialDirective, decorators: [{
                type: i0.Directive,
                args: [{
                        selector: 'ngt-raw-shader-material',
                        exportAs: 'ngtRawShaderMaterial',
                        providers: [
                            { provide: core.ThreeMaterial, useExisting: RawShaderMaterialDirective },
                        ],
                    }]
            }] });

    // GENERATED
    var ThreeRawShaderMaterialModule = /** @class */ (function () {
        function ThreeRawShaderMaterialModule() {
        }
        return ThreeRawShaderMaterialModule;
    }());
    ThreeRawShaderMaterialModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeRawShaderMaterialModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    ThreeRawShaderMaterialModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeRawShaderMaterialModule, declarations: [RawShaderMaterialDirective], exports: [RawShaderMaterialDirective] });
    ThreeRawShaderMaterialModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeRawShaderMaterialModule });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeRawShaderMaterialModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        declarations: [RawShaderMaterialDirective],
                        exports: [RawShaderMaterialDirective],
                    }]
            }] });

    // GENERATED

    var ShaderMaterialDirective = /** @class */ (function (_super) {
        __extends(ShaderMaterialDirective, _super);
        function ShaderMaterialDirective() {
            var _this = _super.apply(this, __spreadArray([], __read(arguments))) || this;
            _this.materialType = three.ShaderMaterial;
            return _this;
        }
        return ShaderMaterialDirective;
    }(core.ThreeMaterial));
    ShaderMaterialDirective.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ShaderMaterialDirective, deps: null, target: i0__namespace.ɵɵFactoryTarget.Directive });
    ShaderMaterialDirective.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: ShaderMaterialDirective, selector: "ngt-shader-material", providers: [{ provide: core.ThreeMaterial, useExisting: ShaderMaterialDirective }], exportAs: ["ngtShaderMaterial"], usesInheritance: true, ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ShaderMaterialDirective, decorators: [{
                type: i0.Directive,
                args: [{
                        selector: 'ngt-shader-material',
                        exportAs: 'ngtShaderMaterial',
                        providers: [{ provide: core.ThreeMaterial, useExisting: ShaderMaterialDirective }],
                    }]
            }] });

    // GENERATED
    var ThreeShaderMaterialModule = /** @class */ (function () {
        function ThreeShaderMaterialModule() {
        }
        return ThreeShaderMaterialModule;
    }());
    ThreeShaderMaterialModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeShaderMaterialModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    ThreeShaderMaterialModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeShaderMaterialModule, declarations: [ShaderMaterialDirective], exports: [ShaderMaterialDirective] });
    ThreeShaderMaterialModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeShaderMaterialModule });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeShaderMaterialModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        declarations: [ShaderMaterialDirective],
                        exports: [ShaderMaterialDirective],
                    }]
            }] });

    // GENERATED

    var PointsMaterialDirective = /** @class */ (function (_super) {
        __extends(PointsMaterialDirective, _super);
        function PointsMaterialDirective() {
            var _this = _super.apply(this, __spreadArray([], __read(arguments))) || this;
            _this.materialType = three.PointsMaterial;
            return _this;
        }
        return PointsMaterialDirective;
    }(core.ThreeMaterial));
    PointsMaterialDirective.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: PointsMaterialDirective, deps: null, target: i0__namespace.ɵɵFactoryTarget.Directive });
    PointsMaterialDirective.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: PointsMaterialDirective, selector: "ngt-points-material", providers: [{ provide: core.ThreeMaterial, useExisting: PointsMaterialDirective }], exportAs: ["ngtPointsMaterial"], usesInheritance: true, ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: PointsMaterialDirective, decorators: [{
                type: i0.Directive,
                args: [{
                        selector: 'ngt-points-material',
                        exportAs: 'ngtPointsMaterial',
                        providers: [{ provide: core.ThreeMaterial, useExisting: PointsMaterialDirective }],
                    }]
            }] });

    // GENERATED
    var ThreePointsMaterialModule = /** @class */ (function () {
        function ThreePointsMaterialModule() {
        }
        return ThreePointsMaterialModule;
    }());
    ThreePointsMaterialModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreePointsMaterialModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    ThreePointsMaterialModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreePointsMaterialModule, declarations: [PointsMaterialDirective], exports: [PointsMaterialDirective] });
    ThreePointsMaterialModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreePointsMaterialModule });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreePointsMaterialModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        declarations: [PointsMaterialDirective],
                        exports: [PointsMaterialDirective],
                    }]
            }] });

    // GENERATED

    var MeshPhysicalMaterialDirective = /** @class */ (function (_super) {
        __extends(MeshPhysicalMaterialDirective, _super);
        function MeshPhysicalMaterialDirective() {
            var _this = _super.apply(this, __spreadArray([], __read(arguments))) || this;
            _this.materialType = three.MeshPhysicalMaterial;
            return _this;
        }
        return MeshPhysicalMaterialDirective;
    }(core.ThreeMaterial));
    MeshPhysicalMaterialDirective.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: MeshPhysicalMaterialDirective, deps: null, target: i0__namespace.ɵɵFactoryTarget.Directive });
    MeshPhysicalMaterialDirective.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: MeshPhysicalMaterialDirective, selector: "ngt-mesh-physical-material", providers: [
            { provide: core.ThreeMaterial, useExisting: MeshPhysicalMaterialDirective },
        ], exportAs: ["ngtMeshPhysicalMaterial"], usesInheritance: true, ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: MeshPhysicalMaterialDirective, decorators: [{
                type: i0.Directive,
                args: [{
                        selector: 'ngt-mesh-physical-material',
                        exportAs: 'ngtMeshPhysicalMaterial',
                        providers: [
                            { provide: core.ThreeMaterial, useExisting: MeshPhysicalMaterialDirective },
                        ],
                    }]
            }] });

    // GENERATED
    var ThreeMeshPhysicalMaterialModule = /** @class */ (function () {
        function ThreeMeshPhysicalMaterialModule() {
        }
        return ThreeMeshPhysicalMaterialModule;
    }());
    ThreeMeshPhysicalMaterialModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeMeshPhysicalMaterialModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    ThreeMeshPhysicalMaterialModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeMeshPhysicalMaterialModule, declarations: [MeshPhysicalMaterialDirective], exports: [MeshPhysicalMaterialDirective] });
    ThreeMeshPhysicalMaterialModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeMeshPhysicalMaterialModule });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeMeshPhysicalMaterialModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        declarations: [MeshPhysicalMaterialDirective],
                        exports: [MeshPhysicalMaterialDirective],
                    }]
            }] });

    // GENERATED

    var MeshStandardMaterialDirective = /** @class */ (function (_super) {
        __extends(MeshStandardMaterialDirective, _super);
        function MeshStandardMaterialDirective() {
            var _this = _super.apply(this, __spreadArray([], __read(arguments))) || this;
            _this.materialType = three.MeshStandardMaterial;
            return _this;
        }
        return MeshStandardMaterialDirective;
    }(core.ThreeMaterial));
    MeshStandardMaterialDirective.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: MeshStandardMaterialDirective, deps: null, target: i0__namespace.ɵɵFactoryTarget.Directive });
    MeshStandardMaterialDirective.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: MeshStandardMaterialDirective, selector: "ngt-mesh-standard-material", providers: [
            { provide: core.ThreeMaterial, useExisting: MeshStandardMaterialDirective },
        ], exportAs: ["ngtMeshStandardMaterial"], usesInheritance: true, ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: MeshStandardMaterialDirective, decorators: [{
                type: i0.Directive,
                args: [{
                        selector: 'ngt-mesh-standard-material',
                        exportAs: 'ngtMeshStandardMaterial',
                        providers: [
                            { provide: core.ThreeMaterial, useExisting: MeshStandardMaterialDirective },
                        ],
                    }]
            }] });

    // GENERATED
    var ThreeMeshStandardMaterialModule = /** @class */ (function () {
        function ThreeMeshStandardMaterialModule() {
        }
        return ThreeMeshStandardMaterialModule;
    }());
    ThreeMeshStandardMaterialModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeMeshStandardMaterialModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    ThreeMeshStandardMaterialModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeMeshStandardMaterialModule, declarations: [MeshStandardMaterialDirective], exports: [MeshStandardMaterialDirective] });
    ThreeMeshStandardMaterialModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeMeshStandardMaterialModule });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeMeshStandardMaterialModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        declarations: [MeshStandardMaterialDirective],
                        exports: [MeshStandardMaterialDirective],
                    }]
            }] });

    // GENERATED

    var MeshPhongMaterialDirective = /** @class */ (function (_super) {
        __extends(MeshPhongMaterialDirective, _super);
        function MeshPhongMaterialDirective() {
            var _this = _super.apply(this, __spreadArray([], __read(arguments))) || this;
            _this.materialType = three.MeshPhongMaterial;
            return _this;
        }
        return MeshPhongMaterialDirective;
    }(core.ThreeMaterial));
    MeshPhongMaterialDirective.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: MeshPhongMaterialDirective, deps: null, target: i0__namespace.ɵɵFactoryTarget.Directive });
    MeshPhongMaterialDirective.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: MeshPhongMaterialDirective, selector: "ngt-mesh-phong-material", providers: [
            { provide: core.ThreeMaterial, useExisting: MeshPhongMaterialDirective },
        ], exportAs: ["ngtMeshPhongMaterial"], usesInheritance: true, ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: MeshPhongMaterialDirective, decorators: [{
                type: i0.Directive,
                args: [{
                        selector: 'ngt-mesh-phong-material',
                        exportAs: 'ngtMeshPhongMaterial',
                        providers: [
                            { provide: core.ThreeMaterial, useExisting: MeshPhongMaterialDirective },
                        ],
                    }]
            }] });

    // GENERATED
    var ThreeMeshPhongMaterialModule = /** @class */ (function () {
        function ThreeMeshPhongMaterialModule() {
        }
        return ThreeMeshPhongMaterialModule;
    }());
    ThreeMeshPhongMaterialModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeMeshPhongMaterialModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    ThreeMeshPhongMaterialModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeMeshPhongMaterialModule, declarations: [MeshPhongMaterialDirective], exports: [MeshPhongMaterialDirective] });
    ThreeMeshPhongMaterialModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeMeshPhongMaterialModule });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeMeshPhongMaterialModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        declarations: [MeshPhongMaterialDirective],
                        exports: [MeshPhongMaterialDirective],
                    }]
            }] });

    // GENERATED

    var MeshToonMaterialDirective = /** @class */ (function (_super) {
        __extends(MeshToonMaterialDirective, _super);
        function MeshToonMaterialDirective() {
            var _this = _super.apply(this, __spreadArray([], __read(arguments))) || this;
            _this.materialType = three.MeshToonMaterial;
            return _this;
        }
        return MeshToonMaterialDirective;
    }(core.ThreeMaterial));
    MeshToonMaterialDirective.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: MeshToonMaterialDirective, deps: null, target: i0__namespace.ɵɵFactoryTarget.Directive });
    MeshToonMaterialDirective.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: MeshToonMaterialDirective, selector: "ngt-mesh-toon-material", providers: [
            { provide: core.ThreeMaterial, useExisting: MeshToonMaterialDirective },
        ], exportAs: ["ngtMeshToonMaterial"], usesInheritance: true, ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: MeshToonMaterialDirective, decorators: [{
                type: i0.Directive,
                args: [{
                        selector: 'ngt-mesh-toon-material',
                        exportAs: 'ngtMeshToonMaterial',
                        providers: [
                            { provide: core.ThreeMaterial, useExisting: MeshToonMaterialDirective },
                        ],
                    }]
            }] });

    // GENERATED
    var ThreeMeshToonMaterialModule = /** @class */ (function () {
        function ThreeMeshToonMaterialModule() {
        }
        return ThreeMeshToonMaterialModule;
    }());
    ThreeMeshToonMaterialModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeMeshToonMaterialModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    ThreeMeshToonMaterialModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeMeshToonMaterialModule, declarations: [MeshToonMaterialDirective], exports: [MeshToonMaterialDirective] });
    ThreeMeshToonMaterialModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeMeshToonMaterialModule });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeMeshToonMaterialModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        declarations: [MeshToonMaterialDirective],
                        exports: [MeshToonMaterialDirective],
                    }]
            }] });

    // GENERATED

    var MeshNormalMaterialDirective = /** @class */ (function (_super) {
        __extends(MeshNormalMaterialDirective, _super);
        function MeshNormalMaterialDirective() {
            var _this = _super.apply(this, __spreadArray([], __read(arguments))) || this;
            _this.materialType = three.MeshNormalMaterial;
            return _this;
        }
        return MeshNormalMaterialDirective;
    }(core.ThreeMaterial));
    MeshNormalMaterialDirective.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: MeshNormalMaterialDirective, deps: null, target: i0__namespace.ɵɵFactoryTarget.Directive });
    MeshNormalMaterialDirective.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: MeshNormalMaterialDirective, selector: "ngt-mesh-normal-material", providers: [
            { provide: core.ThreeMaterial, useExisting: MeshNormalMaterialDirective },
        ], exportAs: ["ngtMeshNormalMaterial"], usesInheritance: true, ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: MeshNormalMaterialDirective, decorators: [{
                type: i0.Directive,
                args: [{
                        selector: 'ngt-mesh-normal-material',
                        exportAs: 'ngtMeshNormalMaterial',
                        providers: [
                            { provide: core.ThreeMaterial, useExisting: MeshNormalMaterialDirective },
                        ],
                    }]
            }] });

    // GENERATED
    var ThreeMeshNormalMaterialModule = /** @class */ (function () {
        function ThreeMeshNormalMaterialModule() {
        }
        return ThreeMeshNormalMaterialModule;
    }());
    ThreeMeshNormalMaterialModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeMeshNormalMaterialModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    ThreeMeshNormalMaterialModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeMeshNormalMaterialModule, declarations: [MeshNormalMaterialDirective], exports: [MeshNormalMaterialDirective] });
    ThreeMeshNormalMaterialModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeMeshNormalMaterialModule });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeMeshNormalMaterialModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        declarations: [MeshNormalMaterialDirective],
                        exports: [MeshNormalMaterialDirective],
                    }]
            }] });

    // GENERATED

    var MeshLambertMaterialDirective = /** @class */ (function (_super) {
        __extends(MeshLambertMaterialDirective, _super);
        function MeshLambertMaterialDirective() {
            var _this = _super.apply(this, __spreadArray([], __read(arguments))) || this;
            _this.materialType = three.MeshLambertMaterial;
            return _this;
        }
        return MeshLambertMaterialDirective;
    }(core.ThreeMaterial));
    MeshLambertMaterialDirective.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: MeshLambertMaterialDirective, deps: null, target: i0__namespace.ɵɵFactoryTarget.Directive });
    MeshLambertMaterialDirective.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: MeshLambertMaterialDirective, selector: "ngt-mesh-lambert-material", providers: [
            { provide: core.ThreeMaterial, useExisting: MeshLambertMaterialDirective },
        ], exportAs: ["ngtMeshLambertMaterial"], usesInheritance: true, ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: MeshLambertMaterialDirective, decorators: [{
                type: i0.Directive,
                args: [{
                        selector: 'ngt-mesh-lambert-material',
                        exportAs: 'ngtMeshLambertMaterial',
                        providers: [
                            { provide: core.ThreeMaterial, useExisting: MeshLambertMaterialDirective },
                        ],
                    }]
            }] });

    // GENERATED
    var ThreeMeshLambertMaterialModule = /** @class */ (function () {
        function ThreeMeshLambertMaterialModule() {
        }
        return ThreeMeshLambertMaterialModule;
    }());
    ThreeMeshLambertMaterialModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeMeshLambertMaterialModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    ThreeMeshLambertMaterialModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeMeshLambertMaterialModule, declarations: [MeshLambertMaterialDirective], exports: [MeshLambertMaterialDirective] });
    ThreeMeshLambertMaterialModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeMeshLambertMaterialModule });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeMeshLambertMaterialModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        declarations: [MeshLambertMaterialDirective],
                        exports: [MeshLambertMaterialDirective],
                    }]
            }] });

    // GENERATED

    var MeshDepthMaterialDirective = /** @class */ (function (_super) {
        __extends(MeshDepthMaterialDirective, _super);
        function MeshDepthMaterialDirective() {
            var _this = _super.apply(this, __spreadArray([], __read(arguments))) || this;
            _this.materialType = three.MeshDepthMaterial;
            return _this;
        }
        return MeshDepthMaterialDirective;
    }(core.ThreeMaterial));
    MeshDepthMaterialDirective.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: MeshDepthMaterialDirective, deps: null, target: i0__namespace.ɵɵFactoryTarget.Directive });
    MeshDepthMaterialDirective.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: MeshDepthMaterialDirective, selector: "ngt-mesh-depth-material", providers: [
            { provide: core.ThreeMaterial, useExisting: MeshDepthMaterialDirective },
        ], exportAs: ["ngtMeshDepthMaterial"], usesInheritance: true, ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: MeshDepthMaterialDirective, decorators: [{
                type: i0.Directive,
                args: [{
                        selector: 'ngt-mesh-depth-material',
                        exportAs: 'ngtMeshDepthMaterial',
                        providers: [
                            { provide: core.ThreeMaterial, useExisting: MeshDepthMaterialDirective },
                        ],
                    }]
            }] });

    // GENERATED
    var ThreeMeshDepthMaterialModule = /** @class */ (function () {
        function ThreeMeshDepthMaterialModule() {
        }
        return ThreeMeshDepthMaterialModule;
    }());
    ThreeMeshDepthMaterialModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeMeshDepthMaterialModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    ThreeMeshDepthMaterialModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeMeshDepthMaterialModule, declarations: [MeshDepthMaterialDirective], exports: [MeshDepthMaterialDirective] });
    ThreeMeshDepthMaterialModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeMeshDepthMaterialModule });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeMeshDepthMaterialModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        declarations: [MeshDepthMaterialDirective],
                        exports: [MeshDepthMaterialDirective],
                    }]
            }] });

    // GENERATED

    var MeshDistanceMaterialDirective = /** @class */ (function (_super) {
        __extends(MeshDistanceMaterialDirective, _super);
        function MeshDistanceMaterialDirective() {
            var _this = _super.apply(this, __spreadArray([], __read(arguments))) || this;
            _this.materialType = three.MeshDistanceMaterial;
            return _this;
        }
        return MeshDistanceMaterialDirective;
    }(core.ThreeMaterial));
    MeshDistanceMaterialDirective.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: MeshDistanceMaterialDirective, deps: null, target: i0__namespace.ɵɵFactoryTarget.Directive });
    MeshDistanceMaterialDirective.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: MeshDistanceMaterialDirective, selector: "ngt-mesh-distance-material", providers: [
            { provide: core.ThreeMaterial, useExisting: MeshDistanceMaterialDirective },
        ], exportAs: ["ngtMeshDistanceMaterial"], usesInheritance: true, ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: MeshDistanceMaterialDirective, decorators: [{
                type: i0.Directive,
                args: [{
                        selector: 'ngt-mesh-distance-material',
                        exportAs: 'ngtMeshDistanceMaterial',
                        providers: [
                            { provide: core.ThreeMaterial, useExisting: MeshDistanceMaterialDirective },
                        ],
                    }]
            }] });

    // GENERATED
    var ThreeMeshDistanceMaterialModule = /** @class */ (function () {
        function ThreeMeshDistanceMaterialModule() {
        }
        return ThreeMeshDistanceMaterialModule;
    }());
    ThreeMeshDistanceMaterialModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeMeshDistanceMaterialModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    ThreeMeshDistanceMaterialModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeMeshDistanceMaterialModule, declarations: [MeshDistanceMaterialDirective], exports: [MeshDistanceMaterialDirective] });
    ThreeMeshDistanceMaterialModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeMeshDistanceMaterialModule });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeMeshDistanceMaterialModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        declarations: [MeshDistanceMaterialDirective],
                        exports: [MeshDistanceMaterialDirective],
                    }]
            }] });

    // GENERATED

    var MeshBasicMaterialDirective = /** @class */ (function (_super) {
        __extends(MeshBasicMaterialDirective, _super);
        function MeshBasicMaterialDirective() {
            var _this = _super.apply(this, __spreadArray([], __read(arguments))) || this;
            _this.materialType = three.MeshBasicMaterial;
            return _this;
        }
        return MeshBasicMaterialDirective;
    }(core.ThreeMaterial));
    MeshBasicMaterialDirective.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: MeshBasicMaterialDirective, deps: null, target: i0__namespace.ɵɵFactoryTarget.Directive });
    MeshBasicMaterialDirective.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: MeshBasicMaterialDirective, selector: "ngt-mesh-basic-material", providers: [
            { provide: core.ThreeMaterial, useExisting: MeshBasicMaterialDirective },
        ], exportAs: ["ngtMeshBasicMaterial"], usesInheritance: true, ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: MeshBasicMaterialDirective, decorators: [{
                type: i0.Directive,
                args: [{
                        selector: 'ngt-mesh-basic-material',
                        exportAs: 'ngtMeshBasicMaterial',
                        providers: [
                            { provide: core.ThreeMaterial, useExisting: MeshBasicMaterialDirective },
                        ],
                    }]
            }] });

    // GENERATED
    var ThreeMeshBasicMaterialModule = /** @class */ (function () {
        function ThreeMeshBasicMaterialModule() {
        }
        return ThreeMeshBasicMaterialModule;
    }());
    ThreeMeshBasicMaterialModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeMeshBasicMaterialModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    ThreeMeshBasicMaterialModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeMeshBasicMaterialModule, declarations: [MeshBasicMaterialDirective], exports: [MeshBasicMaterialDirective] });
    ThreeMeshBasicMaterialModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeMeshBasicMaterialModule });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeMeshBasicMaterialModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        declarations: [MeshBasicMaterialDirective],
                        exports: [MeshBasicMaterialDirective],
                    }]
            }] });

    // GENERATED

    var MeshMatcapMaterialDirective = /** @class */ (function (_super) {
        __extends(MeshMatcapMaterialDirective, _super);
        function MeshMatcapMaterialDirective() {
            var _this = _super.apply(this, __spreadArray([], __read(arguments))) || this;
            _this.materialType = three.MeshMatcapMaterial;
            return _this;
        }
        return MeshMatcapMaterialDirective;
    }(core.ThreeMaterial));
    MeshMatcapMaterialDirective.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: MeshMatcapMaterialDirective, deps: null, target: i0__namespace.ɵɵFactoryTarget.Directive });
    MeshMatcapMaterialDirective.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: MeshMatcapMaterialDirective, selector: "ngt-mesh-matcap-material", providers: [
            { provide: core.ThreeMaterial, useExisting: MeshMatcapMaterialDirective },
        ], exportAs: ["ngtMeshMatcapMaterial"], usesInheritance: true, ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: MeshMatcapMaterialDirective, decorators: [{
                type: i0.Directive,
                args: [{
                        selector: 'ngt-mesh-matcap-material',
                        exportAs: 'ngtMeshMatcapMaterial',
                        providers: [
                            { provide: core.ThreeMaterial, useExisting: MeshMatcapMaterialDirective },
                        ],
                    }]
            }] });

    // GENERATED
    var ThreeMeshMatcapMaterialModule = /** @class */ (function () {
        function ThreeMeshMatcapMaterialModule() {
        }
        return ThreeMeshMatcapMaterialModule;
    }());
    ThreeMeshMatcapMaterialModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeMeshMatcapMaterialModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    ThreeMeshMatcapMaterialModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeMeshMatcapMaterialModule, declarations: [MeshMatcapMaterialDirective], exports: [MeshMatcapMaterialDirective] });
    ThreeMeshMatcapMaterialModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeMeshMatcapMaterialModule });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeMeshMatcapMaterialModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        declarations: [MeshMatcapMaterialDirective],
                        exports: [MeshMatcapMaterialDirective],
                    }]
            }] });

    // GENERATED

    var LineDashedMaterialDirective = /** @class */ (function (_super) {
        __extends(LineDashedMaterialDirective, _super);
        function LineDashedMaterialDirective() {
            var _this = _super.apply(this, __spreadArray([], __read(arguments))) || this;
            _this.materialType = three.LineDashedMaterial;
            return _this;
        }
        return LineDashedMaterialDirective;
    }(core.ThreeMaterial));
    LineDashedMaterialDirective.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: LineDashedMaterialDirective, deps: null, target: i0__namespace.ɵɵFactoryTarget.Directive });
    LineDashedMaterialDirective.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: LineDashedMaterialDirective, selector: "ngt-line-dashed-material", providers: [
            { provide: core.ThreeMaterial, useExisting: LineDashedMaterialDirective },
        ], exportAs: ["ngtLineDashedMaterial"], usesInheritance: true, ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: LineDashedMaterialDirective, decorators: [{
                type: i0.Directive,
                args: [{
                        selector: 'ngt-line-dashed-material',
                        exportAs: 'ngtLineDashedMaterial',
                        providers: [
                            { provide: core.ThreeMaterial, useExisting: LineDashedMaterialDirective },
                        ],
                    }]
            }] });

    // GENERATED
    var ThreeLineDashedMaterialModule = /** @class */ (function () {
        function ThreeLineDashedMaterialModule() {
        }
        return ThreeLineDashedMaterialModule;
    }());
    ThreeLineDashedMaterialModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeLineDashedMaterialModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    ThreeLineDashedMaterialModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeLineDashedMaterialModule, declarations: [LineDashedMaterialDirective], exports: [LineDashedMaterialDirective] });
    ThreeLineDashedMaterialModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeLineDashedMaterialModule });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeLineDashedMaterialModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        declarations: [LineDashedMaterialDirective],
                        exports: [LineDashedMaterialDirective],
                    }]
            }] });

    // GENERATED

    var LineBasicMaterialDirective = /** @class */ (function (_super) {
        __extends(LineBasicMaterialDirective, _super);
        function LineBasicMaterialDirective() {
            var _this = _super.apply(this, __spreadArray([], __read(arguments))) || this;
            _this.materialType = three.LineBasicMaterial;
            return _this;
        }
        return LineBasicMaterialDirective;
    }(core.ThreeMaterial));
    LineBasicMaterialDirective.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: LineBasicMaterialDirective, deps: null, target: i0__namespace.ɵɵFactoryTarget.Directive });
    LineBasicMaterialDirective.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: LineBasicMaterialDirective, selector: "ngt-line-basic-material", providers: [
            { provide: core.ThreeMaterial, useExisting: LineBasicMaterialDirective },
        ], exportAs: ["ngtLineBasicMaterial"], usesInheritance: true, ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: LineBasicMaterialDirective, decorators: [{
                type: i0.Directive,
                args: [{
                        selector: 'ngt-line-basic-material',
                        exportAs: 'ngtLineBasicMaterial',
                        providers: [
                            { provide: core.ThreeMaterial, useExisting: LineBasicMaterialDirective },
                        ],
                    }]
            }] });

    // GENERATED
    var ThreeLineBasicMaterialModule = /** @class */ (function () {
        function ThreeLineBasicMaterialModule() {
        }
        return ThreeLineBasicMaterialModule;
    }());
    ThreeLineBasicMaterialModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeLineBasicMaterialModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    ThreeLineBasicMaterialModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeLineBasicMaterialModule, declarations: [LineBasicMaterialDirective], exports: [LineBasicMaterialDirective] });
    ThreeLineBasicMaterialModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeLineBasicMaterialModule });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeLineBasicMaterialModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        declarations: [LineBasicMaterialDirective],
                        exports: [LineBasicMaterialDirective],
                    }]
            }] });

    // GENERATED

    // GENERATED

    /**
     * Generated bundle index. Do not edit.
     */

    exports.LineBasicMaterialDirective = LineBasicMaterialDirective;
    exports.LineDashedMaterialDirective = LineDashedMaterialDirective;
    exports.MeshBasicMaterialDirective = MeshBasicMaterialDirective;
    exports.MeshDepthMaterialDirective = MeshDepthMaterialDirective;
    exports.MeshDistanceMaterialDirective = MeshDistanceMaterialDirective;
    exports.MeshLambertMaterialDirective = MeshLambertMaterialDirective;
    exports.MeshMatcapMaterialDirective = MeshMatcapMaterialDirective;
    exports.MeshNormalMaterialDirective = MeshNormalMaterialDirective;
    exports.MeshPhongMaterialDirective = MeshPhongMaterialDirective;
    exports.MeshPhysicalMaterialDirective = MeshPhysicalMaterialDirective;
    exports.MeshStandardMaterialDirective = MeshStandardMaterialDirective;
    exports.MeshToonMaterialDirective = MeshToonMaterialDirective;
    exports.PointsMaterialDirective = PointsMaterialDirective;
    exports.RawShaderMaterialDirective = RawShaderMaterialDirective;
    exports.ShaderMaterialDirective = ShaderMaterialDirective;
    exports.ShadowMaterialDirective = ShadowMaterialDirective;
    exports.SpriteMaterialDirective = SpriteMaterialDirective;
    exports.ThreeLineBasicMaterialModule = ThreeLineBasicMaterialModule;
    exports.ThreeLineDashedMaterialModule = ThreeLineDashedMaterialModule;
    exports.ThreeMeshBasicMaterialModule = ThreeMeshBasicMaterialModule;
    exports.ThreeMeshDepthMaterialModule = ThreeMeshDepthMaterialModule;
    exports.ThreeMeshDistanceMaterialModule = ThreeMeshDistanceMaterialModule;
    exports.ThreeMeshLambertMaterialModule = ThreeMeshLambertMaterialModule;
    exports.ThreeMeshMatcapMaterialModule = ThreeMeshMatcapMaterialModule;
    exports.ThreeMeshNormalMaterialModule = ThreeMeshNormalMaterialModule;
    exports.ThreeMeshPhongMaterialModule = ThreeMeshPhongMaterialModule;
    exports.ThreeMeshPhysicalMaterialModule = ThreeMeshPhysicalMaterialModule;
    exports.ThreeMeshStandardMaterialModule = ThreeMeshStandardMaterialModule;
    exports.ThreeMeshToonMaterialModule = ThreeMeshToonMaterialModule;
    exports.ThreePointsMaterialModule = ThreePointsMaterialModule;
    exports.ThreeRawShaderMaterialModule = ThreeRawShaderMaterialModule;
    exports.ThreeShaderMaterialModule = ThreeShaderMaterialModule;
    exports.ThreeShadowMaterialModule = ThreeShadowMaterialModule;
    exports.ThreeSpriteMaterialModule = ThreeSpriteMaterialModule;

    Object.defineProperty(exports, '__esModule', { value: true });

}));
