// @angular-three/core/bundles/angular-three-core-lights.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular-three/core'), require('@angular/core'), require('three')) :
    typeof define === 'function' && define.amd ? define('@angular-three/core/lights', ['exports', '@angular-three/core', '@angular/core', 'three'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory((global["angular-three"] = global["angular-three"] || {}, global["angular-three"].core = global["angular-three"].core || {}, global["angular-three"].core.lights = {}), global["angular-three"].core, global.ng.core, global.three));
})(this, (function (exports, core, i0, three) { 'use strict';

    function _interopNamespace(e) {
        if (e && e.__esModule) return e;
        var n = Object.create(null);
        if (e) {
            Object.keys(e).forEach(function (k) {
                if (k !== 'default') {
                    var d = Object.getOwnPropertyDescriptor(e, k);
                    Object.defineProperty(n, k, d.get ? d : {
                        enumerable: true,
                        get: function () { return e[k]; }
                    });
                }
            });
        }
        n["default"] = e;
        return Object.freeze(n);
    }

    var i0__namespace = /*#__PURE__*/_interopNamespace(i0);

    /*! *****************************************************************************
    Copyright (c) Microsoft Corporation.

    Permission to use, copy, modify, and/or distribute this software for any
    purpose with or without fee is hereby granted.

    THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES WITH
    REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY
    AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT,
    INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM
    LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR
    OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
    PERFORMANCE OF THIS SOFTWARE.
    ***************************************************************************** */
    /* global Reflect, Promise */
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b)
                if (Object.prototype.hasOwnProperty.call(b, p))
                    d[p] = b[p]; };
        return extendStatics(d, b);
    };
    function __extends(d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    }
    var __assign = function () {
        __assign = Object.assign || function __assign(t) {
            for (var s, i = 1, n = arguments.length; i < n; i++) {
                s = arguments[i];
                for (var p in s)
                    if (Object.prototype.hasOwnProperty.call(s, p))
                        t[p] = s[p];
            }
            return t;
        };
        return __assign.apply(this, arguments);
    };
    function __rest(s, e) {
        var t = {};
        for (var p in s)
            if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
                t[p] = s[p];
        if (s != null && typeof Object.getOwnPropertySymbols === "function")
            for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
                if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                    t[p[i]] = s[p[i]];
            }
        return t;
    }
    function __decorate(decorators, target, key, desc) {
        var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
            r = Reflect.decorate(decorators, target, key, desc);
        else
            for (var i = decorators.length - 1; i >= 0; i--)
                if (d = decorators[i])
                    r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    }
    function __param(paramIndex, decorator) {
        return function (target, key) { decorator(target, key, paramIndex); };
    }
    function __metadata(metadataKey, metadataValue) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function")
            return Reflect.metadata(metadataKey, metadataValue);
    }
    function __awaiter(thisArg, _arguments, P, generator) {
        function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
        return new (P || (P = Promise))(function (resolve, reject) {
            function fulfilled(value) { try {
                step(generator.next(value));
            }
            catch (e) {
                reject(e);
            } }
            function rejected(value) { try {
                step(generator["throw"](value));
            }
            catch (e) {
                reject(e);
            } }
            function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
            step((generator = generator.apply(thisArg, _arguments || [])).next());
        });
    }
    function __generator(thisArg, body) {
        var _ = { label: 0, sent: function () { if (t[0] & 1)
                throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
        return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function () { return this; }), g;
        function verb(n) { return function (v) { return step([n, v]); }; }
        function step(op) {
            if (f)
                throw new TypeError("Generator is already executing.");
            while (_)
                try {
                    if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done)
                        return t;
                    if (y = 0, t)
                        op = [op[0] & 2, t.value];
                    switch (op[0]) {
                        case 0:
                        case 1:
                            t = op;
                            break;
                        case 4:
                            _.label++;
                            return { value: op[1], done: false };
                        case 5:
                            _.label++;
                            y = op[1];
                            op = [0];
                            continue;
                        case 7:
                            op = _.ops.pop();
                            _.trys.pop();
                            continue;
                        default:
                            if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) {
                                _ = 0;
                                continue;
                            }
                            if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) {
                                _.label = op[1];
                                break;
                            }
                            if (op[0] === 6 && _.label < t[1]) {
                                _.label = t[1];
                                t = op;
                                break;
                            }
                            if (t && _.label < t[2]) {
                                _.label = t[2];
                                _.ops.push(op);
                                break;
                            }
                            if (t[2])
                                _.ops.pop();
                            _.trys.pop();
                            continue;
                    }
                    op = body.call(thisArg, _);
                }
                catch (e) {
                    op = [6, e];
                    y = 0;
                }
                finally {
                    f = t = 0;
                }
            if (op[0] & 5)
                throw op[1];
            return { value: op[0] ? op[1] : void 0, done: true };
        }
    }
    var __createBinding = Object.create ? (function (o, m, k, k2) {
        if (k2 === undefined)
            k2 = k;
        Object.defineProperty(o, k2, { enumerable: true, get: function () { return m[k]; } });
    }) : (function (o, m, k, k2) {
        if (k2 === undefined)
            k2 = k;
        o[k2] = m[k];
    });
    function __exportStar(m, o) {
        for (var p in m)
            if (p !== "default" && !Object.prototype.hasOwnProperty.call(o, p))
                __createBinding(o, m, p);
    }
    function __values(o) {
        var s = typeof Symbol === "function" && Symbol.iterator, m = s && o[s], i = 0;
        if (m)
            return m.call(o);
        if (o && typeof o.length === "number")
            return {
                next: function () {
                    if (o && i >= o.length)
                        o = void 0;
                    return { value: o && o[i++], done: !o };
                }
            };
        throw new TypeError(s ? "Object is not iterable." : "Symbol.iterator is not defined.");
    }
    function __read(o, n) {
        var m = typeof Symbol === "function" && o[Symbol.iterator];
        if (!m)
            return o;
        var i = m.call(o), r, ar = [], e;
        try {
            while ((n === void 0 || n-- > 0) && !(r = i.next()).done)
                ar.push(r.value);
        }
        catch (error) {
            e = { error: error };
        }
        finally {
            try {
                if (r && !r.done && (m = i["return"]))
                    m.call(i);
            }
            finally {
                if (e)
                    throw e.error;
            }
        }
        return ar;
    }
    /** @deprecated */
    function __spread() {
        for (var ar = [], i = 0; i < arguments.length; i++)
            ar = ar.concat(__read(arguments[i]));
        return ar;
    }
    /** @deprecated */
    function __spreadArrays() {
        for (var s = 0, i = 0, il = arguments.length; i < il; i++)
            s += arguments[i].length;
        for (var r = Array(s), k = 0, i = 0; i < il; i++)
            for (var a = arguments[i], j = 0, jl = a.length; j < jl; j++, k++)
                r[k] = a[j];
        return r;
    }
    function __spreadArray(to, from, pack) {
        if (pack || arguments.length === 2)
            for (var i = 0, l = from.length, ar; i < l; i++) {
                if (ar || !(i in from)) {
                    if (!ar)
                        ar = Array.prototype.slice.call(from, 0, i);
                    ar[i] = from[i];
                }
            }
        return to.concat(ar || Array.prototype.slice.call(from));
    }
    function __await(v) {
        return this instanceof __await ? (this.v = v, this) : new __await(v);
    }
    function __asyncGenerator(thisArg, _arguments, generator) {
        if (!Symbol.asyncIterator)
            throw new TypeError("Symbol.asyncIterator is not defined.");
        var g = generator.apply(thisArg, _arguments || []), i, q = [];
        return i = {}, verb("next"), verb("throw"), verb("return"), i[Symbol.asyncIterator] = function () { return this; }, i;
        function verb(n) { if (g[n])
            i[n] = function (v) { return new Promise(function (a, b) { q.push([n, v, a, b]) > 1 || resume(n, v); }); }; }
        function resume(n, v) { try {
            step(g[n](v));
        }
        catch (e) {
            settle(q[0][3], e);
        } }
        function step(r) { r.value instanceof __await ? Promise.resolve(r.value.v).then(fulfill, reject) : settle(q[0][2], r); }
        function fulfill(value) { resume("next", value); }
        function reject(value) { resume("throw", value); }
        function settle(f, v) { if (f(v), q.shift(), q.length)
            resume(q[0][0], q[0][1]); }
    }
    function __asyncDelegator(o) {
        var i, p;
        return i = {}, verb("next"), verb("throw", function (e) { throw e; }), verb("return"), i[Symbol.iterator] = function () { return this; }, i;
        function verb(n, f) { i[n] = o[n] ? function (v) { return (p = !p) ? { value: __await(o[n](v)), done: n === "return" } : f ? f(v) : v; } : f; }
    }
    function __asyncValues(o) {
        if (!Symbol.asyncIterator)
            throw new TypeError("Symbol.asyncIterator is not defined.");
        var m = o[Symbol.asyncIterator], i;
        return m ? m.call(o) : (o = typeof __values === "function" ? __values(o) : o[Symbol.iterator](), i = {}, verb("next"), verb("throw"), verb("return"), i[Symbol.asyncIterator] = function () { return this; }, i);
        function verb(n) { i[n] = o[n] && function (v) { return new Promise(function (resolve, reject) { v = o[n](v), settle(resolve, reject, v.done, v.value); }); }; }
        function settle(resolve, reject, d, v) { Promise.resolve(v).then(function (v) { resolve({ value: v, done: d }); }, reject); }
    }
    function __makeTemplateObject(cooked, raw) {
        if (Object.defineProperty) {
            Object.defineProperty(cooked, "raw", { value: raw });
        }
        else {
            cooked.raw = raw;
        }
        return cooked;
    }
    ;
    var __setModuleDefault = Object.create ? (function (o, v) {
        Object.defineProperty(o, "default", { enumerable: true, value: v });
    }) : function (o, v) {
        o["default"] = v;
    };
    function __importStar(mod) {
        if (mod && mod.__esModule)
            return mod;
        var result = {};
        if (mod != null)
            for (var k in mod)
                if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k))
                    __createBinding(result, mod, k);
        __setModuleDefault(result, mod);
        return result;
    }
    function __importDefault(mod) {
        return (mod && mod.__esModule) ? mod : { default: mod };
    }
    function __classPrivateFieldGet(receiver, state, kind, f) {
        if (kind === "a" && !f)
            throw new TypeError("Private accessor was defined without a getter");
        if (typeof state === "function" ? receiver !== state || !f : !state.has(receiver))
            throw new TypeError("Cannot read private member from an object whose class did not declare it");
        return kind === "m" ? f : kind === "a" ? f.call(receiver) : f ? f.value : state.get(receiver);
    }
    function __classPrivateFieldSet(receiver, state, value, kind, f) {
        if (kind === "m")
            throw new TypeError("Private method is not writable");
        if (kind === "a" && !f)
            throw new TypeError("Private accessor was defined without a setter");
        if (typeof state === "function" ? receiver !== state || !f : !state.has(receiver))
            throw new TypeError("Cannot write private member to an object whose class did not declare it");
        return (kind === "a" ? f.call(receiver, value) : f ? f.value = value : state.set(receiver, value)), value;
    }

    var LightProbeDirective = /** @class */ (function (_super) {
        __extends(LightProbeDirective, _super);
        function LightProbeDirective() {
            var _this = _super.apply(this, __spreadArray([], __read(arguments))) || this;
            _this.lightType = three.LightProbe;
            return _this;
        }
        Object.defineProperty(LightProbeDirective.prototype, "args", {
            set: function (v) {
                this.extraArgs = v;
            },
            enumerable: false,
            configurable: true
        });
        return LightProbeDirective;
    }(core.ThreeLight));
    LightProbeDirective.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: LightProbeDirective, deps: null, target: i0__namespace.ɵɵFactoryTarget.Directive });
    LightProbeDirective.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: LightProbeDirective, selector: "ngt-light-probe", inputs: { args: "args" }, providers: [
            {
                provide: core.ThreeObject3d,
                useExisting: LightProbeDirective,
            },
            core.OBJECT_3D_CONTROLLER_PROVIDER,
        ], exportAs: ["ngtLightProbe"], usesInheritance: true, ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: LightProbeDirective, decorators: [{
                type: i0.Directive,
                args: [{
                        selector: 'ngt-light-probe',
                        exportAs: 'ngtLightProbe',
                        providers: [
                            {
                                provide: core.ThreeObject3d,
                                useExisting: LightProbeDirective,
                            },
                            core.OBJECT_3D_CONTROLLER_PROVIDER,
                        ],
                    }]
            }], propDecorators: { args: [{
                    type: i0.Input
                }] } });

    // GENERATED
    var ThreeLightProbeModule = /** @class */ (function () {
        function ThreeLightProbeModule() {
        }
        return ThreeLightProbeModule;
    }());
    ThreeLightProbeModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeLightProbeModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    ThreeLightProbeModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeLightProbeModule, declarations: [LightProbeDirective], exports: [LightProbeDirective] });
    ThreeLightProbeModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeLightProbeModule });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeLightProbeModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        declarations: [LightProbeDirective],
                        exports: [LightProbeDirective],
                    }]
            }] });

    // GENERATED

    var AmbientLightDirective = /** @class */ (function (_super) {
        __extends(AmbientLightDirective, _super);
        function AmbientLightDirective() {
            var _this = _super.apply(this, __spreadArray([], __read(arguments))) || this;
            _this.lightType = three.AmbientLight;
            return _this;
        }
        Object.defineProperty(AmbientLightDirective.prototype, "args", {
            set: function (v) {
                this.extraArgs = v;
            },
            enumerable: false,
            configurable: true
        });
        return AmbientLightDirective;
    }(core.ThreeLight));
    AmbientLightDirective.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: AmbientLightDirective, deps: null, target: i0__namespace.ɵɵFactoryTarget.Directive });
    AmbientLightDirective.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: AmbientLightDirective, selector: "ngt-ambient-light", inputs: { args: "args" }, providers: [
            {
                provide: core.ThreeObject3d,
                useExisting: AmbientLightDirective,
            },
            core.OBJECT_3D_CONTROLLER_PROVIDER,
        ], exportAs: ["ngtAmbientLight"], usesInheritance: true, ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: AmbientLightDirective, decorators: [{
                type: i0.Directive,
                args: [{
                        selector: 'ngt-ambient-light',
                        exportAs: 'ngtAmbientLight',
                        providers: [
                            {
                                provide: core.ThreeObject3d,
                                useExisting: AmbientLightDirective,
                            },
                            core.OBJECT_3D_CONTROLLER_PROVIDER,
                        ],
                    }]
            }], propDecorators: { args: [{
                    type: i0.Input
                }] } });

    // GENERATED
    var ThreeAmbientLightModule = /** @class */ (function () {
        function ThreeAmbientLightModule() {
        }
        return ThreeAmbientLightModule;
    }());
    ThreeAmbientLightModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeAmbientLightModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    ThreeAmbientLightModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeAmbientLightModule, declarations: [AmbientLightDirective], exports: [AmbientLightDirective] });
    ThreeAmbientLightModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeAmbientLightModule });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeAmbientLightModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        declarations: [AmbientLightDirective],
                        exports: [AmbientLightDirective],
                    }]
            }] });

    // GENERATED

    var AmbientLightProbeDirective = /** @class */ (function (_super) {
        __extends(AmbientLightProbeDirective, _super);
        function AmbientLightProbeDirective() {
            var _this = _super.apply(this, __spreadArray([], __read(arguments))) || this;
            _this.lightType = three.AmbientLightProbe;
            return _this;
        }
        Object.defineProperty(AmbientLightProbeDirective.prototype, "args", {
            set: function (v) {
                this.extraArgs = v;
            },
            enumerable: false,
            configurable: true
        });
        return AmbientLightProbeDirective;
    }(core.ThreeLight));
    AmbientLightProbeDirective.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: AmbientLightProbeDirective, deps: null, target: i0__namespace.ɵɵFactoryTarget.Directive });
    AmbientLightProbeDirective.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: AmbientLightProbeDirective, selector: "ngt-ambient-light-probe", inputs: { args: "args" }, providers: [
            {
                provide: core.ThreeObject3d,
                useExisting: AmbientLightProbeDirective,
            },
            core.OBJECT_3D_CONTROLLER_PROVIDER,
        ], exportAs: ["ngtAmbientLightProbe"], usesInheritance: true, ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: AmbientLightProbeDirective, decorators: [{
                type: i0.Directive,
                args: [{
                        selector: 'ngt-ambient-light-probe',
                        exportAs: 'ngtAmbientLightProbe',
                        providers: [
                            {
                                provide: core.ThreeObject3d,
                                useExisting: AmbientLightProbeDirective,
                            },
                            core.OBJECT_3D_CONTROLLER_PROVIDER,
                        ],
                    }]
            }], propDecorators: { args: [{
                    type: i0.Input
                }] } });

    // GENERATED
    var ThreeAmbientLightProbeModule = /** @class */ (function () {
        function ThreeAmbientLightProbeModule() {
        }
        return ThreeAmbientLightProbeModule;
    }());
    ThreeAmbientLightProbeModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeAmbientLightProbeModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    ThreeAmbientLightProbeModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeAmbientLightProbeModule, declarations: [AmbientLightProbeDirective], exports: [AmbientLightProbeDirective] });
    ThreeAmbientLightProbeModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeAmbientLightProbeModule });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeAmbientLightProbeModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        declarations: [AmbientLightProbeDirective],
                        exports: [AmbientLightProbeDirective],
                    }]
            }] });

    // GENERATED

    var HemisphereLightDirective = /** @class */ (function (_super) {
        __extends(HemisphereLightDirective, _super);
        function HemisphereLightDirective() {
            var _this = _super.apply(this, __spreadArray([], __read(arguments))) || this;
            _this.lightType = three.HemisphereLight;
            return _this;
        }
        Object.defineProperty(HemisphereLightDirective.prototype, "args", {
            set: function (v) {
                this.extraArgs = v;
            },
            enumerable: false,
            configurable: true
        });
        return HemisphereLightDirective;
    }(core.ThreeLight));
    HemisphereLightDirective.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: HemisphereLightDirective, deps: null, target: i0__namespace.ɵɵFactoryTarget.Directive });
    HemisphereLightDirective.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: HemisphereLightDirective, selector: "ngt-hemisphere-light", inputs: { args: "args" }, providers: [
            {
                provide: core.ThreeObject3d,
                useExisting: HemisphereLightDirective,
            },
            core.OBJECT_3D_CONTROLLER_PROVIDER,
        ], exportAs: ["ngtHemisphereLight"], usesInheritance: true, ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: HemisphereLightDirective, decorators: [{
                type: i0.Directive,
                args: [{
                        selector: 'ngt-hemisphere-light',
                        exportAs: 'ngtHemisphereLight',
                        providers: [
                            {
                                provide: core.ThreeObject3d,
                                useExisting: HemisphereLightDirective,
                            },
                            core.OBJECT_3D_CONTROLLER_PROVIDER,
                        ],
                    }]
            }], propDecorators: { args: [{
                    type: i0.Input
                }] } });

    // GENERATED
    var ThreeHemisphereLightModule = /** @class */ (function () {
        function ThreeHemisphereLightModule() {
        }
        return ThreeHemisphereLightModule;
    }());
    ThreeHemisphereLightModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeHemisphereLightModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    ThreeHemisphereLightModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeHemisphereLightModule, declarations: [HemisphereLightDirective], exports: [HemisphereLightDirective] });
    ThreeHemisphereLightModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeHemisphereLightModule });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeHemisphereLightModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        declarations: [HemisphereLightDirective],
                        exports: [HemisphereLightDirective],
                    }]
            }] });

    // GENERATED

    var HemisphereLightProbeDirective = /** @class */ (function (_super) {
        __extends(HemisphereLightProbeDirective, _super);
        function HemisphereLightProbeDirective() {
            var _this = _super.apply(this, __spreadArray([], __read(arguments))) || this;
            _this.lightType = three.HemisphereLightProbe;
            return _this;
        }
        Object.defineProperty(HemisphereLightProbeDirective.prototype, "args", {
            set: function (v) {
                this.extraArgs = v;
            },
            enumerable: false,
            configurable: true
        });
        return HemisphereLightProbeDirective;
    }(core.ThreeLight));
    HemisphereLightProbeDirective.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: HemisphereLightProbeDirective, deps: null, target: i0__namespace.ɵɵFactoryTarget.Directive });
    HemisphereLightProbeDirective.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: HemisphereLightProbeDirective, selector: "ngt-hemisphere-light-probe", inputs: { args: "args" }, providers: [
            {
                provide: core.ThreeObject3d,
                useExisting: HemisphereLightProbeDirective,
            },
            core.OBJECT_3D_CONTROLLER_PROVIDER,
        ], exportAs: ["ngtHemisphereLightProbe"], usesInheritance: true, ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: HemisphereLightProbeDirective, decorators: [{
                type: i0.Directive,
                args: [{
                        selector: 'ngt-hemisphere-light-probe',
                        exportAs: 'ngtHemisphereLightProbe',
                        providers: [
                            {
                                provide: core.ThreeObject3d,
                                useExisting: HemisphereLightProbeDirective,
                            },
                            core.OBJECT_3D_CONTROLLER_PROVIDER,
                        ],
                    }]
            }], propDecorators: { args: [{
                    type: i0.Input
                }] } });

    // GENERATED
    var ThreeHemisphereLightProbeModule = /** @class */ (function () {
        function ThreeHemisphereLightProbeModule() {
        }
        return ThreeHemisphereLightProbeModule;
    }());
    ThreeHemisphereLightProbeModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeHemisphereLightProbeModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    ThreeHemisphereLightProbeModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeHemisphereLightProbeModule, declarations: [HemisphereLightProbeDirective], exports: [HemisphereLightProbeDirective] });
    ThreeHemisphereLightProbeModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeHemisphereLightProbeModule });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeHemisphereLightProbeModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        declarations: [HemisphereLightProbeDirective],
                        exports: [HemisphereLightProbeDirective],
                    }]
            }] });

    // GENERATED

    var DirectionalLightDirective = /** @class */ (function (_super) {
        __extends(DirectionalLightDirective, _super);
        function DirectionalLightDirective() {
            var _this = _super.apply(this, __spreadArray([], __read(arguments))) || this;
            _this.lightType = three.DirectionalLight;
            return _this;
        }
        Object.defineProperty(DirectionalLightDirective.prototype, "args", {
            set: function (v) {
                this.extraArgs = v;
            },
            enumerable: false,
            configurable: true
        });
        return DirectionalLightDirective;
    }(core.ThreeLight));
    DirectionalLightDirective.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: DirectionalLightDirective, deps: null, target: i0__namespace.ɵɵFactoryTarget.Directive });
    DirectionalLightDirective.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: DirectionalLightDirective, selector: "ngt-directional-light", inputs: { args: "args" }, providers: [
            {
                provide: core.ThreeObject3d,
                useExisting: DirectionalLightDirective,
            },
            core.OBJECT_3D_CONTROLLER_PROVIDER,
        ], exportAs: ["ngtDirectionalLight"], usesInheritance: true, ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: DirectionalLightDirective, decorators: [{
                type: i0.Directive,
                args: [{
                        selector: 'ngt-directional-light',
                        exportAs: 'ngtDirectionalLight',
                        providers: [
                            {
                                provide: core.ThreeObject3d,
                                useExisting: DirectionalLightDirective,
                            },
                            core.OBJECT_3D_CONTROLLER_PROVIDER,
                        ],
                    }]
            }], propDecorators: { args: [{
                    type: i0.Input
                }] } });

    // GENERATED
    var ThreeDirectionalLightModule = /** @class */ (function () {
        function ThreeDirectionalLightModule() {
        }
        return ThreeDirectionalLightModule;
    }());
    ThreeDirectionalLightModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeDirectionalLightModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    ThreeDirectionalLightModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeDirectionalLightModule, declarations: [DirectionalLightDirective], exports: [DirectionalLightDirective] });
    ThreeDirectionalLightModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeDirectionalLightModule });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeDirectionalLightModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        declarations: [DirectionalLightDirective],
                        exports: [DirectionalLightDirective],
                    }]
            }] });

    // GENERATED

    var PointLightDirective = /** @class */ (function (_super) {
        __extends(PointLightDirective, _super);
        function PointLightDirective() {
            var _this = _super.apply(this, __spreadArray([], __read(arguments))) || this;
            _this.lightType = three.PointLight;
            return _this;
        }
        Object.defineProperty(PointLightDirective.prototype, "args", {
            set: function (v) {
                this.extraArgs = v;
            },
            enumerable: false,
            configurable: true
        });
        return PointLightDirective;
    }(core.ThreeLight));
    PointLightDirective.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: PointLightDirective, deps: null, target: i0__namespace.ɵɵFactoryTarget.Directive });
    PointLightDirective.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: PointLightDirective, selector: "ngt-point-light", inputs: { args: "args" }, providers: [
            {
                provide: core.ThreeObject3d,
                useExisting: PointLightDirective,
            },
            core.OBJECT_3D_CONTROLLER_PROVIDER,
        ], exportAs: ["ngtPointLight"], usesInheritance: true, ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: PointLightDirective, decorators: [{
                type: i0.Directive,
                args: [{
                        selector: 'ngt-point-light',
                        exportAs: 'ngtPointLight',
                        providers: [
                            {
                                provide: core.ThreeObject3d,
                                useExisting: PointLightDirective,
                            },
                            core.OBJECT_3D_CONTROLLER_PROVIDER,
                        ],
                    }]
            }], propDecorators: { args: [{
                    type: i0.Input
                }] } });

    // GENERATED
    var ThreePointLightModule = /** @class */ (function () {
        function ThreePointLightModule() {
        }
        return ThreePointLightModule;
    }());
    ThreePointLightModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreePointLightModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    ThreePointLightModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreePointLightModule, declarations: [PointLightDirective], exports: [PointLightDirective] });
    ThreePointLightModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreePointLightModule });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreePointLightModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        declarations: [PointLightDirective],
                        exports: [PointLightDirective],
                    }]
            }] });

    // GENERATED

    var SpotLightDirective = /** @class */ (function (_super) {
        __extends(SpotLightDirective, _super);
        function SpotLightDirective() {
            var _this = _super.apply(this, __spreadArray([], __read(arguments))) || this;
            _this.lightType = three.SpotLight;
            return _this;
        }
        Object.defineProperty(SpotLightDirective.prototype, "args", {
            set: function (v) {
                this.extraArgs = v;
            },
            enumerable: false,
            configurable: true
        });
        return SpotLightDirective;
    }(core.ThreeLight));
    SpotLightDirective.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: SpotLightDirective, deps: null, target: i0__namespace.ɵɵFactoryTarget.Directive });
    SpotLightDirective.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: SpotLightDirective, selector: "ngt-spot-light", inputs: { args: "args" }, providers: [
            {
                provide: core.ThreeObject3d,
                useExisting: SpotLightDirective,
            },
            core.OBJECT_3D_CONTROLLER_PROVIDER,
        ], exportAs: ["ngtSpotLight"], usesInheritance: true, ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: SpotLightDirective, decorators: [{
                type: i0.Directive,
                args: [{
                        selector: 'ngt-spot-light',
                        exportAs: 'ngtSpotLight',
                        providers: [
                            {
                                provide: core.ThreeObject3d,
                                useExisting: SpotLightDirective,
                            },
                            core.OBJECT_3D_CONTROLLER_PROVIDER,
                        ],
                    }]
            }], propDecorators: { args: [{
                    type: i0.Input
                }] } });

    // GENERATED
    var ThreeSpotLightModule = /** @class */ (function () {
        function ThreeSpotLightModule() {
        }
        return ThreeSpotLightModule;
    }());
    ThreeSpotLightModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeSpotLightModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    ThreeSpotLightModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeSpotLightModule, declarations: [SpotLightDirective], exports: [SpotLightDirective] });
    ThreeSpotLightModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeSpotLightModule });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeSpotLightModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        declarations: [SpotLightDirective],
                        exports: [SpotLightDirective],
                    }]
            }] });

    // GENERATED

    var RectAreaLightDirective = /** @class */ (function (_super) {
        __extends(RectAreaLightDirective, _super);
        function RectAreaLightDirective() {
            var _this = _super.apply(this, __spreadArray([], __read(arguments))) || this;
            _this.lightType = three.RectAreaLight;
            return _this;
        }
        Object.defineProperty(RectAreaLightDirective.prototype, "args", {
            set: function (v) {
                this.extraArgs = v;
            },
            enumerable: false,
            configurable: true
        });
        return RectAreaLightDirective;
    }(core.ThreeLight));
    RectAreaLightDirective.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: RectAreaLightDirective, deps: null, target: i0__namespace.ɵɵFactoryTarget.Directive });
    RectAreaLightDirective.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: RectAreaLightDirective, selector: "ngt-rect-area-light", inputs: { args: "args" }, providers: [
            {
                provide: core.ThreeObject3d,
                useExisting: RectAreaLightDirective,
            },
            core.OBJECT_3D_CONTROLLER_PROVIDER,
        ], exportAs: ["ngtRectAreaLight"], usesInheritance: true, ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: RectAreaLightDirective, decorators: [{
                type: i0.Directive,
                args: [{
                        selector: 'ngt-rect-area-light',
                        exportAs: 'ngtRectAreaLight',
                        providers: [
                            {
                                provide: core.ThreeObject3d,
                                useExisting: RectAreaLightDirective,
                            },
                            core.OBJECT_3D_CONTROLLER_PROVIDER,
                        ],
                    }]
            }], propDecorators: { args: [{
                    type: i0.Input
                }] } });

    // GENERATED
    var ThreeRectAreaLightModule = /** @class */ (function () {
        function ThreeRectAreaLightModule() {
        }
        return ThreeRectAreaLightModule;
    }());
    ThreeRectAreaLightModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeRectAreaLightModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    ThreeRectAreaLightModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeRectAreaLightModule, declarations: [RectAreaLightDirective], exports: [RectAreaLightDirective] });
    ThreeRectAreaLightModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeRectAreaLightModule });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeRectAreaLightModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        declarations: [RectAreaLightDirective],
                        exports: [RectAreaLightDirective],
                    }]
            }] });

    // GENERATED

    // GENERATED

    /**
     * Generated bundle index. Do not edit.
     */

    exports.AmbientLightDirective = AmbientLightDirective;
    exports.AmbientLightProbeDirective = AmbientLightProbeDirective;
    exports.DirectionalLightDirective = DirectionalLightDirective;
    exports.HemisphereLightDirective = HemisphereLightDirective;
    exports.HemisphereLightProbeDirective = HemisphereLightProbeDirective;
    exports.LightProbeDirective = LightProbeDirective;
    exports.PointLightDirective = PointLightDirective;
    exports.RectAreaLightDirective = RectAreaLightDirective;
    exports.SpotLightDirective = SpotLightDirective;
    exports.ThreeAmbientLightModule = ThreeAmbientLightModule;
    exports.ThreeAmbientLightProbeModule = ThreeAmbientLightProbeModule;
    exports.ThreeDirectionalLightModule = ThreeDirectionalLightModule;
    exports.ThreeHemisphereLightModule = ThreeHemisphereLightModule;
    exports.ThreeHemisphereLightProbeModule = ThreeHemisphereLightProbeModule;
    exports.ThreeLightProbeModule = ThreeLightProbeModule;
    exports.ThreePointLightModule = ThreePointLightModule;
    exports.ThreeRectAreaLightModule = ThreeRectAreaLightModule;
    exports.ThreeSpotLightModule = ThreeSpotLightModule;

    Object.defineProperty(exports, '__esModule', { value: true });

}));
