// @angular-three/core/bundles/angular-three-core-curves.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular-three/core'), require('@angular/core'), require('three')) :
    typeof define === 'function' && define.amd ? define('@angular-three/core/curves', ['exports', '@angular-three/core', '@angular/core', 'three'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory((global["angular-three"] = global["angular-three"] || {}, global["angular-three"].core = global["angular-three"].core || {}, global["angular-three"].core.curves = {}), global["angular-three"].core, global.ng.core, global.three));
})(this, (function (exports, core, i0, three) { 'use strict';

    function _interopNamespace(e) {
        if (e && e.__esModule) return e;
        var n = Object.create(null);
        if (e) {
            Object.keys(e).forEach(function (k) {
                if (k !== 'default') {
                    var d = Object.getOwnPropertyDescriptor(e, k);
                    Object.defineProperty(n, k, d.get ? d : {
                        enumerable: true,
                        get: function () { return e[k]; }
                    });
                }
            });
        }
        n["default"] = e;
        return Object.freeze(n);
    }

    var i0__namespace = /*#__PURE__*/_interopNamespace(i0);

    /*! *****************************************************************************
    Copyright (c) Microsoft Corporation.

    Permission to use, copy, modify, and/or distribute this software for any
    purpose with or without fee is hereby granted.

    THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES WITH
    REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY
    AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT,
    INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM
    LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR
    OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
    PERFORMANCE OF THIS SOFTWARE.
    ***************************************************************************** */
    /* global Reflect, Promise */
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b)
                if (Object.prototype.hasOwnProperty.call(b, p))
                    d[p] = b[p]; };
        return extendStatics(d, b);
    };
    function __extends(d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    }
    var __assign = function () {
        __assign = Object.assign || function __assign(t) {
            for (var s, i = 1, n = arguments.length; i < n; i++) {
                s = arguments[i];
                for (var p in s)
                    if (Object.prototype.hasOwnProperty.call(s, p))
                        t[p] = s[p];
            }
            return t;
        };
        return __assign.apply(this, arguments);
    };
    function __rest(s, e) {
        var t = {};
        for (var p in s)
            if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
                t[p] = s[p];
        if (s != null && typeof Object.getOwnPropertySymbols === "function")
            for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
                if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                    t[p[i]] = s[p[i]];
            }
        return t;
    }
    function __decorate(decorators, target, key, desc) {
        var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
            r = Reflect.decorate(decorators, target, key, desc);
        else
            for (var i = decorators.length - 1; i >= 0; i--)
                if (d = decorators[i])
                    r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    }
    function __param(paramIndex, decorator) {
        return function (target, key) { decorator(target, key, paramIndex); };
    }
    function __metadata(metadataKey, metadataValue) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function")
            return Reflect.metadata(metadataKey, metadataValue);
    }
    function __awaiter(thisArg, _arguments, P, generator) {
        function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
        return new (P || (P = Promise))(function (resolve, reject) {
            function fulfilled(value) { try {
                step(generator.next(value));
            }
            catch (e) {
                reject(e);
            } }
            function rejected(value) { try {
                step(generator["throw"](value));
            }
            catch (e) {
                reject(e);
            } }
            function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
            step((generator = generator.apply(thisArg, _arguments || [])).next());
        });
    }
    function __generator(thisArg, body) {
        var _ = { label: 0, sent: function () { if (t[0] & 1)
                throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
        return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function () { return this; }), g;
        function verb(n) { return function (v) { return step([n, v]); }; }
        function step(op) {
            if (f)
                throw new TypeError("Generator is already executing.");
            while (_)
                try {
                    if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done)
                        return t;
                    if (y = 0, t)
                        op = [op[0] & 2, t.value];
                    switch (op[0]) {
                        case 0:
                        case 1:
                            t = op;
                            break;
                        case 4:
                            _.label++;
                            return { value: op[1], done: false };
                        case 5:
                            _.label++;
                            y = op[1];
                            op = [0];
                            continue;
                        case 7:
                            op = _.ops.pop();
                            _.trys.pop();
                            continue;
                        default:
                            if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) {
                                _ = 0;
                                continue;
                            }
                            if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) {
                                _.label = op[1];
                                break;
                            }
                            if (op[0] === 6 && _.label < t[1]) {
                                _.label = t[1];
                                t = op;
                                break;
                            }
                            if (t && _.label < t[2]) {
                                _.label = t[2];
                                _.ops.push(op);
                                break;
                            }
                            if (t[2])
                                _.ops.pop();
                            _.trys.pop();
                            continue;
                    }
                    op = body.call(thisArg, _);
                }
                catch (e) {
                    op = [6, e];
                    y = 0;
                }
                finally {
                    f = t = 0;
                }
            if (op[0] & 5)
                throw op[1];
            return { value: op[0] ? op[1] : void 0, done: true };
        }
    }
    var __createBinding = Object.create ? (function (o, m, k, k2) {
        if (k2 === undefined)
            k2 = k;
        Object.defineProperty(o, k2, { enumerable: true, get: function () { return m[k]; } });
    }) : (function (o, m, k, k2) {
        if (k2 === undefined)
            k2 = k;
        o[k2] = m[k];
    });
    function __exportStar(m, o) {
        for (var p in m)
            if (p !== "default" && !Object.prototype.hasOwnProperty.call(o, p))
                __createBinding(o, m, p);
    }
    function __values(o) {
        var s = typeof Symbol === "function" && Symbol.iterator, m = s && o[s], i = 0;
        if (m)
            return m.call(o);
        if (o && typeof o.length === "number")
            return {
                next: function () {
                    if (o && i >= o.length)
                        o = void 0;
                    return { value: o && o[i++], done: !o };
                }
            };
        throw new TypeError(s ? "Object is not iterable." : "Symbol.iterator is not defined.");
    }
    function __read(o, n) {
        var m = typeof Symbol === "function" && o[Symbol.iterator];
        if (!m)
            return o;
        var i = m.call(o), r, ar = [], e;
        try {
            while ((n === void 0 || n-- > 0) && !(r = i.next()).done)
                ar.push(r.value);
        }
        catch (error) {
            e = { error: error };
        }
        finally {
            try {
                if (r && !r.done && (m = i["return"]))
                    m.call(i);
            }
            finally {
                if (e)
                    throw e.error;
            }
        }
        return ar;
    }
    /** @deprecated */
    function __spread() {
        for (var ar = [], i = 0; i < arguments.length; i++)
            ar = ar.concat(__read(arguments[i]));
        return ar;
    }
    /** @deprecated */
    function __spreadArrays() {
        for (var s = 0, i = 0, il = arguments.length; i < il; i++)
            s += arguments[i].length;
        for (var r = Array(s), k = 0, i = 0; i < il; i++)
            for (var a = arguments[i], j = 0, jl = a.length; j < jl; j++, k++)
                r[k] = a[j];
        return r;
    }
    function __spreadArray(to, from, pack) {
        if (pack || arguments.length === 2)
            for (var i = 0, l = from.length, ar; i < l; i++) {
                if (ar || !(i in from)) {
                    if (!ar)
                        ar = Array.prototype.slice.call(from, 0, i);
                    ar[i] = from[i];
                }
            }
        return to.concat(ar || Array.prototype.slice.call(from));
    }
    function __await(v) {
        return this instanceof __await ? (this.v = v, this) : new __await(v);
    }
    function __asyncGenerator(thisArg, _arguments, generator) {
        if (!Symbol.asyncIterator)
            throw new TypeError("Symbol.asyncIterator is not defined.");
        var g = generator.apply(thisArg, _arguments || []), i, q = [];
        return i = {}, verb("next"), verb("throw"), verb("return"), i[Symbol.asyncIterator] = function () { return this; }, i;
        function verb(n) { if (g[n])
            i[n] = function (v) { return new Promise(function (a, b) { q.push([n, v, a, b]) > 1 || resume(n, v); }); }; }
        function resume(n, v) { try {
            step(g[n](v));
        }
        catch (e) {
            settle(q[0][3], e);
        } }
        function step(r) { r.value instanceof __await ? Promise.resolve(r.value.v).then(fulfill, reject) : settle(q[0][2], r); }
        function fulfill(value) { resume("next", value); }
        function reject(value) { resume("throw", value); }
        function settle(f, v) { if (f(v), q.shift(), q.length)
            resume(q[0][0], q[0][1]); }
    }
    function __asyncDelegator(o) {
        var i, p;
        return i = {}, verb("next"), verb("throw", function (e) { throw e; }), verb("return"), i[Symbol.iterator] = function () { return this; }, i;
        function verb(n, f) { i[n] = o[n] ? function (v) { return (p = !p) ? { value: __await(o[n](v)), done: n === "return" } : f ? f(v) : v; } : f; }
    }
    function __asyncValues(o) {
        if (!Symbol.asyncIterator)
            throw new TypeError("Symbol.asyncIterator is not defined.");
        var m = o[Symbol.asyncIterator], i;
        return m ? m.call(o) : (o = typeof __values === "function" ? __values(o) : o[Symbol.iterator](), i = {}, verb("next"), verb("throw"), verb("return"), i[Symbol.asyncIterator] = function () { return this; }, i);
        function verb(n) { i[n] = o[n] && function (v) { return new Promise(function (resolve, reject) { v = o[n](v), settle(resolve, reject, v.done, v.value); }); }; }
        function settle(resolve, reject, d, v) { Promise.resolve(v).then(function (v) { resolve({ value: v, done: d }); }, reject); }
    }
    function __makeTemplateObject(cooked, raw) {
        if (Object.defineProperty) {
            Object.defineProperty(cooked, "raw", { value: raw });
        }
        else {
            cooked.raw = raw;
        }
        return cooked;
    }
    ;
    var __setModuleDefault = Object.create ? (function (o, v) {
        Object.defineProperty(o, "default", { enumerable: true, value: v });
    }) : function (o, v) {
        o["default"] = v;
    };
    function __importStar(mod) {
        if (mod && mod.__esModule)
            return mod;
        var result = {};
        if (mod != null)
            for (var k in mod)
                if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k))
                    __createBinding(result, mod, k);
        __setModuleDefault(result, mod);
        return result;
    }
    function __importDefault(mod) {
        return (mod && mod.__esModule) ? mod : { default: mod };
    }
    function __classPrivateFieldGet(receiver, state, kind, f) {
        if (kind === "a" && !f)
            throw new TypeError("Private accessor was defined without a getter");
        if (typeof state === "function" ? receiver !== state || !f : !state.has(receiver))
            throw new TypeError("Cannot read private member from an object whose class did not declare it");
        return kind === "m" ? f : kind === "a" ? f.call(receiver) : f ? f.value : state.get(receiver);
    }
    function __classPrivateFieldSet(receiver, state, value, kind, f) {
        if (kind === "m")
            throw new TypeError("Private method is not writable");
        if (kind === "a" && !f)
            throw new TypeError("Private accessor was defined without a setter");
        if (typeof state === "function" ? receiver !== state || !f : !state.has(receiver))
            throw new TypeError("Cannot write private member to an object whose class did not declare it");
        return (kind === "a" ? f.call(receiver, value) : f ? f.value = value : state.set(receiver, value)), value;
    }

    var CatmullRomCurve3Directive = /** @class */ (function (_super) {
        __extends(CatmullRomCurve3Directive, _super);
        function CatmullRomCurve3Directive() {
            var _this = _super.apply(this, __spreadArray([], __read(arguments))) || this;
            _this.curveType = three.CatmullRomCurve3;
            return _this;
        }
        Object.defineProperty(CatmullRomCurve3Directive.prototype, "args", {
            set: function (v) {
                this.extraArgs = v;
            },
            enumerable: false,
            configurable: true
        });
        return CatmullRomCurve3Directive;
    }(core.ThreeCurve));
    CatmullRomCurve3Directive.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: CatmullRomCurve3Directive, deps: null, target: i0__namespace.ɵɵFactoryTarget.Directive });
    CatmullRomCurve3Directive.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: CatmullRomCurve3Directive, selector: "ngt-catmull-rom-curve3", inputs: { args: "args" }, providers: [
            {
                provide: core.ThreeCurve,
                useExisting: CatmullRomCurve3Directive,
            },
        ], exportAs: ["ngtCatmullRomCurve3"], usesInheritance: true, ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: CatmullRomCurve3Directive, decorators: [{
                type: i0.Directive,
                args: [{
                        selector: 'ngt-catmull-rom-curve3',
                        exportAs: 'ngtCatmullRomCurve3',
                        providers: [
                            {
                                provide: core.ThreeCurve,
                                useExisting: CatmullRomCurve3Directive,
                            },
                        ],
                    }]
            }], propDecorators: { args: [{
                    type: i0.Input
                }] } });

    // GENERATED
    var ThreeCatmullRomCurve3Module = /** @class */ (function () {
        function ThreeCatmullRomCurve3Module() {
        }
        return ThreeCatmullRomCurve3Module;
    }());
    ThreeCatmullRomCurve3Module.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeCatmullRomCurve3Module, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    ThreeCatmullRomCurve3Module.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeCatmullRomCurve3Module, declarations: [CatmullRomCurve3Directive], exports: [CatmullRomCurve3Directive] });
    ThreeCatmullRomCurve3Module.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeCatmullRomCurve3Module });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeCatmullRomCurve3Module, decorators: [{
                type: i0.NgModule,
                args: [{
                        declarations: [CatmullRomCurve3Directive],
                        exports: [CatmullRomCurve3Directive],
                    }]
            }] });

    // GENERATED

    var CubicBezierCurveDirective = /** @class */ (function (_super) {
        __extends(CubicBezierCurveDirective, _super);
        function CubicBezierCurveDirective() {
            var _this = _super.apply(this, __spreadArray([], __read(arguments))) || this;
            _this.curveType = three.CubicBezierCurve;
            return _this;
        }
        Object.defineProperty(CubicBezierCurveDirective.prototype, "args", {
            set: function (v) {
                this.extraArgs = v;
            },
            enumerable: false,
            configurable: true
        });
        return CubicBezierCurveDirective;
    }(core.ThreeCurve));
    CubicBezierCurveDirective.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: CubicBezierCurveDirective, deps: null, target: i0__namespace.ɵɵFactoryTarget.Directive });
    CubicBezierCurveDirective.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: CubicBezierCurveDirective, selector: "ngt-cubic-bezier-curve", inputs: { args: "args" }, providers: [
            {
                provide: core.ThreeCurve,
                useExisting: CubicBezierCurveDirective,
            },
        ], exportAs: ["ngtCubicBezierCurve"], usesInheritance: true, ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: CubicBezierCurveDirective, decorators: [{
                type: i0.Directive,
                args: [{
                        selector: 'ngt-cubic-bezier-curve',
                        exportAs: 'ngtCubicBezierCurve',
                        providers: [
                            {
                                provide: core.ThreeCurve,
                                useExisting: CubicBezierCurveDirective,
                            },
                        ],
                    }]
            }], propDecorators: { args: [{
                    type: i0.Input
                }] } });

    // GENERATED
    var ThreeCubicBezierCurveModule = /** @class */ (function () {
        function ThreeCubicBezierCurveModule() {
        }
        return ThreeCubicBezierCurveModule;
    }());
    ThreeCubicBezierCurveModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeCubicBezierCurveModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    ThreeCubicBezierCurveModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeCubicBezierCurveModule, declarations: [CubicBezierCurveDirective], exports: [CubicBezierCurveDirective] });
    ThreeCubicBezierCurveModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeCubicBezierCurveModule });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeCubicBezierCurveModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        declarations: [CubicBezierCurveDirective],
                        exports: [CubicBezierCurveDirective],
                    }]
            }] });

    // GENERATED

    var CubicBezierCurve3Directive = /** @class */ (function (_super) {
        __extends(CubicBezierCurve3Directive, _super);
        function CubicBezierCurve3Directive() {
            var _this = _super.apply(this, __spreadArray([], __read(arguments))) || this;
            _this.curveType = three.CubicBezierCurve3;
            return _this;
        }
        Object.defineProperty(CubicBezierCurve3Directive.prototype, "args", {
            set: function (v) {
                this.extraArgs = v;
            },
            enumerable: false,
            configurable: true
        });
        return CubicBezierCurve3Directive;
    }(core.ThreeCurve));
    CubicBezierCurve3Directive.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: CubicBezierCurve3Directive, deps: null, target: i0__namespace.ɵɵFactoryTarget.Directive });
    CubicBezierCurve3Directive.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: CubicBezierCurve3Directive, selector: "ngt-cubic-bezier-curve3", inputs: { args: "args" }, providers: [
            {
                provide: core.ThreeCurve,
                useExisting: CubicBezierCurve3Directive,
            },
        ], exportAs: ["ngtCubicBezierCurve3"], usesInheritance: true, ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: CubicBezierCurve3Directive, decorators: [{
                type: i0.Directive,
                args: [{
                        selector: 'ngt-cubic-bezier-curve3',
                        exportAs: 'ngtCubicBezierCurve3',
                        providers: [
                            {
                                provide: core.ThreeCurve,
                                useExisting: CubicBezierCurve3Directive,
                            },
                        ],
                    }]
            }], propDecorators: { args: [{
                    type: i0.Input
                }] } });

    // GENERATED
    var ThreeCubicBezierCurve3Module = /** @class */ (function () {
        function ThreeCubicBezierCurve3Module() {
        }
        return ThreeCubicBezierCurve3Module;
    }());
    ThreeCubicBezierCurve3Module.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeCubicBezierCurve3Module, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    ThreeCubicBezierCurve3Module.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeCubicBezierCurve3Module, declarations: [CubicBezierCurve3Directive], exports: [CubicBezierCurve3Directive] });
    ThreeCubicBezierCurve3Module.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeCubicBezierCurve3Module });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeCubicBezierCurve3Module, decorators: [{
                type: i0.NgModule,
                args: [{
                        declarations: [CubicBezierCurve3Directive],
                        exports: [CubicBezierCurve3Directive],
                    }]
            }] });

    // GENERATED

    var EllipseCurveDirective = /** @class */ (function (_super) {
        __extends(EllipseCurveDirective, _super);
        function EllipseCurveDirective() {
            var _this = _super.apply(this, __spreadArray([], __read(arguments))) || this;
            _this.curveType = three.EllipseCurve;
            return _this;
        }
        Object.defineProperty(EllipseCurveDirective.prototype, "args", {
            set: function (v) {
                this.extraArgs = v;
            },
            enumerable: false,
            configurable: true
        });
        return EllipseCurveDirective;
    }(core.ThreeCurve));
    EllipseCurveDirective.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: EllipseCurveDirective, deps: null, target: i0__namespace.ɵɵFactoryTarget.Directive });
    EllipseCurveDirective.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: EllipseCurveDirective, selector: "ngt-ellipse-curve", inputs: { args: "args" }, providers: [
            {
                provide: core.ThreeCurve,
                useExisting: EllipseCurveDirective,
            },
        ], exportAs: ["ngtEllipseCurve"], usesInheritance: true, ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: EllipseCurveDirective, decorators: [{
                type: i0.Directive,
                args: [{
                        selector: 'ngt-ellipse-curve',
                        exportAs: 'ngtEllipseCurve',
                        providers: [
                            {
                                provide: core.ThreeCurve,
                                useExisting: EllipseCurveDirective,
                            },
                        ],
                    }]
            }], propDecorators: { args: [{
                    type: i0.Input
                }] } });

    // GENERATED
    var ThreeEllipseCurveModule = /** @class */ (function () {
        function ThreeEllipseCurveModule() {
        }
        return ThreeEllipseCurveModule;
    }());
    ThreeEllipseCurveModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeEllipseCurveModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    ThreeEllipseCurveModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeEllipseCurveModule, declarations: [EllipseCurveDirective], exports: [EllipseCurveDirective] });
    ThreeEllipseCurveModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeEllipseCurveModule });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeEllipseCurveModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        declarations: [EllipseCurveDirective],
                        exports: [EllipseCurveDirective],
                    }]
            }] });

    // GENERATED

    var LineCurveDirective = /** @class */ (function (_super) {
        __extends(LineCurveDirective, _super);
        function LineCurveDirective() {
            var _this = _super.apply(this, __spreadArray([], __read(arguments))) || this;
            _this.curveType = three.LineCurve;
            return _this;
        }
        Object.defineProperty(LineCurveDirective.prototype, "args", {
            set: function (v) {
                this.extraArgs = v;
            },
            enumerable: false,
            configurable: true
        });
        return LineCurveDirective;
    }(core.ThreeCurve));
    LineCurveDirective.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: LineCurveDirective, deps: null, target: i0__namespace.ɵɵFactoryTarget.Directive });
    LineCurveDirective.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: LineCurveDirective, selector: "ngt-line-curve", inputs: { args: "args" }, providers: [
            {
                provide: core.ThreeCurve,
                useExisting: LineCurveDirective,
            },
        ], exportAs: ["ngtLineCurve"], usesInheritance: true, ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: LineCurveDirective, decorators: [{
                type: i0.Directive,
                args: [{
                        selector: 'ngt-line-curve',
                        exportAs: 'ngtLineCurve',
                        providers: [
                            {
                                provide: core.ThreeCurve,
                                useExisting: LineCurveDirective,
                            },
                        ],
                    }]
            }], propDecorators: { args: [{
                    type: i0.Input
                }] } });

    // GENERATED
    var ThreeLineCurveModule = /** @class */ (function () {
        function ThreeLineCurveModule() {
        }
        return ThreeLineCurveModule;
    }());
    ThreeLineCurveModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeLineCurveModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    ThreeLineCurveModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeLineCurveModule, declarations: [LineCurveDirective], exports: [LineCurveDirective] });
    ThreeLineCurveModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeLineCurveModule });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeLineCurveModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        declarations: [LineCurveDirective],
                        exports: [LineCurveDirective],
                    }]
            }] });

    // GENERATED

    var LineCurve3Directive = /** @class */ (function (_super) {
        __extends(LineCurve3Directive, _super);
        function LineCurve3Directive() {
            var _this = _super.apply(this, __spreadArray([], __read(arguments))) || this;
            _this.curveType = three.LineCurve3;
            return _this;
        }
        Object.defineProperty(LineCurve3Directive.prototype, "args", {
            set: function (v) {
                this.extraArgs = v;
            },
            enumerable: false,
            configurable: true
        });
        return LineCurve3Directive;
    }(core.ThreeCurve));
    LineCurve3Directive.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: LineCurve3Directive, deps: null, target: i0__namespace.ɵɵFactoryTarget.Directive });
    LineCurve3Directive.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: LineCurve3Directive, selector: "ngt-line-curve3", inputs: { args: "args" }, providers: [
            {
                provide: core.ThreeCurve,
                useExisting: LineCurve3Directive,
            },
        ], exportAs: ["ngtLineCurve3"], usesInheritance: true, ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: LineCurve3Directive, decorators: [{
                type: i0.Directive,
                args: [{
                        selector: 'ngt-line-curve3',
                        exportAs: 'ngtLineCurve3',
                        providers: [
                            {
                                provide: core.ThreeCurve,
                                useExisting: LineCurve3Directive,
                            },
                        ],
                    }]
            }], propDecorators: { args: [{
                    type: i0.Input
                }] } });

    // GENERATED
    var ThreeLineCurve3Module = /** @class */ (function () {
        function ThreeLineCurve3Module() {
        }
        return ThreeLineCurve3Module;
    }());
    ThreeLineCurve3Module.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeLineCurve3Module, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    ThreeLineCurve3Module.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeLineCurve3Module, declarations: [LineCurve3Directive], exports: [LineCurve3Directive] });
    ThreeLineCurve3Module.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeLineCurve3Module });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeLineCurve3Module, decorators: [{
                type: i0.NgModule,
                args: [{
                        declarations: [LineCurve3Directive],
                        exports: [LineCurve3Directive],
                    }]
            }] });

    // GENERATED

    var QuadraticBezierCurveDirective = /** @class */ (function (_super) {
        __extends(QuadraticBezierCurveDirective, _super);
        function QuadraticBezierCurveDirective() {
            var _this = _super.apply(this, __spreadArray([], __read(arguments))) || this;
            _this.curveType = three.QuadraticBezierCurve;
            return _this;
        }
        Object.defineProperty(QuadraticBezierCurveDirective.prototype, "args", {
            set: function (v) {
                this.extraArgs = v;
            },
            enumerable: false,
            configurable: true
        });
        return QuadraticBezierCurveDirective;
    }(core.ThreeCurve));
    QuadraticBezierCurveDirective.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: QuadraticBezierCurveDirective, deps: null, target: i0__namespace.ɵɵFactoryTarget.Directive });
    QuadraticBezierCurveDirective.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: QuadraticBezierCurveDirective, selector: "ngt-quadratic-bezier-curve", inputs: { args: "args" }, providers: [
            {
                provide: core.ThreeCurve,
                useExisting: QuadraticBezierCurveDirective,
            },
        ], exportAs: ["ngtQuadraticBezierCurve"], usesInheritance: true, ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: QuadraticBezierCurveDirective, decorators: [{
                type: i0.Directive,
                args: [{
                        selector: 'ngt-quadratic-bezier-curve',
                        exportAs: 'ngtQuadraticBezierCurve',
                        providers: [
                            {
                                provide: core.ThreeCurve,
                                useExisting: QuadraticBezierCurveDirective,
                            },
                        ],
                    }]
            }], propDecorators: { args: [{
                    type: i0.Input
                }] } });

    // GENERATED
    var ThreeQuadraticBezierCurveModule = /** @class */ (function () {
        function ThreeQuadraticBezierCurveModule() {
        }
        return ThreeQuadraticBezierCurveModule;
    }());
    ThreeQuadraticBezierCurveModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeQuadraticBezierCurveModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    ThreeQuadraticBezierCurveModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeQuadraticBezierCurveModule, declarations: [QuadraticBezierCurveDirective], exports: [QuadraticBezierCurveDirective] });
    ThreeQuadraticBezierCurveModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeQuadraticBezierCurveModule });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeQuadraticBezierCurveModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        declarations: [QuadraticBezierCurveDirective],
                        exports: [QuadraticBezierCurveDirective],
                    }]
            }] });

    // GENERATED

    var QuadraticBezierCurve3Directive = /** @class */ (function (_super) {
        __extends(QuadraticBezierCurve3Directive, _super);
        function QuadraticBezierCurve3Directive() {
            var _this = _super.apply(this, __spreadArray([], __read(arguments))) || this;
            _this.curveType = three.QuadraticBezierCurve3;
            return _this;
        }
        Object.defineProperty(QuadraticBezierCurve3Directive.prototype, "args", {
            set: function (v) {
                this.extraArgs = v;
            },
            enumerable: false,
            configurable: true
        });
        return QuadraticBezierCurve3Directive;
    }(core.ThreeCurve));
    QuadraticBezierCurve3Directive.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: QuadraticBezierCurve3Directive, deps: null, target: i0__namespace.ɵɵFactoryTarget.Directive });
    QuadraticBezierCurve3Directive.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: QuadraticBezierCurve3Directive, selector: "ngt-quadratic-bezier-curve3", inputs: { args: "args" }, providers: [
            {
                provide: core.ThreeCurve,
                useExisting: QuadraticBezierCurve3Directive,
            },
        ], exportAs: ["ngtQuadraticBezierCurve3"], usesInheritance: true, ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: QuadraticBezierCurve3Directive, decorators: [{
                type: i0.Directive,
                args: [{
                        selector: 'ngt-quadratic-bezier-curve3',
                        exportAs: 'ngtQuadraticBezierCurve3',
                        providers: [
                            {
                                provide: core.ThreeCurve,
                                useExisting: QuadraticBezierCurve3Directive,
                            },
                        ],
                    }]
            }], propDecorators: { args: [{
                    type: i0.Input
                }] } });

    // GENERATED
    var ThreeQuadraticBezierCurve3Module = /** @class */ (function () {
        function ThreeQuadraticBezierCurve3Module() {
        }
        return ThreeQuadraticBezierCurve3Module;
    }());
    ThreeQuadraticBezierCurve3Module.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeQuadraticBezierCurve3Module, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    ThreeQuadraticBezierCurve3Module.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeQuadraticBezierCurve3Module, declarations: [QuadraticBezierCurve3Directive], exports: [QuadraticBezierCurve3Directive] });
    ThreeQuadraticBezierCurve3Module.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeQuadraticBezierCurve3Module });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeQuadraticBezierCurve3Module, decorators: [{
                type: i0.NgModule,
                args: [{
                        declarations: [QuadraticBezierCurve3Directive],
                        exports: [QuadraticBezierCurve3Directive],
                    }]
            }] });

    // GENERATED

    var SplineCurveDirective = /** @class */ (function (_super) {
        __extends(SplineCurveDirective, _super);
        function SplineCurveDirective() {
            var _this = _super.apply(this, __spreadArray([], __read(arguments))) || this;
            _this.curveType = three.SplineCurve;
            return _this;
        }
        Object.defineProperty(SplineCurveDirective.prototype, "args", {
            set: function (v) {
                this.extraArgs = v;
            },
            enumerable: false,
            configurable: true
        });
        return SplineCurveDirective;
    }(core.ThreeCurve));
    SplineCurveDirective.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: SplineCurveDirective, deps: null, target: i0__namespace.ɵɵFactoryTarget.Directive });
    SplineCurveDirective.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: SplineCurveDirective, selector: "ngt-spline-curve", inputs: { args: "args" }, providers: [
            {
                provide: core.ThreeCurve,
                useExisting: SplineCurveDirective,
            },
        ], exportAs: ["ngtSplineCurve"], usesInheritance: true, ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: SplineCurveDirective, decorators: [{
                type: i0.Directive,
                args: [{
                        selector: 'ngt-spline-curve',
                        exportAs: 'ngtSplineCurve',
                        providers: [
                            {
                                provide: core.ThreeCurve,
                                useExisting: SplineCurveDirective,
                            },
                        ],
                    }]
            }], propDecorators: { args: [{
                    type: i0.Input
                }] } });

    // GENERATED
    var ThreeSplineCurveModule = /** @class */ (function () {
        function ThreeSplineCurveModule() {
        }
        return ThreeSplineCurveModule;
    }());
    ThreeSplineCurveModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeSplineCurveModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    ThreeSplineCurveModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeSplineCurveModule, declarations: [SplineCurveDirective], exports: [SplineCurveDirective] });
    ThreeSplineCurveModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeSplineCurveModule });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeSplineCurveModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        declarations: [SplineCurveDirective],
                        exports: [SplineCurveDirective],
                    }]
            }] });

    // GENERATED

    // GENERATED

    /**
     * Generated bundle index. Do not edit.
     */

    exports.CatmullRomCurve3Directive = CatmullRomCurve3Directive;
    exports.CubicBezierCurve3Directive = CubicBezierCurve3Directive;
    exports.CubicBezierCurveDirective = CubicBezierCurveDirective;
    exports.EllipseCurveDirective = EllipseCurveDirective;
    exports.LineCurve3Directive = LineCurve3Directive;
    exports.LineCurveDirective = LineCurveDirective;
    exports.QuadraticBezierCurve3Directive = QuadraticBezierCurve3Directive;
    exports.QuadraticBezierCurveDirective = QuadraticBezierCurveDirective;
    exports.SplineCurveDirective = SplineCurveDirective;
    exports.ThreeCatmullRomCurve3Module = ThreeCatmullRomCurve3Module;
    exports.ThreeCubicBezierCurve3Module = ThreeCubicBezierCurve3Module;
    exports.ThreeCubicBezierCurveModule = ThreeCubicBezierCurveModule;
    exports.ThreeEllipseCurveModule = ThreeEllipseCurveModule;
    exports.ThreeLineCurve3Module = ThreeLineCurve3Module;
    exports.ThreeLineCurveModule = ThreeLineCurveModule;
    exports.ThreeQuadraticBezierCurve3Module = ThreeQuadraticBezierCurve3Module;
    exports.ThreeQuadraticBezierCurveModule = ThreeQuadraticBezierCurveModule;
    exports.ThreeSplineCurveModule = ThreeSplineCurveModule;

    Object.defineProperty(exports, '__esModule', { value: true });

}));
