// @angular-three/core/bundles/angular-three-core-helpers.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular-three/core'), require('@angular/core'), require('three')) :
    typeof define === 'function' && define.amd ? define('@angular-three/core/helpers', ['exports', '@angular-three/core', '@angular/core', 'three'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory((global["angular-three"] = global["angular-three"] || {}, global["angular-three"].core = global["angular-three"].core || {}, global["angular-three"].core.helpers = {}), global["angular-three"].core, global.ng.core, global.three));
})(this, (function (exports, core, i0, three) { 'use strict';

    function _interopNamespace(e) {
        if (e && e.__esModule) return e;
        var n = Object.create(null);
        if (e) {
            Object.keys(e).forEach(function (k) {
                if (k !== 'default') {
                    var d = Object.getOwnPropertyDescriptor(e, k);
                    Object.defineProperty(n, k, d.get ? d : {
                        enumerable: true,
                        get: function () { return e[k]; }
                    });
                }
            });
        }
        n["default"] = e;
        return Object.freeze(n);
    }

    var i0__namespace = /*#__PURE__*/_interopNamespace(i0);

    /*! *****************************************************************************
    Copyright (c) Microsoft Corporation.

    Permission to use, copy, modify, and/or distribute this software for any
    purpose with or without fee is hereby granted.

    THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES WITH
    REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY
    AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT,
    INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM
    LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR
    OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
    PERFORMANCE OF THIS SOFTWARE.
    ***************************************************************************** */
    /* global Reflect, Promise */
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b)
                if (Object.prototype.hasOwnProperty.call(b, p))
                    d[p] = b[p]; };
        return extendStatics(d, b);
    };
    function __extends(d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    }
    var __assign = function () {
        __assign = Object.assign || function __assign(t) {
            for (var s, i = 1, n = arguments.length; i < n; i++) {
                s = arguments[i];
                for (var p in s)
                    if (Object.prototype.hasOwnProperty.call(s, p))
                        t[p] = s[p];
            }
            return t;
        };
        return __assign.apply(this, arguments);
    };
    function __rest(s, e) {
        var t = {};
        for (var p in s)
            if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
                t[p] = s[p];
        if (s != null && typeof Object.getOwnPropertySymbols === "function")
            for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
                if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                    t[p[i]] = s[p[i]];
            }
        return t;
    }
    function __decorate(decorators, target, key, desc) {
        var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
            r = Reflect.decorate(decorators, target, key, desc);
        else
            for (var i = decorators.length - 1; i >= 0; i--)
                if (d = decorators[i])
                    r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    }
    function __param(paramIndex, decorator) {
        return function (target, key) { decorator(target, key, paramIndex); };
    }
    function __metadata(metadataKey, metadataValue) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function")
            return Reflect.metadata(metadataKey, metadataValue);
    }
    function __awaiter(thisArg, _arguments, P, generator) {
        function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
        return new (P || (P = Promise))(function (resolve, reject) {
            function fulfilled(value) { try {
                step(generator.next(value));
            }
            catch (e) {
                reject(e);
            } }
            function rejected(value) { try {
                step(generator["throw"](value));
            }
            catch (e) {
                reject(e);
            } }
            function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
            step((generator = generator.apply(thisArg, _arguments || [])).next());
        });
    }
    function __generator(thisArg, body) {
        var _ = { label: 0, sent: function () { if (t[0] & 1)
                throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
        return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function () { return this; }), g;
        function verb(n) { return function (v) { return step([n, v]); }; }
        function step(op) {
            if (f)
                throw new TypeError("Generator is already executing.");
            while (_)
                try {
                    if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done)
                        return t;
                    if (y = 0, t)
                        op = [op[0] & 2, t.value];
                    switch (op[0]) {
                        case 0:
                        case 1:
                            t = op;
                            break;
                        case 4:
                            _.label++;
                            return { value: op[1], done: false };
                        case 5:
                            _.label++;
                            y = op[1];
                            op = [0];
                            continue;
                        case 7:
                            op = _.ops.pop();
                            _.trys.pop();
                            continue;
                        default:
                            if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) {
                                _ = 0;
                                continue;
                            }
                            if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) {
                                _.label = op[1];
                                break;
                            }
                            if (op[0] === 6 && _.label < t[1]) {
                                _.label = t[1];
                                t = op;
                                break;
                            }
                            if (t && _.label < t[2]) {
                                _.label = t[2];
                                _.ops.push(op);
                                break;
                            }
                            if (t[2])
                                _.ops.pop();
                            _.trys.pop();
                            continue;
                    }
                    op = body.call(thisArg, _);
                }
                catch (e) {
                    op = [6, e];
                    y = 0;
                }
                finally {
                    f = t = 0;
                }
            if (op[0] & 5)
                throw op[1];
            return { value: op[0] ? op[1] : void 0, done: true };
        }
    }
    var __createBinding = Object.create ? (function (o, m, k, k2) {
        if (k2 === undefined)
            k2 = k;
        Object.defineProperty(o, k2, { enumerable: true, get: function () { return m[k]; } });
    }) : (function (o, m, k, k2) {
        if (k2 === undefined)
            k2 = k;
        o[k2] = m[k];
    });
    function __exportStar(m, o) {
        for (var p in m)
            if (p !== "default" && !Object.prototype.hasOwnProperty.call(o, p))
                __createBinding(o, m, p);
    }
    function __values(o) {
        var s = typeof Symbol === "function" && Symbol.iterator, m = s && o[s], i = 0;
        if (m)
            return m.call(o);
        if (o && typeof o.length === "number")
            return {
                next: function () {
                    if (o && i >= o.length)
                        o = void 0;
                    return { value: o && o[i++], done: !o };
                }
            };
        throw new TypeError(s ? "Object is not iterable." : "Symbol.iterator is not defined.");
    }
    function __read(o, n) {
        var m = typeof Symbol === "function" && o[Symbol.iterator];
        if (!m)
            return o;
        var i = m.call(o), r, ar = [], e;
        try {
            while ((n === void 0 || n-- > 0) && !(r = i.next()).done)
                ar.push(r.value);
        }
        catch (error) {
            e = { error: error };
        }
        finally {
            try {
                if (r && !r.done && (m = i["return"]))
                    m.call(i);
            }
            finally {
                if (e)
                    throw e.error;
            }
        }
        return ar;
    }
    /** @deprecated */
    function __spread() {
        for (var ar = [], i = 0; i < arguments.length; i++)
            ar = ar.concat(__read(arguments[i]));
        return ar;
    }
    /** @deprecated */
    function __spreadArrays() {
        for (var s = 0, i = 0, il = arguments.length; i < il; i++)
            s += arguments[i].length;
        for (var r = Array(s), k = 0, i = 0; i < il; i++)
            for (var a = arguments[i], j = 0, jl = a.length; j < jl; j++, k++)
                r[k] = a[j];
        return r;
    }
    function __spreadArray(to, from, pack) {
        if (pack || arguments.length === 2)
            for (var i = 0, l = from.length, ar; i < l; i++) {
                if (ar || !(i in from)) {
                    if (!ar)
                        ar = Array.prototype.slice.call(from, 0, i);
                    ar[i] = from[i];
                }
            }
        return to.concat(ar || Array.prototype.slice.call(from));
    }
    function __await(v) {
        return this instanceof __await ? (this.v = v, this) : new __await(v);
    }
    function __asyncGenerator(thisArg, _arguments, generator) {
        if (!Symbol.asyncIterator)
            throw new TypeError("Symbol.asyncIterator is not defined.");
        var g = generator.apply(thisArg, _arguments || []), i, q = [];
        return i = {}, verb("next"), verb("throw"), verb("return"), i[Symbol.asyncIterator] = function () { return this; }, i;
        function verb(n) { if (g[n])
            i[n] = function (v) { return new Promise(function (a, b) { q.push([n, v, a, b]) > 1 || resume(n, v); }); }; }
        function resume(n, v) { try {
            step(g[n](v));
        }
        catch (e) {
            settle(q[0][3], e);
        } }
        function step(r) { r.value instanceof __await ? Promise.resolve(r.value.v).then(fulfill, reject) : settle(q[0][2], r); }
        function fulfill(value) { resume("next", value); }
        function reject(value) { resume("throw", value); }
        function settle(f, v) { if (f(v), q.shift(), q.length)
            resume(q[0][0], q[0][1]); }
    }
    function __asyncDelegator(o) {
        var i, p;
        return i = {}, verb("next"), verb("throw", function (e) { throw e; }), verb("return"), i[Symbol.iterator] = function () { return this; }, i;
        function verb(n, f) { i[n] = o[n] ? function (v) { return (p = !p) ? { value: __await(o[n](v)), done: n === "return" } : f ? f(v) : v; } : f; }
    }
    function __asyncValues(o) {
        if (!Symbol.asyncIterator)
            throw new TypeError("Symbol.asyncIterator is not defined.");
        var m = o[Symbol.asyncIterator], i;
        return m ? m.call(o) : (o = typeof __values === "function" ? __values(o) : o[Symbol.iterator](), i = {}, verb("next"), verb("throw"), verb("return"), i[Symbol.asyncIterator] = function () { return this; }, i);
        function verb(n) { i[n] = o[n] && function (v) { return new Promise(function (resolve, reject) { v = o[n](v), settle(resolve, reject, v.done, v.value); }); }; }
        function settle(resolve, reject, d, v) { Promise.resolve(v).then(function (v) { resolve({ value: v, done: d }); }, reject); }
    }
    function __makeTemplateObject(cooked, raw) {
        if (Object.defineProperty) {
            Object.defineProperty(cooked, "raw", { value: raw });
        }
        else {
            cooked.raw = raw;
        }
        return cooked;
    }
    ;
    var __setModuleDefault = Object.create ? (function (o, v) {
        Object.defineProperty(o, "default", { enumerable: true, value: v });
    }) : function (o, v) {
        o["default"] = v;
    };
    function __importStar(mod) {
        if (mod && mod.__esModule)
            return mod;
        var result = {};
        if (mod != null)
            for (var k in mod)
                if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k))
                    __createBinding(result, mod, k);
        __setModuleDefault(result, mod);
        return result;
    }
    function __importDefault(mod) {
        return (mod && mod.__esModule) ? mod : { default: mod };
    }
    function __classPrivateFieldGet(receiver, state, kind, f) {
        if (kind === "a" && !f)
            throw new TypeError("Private accessor was defined without a getter");
        if (typeof state === "function" ? receiver !== state || !f : !state.has(receiver))
            throw new TypeError("Cannot read private member from an object whose class did not declare it");
        return kind === "m" ? f : kind === "a" ? f.call(receiver) : f ? f.value : state.get(receiver);
    }
    function __classPrivateFieldSet(receiver, state, value, kind, f) {
        if (kind === "m")
            throw new TypeError("Private method is not writable");
        if (kind === "a" && !f)
            throw new TypeError("Private accessor was defined without a setter");
        if (typeof state === "function" ? receiver !== state || !f : !state.has(receiver))
            throw new TypeError("Cannot write private member to an object whose class did not declare it");
        return (kind === "a" ? f.call(receiver, value) : f ? f.value = value : state.set(receiver, value)), value;
    }

    var ArrowHelperDirective = /** @class */ (function (_super) {
        __extends(ArrowHelperDirective, _super);
        function ArrowHelperDirective() {
            var _this = _super.apply(this, __spreadArray([], __read(arguments))) || this;
            _this.helperType = three.ArrowHelper;
            return _this;
        }
        Object.defineProperty(ArrowHelperDirective.prototype, "args", {
            set: function (v) {
                this.extraArgs = v;
            },
            enumerable: false,
            configurable: true
        });
        return ArrowHelperDirective;
    }(core.ThreeHelper));
    ArrowHelperDirective.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ArrowHelperDirective, deps: null, target: i0__namespace.ɵɵFactoryTarget.Directive });
    ArrowHelperDirective.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: ArrowHelperDirective, selector: "ngt-arrow-helper", inputs: { args: "args" }, providers: [
            {
                provide: core.ThreeObject3d,
                useExisting: ArrowHelperDirective,
            },
            core.OBJECT_3D_CONTROLLER_PROVIDER,
        ], exportAs: ["ngtArrowHelper"], usesInheritance: true, ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ArrowHelperDirective, decorators: [{
                type: i0.Directive,
                args: [{
                        selector: 'ngt-arrow-helper',
                        exportAs: 'ngtArrowHelper',
                        providers: [
                            {
                                provide: core.ThreeObject3d,
                                useExisting: ArrowHelperDirective,
                            },
                            core.OBJECT_3D_CONTROLLER_PROVIDER,
                        ],
                    }]
            }], propDecorators: { args: [{
                    type: i0.Input
                }] } });

    // GENERATED
    var ThreeArrowHelperModule = /** @class */ (function () {
        function ThreeArrowHelperModule() {
        }
        return ThreeArrowHelperModule;
    }());
    ThreeArrowHelperModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeArrowHelperModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    ThreeArrowHelperModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeArrowHelperModule, declarations: [ArrowHelperDirective], exports: [ArrowHelperDirective] });
    ThreeArrowHelperModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeArrowHelperModule });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeArrowHelperModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        declarations: [ArrowHelperDirective],
                        exports: [ArrowHelperDirective],
                    }]
            }] });

    // GENERATED

    var AxesHelperDirective = /** @class */ (function (_super) {
        __extends(AxesHelperDirective, _super);
        function AxesHelperDirective() {
            var _this = _super.apply(this, __spreadArray([], __read(arguments))) || this;
            _this.helperType = three.AxesHelper;
            return _this;
        }
        Object.defineProperty(AxesHelperDirective.prototype, "args", {
            set: function (v) {
                this.extraArgs = v;
            },
            enumerable: false,
            configurable: true
        });
        return AxesHelperDirective;
    }(core.ThreeHelper));
    AxesHelperDirective.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: AxesHelperDirective, deps: null, target: i0__namespace.ɵɵFactoryTarget.Directive });
    AxesHelperDirective.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: AxesHelperDirective, selector: "ngt-axes-helper", inputs: { args: "args" }, providers: [
            {
                provide: core.ThreeObject3d,
                useExisting: AxesHelperDirective,
            },
            core.OBJECT_3D_CONTROLLER_PROVIDER,
        ], exportAs: ["ngtAxesHelper"], usesInheritance: true, ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: AxesHelperDirective, decorators: [{
                type: i0.Directive,
                args: [{
                        selector: 'ngt-axes-helper',
                        exportAs: 'ngtAxesHelper',
                        providers: [
                            {
                                provide: core.ThreeObject3d,
                                useExisting: AxesHelperDirective,
                            },
                            core.OBJECT_3D_CONTROLLER_PROVIDER,
                        ],
                    }]
            }], propDecorators: { args: [{
                    type: i0.Input
                }] } });

    // GENERATED
    var ThreeAxesHelperModule = /** @class */ (function () {
        function ThreeAxesHelperModule() {
        }
        return ThreeAxesHelperModule;
    }());
    ThreeAxesHelperModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeAxesHelperModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    ThreeAxesHelperModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeAxesHelperModule, declarations: [AxesHelperDirective], exports: [AxesHelperDirective] });
    ThreeAxesHelperModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeAxesHelperModule });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeAxesHelperModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        declarations: [AxesHelperDirective],
                        exports: [AxesHelperDirective],
                    }]
            }] });

    // GENERATED

    var BoxHelperDirective = /** @class */ (function (_super) {
        __extends(BoxHelperDirective, _super);
        function BoxHelperDirective() {
            var _this = _super.apply(this, __spreadArray([], __read(arguments))) || this;
            _this.helperType = three.BoxHelper;
            return _this;
        }
        Object.defineProperty(BoxHelperDirective.prototype, "args", {
            set: function (v) {
                this.extraArgs = v;
            },
            enumerable: false,
            configurable: true
        });
        return BoxHelperDirective;
    }(core.ThreeHelper));
    BoxHelperDirective.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: BoxHelperDirective, deps: null, target: i0__namespace.ɵɵFactoryTarget.Directive });
    BoxHelperDirective.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: BoxHelperDirective, selector: "ngt-box-helper", inputs: { args: "args" }, providers: [
            {
                provide: core.ThreeObject3d,
                useExisting: BoxHelperDirective,
            },
            core.OBJECT_3D_CONTROLLER_PROVIDER,
        ], exportAs: ["ngtBoxHelper"], usesInheritance: true, ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: BoxHelperDirective, decorators: [{
                type: i0.Directive,
                args: [{
                        selector: 'ngt-box-helper',
                        exportAs: 'ngtBoxHelper',
                        providers: [
                            {
                                provide: core.ThreeObject3d,
                                useExisting: BoxHelperDirective,
                            },
                            core.OBJECT_3D_CONTROLLER_PROVIDER,
                        ],
                    }]
            }], propDecorators: { args: [{
                    type: i0.Input
                }] } });

    // GENERATED
    var ThreeBoxHelperModule = /** @class */ (function () {
        function ThreeBoxHelperModule() {
        }
        return ThreeBoxHelperModule;
    }());
    ThreeBoxHelperModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeBoxHelperModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    ThreeBoxHelperModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeBoxHelperModule, declarations: [BoxHelperDirective], exports: [BoxHelperDirective] });
    ThreeBoxHelperModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeBoxHelperModule });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeBoxHelperModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        declarations: [BoxHelperDirective],
                        exports: [BoxHelperDirective],
                    }]
            }] });

    // GENERATED

    var Box3HelperDirective = /** @class */ (function (_super) {
        __extends(Box3HelperDirective, _super);
        function Box3HelperDirective() {
            var _this = _super.apply(this, __spreadArray([], __read(arguments))) || this;
            _this.helperType = three.Box3Helper;
            return _this;
        }
        Object.defineProperty(Box3HelperDirective.prototype, "args", {
            set: function (v) {
                this.extraArgs = v;
            },
            enumerable: false,
            configurable: true
        });
        return Box3HelperDirective;
    }(core.ThreeHelper));
    Box3HelperDirective.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: Box3HelperDirective, deps: null, target: i0__namespace.ɵɵFactoryTarget.Directive });
    Box3HelperDirective.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: Box3HelperDirective, selector: "ngt-box3-helper", inputs: { args: "args" }, providers: [
            {
                provide: core.ThreeObject3d,
                useExisting: Box3HelperDirective,
            },
            core.OBJECT_3D_CONTROLLER_PROVIDER,
        ], exportAs: ["ngtBox3Helper"], usesInheritance: true, ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: Box3HelperDirective, decorators: [{
                type: i0.Directive,
                args: [{
                        selector: 'ngt-box3-helper',
                        exportAs: 'ngtBox3Helper',
                        providers: [
                            {
                                provide: core.ThreeObject3d,
                                useExisting: Box3HelperDirective,
                            },
                            core.OBJECT_3D_CONTROLLER_PROVIDER,
                        ],
                    }]
            }], propDecorators: { args: [{
                    type: i0.Input
                }] } });

    // GENERATED
    var ThreeBox3HelperModule = /** @class */ (function () {
        function ThreeBox3HelperModule() {
        }
        return ThreeBox3HelperModule;
    }());
    ThreeBox3HelperModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeBox3HelperModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    ThreeBox3HelperModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeBox3HelperModule, declarations: [Box3HelperDirective], exports: [Box3HelperDirective] });
    ThreeBox3HelperModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeBox3HelperModule });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeBox3HelperModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        declarations: [Box3HelperDirective],
                        exports: [Box3HelperDirective],
                    }]
            }] });

    // GENERATED

    var GridHelperDirective = /** @class */ (function (_super) {
        __extends(GridHelperDirective, _super);
        function GridHelperDirective() {
            var _this = _super.apply(this, __spreadArray([], __read(arguments))) || this;
            _this.helperType = three.GridHelper;
            return _this;
        }
        Object.defineProperty(GridHelperDirective.prototype, "args", {
            set: function (v) {
                this.extraArgs = v;
            },
            enumerable: false,
            configurable: true
        });
        return GridHelperDirective;
    }(core.ThreeHelper));
    GridHelperDirective.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: GridHelperDirective, deps: null, target: i0__namespace.ɵɵFactoryTarget.Directive });
    GridHelperDirective.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: GridHelperDirective, selector: "ngt-grid-helper", inputs: { args: "args" }, providers: [
            {
                provide: core.ThreeObject3d,
                useExisting: GridHelperDirective,
            },
            core.OBJECT_3D_CONTROLLER_PROVIDER,
        ], exportAs: ["ngtGridHelper"], usesInheritance: true, ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: GridHelperDirective, decorators: [{
                type: i0.Directive,
                args: [{
                        selector: 'ngt-grid-helper',
                        exportAs: 'ngtGridHelper',
                        providers: [
                            {
                                provide: core.ThreeObject3d,
                                useExisting: GridHelperDirective,
                            },
                            core.OBJECT_3D_CONTROLLER_PROVIDER,
                        ],
                    }]
            }], propDecorators: { args: [{
                    type: i0.Input
                }] } });

    // GENERATED
    var ThreeGridHelperModule = /** @class */ (function () {
        function ThreeGridHelperModule() {
        }
        return ThreeGridHelperModule;
    }());
    ThreeGridHelperModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeGridHelperModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    ThreeGridHelperModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeGridHelperModule, declarations: [GridHelperDirective], exports: [GridHelperDirective] });
    ThreeGridHelperModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeGridHelperModule });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeGridHelperModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        declarations: [GridHelperDirective],
                        exports: [GridHelperDirective],
                    }]
            }] });

    // GENERATED

    var CameraHelperDirective = /** @class */ (function (_super) {
        __extends(CameraHelperDirective, _super);
        function CameraHelperDirective() {
            var _this = _super.apply(this, __spreadArray([], __read(arguments))) || this;
            _this.helperType = three.CameraHelper;
            return _this;
        }
        Object.defineProperty(CameraHelperDirective.prototype, "args", {
            set: function (v) {
                this.extraArgs = v;
            },
            enumerable: false,
            configurable: true
        });
        return CameraHelperDirective;
    }(core.ThreeHelper));
    CameraHelperDirective.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: CameraHelperDirective, deps: null, target: i0__namespace.ɵɵFactoryTarget.Directive });
    CameraHelperDirective.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: CameraHelperDirective, selector: "ngt-camera-helper", inputs: { args: "args" }, providers: [
            {
                provide: core.ThreeObject3d,
                useExisting: CameraHelperDirective,
            },
            core.OBJECT_3D_CONTROLLER_PROVIDER,
        ], exportAs: ["ngtCameraHelper"], usesInheritance: true, ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: CameraHelperDirective, decorators: [{
                type: i0.Directive,
                args: [{
                        selector: 'ngt-camera-helper',
                        exportAs: 'ngtCameraHelper',
                        providers: [
                            {
                                provide: core.ThreeObject3d,
                                useExisting: CameraHelperDirective,
                            },
                            core.OBJECT_3D_CONTROLLER_PROVIDER,
                        ],
                    }]
            }], propDecorators: { args: [{
                    type: i0.Input
                }] } });

    // GENERATED
    var ThreeCameraHelperModule = /** @class */ (function () {
        function ThreeCameraHelperModule() {
        }
        return ThreeCameraHelperModule;
    }());
    ThreeCameraHelperModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeCameraHelperModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    ThreeCameraHelperModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeCameraHelperModule, declarations: [CameraHelperDirective], exports: [CameraHelperDirective] });
    ThreeCameraHelperModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeCameraHelperModule });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeCameraHelperModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        declarations: [CameraHelperDirective],
                        exports: [CameraHelperDirective],
                    }]
            }] });

    // GENERATED

    var DirectionalLightHelperDirective = /** @class */ (function (_super) {
        __extends(DirectionalLightHelperDirective, _super);
        function DirectionalLightHelperDirective() {
            var _this = _super.apply(this, __spreadArray([], __read(arguments))) || this;
            _this.helperType = three.DirectionalLightHelper;
            return _this;
        }
        Object.defineProperty(DirectionalLightHelperDirective.prototype, "args", {
            set: function (v) {
                this.extraArgs = v;
            },
            enumerable: false,
            configurable: true
        });
        return DirectionalLightHelperDirective;
    }(core.ThreeHelper));
    DirectionalLightHelperDirective.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: DirectionalLightHelperDirective, deps: null, target: i0__namespace.ɵɵFactoryTarget.Directive });
    DirectionalLightHelperDirective.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: DirectionalLightHelperDirective, selector: "ngt-directional-light-helper", inputs: { args: "args" }, providers: [
            {
                provide: core.ThreeObject3d,
                useExisting: DirectionalLightHelperDirective,
            },
            core.OBJECT_3D_CONTROLLER_PROVIDER,
        ], exportAs: ["ngtDirectionalLightHelper"], usesInheritance: true, ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: DirectionalLightHelperDirective, decorators: [{
                type: i0.Directive,
                args: [{
                        selector: 'ngt-directional-light-helper',
                        exportAs: 'ngtDirectionalLightHelper',
                        providers: [
                            {
                                provide: core.ThreeObject3d,
                                useExisting: DirectionalLightHelperDirective,
                            },
                            core.OBJECT_3D_CONTROLLER_PROVIDER,
                        ],
                    }]
            }], propDecorators: { args: [{
                    type: i0.Input
                }] } });

    // GENERATED
    var ThreeDirectionalLightHelperModule = /** @class */ (function () {
        function ThreeDirectionalLightHelperModule() {
        }
        return ThreeDirectionalLightHelperModule;
    }());
    ThreeDirectionalLightHelperModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeDirectionalLightHelperModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    ThreeDirectionalLightHelperModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeDirectionalLightHelperModule, declarations: [DirectionalLightHelperDirective], exports: [DirectionalLightHelperDirective] });
    ThreeDirectionalLightHelperModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeDirectionalLightHelperModule });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeDirectionalLightHelperModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        declarations: [DirectionalLightHelperDirective],
                        exports: [DirectionalLightHelperDirective],
                    }]
            }] });

    // GENERATED

    var HemisphereLightHelperDirective = /** @class */ (function (_super) {
        __extends(HemisphereLightHelperDirective, _super);
        function HemisphereLightHelperDirective() {
            var _this = _super.apply(this, __spreadArray([], __read(arguments))) || this;
            _this.helperType = three.HemisphereLightHelper;
            return _this;
        }
        Object.defineProperty(HemisphereLightHelperDirective.prototype, "args", {
            set: function (v) {
                this.extraArgs = v;
            },
            enumerable: false,
            configurable: true
        });
        return HemisphereLightHelperDirective;
    }(core.ThreeHelper));
    HemisphereLightHelperDirective.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: HemisphereLightHelperDirective, deps: null, target: i0__namespace.ɵɵFactoryTarget.Directive });
    HemisphereLightHelperDirective.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: HemisphereLightHelperDirective, selector: "ngt-hemisphere-light-helper", inputs: { args: "args" }, providers: [
            {
                provide: core.ThreeObject3d,
                useExisting: HemisphereLightHelperDirective,
            },
            core.OBJECT_3D_CONTROLLER_PROVIDER,
        ], exportAs: ["ngtHemisphereLightHelper"], usesInheritance: true, ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: HemisphereLightHelperDirective, decorators: [{
                type: i0.Directive,
                args: [{
                        selector: 'ngt-hemisphere-light-helper',
                        exportAs: 'ngtHemisphereLightHelper',
                        providers: [
                            {
                                provide: core.ThreeObject3d,
                                useExisting: HemisphereLightHelperDirective,
                            },
                            core.OBJECT_3D_CONTROLLER_PROVIDER,
                        ],
                    }]
            }], propDecorators: { args: [{
                    type: i0.Input
                }] } });

    // GENERATED
    var ThreeHemisphereLightHelperModule = /** @class */ (function () {
        function ThreeHemisphereLightHelperModule() {
        }
        return ThreeHemisphereLightHelperModule;
    }());
    ThreeHemisphereLightHelperModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeHemisphereLightHelperModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    ThreeHemisphereLightHelperModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeHemisphereLightHelperModule, declarations: [HemisphereLightHelperDirective], exports: [HemisphereLightHelperDirective] });
    ThreeHemisphereLightHelperModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeHemisphereLightHelperModule });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeHemisphereLightHelperModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        declarations: [HemisphereLightHelperDirective],
                        exports: [HemisphereLightHelperDirective],
                    }]
            }] });

    // GENERATED

    var PlaneHelperDirective = /** @class */ (function (_super) {
        __extends(PlaneHelperDirective, _super);
        function PlaneHelperDirective() {
            var _this = _super.apply(this, __spreadArray([], __read(arguments))) || this;
            _this.helperType = three.PlaneHelper;
            return _this;
        }
        Object.defineProperty(PlaneHelperDirective.prototype, "args", {
            set: function (v) {
                this.extraArgs = v;
            },
            enumerable: false,
            configurable: true
        });
        return PlaneHelperDirective;
    }(core.ThreeHelper));
    PlaneHelperDirective.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: PlaneHelperDirective, deps: null, target: i0__namespace.ɵɵFactoryTarget.Directive });
    PlaneHelperDirective.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: PlaneHelperDirective, selector: "ngt-plane-helper", inputs: { args: "args" }, providers: [
            {
                provide: core.ThreeObject3d,
                useExisting: PlaneHelperDirective,
            },
            core.OBJECT_3D_CONTROLLER_PROVIDER,
        ], exportAs: ["ngtPlaneHelper"], usesInheritance: true, ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: PlaneHelperDirective, decorators: [{
                type: i0.Directive,
                args: [{
                        selector: 'ngt-plane-helper',
                        exportAs: 'ngtPlaneHelper',
                        providers: [
                            {
                                provide: core.ThreeObject3d,
                                useExisting: PlaneHelperDirective,
                            },
                            core.OBJECT_3D_CONTROLLER_PROVIDER,
                        ],
                    }]
            }], propDecorators: { args: [{
                    type: i0.Input
                }] } });

    // GENERATED
    var ThreePlaneHelperModule = /** @class */ (function () {
        function ThreePlaneHelperModule() {
        }
        return ThreePlaneHelperModule;
    }());
    ThreePlaneHelperModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreePlaneHelperModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    ThreePlaneHelperModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreePlaneHelperModule, declarations: [PlaneHelperDirective], exports: [PlaneHelperDirective] });
    ThreePlaneHelperModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreePlaneHelperModule });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreePlaneHelperModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        declarations: [PlaneHelperDirective],
                        exports: [PlaneHelperDirective],
                    }]
            }] });

    // GENERATED

    var PointLightHelperDirective = /** @class */ (function (_super) {
        __extends(PointLightHelperDirective, _super);
        function PointLightHelperDirective() {
            var _this = _super.apply(this, __spreadArray([], __read(arguments))) || this;
            _this.helperType = three.PointLightHelper;
            return _this;
        }
        Object.defineProperty(PointLightHelperDirective.prototype, "args", {
            set: function (v) {
                this.extraArgs = v;
            },
            enumerable: false,
            configurable: true
        });
        return PointLightHelperDirective;
    }(core.ThreeHelper));
    PointLightHelperDirective.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: PointLightHelperDirective, deps: null, target: i0__namespace.ɵɵFactoryTarget.Directive });
    PointLightHelperDirective.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: PointLightHelperDirective, selector: "ngt-point-light-helper", inputs: { args: "args" }, providers: [
            {
                provide: core.ThreeObject3d,
                useExisting: PointLightHelperDirective,
            },
            core.OBJECT_3D_CONTROLLER_PROVIDER,
        ], exportAs: ["ngtPointLightHelper"], usesInheritance: true, ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: PointLightHelperDirective, decorators: [{
                type: i0.Directive,
                args: [{
                        selector: 'ngt-point-light-helper',
                        exportAs: 'ngtPointLightHelper',
                        providers: [
                            {
                                provide: core.ThreeObject3d,
                                useExisting: PointLightHelperDirective,
                            },
                            core.OBJECT_3D_CONTROLLER_PROVIDER,
                        ],
                    }]
            }], propDecorators: { args: [{
                    type: i0.Input
                }] } });

    // GENERATED
    var ThreePointLightHelperModule = /** @class */ (function () {
        function ThreePointLightHelperModule() {
        }
        return ThreePointLightHelperModule;
    }());
    ThreePointLightHelperModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreePointLightHelperModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    ThreePointLightHelperModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreePointLightHelperModule, declarations: [PointLightHelperDirective], exports: [PointLightHelperDirective] });
    ThreePointLightHelperModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreePointLightHelperModule });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreePointLightHelperModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        declarations: [PointLightHelperDirective],
                        exports: [PointLightHelperDirective],
                    }]
            }] });

    // GENERATED

    var PolarGridHelperDirective = /** @class */ (function (_super) {
        __extends(PolarGridHelperDirective, _super);
        function PolarGridHelperDirective() {
            var _this = _super.apply(this, __spreadArray([], __read(arguments))) || this;
            _this.helperType = three.PolarGridHelper;
            return _this;
        }
        Object.defineProperty(PolarGridHelperDirective.prototype, "args", {
            set: function (v) {
                this.extraArgs = v;
            },
            enumerable: false,
            configurable: true
        });
        return PolarGridHelperDirective;
    }(core.ThreeHelper));
    PolarGridHelperDirective.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: PolarGridHelperDirective, deps: null, target: i0__namespace.ɵɵFactoryTarget.Directive });
    PolarGridHelperDirective.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: PolarGridHelperDirective, selector: "ngt-polar-grid-helper", inputs: { args: "args" }, providers: [
            {
                provide: core.ThreeObject3d,
                useExisting: PolarGridHelperDirective,
            },
            core.OBJECT_3D_CONTROLLER_PROVIDER,
        ], exportAs: ["ngtPolarGridHelper"], usesInheritance: true, ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: PolarGridHelperDirective, decorators: [{
                type: i0.Directive,
                args: [{
                        selector: 'ngt-polar-grid-helper',
                        exportAs: 'ngtPolarGridHelper',
                        providers: [
                            {
                                provide: core.ThreeObject3d,
                                useExisting: PolarGridHelperDirective,
                            },
                            core.OBJECT_3D_CONTROLLER_PROVIDER,
                        ],
                    }]
            }], propDecorators: { args: [{
                    type: i0.Input
                }] } });

    // GENERATED
    var ThreePolarGridHelperModule = /** @class */ (function () {
        function ThreePolarGridHelperModule() {
        }
        return ThreePolarGridHelperModule;
    }());
    ThreePolarGridHelperModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreePolarGridHelperModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    ThreePolarGridHelperModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreePolarGridHelperModule, declarations: [PolarGridHelperDirective], exports: [PolarGridHelperDirective] });
    ThreePolarGridHelperModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreePolarGridHelperModule });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreePolarGridHelperModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        declarations: [PolarGridHelperDirective],
                        exports: [PolarGridHelperDirective],
                    }]
            }] });

    // GENERATED

    var SkeletonHelperDirective = /** @class */ (function (_super) {
        __extends(SkeletonHelperDirective, _super);
        function SkeletonHelperDirective() {
            var _this = _super.apply(this, __spreadArray([], __read(arguments))) || this;
            _this.helperType = three.SkeletonHelper;
            return _this;
        }
        Object.defineProperty(SkeletonHelperDirective.prototype, "args", {
            set: function (v) {
                this.extraArgs = v;
            },
            enumerable: false,
            configurable: true
        });
        return SkeletonHelperDirective;
    }(core.ThreeHelper));
    SkeletonHelperDirective.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: SkeletonHelperDirective, deps: null, target: i0__namespace.ɵɵFactoryTarget.Directive });
    SkeletonHelperDirective.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: SkeletonHelperDirective, selector: "ngt-skeleton-helper", inputs: { args: "args" }, providers: [
            {
                provide: core.ThreeObject3d,
                useExisting: SkeletonHelperDirective,
            },
            core.OBJECT_3D_CONTROLLER_PROVIDER,
        ], exportAs: ["ngtSkeletonHelper"], usesInheritance: true, ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: SkeletonHelperDirective, decorators: [{
                type: i0.Directive,
                args: [{
                        selector: 'ngt-skeleton-helper',
                        exportAs: 'ngtSkeletonHelper',
                        providers: [
                            {
                                provide: core.ThreeObject3d,
                                useExisting: SkeletonHelperDirective,
                            },
                            core.OBJECT_3D_CONTROLLER_PROVIDER,
                        ],
                    }]
            }], propDecorators: { args: [{
                    type: i0.Input
                }] } });

    // GENERATED
    var ThreeSkeletonHelperModule = /** @class */ (function () {
        function ThreeSkeletonHelperModule() {
        }
        return ThreeSkeletonHelperModule;
    }());
    ThreeSkeletonHelperModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeSkeletonHelperModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    ThreeSkeletonHelperModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeSkeletonHelperModule, declarations: [SkeletonHelperDirective], exports: [SkeletonHelperDirective] });
    ThreeSkeletonHelperModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeSkeletonHelperModule });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeSkeletonHelperModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        declarations: [SkeletonHelperDirective],
                        exports: [SkeletonHelperDirective],
                    }]
            }] });

    // GENERATED

    var SpotLightHelperDirective = /** @class */ (function (_super) {
        __extends(SpotLightHelperDirective, _super);
        function SpotLightHelperDirective() {
            var _this = _super.apply(this, __spreadArray([], __read(arguments))) || this;
            _this.helperType = three.SpotLightHelper;
            return _this;
        }
        Object.defineProperty(SpotLightHelperDirective.prototype, "args", {
            set: function (v) {
                this.extraArgs = v;
            },
            enumerable: false,
            configurable: true
        });
        return SpotLightHelperDirective;
    }(core.ThreeHelper));
    SpotLightHelperDirective.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: SpotLightHelperDirective, deps: null, target: i0__namespace.ɵɵFactoryTarget.Directive });
    SpotLightHelperDirective.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: SpotLightHelperDirective, selector: "ngt-spot-light-helper", inputs: { args: "args" }, providers: [
            {
                provide: core.ThreeObject3d,
                useExisting: SpotLightHelperDirective,
            },
            core.OBJECT_3D_CONTROLLER_PROVIDER,
        ], exportAs: ["ngtSpotLightHelper"], usesInheritance: true, ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: SpotLightHelperDirective, decorators: [{
                type: i0.Directive,
                args: [{
                        selector: 'ngt-spot-light-helper',
                        exportAs: 'ngtSpotLightHelper',
                        providers: [
                            {
                                provide: core.ThreeObject3d,
                                useExisting: SpotLightHelperDirective,
                            },
                            core.OBJECT_3D_CONTROLLER_PROVIDER,
                        ],
                    }]
            }], propDecorators: { args: [{
                    type: i0.Input
                }] } });

    // GENERATED
    var ThreeSpotLightHelperModule = /** @class */ (function () {
        function ThreeSpotLightHelperModule() {
        }
        return ThreeSpotLightHelperModule;
    }());
    ThreeSpotLightHelperModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeSpotLightHelperModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    ThreeSpotLightHelperModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeSpotLightHelperModule, declarations: [SpotLightHelperDirective], exports: [SpotLightHelperDirective] });
    ThreeSpotLightHelperModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeSpotLightHelperModule });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeSpotLightHelperModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        declarations: [SpotLightHelperDirective],
                        exports: [SpotLightHelperDirective],
                    }]
            }] });

    // GENERATED

    // GENERATED

    /**
     * Generated bundle index. Do not edit.
     */

    exports.ArrowHelperDirective = ArrowHelperDirective;
    exports.AxesHelperDirective = AxesHelperDirective;
    exports.Box3HelperDirective = Box3HelperDirective;
    exports.BoxHelperDirective = BoxHelperDirective;
    exports.CameraHelperDirective = CameraHelperDirective;
    exports.DirectionalLightHelperDirective = DirectionalLightHelperDirective;
    exports.GridHelperDirective = GridHelperDirective;
    exports.HemisphereLightHelperDirective = HemisphereLightHelperDirective;
    exports.PlaneHelperDirective = PlaneHelperDirective;
    exports.PointLightHelperDirective = PointLightHelperDirective;
    exports.PolarGridHelperDirective = PolarGridHelperDirective;
    exports.SkeletonHelperDirective = SkeletonHelperDirective;
    exports.SpotLightHelperDirective = SpotLightHelperDirective;
    exports.ThreeArrowHelperModule = ThreeArrowHelperModule;
    exports.ThreeAxesHelperModule = ThreeAxesHelperModule;
    exports.ThreeBox3HelperModule = ThreeBox3HelperModule;
    exports.ThreeBoxHelperModule = ThreeBoxHelperModule;
    exports.ThreeCameraHelperModule = ThreeCameraHelperModule;
    exports.ThreeDirectionalLightHelperModule = ThreeDirectionalLightHelperModule;
    exports.ThreeGridHelperModule = ThreeGridHelperModule;
    exports.ThreeHemisphereLightHelperModule = ThreeHemisphereLightHelperModule;
    exports.ThreePlaneHelperModule = ThreePlaneHelperModule;
    exports.ThreePointLightHelperModule = ThreePointLightHelperModule;
    exports.ThreePolarGridHelperModule = ThreePolarGridHelperModule;
    exports.ThreeSkeletonHelperModule = ThreeSkeletonHelperModule;
    exports.ThreeSpotLightHelperModule = ThreeSpotLightHelperModule;

    Object.defineProperty(exports, '__esModule', { value: true });

}));
