// @angular-three/core/bundles/angular-three-core-attributes.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('three'), require('@angular-three/core')) :
    typeof define === 'function' && define.amd ? define('@angular-three/core/attributes', ['exports', '@angular/core', 'three', '@angular-three/core'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory((global["angular-three"] = global["angular-three"] || {}, global["angular-three"].core = global["angular-three"].core || {}, global["angular-three"].core.attributes = {}), global.ng.core, global.three, global["angular-three"].core));
})(this, (function (exports, i0, three, i1) { 'use strict';

    function _interopNamespace(e) {
        if (e && e.__esModule) return e;
        var n = Object.create(null);
        if (e) {
            Object.keys(e).forEach(function (k) {
                if (k !== 'default') {
                    var d = Object.getOwnPropertyDescriptor(e, k);
                    Object.defineProperty(n, k, d.get ? d : {
                        enumerable: true,
                        get: function () { return e[k]; }
                    });
                }
            });
        }
        n["default"] = e;
        return Object.freeze(n);
    }

    var i0__namespace = /*#__PURE__*/_interopNamespace(i0);
    var i1__namespace = /*#__PURE__*/_interopNamespace(i1);

    /*! *****************************************************************************
    Copyright (c) Microsoft Corporation.

    Permission to use, copy, modify, and/or distribute this software for any
    purpose with or without fee is hereby granted.

    THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES WITH
    REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY
    AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT,
    INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM
    LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR
    OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
    PERFORMANCE OF THIS SOFTWARE.
    ***************************************************************************** */
    /* global Reflect, Promise */
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b)
                if (Object.prototype.hasOwnProperty.call(b, p))
                    d[p] = b[p]; };
        return extendStatics(d, b);
    };
    function __extends(d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    }
    var __assign = function () {
        __assign = Object.assign || function __assign(t) {
            for (var s, i = 1, n = arguments.length; i < n; i++) {
                s = arguments[i];
                for (var p in s)
                    if (Object.prototype.hasOwnProperty.call(s, p))
                        t[p] = s[p];
            }
            return t;
        };
        return __assign.apply(this, arguments);
    };
    function __rest(s, e) {
        var t = {};
        for (var p in s)
            if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
                t[p] = s[p];
        if (s != null && typeof Object.getOwnPropertySymbols === "function")
            for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
                if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                    t[p[i]] = s[p[i]];
            }
        return t;
    }
    function __decorate(decorators, target, key, desc) {
        var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
            r = Reflect.decorate(decorators, target, key, desc);
        else
            for (var i = decorators.length - 1; i >= 0; i--)
                if (d = decorators[i])
                    r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    }
    function __param(paramIndex, decorator) {
        return function (target, key) { decorator(target, key, paramIndex); };
    }
    function __metadata(metadataKey, metadataValue) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function")
            return Reflect.metadata(metadataKey, metadataValue);
    }
    function __awaiter(thisArg, _arguments, P, generator) {
        function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
        return new (P || (P = Promise))(function (resolve, reject) {
            function fulfilled(value) { try {
                step(generator.next(value));
            }
            catch (e) {
                reject(e);
            } }
            function rejected(value) { try {
                step(generator["throw"](value));
            }
            catch (e) {
                reject(e);
            } }
            function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
            step((generator = generator.apply(thisArg, _arguments || [])).next());
        });
    }
    function __generator(thisArg, body) {
        var _ = { label: 0, sent: function () { if (t[0] & 1)
                throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
        return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function () { return this; }), g;
        function verb(n) { return function (v) { return step([n, v]); }; }
        function step(op) {
            if (f)
                throw new TypeError("Generator is already executing.");
            while (_)
                try {
                    if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done)
                        return t;
                    if (y = 0, t)
                        op = [op[0] & 2, t.value];
                    switch (op[0]) {
                        case 0:
                        case 1:
                            t = op;
                            break;
                        case 4:
                            _.label++;
                            return { value: op[1], done: false };
                        case 5:
                            _.label++;
                            y = op[1];
                            op = [0];
                            continue;
                        case 7:
                            op = _.ops.pop();
                            _.trys.pop();
                            continue;
                        default:
                            if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) {
                                _ = 0;
                                continue;
                            }
                            if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) {
                                _.label = op[1];
                                break;
                            }
                            if (op[0] === 6 && _.label < t[1]) {
                                _.label = t[1];
                                t = op;
                                break;
                            }
                            if (t && _.label < t[2]) {
                                _.label = t[2];
                                _.ops.push(op);
                                break;
                            }
                            if (t[2])
                                _.ops.pop();
                            _.trys.pop();
                            continue;
                    }
                    op = body.call(thisArg, _);
                }
                catch (e) {
                    op = [6, e];
                    y = 0;
                }
                finally {
                    f = t = 0;
                }
            if (op[0] & 5)
                throw op[1];
            return { value: op[0] ? op[1] : void 0, done: true };
        }
    }
    var __createBinding = Object.create ? (function (o, m, k, k2) {
        if (k2 === undefined)
            k2 = k;
        Object.defineProperty(o, k2, { enumerable: true, get: function () { return m[k]; } });
    }) : (function (o, m, k, k2) {
        if (k2 === undefined)
            k2 = k;
        o[k2] = m[k];
    });
    function __exportStar(m, o) {
        for (var p in m)
            if (p !== "default" && !Object.prototype.hasOwnProperty.call(o, p))
                __createBinding(o, m, p);
    }
    function __values(o) {
        var s = typeof Symbol === "function" && Symbol.iterator, m = s && o[s], i = 0;
        if (m)
            return m.call(o);
        if (o && typeof o.length === "number")
            return {
                next: function () {
                    if (o && i >= o.length)
                        o = void 0;
                    return { value: o && o[i++], done: !o };
                }
            };
        throw new TypeError(s ? "Object is not iterable." : "Symbol.iterator is not defined.");
    }
    function __read(o, n) {
        var m = typeof Symbol === "function" && o[Symbol.iterator];
        if (!m)
            return o;
        var i = m.call(o), r, ar = [], e;
        try {
            while ((n === void 0 || n-- > 0) && !(r = i.next()).done)
                ar.push(r.value);
        }
        catch (error) {
            e = { error: error };
        }
        finally {
            try {
                if (r && !r.done && (m = i["return"]))
                    m.call(i);
            }
            finally {
                if (e)
                    throw e.error;
            }
        }
        return ar;
    }
    /** @deprecated */
    function __spread() {
        for (var ar = [], i = 0; i < arguments.length; i++)
            ar = ar.concat(__read(arguments[i]));
        return ar;
    }
    /** @deprecated */
    function __spreadArrays() {
        for (var s = 0, i = 0, il = arguments.length; i < il; i++)
            s += arguments[i].length;
        for (var r = Array(s), k = 0, i = 0; i < il; i++)
            for (var a = arguments[i], j = 0, jl = a.length; j < jl; j++, k++)
                r[k] = a[j];
        return r;
    }
    function __spreadArray(to, from, pack) {
        if (pack || arguments.length === 2)
            for (var i = 0, l = from.length, ar; i < l; i++) {
                if (ar || !(i in from)) {
                    if (!ar)
                        ar = Array.prototype.slice.call(from, 0, i);
                    ar[i] = from[i];
                }
            }
        return to.concat(ar || Array.prototype.slice.call(from));
    }
    function __await(v) {
        return this instanceof __await ? (this.v = v, this) : new __await(v);
    }
    function __asyncGenerator(thisArg, _arguments, generator) {
        if (!Symbol.asyncIterator)
            throw new TypeError("Symbol.asyncIterator is not defined.");
        var g = generator.apply(thisArg, _arguments || []), i, q = [];
        return i = {}, verb("next"), verb("throw"), verb("return"), i[Symbol.asyncIterator] = function () { return this; }, i;
        function verb(n) { if (g[n])
            i[n] = function (v) { return new Promise(function (a, b) { q.push([n, v, a, b]) > 1 || resume(n, v); }); }; }
        function resume(n, v) { try {
            step(g[n](v));
        }
        catch (e) {
            settle(q[0][3], e);
        } }
        function step(r) { r.value instanceof __await ? Promise.resolve(r.value.v).then(fulfill, reject) : settle(q[0][2], r); }
        function fulfill(value) { resume("next", value); }
        function reject(value) { resume("throw", value); }
        function settle(f, v) { if (f(v), q.shift(), q.length)
            resume(q[0][0], q[0][1]); }
    }
    function __asyncDelegator(o) {
        var i, p;
        return i = {}, verb("next"), verb("throw", function (e) { throw e; }), verb("return"), i[Symbol.iterator] = function () { return this; }, i;
        function verb(n, f) { i[n] = o[n] ? function (v) { return (p = !p) ? { value: __await(o[n](v)), done: n === "return" } : f ? f(v) : v; } : f; }
    }
    function __asyncValues(o) {
        if (!Symbol.asyncIterator)
            throw new TypeError("Symbol.asyncIterator is not defined.");
        var m = o[Symbol.asyncIterator], i;
        return m ? m.call(o) : (o = typeof __values === "function" ? __values(o) : o[Symbol.iterator](), i = {}, verb("next"), verb("throw"), verb("return"), i[Symbol.asyncIterator] = function () { return this; }, i);
        function verb(n) { i[n] = o[n] && function (v) { return new Promise(function (resolve, reject) { v = o[n](v), settle(resolve, reject, v.done, v.value); }); }; }
        function settle(resolve, reject, d, v) { Promise.resolve(v).then(function (v) { resolve({ value: v, done: d }); }, reject); }
    }
    function __makeTemplateObject(cooked, raw) {
        if (Object.defineProperty) {
            Object.defineProperty(cooked, "raw", { value: raw });
        }
        else {
            cooked.raw = raw;
        }
        return cooked;
    }
    ;
    var __setModuleDefault = Object.create ? (function (o, v) {
        Object.defineProperty(o, "default", { enumerable: true, value: v });
    }) : function (o, v) {
        o["default"] = v;
    };
    function __importStar(mod) {
        if (mod && mod.__esModule)
            return mod;
        var result = {};
        if (mod != null)
            for (var k in mod)
                if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k))
                    __createBinding(result, mod, k);
        __setModuleDefault(result, mod);
        return result;
    }
    function __importDefault(mod) {
        return (mod && mod.__esModule) ? mod : { default: mod };
    }
    function __classPrivateFieldGet(receiver, state, kind, f) {
        if (kind === "a" && !f)
            throw new TypeError("Private accessor was defined without a getter");
        if (typeof state === "function" ? receiver !== state || !f : !state.has(receiver))
            throw new TypeError("Cannot read private member from an object whose class did not declare it");
        return kind === "m" ? f : kind === "a" ? f.call(receiver) : f ? f.value : state.get(receiver);
    }
    function __classPrivateFieldSet(receiver, state, value, kind, f) {
        if (kind === "m")
            throw new TypeError("Private method is not writable");
        if (kind === "a" && !f)
            throw new TypeError("Private accessor was defined without a setter");
        if (typeof state === "function" ? receiver !== state || !f : !state.has(receiver))
            throw new TypeError("Cannot write private member to an object whose class did not declare it");
        return (kind === "a" ? f.call(receiver, value) : f ? f.value = value : state.set(receiver, value)), value;
    }

    var InterleavedBufferAttributeDirective = /** @class */ (function () {
        function InterleavedBufferAttributeDirective(ngZone, geometryDirective) {
            this.ngZone = ngZone;
            this.geometryDirective = geometryDirective;
        }
        InterleavedBufferAttributeDirective.prototype.ngOnChanges = function () {
            var _this = this;
            this.ngZone.runOutsideAngular(function () {
                if (_this.attribute) {
                    _this.attribute.needsUpdate = true;
                }
            });
        };
        InterleavedBufferAttributeDirective.prototype.ngOnInit = function () {
            var _this = this;
            this.ngZone.runOutsideAngular(function () {
                if (_this.geometryDirective && _this.attach) {
                    _this._attribute = new (three.InterleavedBufferAttribute.bind.apply(three.InterleavedBufferAttribute, __spreadArray([void 0], __read(_this.args))))();
                    if (_this.attribute) {
                        _this.geometryDirective.bufferGeometry.setAttribute(_this.attach, _this.attribute);
                    }
                }
            });
        };
        InterleavedBufferAttributeDirective.prototype.ngOnDestroy = function () {
            var _this = this;
            this.ngZone.runOutsideAngular(function () {
                if (_this.geometryDirective && _this.attach) {
                    _this.geometryDirective.bufferGeometry.deleteAttribute(_this.attach);
                }
            });
        };
        Object.defineProperty(InterleavedBufferAttributeDirective.prototype, "attribute", {
            get: function () {
                return this._attribute;
            },
            enumerable: false,
            configurable: true
        });
        return InterleavedBufferAttributeDirective;
    }());
    InterleavedBufferAttributeDirective.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: InterleavedBufferAttributeDirective, deps: [{ token: i0__namespace.NgZone }, { token: i1__namespace.ThreeBufferGeometry, optional: true }], target: i0__namespace.ɵɵFactoryTarget.Directive });
    InterleavedBufferAttributeDirective.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: InterleavedBufferAttributeDirective, selector: "ngt-interleaved-buffer-attribute", inputs: { args: "args", attach: "attach" }, exportAs: ["ngtInterleavedBufferAttribute"], usesOnChanges: true, ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: InterleavedBufferAttributeDirective, decorators: [{
                type: i0.Directive,
                args: [{
                        selector: 'ngt-interleaved-buffer-attribute',
                        exportAs: 'ngtInterleavedBufferAttribute',
                    }]
            }], ctorParameters: function () {
            return [{ type: i0__namespace.NgZone }, { type: i1__namespace.ThreeBufferGeometry, decorators: [{
                            type: i0.Optional
                        }] }];
        }, propDecorators: { args: [{
                    type: i0.Input
                }], attach: [{
                    type: i0.Input
                }] } });

    var InterleavedBufferAttributeModule = /** @class */ (function () {
        function InterleavedBufferAttributeModule() {
        }
        return InterleavedBufferAttributeModule;
    }());
    InterleavedBufferAttributeModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: InterleavedBufferAttributeModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    InterleavedBufferAttributeModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: InterleavedBufferAttributeModule, declarations: [InterleavedBufferAttributeDirective], exports: [InterleavedBufferAttributeDirective] });
    InterleavedBufferAttributeModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: InterleavedBufferAttributeModule });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: InterleavedBufferAttributeModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        declarations: [InterleavedBufferAttributeDirective],
                        exports: [InterleavedBufferAttributeDirective],
                    }]
            }] });

    var BufferAttributeDirective = /** @class */ (function (_super) {
        __extends(BufferAttributeDirective, _super);
        function BufferAttributeDirective() {
            var _this = _super.apply(this, __spreadArray([], __read(arguments))) || this;
            _this.attributeType = three.BufferAttribute;
            return _this;
        }
        Object.defineProperty(BufferAttributeDirective.prototype, "args", {
            set: function (v) {
                this.extraArgs = v;
            },
            enumerable: false,
            configurable: true
        });
        return BufferAttributeDirective;
    }(i1.ThreeAttribute));
    BufferAttributeDirective.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: BufferAttributeDirective, deps: null, target: i0__namespace.ɵɵFactoryTarget.Directive });
    BufferAttributeDirective.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: BufferAttributeDirective, selector: "ngt-buffer-attribute", inputs: { args: "args" }, providers: [
            {
                provide: i1.ThreeAttribute,
                useExisting: BufferAttributeDirective,
            },
        ], exportAs: ["ngtBufferAttribute"], usesInheritance: true, ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: BufferAttributeDirective, decorators: [{
                type: i0.Directive,
                args: [{
                        selector: 'ngt-buffer-attribute',
                        exportAs: 'ngtBufferAttribute',
                        providers: [
                            {
                                provide: i1.ThreeAttribute,
                                useExisting: BufferAttributeDirective,
                            },
                        ],
                    }]
            }], propDecorators: { args: [{
                    type: i0.Input
                }] } });

    // GENERATED
    var ThreeBufferAttributeModule = /** @class */ (function () {
        function ThreeBufferAttributeModule() {
        }
        return ThreeBufferAttributeModule;
    }());
    ThreeBufferAttributeModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeBufferAttributeModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    ThreeBufferAttributeModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeBufferAttributeModule, declarations: [BufferAttributeDirective], exports: [BufferAttributeDirective] });
    ThreeBufferAttributeModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeBufferAttributeModule });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeBufferAttributeModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        declarations: [BufferAttributeDirective],
                        exports: [BufferAttributeDirective],
                    }]
            }] });

    // GENERATED

    var InstancedBufferAttributeDirective = /** @class */ (function (_super) {
        __extends(InstancedBufferAttributeDirective, _super);
        function InstancedBufferAttributeDirective() {
            var _this = _super.apply(this, __spreadArray([], __read(arguments))) || this;
            _this.attributeType = three.InstancedBufferAttribute;
            return _this;
        }
        Object.defineProperty(InstancedBufferAttributeDirective.prototype, "args", {
            set: function (v) {
                this.extraArgs = v;
            },
            enumerable: false,
            configurable: true
        });
        return InstancedBufferAttributeDirective;
    }(i1.ThreeAttribute));
    InstancedBufferAttributeDirective.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: InstancedBufferAttributeDirective, deps: null, target: i0__namespace.ɵɵFactoryTarget.Directive });
    InstancedBufferAttributeDirective.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: InstancedBufferAttributeDirective, selector: "ngt-instanced-buffer-attribute", inputs: { args: "args" }, providers: [
            {
                provide: i1.ThreeAttribute,
                useExisting: InstancedBufferAttributeDirective,
            },
        ], exportAs: ["ngtInstancedBufferAttribute"], usesInheritance: true, ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: InstancedBufferAttributeDirective, decorators: [{
                type: i0.Directive,
                args: [{
                        selector: 'ngt-instanced-buffer-attribute',
                        exportAs: 'ngtInstancedBufferAttribute',
                        providers: [
                            {
                                provide: i1.ThreeAttribute,
                                useExisting: InstancedBufferAttributeDirective,
                            },
                        ],
                    }]
            }], propDecorators: { args: [{
                    type: i0.Input
                }] } });

    // GENERATED
    var ThreeInstancedBufferAttributeModule = /** @class */ (function () {
        function ThreeInstancedBufferAttributeModule() {
        }
        return ThreeInstancedBufferAttributeModule;
    }());
    ThreeInstancedBufferAttributeModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeInstancedBufferAttributeModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    ThreeInstancedBufferAttributeModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeInstancedBufferAttributeModule, declarations: [InstancedBufferAttributeDirective], exports: [InstancedBufferAttributeDirective] });
    ThreeInstancedBufferAttributeModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeInstancedBufferAttributeModule });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeInstancedBufferAttributeModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        declarations: [InstancedBufferAttributeDirective],
                        exports: [InstancedBufferAttributeDirective],
                    }]
            }] });

    // GENERATED

    var Float16BufferAttributeDirective = /** @class */ (function (_super) {
        __extends(Float16BufferAttributeDirective, _super);
        function Float16BufferAttributeDirective() {
            var _this = _super.apply(this, __spreadArray([], __read(arguments))) || this;
            _this.attributeType = three.Float16BufferAttribute;
            return _this;
        }
        Object.defineProperty(Float16BufferAttributeDirective.prototype, "args", {
            set: function (v) {
                this.extraArgs = v;
            },
            enumerable: false,
            configurable: true
        });
        return Float16BufferAttributeDirective;
    }(i1.ThreeAttribute));
    Float16BufferAttributeDirective.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: Float16BufferAttributeDirective, deps: null, target: i0__namespace.ɵɵFactoryTarget.Directive });
    Float16BufferAttributeDirective.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: Float16BufferAttributeDirective, selector: "ngt-float16-buffer-attribute", inputs: { args: "args" }, providers: [
            {
                provide: i1.ThreeAttribute,
                useExisting: Float16BufferAttributeDirective,
            },
        ], exportAs: ["ngtFloat16BufferAttribute"], usesInheritance: true, ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: Float16BufferAttributeDirective, decorators: [{
                type: i0.Directive,
                args: [{
                        selector: 'ngt-float16-buffer-attribute',
                        exportAs: 'ngtFloat16BufferAttribute',
                        providers: [
                            {
                                provide: i1.ThreeAttribute,
                                useExisting: Float16BufferAttributeDirective,
                            },
                        ],
                    }]
            }], propDecorators: { args: [{
                    type: i0.Input
                }] } });

    // GENERATED
    var ThreeFloat16BufferAttributeModule = /** @class */ (function () {
        function ThreeFloat16BufferAttributeModule() {
        }
        return ThreeFloat16BufferAttributeModule;
    }());
    ThreeFloat16BufferAttributeModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeFloat16BufferAttributeModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    ThreeFloat16BufferAttributeModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeFloat16BufferAttributeModule, declarations: [Float16BufferAttributeDirective], exports: [Float16BufferAttributeDirective] });
    ThreeFloat16BufferAttributeModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeFloat16BufferAttributeModule });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeFloat16BufferAttributeModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        declarations: [Float16BufferAttributeDirective],
                        exports: [Float16BufferAttributeDirective],
                    }]
            }] });

    // GENERATED

    var Float32BufferAttributeDirective = /** @class */ (function (_super) {
        __extends(Float32BufferAttributeDirective, _super);
        function Float32BufferAttributeDirective() {
            var _this = _super.apply(this, __spreadArray([], __read(arguments))) || this;
            _this.attributeType = three.Float32BufferAttribute;
            return _this;
        }
        Object.defineProperty(Float32BufferAttributeDirective.prototype, "args", {
            set: function (v) {
                this.extraArgs = v;
            },
            enumerable: false,
            configurable: true
        });
        return Float32BufferAttributeDirective;
    }(i1.ThreeAttribute));
    Float32BufferAttributeDirective.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: Float32BufferAttributeDirective, deps: null, target: i0__namespace.ɵɵFactoryTarget.Directive });
    Float32BufferAttributeDirective.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: Float32BufferAttributeDirective, selector: "ngt-float32-buffer-attribute", inputs: { args: "args" }, providers: [
            {
                provide: i1.ThreeAttribute,
                useExisting: Float32BufferAttributeDirective,
            },
        ], exportAs: ["ngtFloat32BufferAttribute"], usesInheritance: true, ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: Float32BufferAttributeDirective, decorators: [{
                type: i0.Directive,
                args: [{
                        selector: 'ngt-float32-buffer-attribute',
                        exportAs: 'ngtFloat32BufferAttribute',
                        providers: [
                            {
                                provide: i1.ThreeAttribute,
                                useExisting: Float32BufferAttributeDirective,
                            },
                        ],
                    }]
            }], propDecorators: { args: [{
                    type: i0.Input
                }] } });

    // GENERATED
    var ThreeFloat32BufferAttributeModule = /** @class */ (function () {
        function ThreeFloat32BufferAttributeModule() {
        }
        return ThreeFloat32BufferAttributeModule;
    }());
    ThreeFloat32BufferAttributeModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeFloat32BufferAttributeModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    ThreeFloat32BufferAttributeModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeFloat32BufferAttributeModule, declarations: [Float32BufferAttributeDirective], exports: [Float32BufferAttributeDirective] });
    ThreeFloat32BufferAttributeModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeFloat32BufferAttributeModule });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeFloat32BufferAttributeModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        declarations: [Float32BufferAttributeDirective],
                        exports: [Float32BufferAttributeDirective],
                    }]
            }] });

    // GENERATED

    var Float64BufferAttributeDirective = /** @class */ (function (_super) {
        __extends(Float64BufferAttributeDirective, _super);
        function Float64BufferAttributeDirective() {
            var _this = _super.apply(this, __spreadArray([], __read(arguments))) || this;
            _this.attributeType = three.Float64BufferAttribute;
            return _this;
        }
        Object.defineProperty(Float64BufferAttributeDirective.prototype, "args", {
            set: function (v) {
                this.extraArgs = v;
            },
            enumerable: false,
            configurable: true
        });
        return Float64BufferAttributeDirective;
    }(i1.ThreeAttribute));
    Float64BufferAttributeDirective.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: Float64BufferAttributeDirective, deps: null, target: i0__namespace.ɵɵFactoryTarget.Directive });
    Float64BufferAttributeDirective.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: Float64BufferAttributeDirective, selector: "ngt-float64-buffer-attribute", inputs: { args: "args" }, providers: [
            {
                provide: i1.ThreeAttribute,
                useExisting: Float64BufferAttributeDirective,
            },
        ], exportAs: ["ngtFloat64BufferAttribute"], usesInheritance: true, ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: Float64BufferAttributeDirective, decorators: [{
                type: i0.Directive,
                args: [{
                        selector: 'ngt-float64-buffer-attribute',
                        exportAs: 'ngtFloat64BufferAttribute',
                        providers: [
                            {
                                provide: i1.ThreeAttribute,
                                useExisting: Float64BufferAttributeDirective,
                            },
                        ],
                    }]
            }], propDecorators: { args: [{
                    type: i0.Input
                }] } });

    // GENERATED
    var ThreeFloat64BufferAttributeModule = /** @class */ (function () {
        function ThreeFloat64BufferAttributeModule() {
        }
        return ThreeFloat64BufferAttributeModule;
    }());
    ThreeFloat64BufferAttributeModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeFloat64BufferAttributeModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    ThreeFloat64BufferAttributeModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeFloat64BufferAttributeModule, declarations: [Float64BufferAttributeDirective], exports: [Float64BufferAttributeDirective] });
    ThreeFloat64BufferAttributeModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeFloat64BufferAttributeModule });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeFloat64BufferAttributeModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        declarations: [Float64BufferAttributeDirective],
                        exports: [Float64BufferAttributeDirective],
                    }]
            }] });

    // GENERATED

    var Int8BufferAttributeDirective = /** @class */ (function (_super) {
        __extends(Int8BufferAttributeDirective, _super);
        function Int8BufferAttributeDirective() {
            var _this = _super.apply(this, __spreadArray([], __read(arguments))) || this;
            _this.attributeType = three.Int8BufferAttribute;
            return _this;
        }
        Object.defineProperty(Int8BufferAttributeDirective.prototype, "args", {
            set: function (v) {
                this.extraArgs = v;
            },
            enumerable: false,
            configurable: true
        });
        return Int8BufferAttributeDirective;
    }(i1.ThreeAttribute));
    Int8BufferAttributeDirective.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: Int8BufferAttributeDirective, deps: null, target: i0__namespace.ɵɵFactoryTarget.Directive });
    Int8BufferAttributeDirective.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: Int8BufferAttributeDirective, selector: "ngt-int8-buffer-attribute", inputs: { args: "args" }, providers: [
            {
                provide: i1.ThreeAttribute,
                useExisting: Int8BufferAttributeDirective,
            },
        ], exportAs: ["ngtInt8BufferAttribute"], usesInheritance: true, ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: Int8BufferAttributeDirective, decorators: [{
                type: i0.Directive,
                args: [{
                        selector: 'ngt-int8-buffer-attribute',
                        exportAs: 'ngtInt8BufferAttribute',
                        providers: [
                            {
                                provide: i1.ThreeAttribute,
                                useExisting: Int8BufferAttributeDirective,
                            },
                        ],
                    }]
            }], propDecorators: { args: [{
                    type: i0.Input
                }] } });

    // GENERATED
    var ThreeInt8BufferAttributeModule = /** @class */ (function () {
        function ThreeInt8BufferAttributeModule() {
        }
        return ThreeInt8BufferAttributeModule;
    }());
    ThreeInt8BufferAttributeModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeInt8BufferAttributeModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    ThreeInt8BufferAttributeModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeInt8BufferAttributeModule, declarations: [Int8BufferAttributeDirective], exports: [Int8BufferAttributeDirective] });
    ThreeInt8BufferAttributeModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeInt8BufferAttributeModule });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeInt8BufferAttributeModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        declarations: [Int8BufferAttributeDirective],
                        exports: [Int8BufferAttributeDirective],
                    }]
            }] });

    // GENERATED

    var Int16BufferAttributeDirective = /** @class */ (function (_super) {
        __extends(Int16BufferAttributeDirective, _super);
        function Int16BufferAttributeDirective() {
            var _this = _super.apply(this, __spreadArray([], __read(arguments))) || this;
            _this.attributeType = three.Int16BufferAttribute;
            return _this;
        }
        Object.defineProperty(Int16BufferAttributeDirective.prototype, "args", {
            set: function (v) {
                this.extraArgs = v;
            },
            enumerable: false,
            configurable: true
        });
        return Int16BufferAttributeDirective;
    }(i1.ThreeAttribute));
    Int16BufferAttributeDirective.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: Int16BufferAttributeDirective, deps: null, target: i0__namespace.ɵɵFactoryTarget.Directive });
    Int16BufferAttributeDirective.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: Int16BufferAttributeDirective, selector: "ngt-int16-buffer-attribute", inputs: { args: "args" }, providers: [
            {
                provide: i1.ThreeAttribute,
                useExisting: Int16BufferAttributeDirective,
            },
        ], exportAs: ["ngtInt16BufferAttribute"], usesInheritance: true, ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: Int16BufferAttributeDirective, decorators: [{
                type: i0.Directive,
                args: [{
                        selector: 'ngt-int16-buffer-attribute',
                        exportAs: 'ngtInt16BufferAttribute',
                        providers: [
                            {
                                provide: i1.ThreeAttribute,
                                useExisting: Int16BufferAttributeDirective,
                            },
                        ],
                    }]
            }], propDecorators: { args: [{
                    type: i0.Input
                }] } });

    // GENERATED
    var ThreeInt16BufferAttributeModule = /** @class */ (function () {
        function ThreeInt16BufferAttributeModule() {
        }
        return ThreeInt16BufferAttributeModule;
    }());
    ThreeInt16BufferAttributeModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeInt16BufferAttributeModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    ThreeInt16BufferAttributeModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeInt16BufferAttributeModule, declarations: [Int16BufferAttributeDirective], exports: [Int16BufferAttributeDirective] });
    ThreeInt16BufferAttributeModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeInt16BufferAttributeModule });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeInt16BufferAttributeModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        declarations: [Int16BufferAttributeDirective],
                        exports: [Int16BufferAttributeDirective],
                    }]
            }] });

    // GENERATED

    var Int32BufferAttributeDirective = /** @class */ (function (_super) {
        __extends(Int32BufferAttributeDirective, _super);
        function Int32BufferAttributeDirective() {
            var _this = _super.apply(this, __spreadArray([], __read(arguments))) || this;
            _this.attributeType = three.Int32BufferAttribute;
            return _this;
        }
        Object.defineProperty(Int32BufferAttributeDirective.prototype, "args", {
            set: function (v) {
                this.extraArgs = v;
            },
            enumerable: false,
            configurable: true
        });
        return Int32BufferAttributeDirective;
    }(i1.ThreeAttribute));
    Int32BufferAttributeDirective.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: Int32BufferAttributeDirective, deps: null, target: i0__namespace.ɵɵFactoryTarget.Directive });
    Int32BufferAttributeDirective.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: Int32BufferAttributeDirective, selector: "ngt-int32-buffer-attribute", inputs: { args: "args" }, providers: [
            {
                provide: i1.ThreeAttribute,
                useExisting: Int32BufferAttributeDirective,
            },
        ], exportAs: ["ngtInt32BufferAttribute"], usesInheritance: true, ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: Int32BufferAttributeDirective, decorators: [{
                type: i0.Directive,
                args: [{
                        selector: 'ngt-int32-buffer-attribute',
                        exportAs: 'ngtInt32BufferAttribute',
                        providers: [
                            {
                                provide: i1.ThreeAttribute,
                                useExisting: Int32BufferAttributeDirective,
                            },
                        ],
                    }]
            }], propDecorators: { args: [{
                    type: i0.Input
                }] } });

    // GENERATED
    var ThreeInt32BufferAttributeModule = /** @class */ (function () {
        function ThreeInt32BufferAttributeModule() {
        }
        return ThreeInt32BufferAttributeModule;
    }());
    ThreeInt32BufferAttributeModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeInt32BufferAttributeModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    ThreeInt32BufferAttributeModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeInt32BufferAttributeModule, declarations: [Int32BufferAttributeDirective], exports: [Int32BufferAttributeDirective] });
    ThreeInt32BufferAttributeModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeInt32BufferAttributeModule });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeInt32BufferAttributeModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        declarations: [Int32BufferAttributeDirective],
                        exports: [Int32BufferAttributeDirective],
                    }]
            }] });

    // GENERATED

    var Uint8BufferAttributeDirective = /** @class */ (function (_super) {
        __extends(Uint8BufferAttributeDirective, _super);
        function Uint8BufferAttributeDirective() {
            var _this = _super.apply(this, __spreadArray([], __read(arguments))) || this;
            _this.attributeType = three.Uint8BufferAttribute;
            return _this;
        }
        Object.defineProperty(Uint8BufferAttributeDirective.prototype, "args", {
            set: function (v) {
                this.extraArgs = v;
            },
            enumerable: false,
            configurable: true
        });
        return Uint8BufferAttributeDirective;
    }(i1.ThreeAttribute));
    Uint8BufferAttributeDirective.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: Uint8BufferAttributeDirective, deps: null, target: i0__namespace.ɵɵFactoryTarget.Directive });
    Uint8BufferAttributeDirective.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: Uint8BufferAttributeDirective, selector: "ngt-uint8-buffer-attribute", inputs: { args: "args" }, providers: [
            {
                provide: i1.ThreeAttribute,
                useExisting: Uint8BufferAttributeDirective,
            },
        ], exportAs: ["ngtUint8BufferAttribute"], usesInheritance: true, ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: Uint8BufferAttributeDirective, decorators: [{
                type: i0.Directive,
                args: [{
                        selector: 'ngt-uint8-buffer-attribute',
                        exportAs: 'ngtUint8BufferAttribute',
                        providers: [
                            {
                                provide: i1.ThreeAttribute,
                                useExisting: Uint8BufferAttributeDirective,
                            },
                        ],
                    }]
            }], propDecorators: { args: [{
                    type: i0.Input
                }] } });

    // GENERATED
    var ThreeUint8BufferAttributeModule = /** @class */ (function () {
        function ThreeUint8BufferAttributeModule() {
        }
        return ThreeUint8BufferAttributeModule;
    }());
    ThreeUint8BufferAttributeModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeUint8BufferAttributeModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    ThreeUint8BufferAttributeModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeUint8BufferAttributeModule, declarations: [Uint8BufferAttributeDirective], exports: [Uint8BufferAttributeDirective] });
    ThreeUint8BufferAttributeModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeUint8BufferAttributeModule });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeUint8BufferAttributeModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        declarations: [Uint8BufferAttributeDirective],
                        exports: [Uint8BufferAttributeDirective],
                    }]
            }] });

    // GENERATED

    var Uint16BufferAttributeDirective = /** @class */ (function (_super) {
        __extends(Uint16BufferAttributeDirective, _super);
        function Uint16BufferAttributeDirective() {
            var _this = _super.apply(this, __spreadArray([], __read(arguments))) || this;
            _this.attributeType = three.Uint16BufferAttribute;
            return _this;
        }
        Object.defineProperty(Uint16BufferAttributeDirective.prototype, "args", {
            set: function (v) {
                this.extraArgs = v;
            },
            enumerable: false,
            configurable: true
        });
        return Uint16BufferAttributeDirective;
    }(i1.ThreeAttribute));
    Uint16BufferAttributeDirective.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: Uint16BufferAttributeDirective, deps: null, target: i0__namespace.ɵɵFactoryTarget.Directive });
    Uint16BufferAttributeDirective.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: Uint16BufferAttributeDirective, selector: "ngt-uint16-buffer-attribute", inputs: { args: "args" }, providers: [
            {
                provide: i1.ThreeAttribute,
                useExisting: Uint16BufferAttributeDirective,
            },
        ], exportAs: ["ngtUint16BufferAttribute"], usesInheritance: true, ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: Uint16BufferAttributeDirective, decorators: [{
                type: i0.Directive,
                args: [{
                        selector: 'ngt-uint16-buffer-attribute',
                        exportAs: 'ngtUint16BufferAttribute',
                        providers: [
                            {
                                provide: i1.ThreeAttribute,
                                useExisting: Uint16BufferAttributeDirective,
                            },
                        ],
                    }]
            }], propDecorators: { args: [{
                    type: i0.Input
                }] } });

    // GENERATED
    var ThreeUint16BufferAttributeModule = /** @class */ (function () {
        function ThreeUint16BufferAttributeModule() {
        }
        return ThreeUint16BufferAttributeModule;
    }());
    ThreeUint16BufferAttributeModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeUint16BufferAttributeModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    ThreeUint16BufferAttributeModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeUint16BufferAttributeModule, declarations: [Uint16BufferAttributeDirective], exports: [Uint16BufferAttributeDirective] });
    ThreeUint16BufferAttributeModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeUint16BufferAttributeModule });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeUint16BufferAttributeModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        declarations: [Uint16BufferAttributeDirective],
                        exports: [Uint16BufferAttributeDirective],
                    }]
            }] });

    // GENERATED

    var Uint32BufferAttributeDirective = /** @class */ (function (_super) {
        __extends(Uint32BufferAttributeDirective, _super);
        function Uint32BufferAttributeDirective() {
            var _this = _super.apply(this, __spreadArray([], __read(arguments))) || this;
            _this.attributeType = three.Uint32BufferAttribute;
            return _this;
        }
        Object.defineProperty(Uint32BufferAttributeDirective.prototype, "args", {
            set: function (v) {
                this.extraArgs = v;
            },
            enumerable: false,
            configurable: true
        });
        return Uint32BufferAttributeDirective;
    }(i1.ThreeAttribute));
    Uint32BufferAttributeDirective.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: Uint32BufferAttributeDirective, deps: null, target: i0__namespace.ɵɵFactoryTarget.Directive });
    Uint32BufferAttributeDirective.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: Uint32BufferAttributeDirective, selector: "ngt-uint32-buffer-attribute", inputs: { args: "args" }, providers: [
            {
                provide: i1.ThreeAttribute,
                useExisting: Uint32BufferAttributeDirective,
            },
        ], exportAs: ["ngtUint32BufferAttribute"], usesInheritance: true, ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: Uint32BufferAttributeDirective, decorators: [{
                type: i0.Directive,
                args: [{
                        selector: 'ngt-uint32-buffer-attribute',
                        exportAs: 'ngtUint32BufferAttribute',
                        providers: [
                            {
                                provide: i1.ThreeAttribute,
                                useExisting: Uint32BufferAttributeDirective,
                            },
                        ],
                    }]
            }], propDecorators: { args: [{
                    type: i0.Input
                }] } });

    // GENERATED
    var ThreeUint32BufferAttributeModule = /** @class */ (function () {
        function ThreeUint32BufferAttributeModule() {
        }
        return ThreeUint32BufferAttributeModule;
    }());
    ThreeUint32BufferAttributeModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeUint32BufferAttributeModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    ThreeUint32BufferAttributeModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeUint32BufferAttributeModule, declarations: [Uint32BufferAttributeDirective], exports: [Uint32BufferAttributeDirective] });
    ThreeUint32BufferAttributeModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeUint32BufferAttributeModule });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeUint32BufferAttributeModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        declarations: [Uint32BufferAttributeDirective],
                        exports: [Uint32BufferAttributeDirective],
                    }]
            }] });

    // GENERATED

    var Uint8ClampedBufferAttributeDirective = /** @class */ (function (_super) {
        __extends(Uint8ClampedBufferAttributeDirective, _super);
        function Uint8ClampedBufferAttributeDirective() {
            var _this = _super.apply(this, __spreadArray([], __read(arguments))) || this;
            _this.attributeType = three.Uint8ClampedBufferAttribute;
            return _this;
        }
        Object.defineProperty(Uint8ClampedBufferAttributeDirective.prototype, "args", {
            set: function (v) {
                this.extraArgs = v;
            },
            enumerable: false,
            configurable: true
        });
        return Uint8ClampedBufferAttributeDirective;
    }(i1.ThreeAttribute));
    Uint8ClampedBufferAttributeDirective.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: Uint8ClampedBufferAttributeDirective, deps: null, target: i0__namespace.ɵɵFactoryTarget.Directive });
    Uint8ClampedBufferAttributeDirective.ɵdir = i0__namespace.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: Uint8ClampedBufferAttributeDirective, selector: "ngt-uint8-clamped-buffer-attribute", inputs: { args: "args" }, providers: [
            {
                provide: i1.ThreeAttribute,
                useExisting: Uint8ClampedBufferAttributeDirective,
            },
        ], exportAs: ["ngtUint8ClampedBufferAttribute"], usesInheritance: true, ngImport: i0__namespace });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: Uint8ClampedBufferAttributeDirective, decorators: [{
                type: i0.Directive,
                args: [{
                        selector: 'ngt-uint8-clamped-buffer-attribute',
                        exportAs: 'ngtUint8ClampedBufferAttribute',
                        providers: [
                            {
                                provide: i1.ThreeAttribute,
                                useExisting: Uint8ClampedBufferAttributeDirective,
                            },
                        ],
                    }]
            }], propDecorators: { args: [{
                    type: i0.Input
                }] } });

    // GENERATED
    var ThreeUint8ClampedBufferAttributeModule = /** @class */ (function () {
        function ThreeUint8ClampedBufferAttributeModule() {
        }
        return ThreeUint8ClampedBufferAttributeModule;
    }());
    ThreeUint8ClampedBufferAttributeModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeUint8ClampedBufferAttributeModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    ThreeUint8ClampedBufferAttributeModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeUint8ClampedBufferAttributeModule, declarations: [Uint8ClampedBufferAttributeDirective], exports: [Uint8ClampedBufferAttributeDirective] });
    ThreeUint8ClampedBufferAttributeModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeUint8ClampedBufferAttributeModule });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0__namespace, type: ThreeUint8ClampedBufferAttributeModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        declarations: [Uint8ClampedBufferAttributeDirective],
                        exports: [Uint8ClampedBufferAttributeDirective],
                    }]
            }] });

    // GENERATED

    // GENERATED

    /**
     * Generated bundle index. Do not edit.
     */

    exports.BufferAttributeDirective = BufferAttributeDirective;
    exports.Float16BufferAttributeDirective = Float16BufferAttributeDirective;
    exports.Float32BufferAttributeDirective = Float32BufferAttributeDirective;
    exports.Float64BufferAttributeDirective = Float64BufferAttributeDirective;
    exports.InstancedBufferAttributeDirective = InstancedBufferAttributeDirective;
    exports.Int16BufferAttributeDirective = Int16BufferAttributeDirective;
    exports.Int32BufferAttributeDirective = Int32BufferAttributeDirective;
    exports.Int8BufferAttributeDirective = Int8BufferAttributeDirective;
    exports.InterleavedBufferAttributeDirective = InterleavedBufferAttributeDirective;
    exports.InterleavedBufferAttributeModule = InterleavedBufferAttributeModule;
    exports.ThreeBufferAttributeModule = ThreeBufferAttributeModule;
    exports.ThreeFloat16BufferAttributeModule = ThreeFloat16BufferAttributeModule;
    exports.ThreeFloat32BufferAttributeModule = ThreeFloat32BufferAttributeModule;
    exports.ThreeFloat64BufferAttributeModule = ThreeFloat64BufferAttributeModule;
    exports.ThreeInstancedBufferAttributeModule = ThreeInstancedBufferAttributeModule;
    exports.ThreeInt16BufferAttributeModule = ThreeInt16BufferAttributeModule;
    exports.ThreeInt32BufferAttributeModule = ThreeInt32BufferAttributeModule;
    exports.ThreeInt8BufferAttributeModule = ThreeInt8BufferAttributeModule;
    exports.ThreeUint16BufferAttributeModule = ThreeUint16BufferAttributeModule;
    exports.ThreeUint32BufferAttributeModule = ThreeUint32BufferAttributeModule;
    exports.ThreeUint8BufferAttributeModule = ThreeUint8BufferAttributeModule;
    exports.ThreeUint8ClampedBufferAttributeModule = ThreeUint8ClampedBufferAttributeModule;
    exports.Uint16BufferAttributeDirective = Uint16BufferAttributeDirective;
    exports.Uint32BufferAttributeDirective = Uint32BufferAttributeDirective;
    exports.Uint8BufferAttributeDirective = Uint8BufferAttributeDirective;
    exports.Uint8ClampedBufferAttributeDirective = Uint8ClampedBufferAttributeDirective;

    Object.defineProperty(exports, '__esModule', { value: true });

}));
