// @angular-three/core/esm2015/geometries/lib/parametric-geometry/parametric-geometry.directive.js
// GENERATED
import { ThreeBufferGeometry } from '@angular-three/core';
import { Directive, Input } from '@angular/core';
import { ParametricGeometry } from 'three/examples/jsm/geometries/ParametricGeometry';
import * as i0 from "@angular/core";
export class ParametricGeometryDirective extends ThreeBufferGeometry {
    constructor() {
        super(...arguments);
        this.geometryType = ParametricGeometry;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
ParametricGeometryDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ParametricGeometryDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
ParametricGeometryDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: ParametricGeometryDirective, selector: "ngt-parametric-geometry", inputs: { args: "args" }, providers: [
        {
            provide: ThreeBufferGeometry,
            useExisting: ParametricGeometryDirective,
        },
    ], exportAs: ["ngtParametricGeometry"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ParametricGeometryDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-parametric-geometry',
                    exportAs: 'ngtParametricGeometry',
                    providers: [
                        {
                            provide: ThreeBufferGeometry,
                            useExisting: ParametricGeometryDirective,
                        },
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });
