// @angular-three/core/esm2015/curves/lib/quadratic-bezier-curve3/quadratic-bezier-curve3.directive.js
// GENERATED
import { ThreeCurve } from '@angular-three/core';
import { Directive, Input } from '@angular/core';
import { QuadraticBezierCurve3 } from 'three';
import * as i0 from "@angular/core";
export class QuadraticBezierCurve3Directive extends ThreeCurve {
    constructor() {
        super(...arguments);
        this.curveType = QuadraticBezierCurve3;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
QuadraticBezierCurve3Directive.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: QuadraticBezierCurve3Directive, deps: null, target: i0.ɵɵFactoryTarget.Directive });
QuadraticBezierCurve3Directive.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: QuadraticBezierCurve3Directive, selector: "ngt-quadratic-bezier-curve3", inputs: { args: "args" }, providers: [
        {
            provide: ThreeCurve,
            useExisting: QuadraticBezierCurve3Directive,
        },
    ], exportAs: ["ngtQuadraticBezierCurve3"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: QuadraticBezierCurve3Directive, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-quadratic-bezier-curve3',
                    exportAs: 'ngtQuadraticBezierCurve3',
                    providers: [
                        {
                            provide: ThreeCurve,
                            useExisting: QuadraticBezierCurve3Directive,
                        },
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });
