// @angular-three/core/esm2015/attributes/lib/uint32-buffer-attribute/uint32-buffer-attribute.directive.js
// GENERATED
import { ThreeAttribute } from '@angular-three/core';
import { Directive, Input } from '@angular/core';
import { Uint32BufferAttribute } from 'three';
import * as i0 from "@angular/core";
export class Uint32BufferAttributeDirective extends ThreeAttribute {
    constructor() {
        super(...arguments);
        this.attributeType = Uint32BufferAttribute;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
Uint32BufferAttributeDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: Uint32BufferAttributeDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
Uint32BufferAttributeDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: Uint32BufferAttributeDirective, selector: "ngt-uint32-buffer-attribute", inputs: { args: "args" }, providers: [
        {
            provide: ThreeAttribute,
            useExisting: Uint32BufferAttributeDirective,
        },
    ], exportAs: ["ngtUint32BufferAttribute"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: Uint32BufferAttributeDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-uint32-buffer-attribute',
                    exportAs: 'ngtUint32BufferAttribute',
                    providers: [
                        {
                            provide: ThreeAttribute,
                            useExisting: Uint32BufferAttributeDirective,
                        },
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });
