// @angular-three/core/esm2015/lod/lib/lod/lod.directive.js
import { OBJECT_3D_CONTROLLER_PROVIDER, ThreeObject3d, } from '@angular-three/core';
import { Directive } from '@angular/core';
import { LOD } from 'three';
import * as i0 from "@angular/core";
export class LodDirective extends ThreeObject3d {
    ngOnInit() {
        this.init();
    }
    initObject() {
        this._lod = new LOD();
    }
    get object3d() {
        return this._lod;
    }
}
LodDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: LodDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
LodDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: LodDirective, selector: "ngt-lod", providers: [
        { provide: ThreeObject3d, useExisting: LodDirective },
        OBJECT_3D_CONTROLLER_PROVIDER,
    ], exportAs: ["ngtLod"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: LodDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-lod',
                    exportAs: 'ngtLod',
                    providers: [
                        { provide: ThreeObject3d, useExisting: LodDirective },
                        OBJECT_3D_CONTROLLER_PROVIDER,
                    ],
                }]
        }] });
