// @angular-three/core/esm2015/geometries/lib/torus-knot-buffer-geometry/torus-knot-buffer-geometry.directive.js
// GENERATED
import { ThreeBufferGeometry } from '@angular-three/core';
import { Directive, Input } from '@angular/core';
import { TorusKnotBufferGeometry } from 'three';
import * as i0 from "@angular/core";
export class TorusKnotBufferGeometryDirective extends ThreeBufferGeometry {
    constructor() {
        super(...arguments);
        this.geometryType = TorusKnotBufferGeometry;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
TorusKnotBufferGeometryDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: TorusKnotBufferGeometryDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
TorusKnotBufferGeometryDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: TorusKnotBufferGeometryDirective, selector: "ngt-torus-knot-buffer-geometry,ngt-torus-knot-geometry", inputs: { args: "args" }, providers: [
        {
            provide: ThreeBufferGeometry,
            useExisting: TorusKnotBufferGeometryDirective,
        },
    ], exportAs: ["ngtTorusKnotBufferGeometry"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: TorusKnotBufferGeometryDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-torus-knot-buffer-geometry,ngt-torus-knot-geometry',
                    exportAs: 'ngtTorusKnotBufferGeometry',
                    providers: [
                        {
                            provide: ThreeBufferGeometry,
                            useExisting: TorusKnotBufferGeometryDirective,
                        },
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });
