// @angular-three/core/esm2015/lights/lib/directional-light/directional-light.directive.js
// GENERATED
import { ThreeLight, ThreeObject3d, OBJECT_3D_CONTROLLER_PROVIDER, } from '@angular-three/core';
import { Directive, Input } from '@angular/core';
import { DirectionalLight } from 'three';
import * as i0 from "@angular/core";
export class DirectionalLightDirective extends ThreeLight {
    constructor() {
        super(...arguments);
        this.lightType = DirectionalLight;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
DirectionalLightDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: DirectionalLightDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
DirectionalLightDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: DirectionalLightDirective, selector: "ngt-directional-light", inputs: { args: "args" }, providers: [
        {
            provide: ThreeObject3d,
            useExisting: DirectionalLightDirective,
        },
        OBJECT_3D_CONTROLLER_PROVIDER,
    ], exportAs: ["ngtDirectionalLight"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: DirectionalLightDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-directional-light',
                    exportAs: 'ngtDirectionalLight',
                    providers: [
                        {
                            provide: ThreeObject3d,
                            useExisting: DirectionalLightDirective,
                        },
                        OBJECT_3D_CONTROLLER_PROVIDER,
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });
