// @angular-three/core/esm2015/helpers/lib/skeleton-helper/skeleton-helper.directive.js
// GENERATED
import { ThreeHelper, ThreeObject3d, OBJECT_3D_CONTROLLER_PROVIDER, } from '@angular-three/core';
import { Directive, Input } from '@angular/core';
import { SkeletonHelper } from 'three';
import * as i0 from "@angular/core";
export class SkeletonHelperDirective extends ThreeHelper {
    constructor() {
        super(...arguments);
        this.helperType = SkeletonHelper;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
SkeletonHelperDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: SkeletonHelperDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
SkeletonHelperDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: SkeletonHelperDirective, selector: "ngt-skeleton-helper", inputs: { args: "args" }, providers: [
        {
            provide: ThreeObject3d,
            useExisting: SkeletonHelperDirective,
        },
        OBJECT_3D_CONTROLLER_PROVIDER,
    ], exportAs: ["ngtSkeletonHelper"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: SkeletonHelperDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-skeleton-helper',
                    exportAs: 'ngtSkeletonHelper',
                    providers: [
                        {
                            provide: ThreeObject3d,
                            useExisting: SkeletonHelperDirective,
                        },
                        OBJECT_3D_CONTROLLER_PROVIDER,
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });
