// @angular-three/core/esm2015/curves/lib/line-curve3/line-curve3.directive.js
// GENERATED
import { ThreeCurve } from '@angular-three/core';
import { Directive, Input } from '@angular/core';
import { LineCurve3 } from 'three';
import * as i0 from "@angular/core";
export class LineCurve3Directive extends ThreeCurve {
    constructor() {
        super(...arguments);
        this.curveType = LineCurve3;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
LineCurve3Directive.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: LineCurve3Directive, deps: null, target: i0.ɵɵFactoryTarget.Directive });
LineCurve3Directive.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: LineCurve3Directive, selector: "ngt-line-curve3", inputs: { args: "args" }, providers: [
        {
            provide: ThreeCurve,
            useExisting: LineCurve3Directive,
        },
    ], exportAs: ["ngtLineCurve3"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: LineCurve3Directive, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-line-curve3',
                    exportAs: 'ngtLineCurve3',
                    providers: [
                        {
                            provide: ThreeCurve,
                            useExisting: LineCurve3Directive,
                        },
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });
