// @angular-three/core/esm2015/geometries/lib/octahedron-buffer-geometry/octahedron-buffer-geometry.directive.js
// GENERATED
import { ThreeBufferGeometry } from '@angular-three/core';
import { Directive, Input } from '@angular/core';
import { OctahedronBufferGeometry } from 'three';
import * as i0 from "@angular/core";
export class OctahedronBufferGeometryDirective extends ThreeBufferGeometry {
    constructor() {
        super(...arguments);
        this.geometryType = OctahedronBufferGeometry;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
OctahedronBufferGeometryDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: OctahedronBufferGeometryDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
OctahedronBufferGeometryDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: OctahedronBufferGeometryDirective, selector: "ngt-octahedron-buffer-geometry,ngt-octahedron-geometry", inputs: { args: "args" }, providers: [
        {
            provide: ThreeBufferGeometry,
            useExisting: OctahedronBufferGeometryDirective,
        },
    ], exportAs: ["ngtOctahedronBufferGeometry"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: OctahedronBufferGeometryDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-octahedron-buffer-geometry,ngt-octahedron-geometry',
                    exportAs: 'ngtOctahedronBufferGeometry',
                    providers: [
                        {
                            provide: ThreeBufferGeometry,
                            useExisting: OctahedronBufferGeometryDirective,
                        },
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });
