// @angular-three/core/esm2015/curves/lib/line-curve/line-curve.directive.js
// GENERATED
import { ThreeCurve } from '@angular-three/core';
import { Directive, Input } from '@angular/core';
import { LineCurve } from 'three';
import * as i0 from "@angular/core";
export class LineCurveDirective extends ThreeCurve {
    constructor() {
        super(...arguments);
        this.curveType = LineCurve;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
LineCurveDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: LineCurveDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
LineCurveDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: LineCurveDirective, selector: "ngt-line-curve", inputs: { args: "args" }, providers: [
        {
            provide: ThreeCurve,
            useExisting: LineCurveDirective,
        },
    ], exportAs: ["ngtLineCurve"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: LineCurveDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-line-curve',
                    exportAs: 'ngtLineCurve',
                    providers: [
                        {
                            provide: ThreeCurve,
                            useExisting: LineCurveDirective,
                        },
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });
