// @angular-three/core/esm2015/geometries/lib/sphere-buffer-geometry/sphere-buffer-geometry.directive.js
// GENERATED
import { ThreeBufferGeometry } from '@angular-three/core';
import { Directive, Input } from '@angular/core';
import { SphereBufferGeometry } from 'three';
import * as i0 from "@angular/core";
export class SphereBufferGeometryDirective extends ThreeBufferGeometry {
    constructor() {
        super(...arguments);
        this.geometryType = SphereBufferGeometry;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
SphereBufferGeometryDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: SphereBufferGeometryDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
SphereBufferGeometryDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: SphereBufferGeometryDirective, selector: "ngt-sphere-buffer-geometry,ngt-sphere-geometry", inputs: { args: "args" }, providers: [
        {
            provide: ThreeBufferGeometry,
            useExisting: SphereBufferGeometryDirective,
        },
    ], exportAs: ["ngtSphereBufferGeometry"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: SphereBufferGeometryDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-sphere-buffer-geometry,ngt-sphere-geometry',
                    exportAs: 'ngtSphereBufferGeometry',
                    providers: [
                        {
                            provide: ThreeBufferGeometry,
                            useExisting: SphereBufferGeometryDirective,
                        },
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });
