// @angular-three/core/esm2015/materials/lib/mesh-depth-material/mesh-depth-material.directive.js
// GENERATED
import { ThreeMaterial } from '@angular-three/core';
import { Directive } from '@angular/core';
import { MeshDepthMaterial } from 'three';
import * as i0 from "@angular/core";
export class MeshDepthMaterialDirective extends ThreeMaterial {
    constructor() {
        super(...arguments);
        this.materialType = MeshDepthMaterial;
    }
}
MeshDepthMaterialDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: MeshDepthMaterialDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
MeshDepthMaterialDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: MeshDepthMaterialDirective, selector: "ngt-mesh-depth-material", providers: [
        { provide: ThreeMaterial, useExisting: MeshDepthMaterialDirective },
    ], exportAs: ["ngtMeshDepthMaterial"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: MeshDepthMaterialDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-mesh-depth-material',
                    exportAs: 'ngtMeshDepthMaterial',
                    providers: [
                        { provide: ThreeMaterial, useExisting: MeshDepthMaterialDirective },
                    ],
                }]
        }] });
