// @angular-three/core/esm2015/textures/lib/compressed-texture/compressed-texture.directive.js
// GENERATED
import { ThreeTexture } from '@angular-three/core';
import { Directive, Input } from '@angular/core';
import { CompressedTexture } from 'three';
import * as i0 from "@angular/core";
export class CompressedTextureDirective extends ThreeTexture {
    constructor() {
        super(...arguments);
        this.textureType = CompressedTexture;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
CompressedTextureDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: CompressedTextureDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
CompressedTextureDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: CompressedTextureDirective, selector: "ngt-compressed-texture", inputs: { args: "args" }, providers: [
        {
            provide: ThreeTexture,
            useExisting: CompressedTextureDirective,
        },
    ], exportAs: ["ngtCompressedTexture"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: CompressedTextureDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-compressed-texture',
                    exportAs: 'ngtCompressedTexture',
                    providers: [
                        {
                            provide: ThreeTexture,
                            useExisting: CompressedTextureDirective,
                        },
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });
