// @angular-three/core/esm2015/geometries/lib/edges-geometry/edges-geometry.directive.js
// GENERATED
import { ThreeBufferGeometry } from '@angular-three/core';
import { Directive, Input } from '@angular/core';
import { EdgesGeometry } from 'three';
import * as i0 from "@angular/core";
export class EdgesGeometryDirective extends ThreeBufferGeometry {
    constructor() {
        super(...arguments);
        this.geometryType = EdgesGeometry;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
EdgesGeometryDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: EdgesGeometryDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
EdgesGeometryDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: EdgesGeometryDirective, selector: "ngt-edges-geometry", inputs: { args: "args" }, providers: [
        {
            provide: ThreeBufferGeometry,
            useExisting: EdgesGeometryDirective,
        },
    ], exportAs: ["ngtEdgesGeometry"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: EdgesGeometryDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-edges-geometry',
                    exportAs: 'ngtEdgesGeometry',
                    providers: [
                        {
                            provide: ThreeBufferGeometry,
                            useExisting: EdgesGeometryDirective,
                        },
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });
