// @angular-three/core/esm2015/geometries/lib/plane-buffer-geometry/plane-buffer-geometry.directive.js
// GENERATED
import { ThreeBufferGeometry } from '@angular-three/core';
import { Directive, Input } from '@angular/core';
import { PlaneBufferGeometry } from 'three';
import * as i0 from "@angular/core";
export class PlaneBufferGeometryDirective extends ThreeBufferGeometry {
    constructor() {
        super(...arguments);
        this.geometryType = PlaneBufferGeometry;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
PlaneBufferGeometryDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: PlaneBufferGeometryDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
PlaneBufferGeometryDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: PlaneBufferGeometryDirective, selector: "ngt-plane-buffer-geometry,ngt-plane-geometry", inputs: { args: "args" }, providers: [
        {
            provide: ThreeBufferGeometry,
            useExisting: PlaneBufferGeometryDirective,
        },
    ], exportAs: ["ngtPlaneBufferGeometry"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: PlaneBufferGeometryDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-plane-buffer-geometry,ngt-plane-geometry',
                    exportAs: 'ngtPlaneBufferGeometry',
                    providers: [
                        {
                            provide: ThreeBufferGeometry,
                            useExisting: PlaneBufferGeometryDirective,
                        },
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });
