// @angular-three/core/esm2015/curves/lib/cubic-bezier-curve3/cubic-bezier-curve3.directive.js
// GENERATED
import { ThreeCurve } from '@angular-three/core';
import { Directive, Input } from '@angular/core';
import { CubicBezierCurve3 } from 'three';
import * as i0 from "@angular/core";
export class CubicBezierCurve3Directive extends ThreeCurve {
    constructor() {
        super(...arguments);
        this.curveType = CubicBezierCurve3;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
CubicBezierCurve3Directive.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: CubicBezierCurve3Directive, deps: null, target: i0.ɵɵFactoryTarget.Directive });
CubicBezierCurve3Directive.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: CubicBezierCurve3Directive, selector: "ngt-cubic-bezier-curve3", inputs: { args: "args" }, providers: [
        {
            provide: ThreeCurve,
            useExisting: CubicBezierCurve3Directive,
        },
    ], exportAs: ["ngtCubicBezierCurve3"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: CubicBezierCurve3Directive, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-cubic-bezier-curve3',
                    exportAs: 'ngtCubicBezierCurve3',
                    providers: [
                        {
                            provide: ThreeCurve,
                            useExisting: CubicBezierCurve3Directive,
                        },
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });
