// @angular-three/core/esm2015/helpers/lib/camera-helper/camera-helper.directive.js
// GENERATED
import { ThreeHelper, ThreeObject3d, OBJECT_3D_CONTROLLER_PROVIDER, } from '@angular-three/core';
import { Directive, Input } from '@angular/core';
import { CameraHelper } from 'three';
import * as i0 from "@angular/core";
export class CameraHelperDirective extends ThreeHelper {
    constructor() {
        super(...arguments);
        this.helperType = CameraHelper;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
CameraHelperDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: CameraHelperDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
CameraHelperDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: CameraHelperDirective, selector: "ngt-camera-helper", inputs: { args: "args" }, providers: [
        {
            provide: ThreeObject3d,
            useExisting: CameraHelperDirective,
        },
        OBJECT_3D_CONTROLLER_PROVIDER,
    ], exportAs: ["ngtCameraHelper"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: CameraHelperDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-camera-helper',
                    exportAs: 'ngtCameraHelper',
                    providers: [
                        {
                            provide: ThreeObject3d,
                            useExisting: CameraHelperDirective,
                        },
                        OBJECT_3D_CONTROLLER_PROVIDER,
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });
