// @angular-three/core/esm2015/geometries/lib/convex-geometry/convex-geometry.directive.js
// GENERATED
import { ThreeBufferGeometry } from '@angular-three/core';
import { Directive, Input } from '@angular/core';
import { ConvexGeometry } from 'three/examples/jsm/geometries/ConvexGeometry';
import * as i0 from "@angular/core";
export class ConvexGeometryDirective extends ThreeBufferGeometry {
    constructor() {
        super(...arguments);
        this.geometryType = ConvexGeometry;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
ConvexGeometryDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ConvexGeometryDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
ConvexGeometryDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: ConvexGeometryDirective, selector: "ngt-convex-geometry", inputs: { args: "args" }, providers: [
        {
            provide: ThreeBufferGeometry,
            useExisting: ConvexGeometryDirective,
        },
    ], exportAs: ["ngtConvexGeometry"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ConvexGeometryDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-convex-geometry',
                    exportAs: 'ngtConvexGeometry',
                    providers: [
                        {
                            provide: ThreeBufferGeometry,
                            useExisting: ConvexGeometryDirective,
                        },
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });
