// @angular-three/core/esm2015/helpers/lib/box-helper/box-helper.directive.js
// GENERATED
import { ThreeHelper, ThreeObject3d, OBJECT_3D_CONTROLLER_PROVIDER, } from '@angular-three/core';
import { Directive, Input } from '@angular/core';
import { BoxHelper } from 'three';
import * as i0 from "@angular/core";
export class BoxHelperDirective extends ThreeHelper {
    constructor() {
        super(...arguments);
        this.helperType = BoxHelper;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
BoxHelperDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: BoxHelperDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
BoxHelperDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: BoxHelperDirective, selector: "ngt-box-helper", inputs: { args: "args" }, providers: [
        {
            provide: ThreeObject3d,
            useExisting: BoxHelperDirective,
        },
        OBJECT_3D_CONTROLLER_PROVIDER,
    ], exportAs: ["ngtBoxHelper"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: BoxHelperDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-box-helper',
                    exportAs: 'ngtBoxHelper',
                    providers: [
                        {
                            provide: ThreeObject3d,
                            useExisting: BoxHelperDirective,
                        },
                        OBJECT_3D_CONTROLLER_PROVIDER,
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });
