// @angular-three/core/esm2015/materials/lib/mesh-basic-material/mesh-basic-material.directive.js
// GENERATED
import { ThreeMaterial } from '@angular-three/core';
import { Directive } from '@angular/core';
import { MeshBasicMaterial } from 'three';
import * as i0 from "@angular/core";
export class MeshBasicMaterialDirective extends ThreeMaterial {
    constructor() {
        super(...arguments);
        this.materialType = MeshBasicMaterial;
    }
}
MeshBasicMaterialDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: MeshBasicMaterialDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
MeshBasicMaterialDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: MeshBasicMaterialDirective, selector: "ngt-mesh-basic-material", providers: [
        { provide: ThreeMaterial, useExisting: MeshBasicMaterialDirective },
    ], exportAs: ["ngtMeshBasicMaterial"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: MeshBasicMaterialDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-mesh-basic-material',
                    exportAs: 'ngtMeshBasicMaterial',
                    providers: [
                        { provide: ThreeMaterial, useExisting: MeshBasicMaterialDirective },
                    ],
                }]
        }] });
