// @angular-three/core/esm2015/materials/lib/mesh-matcap-material/mesh-matcap-material.directive.js
// GENERATED
import { ThreeMaterial } from '@angular-three/core';
import { Directive } from '@angular/core';
import { MeshMatcapMaterial } from 'three';
import * as i0 from "@angular/core";
export class MeshMatcapMaterialDirective extends ThreeMaterial {
    constructor() {
        super(...arguments);
        this.materialType = MeshMatcapMaterial;
    }
}
MeshMatcapMaterialDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: MeshMatcapMaterialDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
MeshMatcapMaterialDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: MeshMatcapMaterialDirective, selector: "ngt-mesh-matcap-material", providers: [
        { provide: ThreeMaterial, useExisting: MeshMatcapMaterialDirective },
    ], exportAs: ["ngtMeshMatcapMaterial"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: MeshMatcapMaterialDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-mesh-matcap-material',
                    exportAs: 'ngtMeshMatcapMaterial',
                    providers: [
                        { provide: ThreeMaterial, useExisting: MeshMatcapMaterialDirective },
                    ],
                }]
        }] });
