// @angular-three/core/esm2015/lights/lib/ambient-light/ambient-light.directive.js
// GENERATED
import { ThreeLight, ThreeObject3d, OBJECT_3D_CONTROLLER_PROVIDER, } from '@angular-three/core';
import { Directive, Input } from '@angular/core';
import { AmbientLight } from 'three';
import * as i0 from "@angular/core";
export class AmbientLightDirective extends ThreeLight {
    constructor() {
        super(...arguments);
        this.lightType = AmbientLight;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
AmbientLightDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: AmbientLightDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
AmbientLightDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: AmbientLightDirective, selector: "ngt-ambient-light", inputs: { args: "args" }, providers: [
        {
            provide: ThreeObject3d,
            useExisting: AmbientLightDirective,
        },
        OBJECT_3D_CONTROLLER_PROVIDER,
    ], exportAs: ["ngtAmbientLight"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: AmbientLightDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-ambient-light',
                    exportAs: 'ngtAmbientLight',
                    providers: [
                        {
                            provide: ThreeObject3d,
                            useExisting: AmbientLightDirective,
                        },
                        OBJECT_3D_CONTROLLER_PROVIDER,
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });
