// @angular-three/core/esm2015/geometries/lib/dodecahedron-buffer-geometry/dodecahedron-buffer-geometry.directive.js
// GENERATED
import { ThreeBufferGeometry } from '@angular-three/core';
import { Directive, Input } from '@angular/core';
import { DodecahedronBufferGeometry } from 'three';
import * as i0 from "@angular/core";
export class DodecahedronBufferGeometryDirective extends ThreeBufferGeometry {
    constructor() {
        super(...arguments);
        this.geometryType = DodecahedronBufferGeometry;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
DodecahedronBufferGeometryDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: DodecahedronBufferGeometryDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
DodecahedronBufferGeometryDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: DodecahedronBufferGeometryDirective, selector: "ngt-dodecahedron-buffer-geometry,ngt-dodecahedron-geometry", inputs: { args: "args" }, providers: [
        {
            provide: ThreeBufferGeometry,
            useExisting: DodecahedronBufferGeometryDirective,
        },
    ], exportAs: ["ngtDodecahedronBufferGeometry"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: DodecahedronBufferGeometryDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-dodecahedron-buffer-geometry,ngt-dodecahedron-geometry',
                    exportAs: 'ngtDodecahedronBufferGeometry',
                    providers: [
                        {
                            provide: ThreeBufferGeometry,
                            useExisting: DodecahedronBufferGeometryDirective,
                        },
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });
