// @angular-three/core/esm2015/geometries/lib/cylinder-buffer-geometry/cylinder-buffer-geometry.directive.js
// GENERATED
import { ThreeBufferGeometry } from '@angular-three/core';
import { Directive, Input } from '@angular/core';
import { CylinderBufferGeometry } from 'three';
import * as i0 from "@angular/core";
export class CylinderBufferGeometryDirective extends ThreeBufferGeometry {
    constructor() {
        super(...arguments);
        this.geometryType = CylinderBufferGeometry;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
CylinderBufferGeometryDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: CylinderBufferGeometryDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
CylinderBufferGeometryDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: CylinderBufferGeometryDirective, selector: "ngt-cylinder-buffer-geometry,ngt-cylinder-geometry", inputs: { args: "args" }, providers: [
        {
            provide: ThreeBufferGeometry,
            useExisting: CylinderBufferGeometryDirective,
        },
    ], exportAs: ["ngtCylinderBufferGeometry"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: CylinderBufferGeometryDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-cylinder-buffer-geometry,ngt-cylinder-geometry',
                    exportAs: 'ngtCylinderBufferGeometry',
                    providers: [
                        {
                            provide: ThreeBufferGeometry,
                            useExisting: CylinderBufferGeometryDirective,
                        },
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });
