// @angular-three/core/esm2015/lines/lib/line-segments/line-segments.directive.js
// GENERATED
import { ThreeLine, ThreeObject3d, OBJECT_3D_CONTROLLER_PROVIDER, } from '@angular-three/core';
import { Directive } from '@angular/core';
import { LineSegments } from 'three';
import * as i0 from "@angular/core";
export class LineSegmentsDirective extends ThreeLine {
    constructor() {
        super(...arguments);
        this.lineType = LineSegments;
    }
}
LineSegmentsDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: LineSegmentsDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
LineSegmentsDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: LineSegmentsDirective, selector: "ngt-line-segments", providers: [
        {
            provide: ThreeObject3d,
            useExisting: LineSegmentsDirective,
        },
        OBJECT_3D_CONTROLLER_PROVIDER,
    ], exportAs: ["ngtLineSegments"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: LineSegmentsDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-line-segments',
                    exportAs: 'ngtLineSegments',
                    providers: [
                        {
                            provide: ThreeObject3d,
                            useExisting: LineSegmentsDirective,
                        },
                        OBJECT_3D_CONTROLLER_PROVIDER,
                    ],
                }]
        }] });
