// @angular-three/core/esm2015/materials/lib/shader-material/shader-material.directive.js
// GENERATED
import { ThreeMaterial } from '@angular-three/core';
import { Directive } from '@angular/core';
import { ShaderMaterial } from 'three';
import * as i0 from "@angular/core";
export class ShaderMaterialDirective extends ThreeMaterial {
    constructor() {
        super(...arguments);
        this.materialType = ShaderMaterial;
    }
}
ShaderMaterialDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ShaderMaterialDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
ShaderMaterialDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: ShaderMaterialDirective, selector: "ngt-shader-material", providers: [{ provide: ThreeMaterial, useExisting: ShaderMaterialDirective }], exportAs: ["ngtShaderMaterial"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ShaderMaterialDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-shader-material',
                    exportAs: 'ngtShaderMaterial',
                    providers: [{ provide: ThreeMaterial, useExisting: ShaderMaterialDirective }],
                }]
        }] });
