// @angular-three/core/esm2015/geometries/lib/box-line-geometry/box-line-geometry.directive.js
// GENERATED
import { ThreeBufferGeometry } from '@angular-three/core';
import { Directive, Input } from '@angular/core';
import { BoxLineGeometry } from 'three/examples/jsm/geometries/BoxLineGeometry';
import * as i0 from "@angular/core";
export class BoxLineGeometryDirective extends ThreeBufferGeometry {
    constructor() {
        super(...arguments);
        this.geometryType = BoxLineGeometry;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
BoxLineGeometryDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: BoxLineGeometryDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
BoxLineGeometryDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: BoxLineGeometryDirective, selector: "ngt-box-line-geometry", inputs: { args: "args" }, providers: [
        {
            provide: ThreeBufferGeometry,
            useExisting: BoxLineGeometryDirective,
        },
    ], exportAs: ["ngtBoxLineGeometry"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: BoxLineGeometryDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-box-line-geometry',
                    exportAs: 'ngtBoxLineGeometry',
                    providers: [
                        {
                            provide: ThreeBufferGeometry,
                            useExisting: BoxLineGeometryDirective,
                        },
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });
