// @angular-three/core/esm2015/lib/abstracts/animation-loop-participant.abstract.js
import { Directive, EventEmitter, NgZone, Output, } from '@angular/core';
import { Object3D } from 'three';
import { AnimationStore } from '../stores';
import * as i0 from "@angular/core";
import * as i1 from "../stores";
export class AnimationLoopParticipant {
    constructor(animationStore, ngZone) {
        this.animationStore = animationStore;
        this.ngZone = ngZone;
        this.animateReady = new EventEmitter();
    }
    participate(animateObject) {
        this.ngZone.runOutsideAngular(() => {
            if (this.animateReady.observed) {
                if (animateObject instanceof Object3D) {
                    this.animateTeardown = this.animationStore.registerAnimation(animateObject, (obj, state) => {
                        this.animateReady.emit({
                            animateObject: obj,
                            renderState: state,
                        });
                    });
                }
                else {
                    this.animateTeardown = this.animationStore.registerAnimation((state) => {
                        this.animateReady.emit({
                            animateObject,
                            renderState: state,
                        });
                    });
                }
            }
        });
    }
    ngOnDestroy() {
        this.ngZone.runOutsideAngular(() => {
            if (this.animateTeardown) {
                this.animateTeardown();
            }
        });
    }
}
AnimationLoopParticipant.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: AnimationLoopParticipant, deps: [{ token: i1.AnimationStore }, { token: i0.NgZone }], target: i0.ɵɵFactoryTarget.Directive });
AnimationLoopParticipant.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: AnimationLoopParticipant, outputs: { animateReady: "animateReady" }, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: AnimationLoopParticipant, decorators: [{
            type: Directive
        }], ctorParameters: function () { return [{ type: i1.AnimationStore }, { type: i0.NgZone }]; }, propDecorators: { animateReady: [{
                type: Output
            }] } });
