// @angular-three/core/esm2015/lib/abstracts/camera.abstract.js
import { Directive } from '@angular/core';
import { ThreeObject3d } from './object-3d.abstract';
import * as i0 from "@angular/core";
export class ThreeCamera extends ThreeObject3d {
    constructor() {
        super(...arguments);
        this._extraArgs = [];
    }
    set extraArgs(v) {
        this._extraArgs = v;
        this.ngZone.runOutsideAngular(() => {
            this.init();
        });
    }
    ngOnInit() {
        if (!this.object3d) {
            this.init();
        }
    }
    initObject() {
        this._camera = new this.cameraType(...this._extraArgs);
    }
    get object3d() {
        return this._camera;
    }
}
ThreeCamera.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeCamera, deps: null, target: i0.ɵɵFactoryTarget.Directive });
ThreeCamera.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: ThreeCamera, usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeCamera, decorators: [{
            type: Directive
        }] });
