// @angular-three/core/esm2015/lib/abstracts/light.abstract.js
import { Directive, Input } from '@angular/core';
import { ThreeObject3d } from './object-3d.abstract';
import * as i0 from "@angular/core";
export class ThreeLight extends ThreeObject3d {
    constructor() {
        super(...arguments);
        this._extraArgs = [];
    }
    set extraArgs(v) {
        this._extraArgs = v;
        this.ngZone.runOutsideAngular(() => {
            this.init();
        });
    }
    ngOnInit() {
        if (!this.object3d) {
            this.init();
        }
    }
    initObject() {
        if (this.intensity) {
            this._extraArgs[1] = this.intensity;
        }
        this._light = new this.lightType(...this._extraArgs);
    }
    get object3d() {
        return this._light;
    }
}
ThreeLight.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeLight, deps: null, target: i0.ɵɵFactoryTarget.Directive });
ThreeLight.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: ThreeLight, inputs: { intensity: "intensity" }, usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeLight, decorators: [{
            type: Directive
        }], propDecorators: { intensity: [{
                type: Input
            }] } });
