// @angular-three/core/esm2015/lib/abstracts/texture.abstract.js
import { Directive, NgZone } from '@angular/core';
import * as i0 from "@angular/core";
export class ThreeTexture {
    constructor(ngZone) {
        this.ngZone = ngZone;
        this._extraArgs = [];
    }
    set extraArgs(v) {
        this._extraArgs = v;
        this.ngZone.runOutsideAngular(() => {
            this._texture = new this.textureType(...this._extraArgs);
        });
    }
    ngOnInit() {
        this.ngZone.runOutsideAngular(() => {
            if (!this.texture) {
                this._texture = new this.textureType(...this._extraArgs);
            }
        });
    }
    get texture() {
        return this._texture;
    }
    ngOnDestroy() {
        this.ngZone.runOutsideAngular(() => {
            if (this.texture) {
                this.texture.dispose();
            }
        });
    }
}
ThreeTexture.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeTexture, deps: [{ token: i0.NgZone }], target: i0.ɵɵFactoryTarget.Directive });
ThreeTexture.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: ThreeTexture, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeTexture, decorators: [{
            type: Directive
        }], ctorParameters: function () { return [{ type: i0.NgZone }]; } });
