// @angular-three/core/esm2015/helpers/lib/axes-helper/axes-helper.directive.js
// GENERATED
import { ThreeHelper, ThreeObject3d, OBJECT_3D_CONTROLLER_PROVIDER, } from '@angular-three/core';
import { Directive, Input } from '@angular/core';
import { AxesHelper } from 'three';
import * as i0 from "@angular/core";
export class AxesHelperDirective extends ThreeHelper {
    constructor() {
        super(...arguments);
        this.helperType = AxesHelper;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
AxesHelperDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: AxesHelperDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
AxesHelperDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: AxesHelperDirective, selector: "ngt-axes-helper", inputs: { args: "args" }, providers: [
        {
            provide: ThreeObject3d,
            useExisting: AxesHelperDirective,
        },
        OBJECT_3D_CONTROLLER_PROVIDER,
    ], exportAs: ["ngtAxesHelper"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: AxesHelperDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-axes-helper',
                    exportAs: 'ngtAxesHelper',
                    providers: [
                        {
                            provide: ThreeObject3d,
                            useExisting: AxesHelperDirective,
                        },
                        OBJECT_3D_CONTROLLER_PROVIDER,
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });
