// @angular-three/core/esm2015/scene/lib/scene.directive.js
import { OBJECT_3D_CONTROLLER_PROVIDER, ThreeObject3d, } from '@angular-three/core';
import { Directive } from '@angular/core';
import { Scene } from 'three';
import * as i0 from "@angular/core";
export class SceneDirective extends ThreeObject3d {
    ngOnInit() {
        this.init();
    }
    initObject() {
        this._scene = new Scene();
    }
    get object3d() {
        return this._scene;
    }
}
SceneDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: SceneDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
SceneDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: SceneDirective, selector: "ngt-scene", providers: [
        { provide: ThreeObject3d, useExisting: SceneDirective },
        OBJECT_3D_CONTROLLER_PROVIDER,
    ], exportAs: ["ngtScene"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: SceneDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-scene',
                    exportAs: 'ngtScene',
                    providers: [
                        { provide: ThreeObject3d, useExisting: SceneDirective },
                        OBJECT_3D_CONTROLLER_PROVIDER,
                    ],
                }]
        }] });
