// @angular-three/core/esm2015/materials/lib/sprite-material/sprite-material.directive.js
// GENERATED
import { ThreeMaterial } from '@angular-three/core';
import { Directive } from '@angular/core';
import { SpriteMaterial } from 'three';
import * as i0 from "@angular/core";
export class SpriteMaterialDirective extends ThreeMaterial {
    constructor() {
        super(...arguments);
        this.materialType = SpriteMaterial;
    }
}
SpriteMaterialDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: SpriteMaterialDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
SpriteMaterialDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: SpriteMaterialDirective, selector: "ngt-sprite-material", providers: [{ provide: ThreeMaterial, useExisting: SpriteMaterialDirective }], exportAs: ["ngtSpriteMaterial"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: SpriteMaterialDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-sprite-material',
                    exportAs: 'ngtSpriteMaterial',
                    providers: [{ provide: ThreeMaterial, useExisting: SpriteMaterialDirective }],
                }]
        }] });
