// @angular-three/core/esm2015/materials/lib/mesh-distance-material/mesh-distance-material.directive.js
// GENERATED
import { ThreeMaterial } from '@angular-three/core';
import { Directive } from '@angular/core';
import { MeshDistanceMaterial } from 'three';
import * as i0 from "@angular/core";
export class MeshDistanceMaterialDirective extends ThreeMaterial {
    constructor() {
        super(...arguments);
        this.materialType = MeshDistanceMaterial;
    }
}
MeshDistanceMaterialDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: MeshDistanceMaterialDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
MeshDistanceMaterialDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: MeshDistanceMaterialDirective, selector: "ngt-mesh-distance-material", providers: [
        { provide: ThreeMaterial, useExisting: MeshDistanceMaterialDirective },
    ], exportAs: ["ngtMeshDistanceMaterial"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: MeshDistanceMaterialDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-mesh-distance-material',
                    exportAs: 'ngtMeshDistanceMaterial',
                    providers: [
                        { provide: ThreeMaterial, useExisting: MeshDistanceMaterialDirective },
                    ],
                }]
        }] });
