// @angular-three/core/esm2015/materials/lib/mesh-standard-material/mesh-standard-material.directive.js
// GENERATED
import { ThreeMaterial } from '@angular-three/core';
import { Directive } from '@angular/core';
import { MeshStandardMaterial } from 'three';
import * as i0 from "@angular/core";
export class MeshStandardMaterialDirective extends ThreeMaterial {
    constructor() {
        super(...arguments);
        this.materialType = MeshStandardMaterial;
    }
}
MeshStandardMaterialDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: MeshStandardMaterialDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
MeshStandardMaterialDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: MeshStandardMaterialDirective, selector: "ngt-mesh-standard-material", providers: [
        { provide: ThreeMaterial, useExisting: MeshStandardMaterialDirective },
    ], exportAs: ["ngtMeshStandardMaterial"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: MeshStandardMaterialDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-mesh-standard-material',
                    exportAs: 'ngtMeshStandardMaterial',
                    providers: [
                        { provide: ThreeMaterial, useExisting: MeshStandardMaterialDirective },
                    ],
                }]
        }] });
