// @angular-three/core/esm2015/attributes/lib/buffer-attribute/buffer-attribute.directive.js
// GENERATED
import { ThreeAttribute } from '@angular-three/core';
import { Directive, Input } from '@angular/core';
import { BufferAttribute } from 'three';
import * as i0 from "@angular/core";
export class BufferAttributeDirective extends ThreeAttribute {
    constructor() {
        super(...arguments);
        this.attributeType = BufferAttribute;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
BufferAttributeDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: BufferAttributeDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
BufferAttributeDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: BufferAttributeDirective, selector: "ngt-buffer-attribute", inputs: { args: "args" }, providers: [
        {
            provide: ThreeAttribute,
            useExisting: BufferAttributeDirective,
        },
    ], exportAs: ["ngtBufferAttribute"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: BufferAttributeDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-buffer-attribute',
                    exportAs: 'ngtBufferAttribute',
                    providers: [
                        {
                            provide: ThreeAttribute,
                            useExisting: BufferAttributeDirective,
                        },
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });
