// @angular-three/core/esm2015/cameras/lib/perspective-camera/perspective-camera.directive.js
// GENERATED
import { ThreeCamera, ThreeObject3d, OBJECT_3D_CONTROLLER_PROVIDER, } from '@angular-three/core';
import { Directive, Input } from '@angular/core';
import { PerspectiveCamera } from 'three';
import * as i0 from "@angular/core";
export class PerspectiveCameraDirective extends ThreeCamera {
    constructor() {
        super(...arguments);
        this.cameraType = PerspectiveCamera;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
PerspectiveCameraDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: PerspectiveCameraDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
PerspectiveCameraDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: PerspectiveCameraDirective, selector: "ngt-perspective-camera", inputs: { args: "args" }, providers: [
        {
            provide: ThreeObject3d,
            useExisting: PerspectiveCameraDirective,
        },
        OBJECT_3D_CONTROLLER_PROVIDER,
    ], exportAs: ["ngtPerspectiveCamera"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: PerspectiveCameraDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-perspective-camera',
                    exportAs: 'ngtPerspectiveCamera',
                    providers: [
                        {
                            provide: ThreeObject3d,
                            useExisting: PerspectiveCameraDirective,
                        },
                        OBJECT_3D_CONTROLLER_PROVIDER,
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });
