// @angular-three/core/esm2015/geometries/lib/ring-buffer-geometry/ring-buffer-geometry.directive.js
// GENERATED
import { ThreeBufferGeometry } from '@angular-three/core';
import { Directive, Input } from '@angular/core';
import { RingBufferGeometry } from 'three';
import * as i0 from "@angular/core";
export class RingBufferGeometryDirective extends ThreeBufferGeometry {
    constructor() {
        super(...arguments);
        this.geometryType = RingBufferGeometry;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
RingBufferGeometryDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: RingBufferGeometryDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
RingBufferGeometryDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: RingBufferGeometryDirective, selector: "ngt-ring-buffer-geometry,ngt-ring-geometry", inputs: { args: "args" }, providers: [
        {
            provide: ThreeBufferGeometry,
            useExisting: RingBufferGeometryDirective,
        },
    ], exportAs: ["ngtRingBufferGeometry"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: RingBufferGeometryDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-ring-buffer-geometry,ngt-ring-geometry',
                    exportAs: 'ngtRingBufferGeometry',
                    providers: [
                        {
                            provide: ThreeBufferGeometry,
                            useExisting: RingBufferGeometryDirective,
                        },
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });
