// @angular-three/core/esm2015/lights/lib/hemisphere-light/hemisphere-light.directive.js
// GENERATED
import { ThreeLight, ThreeObject3d, OBJECT_3D_CONTROLLER_PROVIDER, } from '@angular-three/core';
import { Directive, Input } from '@angular/core';
import { HemisphereLight } from 'three';
import * as i0 from "@angular/core";
export class HemisphereLightDirective extends ThreeLight {
    constructor() {
        super(...arguments);
        this.lightType = HemisphereLight;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
HemisphereLightDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: HemisphereLightDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
HemisphereLightDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: HemisphereLightDirective, selector: "ngt-hemisphere-light", inputs: { args: "args" }, providers: [
        {
            provide: ThreeObject3d,
            useExisting: HemisphereLightDirective,
        },
        OBJECT_3D_CONTROLLER_PROVIDER,
    ], exportAs: ["ngtHemisphereLight"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: HemisphereLightDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-hemisphere-light',
                    exportAs: 'ngtHemisphereLight',
                    providers: [
                        {
                            provide: ThreeObject3d,
                            useExisting: HemisphereLightDirective,
                        },
                        OBJECT_3D_CONTROLLER_PROVIDER,
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });
