// @angular-three/core/esm2015/textures/lib/data-texture3-d/data-texture3-d.directive.js
// GENERATED
import { ThreeTexture } from '@angular-three/core';
import { Directive, Input } from '@angular/core';
import { DataTexture3D } from 'three';
import * as i0 from "@angular/core";
export class DataTexture3DDirective extends ThreeTexture {
    constructor() {
        super(...arguments);
        this.textureType = DataTexture3D;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
DataTexture3DDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: DataTexture3DDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
DataTexture3DDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: DataTexture3DDirective, selector: "ngt-data-texture3-d", inputs: { args: "args" }, providers: [
        {
            provide: ThreeTexture,
            useExisting: DataTexture3DDirective,
        },
    ], exportAs: ["ngtDataTexture3D"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: DataTexture3DDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-data-texture3-d',
                    exportAs: 'ngtDataTexture3D',
                    providers: [
                        {
                            provide: ThreeTexture,
                            useExisting: DataTexture3DDirective,
                        },
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });
