// @angular-three/core/esm2015/textures/lib/data-texture/data-texture.directive.js
// GENERATED
import { ThreeTexture } from '@angular-three/core';
import { Directive, Input } from '@angular/core';
import { DataTexture } from 'three';
import * as i0 from "@angular/core";
export class DataTextureDirective extends ThreeTexture {
    constructor() {
        super(...arguments);
        this.textureType = DataTexture;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
DataTextureDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: DataTextureDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
DataTextureDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: DataTextureDirective, selector: "ngt-data-texture", inputs: { args: "args" }, providers: [
        {
            provide: ThreeTexture,
            useExisting: DataTextureDirective,
        },
    ], exportAs: ["ngtDataTexture"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: DataTextureDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-data-texture',
                    exportAs: 'ngtDataTexture',
                    providers: [
                        {
                            provide: ThreeTexture,
                            useExisting: DataTextureDirective,
                        },
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });
