// @angular-three/core/fesm2015/angular-three-core-cameras.js
import { ThreeObject3d, OBJECT_3D_CONTROLLER_PROVIDER, ThreeCamera } from '@angular-three/core';
import * as i0 from '@angular/core';
import { Directive, Input, NgModule } from '@angular/core';
import { CubeCamera, Camera, PerspectiveCamera, OrthographicCamera, ArrayCamera, StereoCamera } from 'three';

class CubeCameraDirective extends ThreeObject3d {
    ngOnInit() {
        this.init();
    }
    initObject() {
        this._camera = new CubeCamera(...this.args);
    }
    get object3d() {
        return this._camera;
    }
}
CubeCameraDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: CubeCameraDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
CubeCameraDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: CubeCameraDirective, selector: "ngt-cube-camera", inputs: { args: "args" }, providers: [
        {
            provide: ThreeObject3d,
            useExisting: CubeCameraDirective,
        },
        OBJECT_3D_CONTROLLER_PROVIDER,
    ], exportAs: ["ngtCubeCamera"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: CubeCameraDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-cube-camera',
                    exportAs: 'ngtCubeCamera',
                    providers: [
                        {
                            provide: ThreeObject3d,
                            useExisting: CubeCameraDirective,
                        },
                        OBJECT_3D_CONTROLLER_PROVIDER,
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });

class ThreeCubeCameraModule {
}
ThreeCubeCameraModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeCubeCameraModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ThreeCubeCameraModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeCubeCameraModule, declarations: [CubeCameraDirective], exports: [CubeCameraDirective] });
ThreeCubeCameraModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeCubeCameraModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeCubeCameraModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [CubeCameraDirective],
                    exports: [CubeCameraDirective],
                }]
        }] });

// GENERATED
class CameraDirective extends ThreeCamera {
    constructor() {
        super(...arguments);
        this.cameraType = Camera;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
CameraDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: CameraDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
CameraDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: CameraDirective, selector: "ngt-camera", inputs: { args: "args" }, providers: [
        {
            provide: ThreeObject3d,
            useExisting: CameraDirective,
        },
        OBJECT_3D_CONTROLLER_PROVIDER,
    ], exportAs: ["ngtCamera"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: CameraDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-camera',
                    exportAs: 'ngtCamera',
                    providers: [
                        {
                            provide: ThreeObject3d,
                            useExisting: CameraDirective,
                        },
                        OBJECT_3D_CONTROLLER_PROVIDER,
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });

// GENERATED
class ThreeCameraModule {
}
ThreeCameraModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeCameraModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ThreeCameraModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeCameraModule, declarations: [CameraDirective], exports: [CameraDirective] });
ThreeCameraModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeCameraModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeCameraModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [CameraDirective],
                    exports: [CameraDirective],
                }]
        }] });

// GENERATED

// GENERATED
class PerspectiveCameraDirective extends ThreeCamera {
    constructor() {
        super(...arguments);
        this.cameraType = PerspectiveCamera;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
PerspectiveCameraDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: PerspectiveCameraDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
PerspectiveCameraDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: PerspectiveCameraDirective, selector: "ngt-perspective-camera", inputs: { args: "args" }, providers: [
        {
            provide: ThreeObject3d,
            useExisting: PerspectiveCameraDirective,
        },
        OBJECT_3D_CONTROLLER_PROVIDER,
    ], exportAs: ["ngtPerspectiveCamera"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: PerspectiveCameraDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-perspective-camera',
                    exportAs: 'ngtPerspectiveCamera',
                    providers: [
                        {
                            provide: ThreeObject3d,
                            useExisting: PerspectiveCameraDirective,
                        },
                        OBJECT_3D_CONTROLLER_PROVIDER,
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });

// GENERATED
class ThreePerspectiveCameraModule {
}
ThreePerspectiveCameraModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreePerspectiveCameraModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ThreePerspectiveCameraModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreePerspectiveCameraModule, declarations: [PerspectiveCameraDirective], exports: [PerspectiveCameraDirective] });
ThreePerspectiveCameraModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreePerspectiveCameraModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreePerspectiveCameraModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [PerspectiveCameraDirective],
                    exports: [PerspectiveCameraDirective],
                }]
        }] });

// GENERATED

// GENERATED
class OrthographicCameraDirective extends ThreeCamera {
    constructor() {
        super(...arguments);
        this.cameraType = OrthographicCamera;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
OrthographicCameraDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: OrthographicCameraDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
OrthographicCameraDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: OrthographicCameraDirective, selector: "ngt-orthographic-camera", inputs: { args: "args" }, providers: [
        {
            provide: ThreeObject3d,
            useExisting: OrthographicCameraDirective,
        },
        OBJECT_3D_CONTROLLER_PROVIDER,
    ], exportAs: ["ngtOrthographicCamera"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: OrthographicCameraDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-orthographic-camera',
                    exportAs: 'ngtOrthographicCamera',
                    providers: [
                        {
                            provide: ThreeObject3d,
                            useExisting: OrthographicCameraDirective,
                        },
                        OBJECT_3D_CONTROLLER_PROVIDER,
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });

// GENERATED
class ThreeOrthographicCameraModule {
}
ThreeOrthographicCameraModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeOrthographicCameraModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ThreeOrthographicCameraModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeOrthographicCameraModule, declarations: [OrthographicCameraDirective], exports: [OrthographicCameraDirective] });
ThreeOrthographicCameraModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeOrthographicCameraModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeOrthographicCameraModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [OrthographicCameraDirective],
                    exports: [OrthographicCameraDirective],
                }]
        }] });

// GENERATED

// GENERATED
class ArrayCameraDirective extends ThreeCamera {
    constructor() {
        super(...arguments);
        this.cameraType = ArrayCamera;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
ArrayCameraDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ArrayCameraDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
ArrayCameraDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: ArrayCameraDirective, selector: "ngt-array-camera", inputs: { args: "args" }, providers: [
        {
            provide: ThreeObject3d,
            useExisting: ArrayCameraDirective,
        },
        OBJECT_3D_CONTROLLER_PROVIDER,
    ], exportAs: ["ngtArrayCamera"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ArrayCameraDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-array-camera',
                    exportAs: 'ngtArrayCamera',
                    providers: [
                        {
                            provide: ThreeObject3d,
                            useExisting: ArrayCameraDirective,
                        },
                        OBJECT_3D_CONTROLLER_PROVIDER,
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });

// GENERATED
class ThreeArrayCameraModule {
}
ThreeArrayCameraModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeArrayCameraModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ThreeArrayCameraModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeArrayCameraModule, declarations: [ArrayCameraDirective], exports: [ArrayCameraDirective] });
ThreeArrayCameraModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeArrayCameraModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeArrayCameraModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [ArrayCameraDirective],
                    exports: [ArrayCameraDirective],
                }]
        }] });

// GENERATED

// GENERATED
class StereoCameraDirective extends ThreeCamera {
    constructor() {
        super(...arguments);
        this.cameraType = StereoCamera;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
StereoCameraDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: StereoCameraDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
StereoCameraDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: StereoCameraDirective, selector: "ngt-stereo-camera", inputs: { args: "args" }, providers: [
        {
            provide: ThreeObject3d,
            useExisting: StereoCameraDirective,
        },
        OBJECT_3D_CONTROLLER_PROVIDER,
    ], exportAs: ["ngtStereoCamera"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: StereoCameraDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-stereo-camera',
                    exportAs: 'ngtStereoCamera',
                    providers: [
                        {
                            provide: ThreeObject3d,
                            useExisting: StereoCameraDirective,
                        },
                        OBJECT_3D_CONTROLLER_PROVIDER,
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });

// GENERATED
class ThreeStereoCameraModule {
}
ThreeStereoCameraModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeStereoCameraModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ThreeStereoCameraModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeStereoCameraModule, declarations: [StereoCameraDirective], exports: [StereoCameraDirective] });
ThreeStereoCameraModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeStereoCameraModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeStereoCameraModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [StereoCameraDirective],
                    exports: [StereoCameraDirective],
                }]
        }] });

// GENERATED

// GENERATED

/**
 * Generated bundle index. Do not edit.
 */

export { ArrayCameraDirective, CameraDirective, CubeCameraDirective, OrthographicCameraDirective, PerspectiveCameraDirective, StereoCameraDirective, ThreeArrayCameraModule, ThreeCameraModule, ThreeCubeCameraModule, ThreeOrthographicCameraModule, ThreePerspectiveCameraModule, ThreeStereoCameraModule };
