// @angular-three/core/fesm2015/angular-three-core-materials.js
import { ThreeMaterial } from '@angular-three/core';
import * as i0 from '@angular/core';
import { Directive, NgModule } from '@angular/core';
import { ShadowMaterial, SpriteMaterial, RawShaderMaterial, ShaderMaterial, PointsMaterial, MeshPhysicalMaterial, MeshStandardMaterial, MeshPhongMaterial, MeshToonMaterial, MeshNormalMaterial, MeshLambertMaterial, MeshDepthMaterial, MeshDistanceMaterial, MeshBasicMaterial, MeshMatcapMaterial, LineDashedMaterial, LineBasicMaterial } from 'three';

// GENERATED
class ShadowMaterialDirective extends ThreeMaterial {
    constructor() {
        super(...arguments);
        this.materialType = ShadowMaterial;
    }
}
ShadowMaterialDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ShadowMaterialDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
ShadowMaterialDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: ShadowMaterialDirective, selector: "ngt-shadow-material", providers: [{ provide: ThreeMaterial, useExisting: ShadowMaterialDirective }], exportAs: ["ngtShadowMaterial"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ShadowMaterialDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-shadow-material',
                    exportAs: 'ngtShadowMaterial',
                    providers: [{ provide: ThreeMaterial, useExisting: ShadowMaterialDirective }],
                }]
        }] });

// GENERATED
class ThreeShadowMaterialModule {
}
ThreeShadowMaterialModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeShadowMaterialModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ThreeShadowMaterialModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeShadowMaterialModule, declarations: [ShadowMaterialDirective], exports: [ShadowMaterialDirective] });
ThreeShadowMaterialModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeShadowMaterialModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeShadowMaterialModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [ShadowMaterialDirective],
                    exports: [ShadowMaterialDirective],
                }]
        }] });

// GENERATED

// GENERATED
class SpriteMaterialDirective extends ThreeMaterial {
    constructor() {
        super(...arguments);
        this.materialType = SpriteMaterial;
    }
}
SpriteMaterialDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: SpriteMaterialDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
SpriteMaterialDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: SpriteMaterialDirective, selector: "ngt-sprite-material", providers: [{ provide: ThreeMaterial, useExisting: SpriteMaterialDirective }], exportAs: ["ngtSpriteMaterial"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: SpriteMaterialDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-sprite-material',
                    exportAs: 'ngtSpriteMaterial',
                    providers: [{ provide: ThreeMaterial, useExisting: SpriteMaterialDirective }],
                }]
        }] });

// GENERATED
class ThreeSpriteMaterialModule {
}
ThreeSpriteMaterialModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeSpriteMaterialModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ThreeSpriteMaterialModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeSpriteMaterialModule, declarations: [SpriteMaterialDirective], exports: [SpriteMaterialDirective] });
ThreeSpriteMaterialModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeSpriteMaterialModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeSpriteMaterialModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [SpriteMaterialDirective],
                    exports: [SpriteMaterialDirective],
                }]
        }] });

// GENERATED

// GENERATED
class RawShaderMaterialDirective extends ThreeMaterial {
    constructor() {
        super(...arguments);
        this.materialType = RawShaderMaterial;
    }
}
RawShaderMaterialDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: RawShaderMaterialDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
RawShaderMaterialDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: RawShaderMaterialDirective, selector: "ngt-raw-shader-material", providers: [
        { provide: ThreeMaterial, useExisting: RawShaderMaterialDirective },
    ], exportAs: ["ngtRawShaderMaterial"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: RawShaderMaterialDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-raw-shader-material',
                    exportAs: 'ngtRawShaderMaterial',
                    providers: [
                        { provide: ThreeMaterial, useExisting: RawShaderMaterialDirective },
                    ],
                }]
        }] });

// GENERATED
class ThreeRawShaderMaterialModule {
}
ThreeRawShaderMaterialModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeRawShaderMaterialModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ThreeRawShaderMaterialModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeRawShaderMaterialModule, declarations: [RawShaderMaterialDirective], exports: [RawShaderMaterialDirective] });
ThreeRawShaderMaterialModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeRawShaderMaterialModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeRawShaderMaterialModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [RawShaderMaterialDirective],
                    exports: [RawShaderMaterialDirective],
                }]
        }] });

// GENERATED

// GENERATED
class ShaderMaterialDirective extends ThreeMaterial {
    constructor() {
        super(...arguments);
        this.materialType = ShaderMaterial;
    }
}
ShaderMaterialDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ShaderMaterialDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
ShaderMaterialDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: ShaderMaterialDirective, selector: "ngt-shader-material", providers: [{ provide: ThreeMaterial, useExisting: ShaderMaterialDirective }], exportAs: ["ngtShaderMaterial"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ShaderMaterialDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-shader-material',
                    exportAs: 'ngtShaderMaterial',
                    providers: [{ provide: ThreeMaterial, useExisting: ShaderMaterialDirective }],
                }]
        }] });

// GENERATED
class ThreeShaderMaterialModule {
}
ThreeShaderMaterialModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeShaderMaterialModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ThreeShaderMaterialModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeShaderMaterialModule, declarations: [ShaderMaterialDirective], exports: [ShaderMaterialDirective] });
ThreeShaderMaterialModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeShaderMaterialModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeShaderMaterialModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [ShaderMaterialDirective],
                    exports: [ShaderMaterialDirective],
                }]
        }] });

// GENERATED

// GENERATED
class PointsMaterialDirective extends ThreeMaterial {
    constructor() {
        super(...arguments);
        this.materialType = PointsMaterial;
    }
}
PointsMaterialDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: PointsMaterialDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
PointsMaterialDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: PointsMaterialDirective, selector: "ngt-points-material", providers: [{ provide: ThreeMaterial, useExisting: PointsMaterialDirective }], exportAs: ["ngtPointsMaterial"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: PointsMaterialDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-points-material',
                    exportAs: 'ngtPointsMaterial',
                    providers: [{ provide: ThreeMaterial, useExisting: PointsMaterialDirective }],
                }]
        }] });

// GENERATED
class ThreePointsMaterialModule {
}
ThreePointsMaterialModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreePointsMaterialModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ThreePointsMaterialModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreePointsMaterialModule, declarations: [PointsMaterialDirective], exports: [PointsMaterialDirective] });
ThreePointsMaterialModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreePointsMaterialModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreePointsMaterialModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [PointsMaterialDirective],
                    exports: [PointsMaterialDirective],
                }]
        }] });

// GENERATED

// GENERATED
class MeshPhysicalMaterialDirective extends ThreeMaterial {
    constructor() {
        super(...arguments);
        this.materialType = MeshPhysicalMaterial;
    }
}
MeshPhysicalMaterialDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: MeshPhysicalMaterialDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
MeshPhysicalMaterialDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: MeshPhysicalMaterialDirective, selector: "ngt-mesh-physical-material", providers: [
        { provide: ThreeMaterial, useExisting: MeshPhysicalMaterialDirective },
    ], exportAs: ["ngtMeshPhysicalMaterial"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: MeshPhysicalMaterialDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-mesh-physical-material',
                    exportAs: 'ngtMeshPhysicalMaterial',
                    providers: [
                        { provide: ThreeMaterial, useExisting: MeshPhysicalMaterialDirective },
                    ],
                }]
        }] });

// GENERATED
class ThreeMeshPhysicalMaterialModule {
}
ThreeMeshPhysicalMaterialModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeMeshPhysicalMaterialModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ThreeMeshPhysicalMaterialModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeMeshPhysicalMaterialModule, declarations: [MeshPhysicalMaterialDirective], exports: [MeshPhysicalMaterialDirective] });
ThreeMeshPhysicalMaterialModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeMeshPhysicalMaterialModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeMeshPhysicalMaterialModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [MeshPhysicalMaterialDirective],
                    exports: [MeshPhysicalMaterialDirective],
                }]
        }] });

// GENERATED

// GENERATED
class MeshStandardMaterialDirective extends ThreeMaterial {
    constructor() {
        super(...arguments);
        this.materialType = MeshStandardMaterial;
    }
}
MeshStandardMaterialDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: MeshStandardMaterialDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
MeshStandardMaterialDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: MeshStandardMaterialDirective, selector: "ngt-mesh-standard-material", providers: [
        { provide: ThreeMaterial, useExisting: MeshStandardMaterialDirective },
    ], exportAs: ["ngtMeshStandardMaterial"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: MeshStandardMaterialDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-mesh-standard-material',
                    exportAs: 'ngtMeshStandardMaterial',
                    providers: [
                        { provide: ThreeMaterial, useExisting: MeshStandardMaterialDirective },
                    ],
                }]
        }] });

// GENERATED
class ThreeMeshStandardMaterialModule {
}
ThreeMeshStandardMaterialModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeMeshStandardMaterialModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ThreeMeshStandardMaterialModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeMeshStandardMaterialModule, declarations: [MeshStandardMaterialDirective], exports: [MeshStandardMaterialDirective] });
ThreeMeshStandardMaterialModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeMeshStandardMaterialModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeMeshStandardMaterialModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [MeshStandardMaterialDirective],
                    exports: [MeshStandardMaterialDirective],
                }]
        }] });

// GENERATED

// GENERATED
class MeshPhongMaterialDirective extends ThreeMaterial {
    constructor() {
        super(...arguments);
        this.materialType = MeshPhongMaterial;
    }
}
MeshPhongMaterialDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: MeshPhongMaterialDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
MeshPhongMaterialDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: MeshPhongMaterialDirective, selector: "ngt-mesh-phong-material", providers: [
        { provide: ThreeMaterial, useExisting: MeshPhongMaterialDirective },
    ], exportAs: ["ngtMeshPhongMaterial"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: MeshPhongMaterialDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-mesh-phong-material',
                    exportAs: 'ngtMeshPhongMaterial',
                    providers: [
                        { provide: ThreeMaterial, useExisting: MeshPhongMaterialDirective },
                    ],
                }]
        }] });

// GENERATED
class ThreeMeshPhongMaterialModule {
}
ThreeMeshPhongMaterialModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeMeshPhongMaterialModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ThreeMeshPhongMaterialModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeMeshPhongMaterialModule, declarations: [MeshPhongMaterialDirective], exports: [MeshPhongMaterialDirective] });
ThreeMeshPhongMaterialModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeMeshPhongMaterialModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeMeshPhongMaterialModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [MeshPhongMaterialDirective],
                    exports: [MeshPhongMaterialDirective],
                }]
        }] });

// GENERATED

// GENERATED
class MeshToonMaterialDirective extends ThreeMaterial {
    constructor() {
        super(...arguments);
        this.materialType = MeshToonMaterial;
    }
}
MeshToonMaterialDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: MeshToonMaterialDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
MeshToonMaterialDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: MeshToonMaterialDirective, selector: "ngt-mesh-toon-material", providers: [
        { provide: ThreeMaterial, useExisting: MeshToonMaterialDirective },
    ], exportAs: ["ngtMeshToonMaterial"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: MeshToonMaterialDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-mesh-toon-material',
                    exportAs: 'ngtMeshToonMaterial',
                    providers: [
                        { provide: ThreeMaterial, useExisting: MeshToonMaterialDirective },
                    ],
                }]
        }] });

// GENERATED
class ThreeMeshToonMaterialModule {
}
ThreeMeshToonMaterialModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeMeshToonMaterialModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ThreeMeshToonMaterialModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeMeshToonMaterialModule, declarations: [MeshToonMaterialDirective], exports: [MeshToonMaterialDirective] });
ThreeMeshToonMaterialModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeMeshToonMaterialModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeMeshToonMaterialModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [MeshToonMaterialDirective],
                    exports: [MeshToonMaterialDirective],
                }]
        }] });

// GENERATED

// GENERATED
class MeshNormalMaterialDirective extends ThreeMaterial {
    constructor() {
        super(...arguments);
        this.materialType = MeshNormalMaterial;
    }
}
MeshNormalMaterialDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: MeshNormalMaterialDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
MeshNormalMaterialDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: MeshNormalMaterialDirective, selector: "ngt-mesh-normal-material", providers: [
        { provide: ThreeMaterial, useExisting: MeshNormalMaterialDirective },
    ], exportAs: ["ngtMeshNormalMaterial"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: MeshNormalMaterialDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-mesh-normal-material',
                    exportAs: 'ngtMeshNormalMaterial',
                    providers: [
                        { provide: ThreeMaterial, useExisting: MeshNormalMaterialDirective },
                    ],
                }]
        }] });

// GENERATED
class ThreeMeshNormalMaterialModule {
}
ThreeMeshNormalMaterialModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeMeshNormalMaterialModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ThreeMeshNormalMaterialModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeMeshNormalMaterialModule, declarations: [MeshNormalMaterialDirective], exports: [MeshNormalMaterialDirective] });
ThreeMeshNormalMaterialModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeMeshNormalMaterialModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeMeshNormalMaterialModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [MeshNormalMaterialDirective],
                    exports: [MeshNormalMaterialDirective],
                }]
        }] });

// GENERATED

// GENERATED
class MeshLambertMaterialDirective extends ThreeMaterial {
    constructor() {
        super(...arguments);
        this.materialType = MeshLambertMaterial;
    }
}
MeshLambertMaterialDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: MeshLambertMaterialDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
MeshLambertMaterialDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: MeshLambertMaterialDirective, selector: "ngt-mesh-lambert-material", providers: [
        { provide: ThreeMaterial, useExisting: MeshLambertMaterialDirective },
    ], exportAs: ["ngtMeshLambertMaterial"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: MeshLambertMaterialDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-mesh-lambert-material',
                    exportAs: 'ngtMeshLambertMaterial',
                    providers: [
                        { provide: ThreeMaterial, useExisting: MeshLambertMaterialDirective },
                    ],
                }]
        }] });

// GENERATED
class ThreeMeshLambertMaterialModule {
}
ThreeMeshLambertMaterialModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeMeshLambertMaterialModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ThreeMeshLambertMaterialModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeMeshLambertMaterialModule, declarations: [MeshLambertMaterialDirective], exports: [MeshLambertMaterialDirective] });
ThreeMeshLambertMaterialModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeMeshLambertMaterialModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeMeshLambertMaterialModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [MeshLambertMaterialDirective],
                    exports: [MeshLambertMaterialDirective],
                }]
        }] });

// GENERATED

// GENERATED
class MeshDepthMaterialDirective extends ThreeMaterial {
    constructor() {
        super(...arguments);
        this.materialType = MeshDepthMaterial;
    }
}
MeshDepthMaterialDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: MeshDepthMaterialDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
MeshDepthMaterialDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: MeshDepthMaterialDirective, selector: "ngt-mesh-depth-material", providers: [
        { provide: ThreeMaterial, useExisting: MeshDepthMaterialDirective },
    ], exportAs: ["ngtMeshDepthMaterial"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: MeshDepthMaterialDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-mesh-depth-material',
                    exportAs: 'ngtMeshDepthMaterial',
                    providers: [
                        { provide: ThreeMaterial, useExisting: MeshDepthMaterialDirective },
                    ],
                }]
        }] });

// GENERATED
class ThreeMeshDepthMaterialModule {
}
ThreeMeshDepthMaterialModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeMeshDepthMaterialModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ThreeMeshDepthMaterialModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeMeshDepthMaterialModule, declarations: [MeshDepthMaterialDirective], exports: [MeshDepthMaterialDirective] });
ThreeMeshDepthMaterialModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeMeshDepthMaterialModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeMeshDepthMaterialModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [MeshDepthMaterialDirective],
                    exports: [MeshDepthMaterialDirective],
                }]
        }] });

// GENERATED

// GENERATED
class MeshDistanceMaterialDirective extends ThreeMaterial {
    constructor() {
        super(...arguments);
        this.materialType = MeshDistanceMaterial;
    }
}
MeshDistanceMaterialDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: MeshDistanceMaterialDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
MeshDistanceMaterialDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: MeshDistanceMaterialDirective, selector: "ngt-mesh-distance-material", providers: [
        { provide: ThreeMaterial, useExisting: MeshDistanceMaterialDirective },
    ], exportAs: ["ngtMeshDistanceMaterial"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: MeshDistanceMaterialDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-mesh-distance-material',
                    exportAs: 'ngtMeshDistanceMaterial',
                    providers: [
                        { provide: ThreeMaterial, useExisting: MeshDistanceMaterialDirective },
                    ],
                }]
        }] });

// GENERATED
class ThreeMeshDistanceMaterialModule {
}
ThreeMeshDistanceMaterialModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeMeshDistanceMaterialModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ThreeMeshDistanceMaterialModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeMeshDistanceMaterialModule, declarations: [MeshDistanceMaterialDirective], exports: [MeshDistanceMaterialDirective] });
ThreeMeshDistanceMaterialModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeMeshDistanceMaterialModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeMeshDistanceMaterialModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [MeshDistanceMaterialDirective],
                    exports: [MeshDistanceMaterialDirective],
                }]
        }] });

// GENERATED

// GENERATED
class MeshBasicMaterialDirective extends ThreeMaterial {
    constructor() {
        super(...arguments);
        this.materialType = MeshBasicMaterial;
    }
}
MeshBasicMaterialDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: MeshBasicMaterialDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
MeshBasicMaterialDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: MeshBasicMaterialDirective, selector: "ngt-mesh-basic-material", providers: [
        { provide: ThreeMaterial, useExisting: MeshBasicMaterialDirective },
    ], exportAs: ["ngtMeshBasicMaterial"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: MeshBasicMaterialDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-mesh-basic-material',
                    exportAs: 'ngtMeshBasicMaterial',
                    providers: [
                        { provide: ThreeMaterial, useExisting: MeshBasicMaterialDirective },
                    ],
                }]
        }] });

// GENERATED
class ThreeMeshBasicMaterialModule {
}
ThreeMeshBasicMaterialModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeMeshBasicMaterialModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ThreeMeshBasicMaterialModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeMeshBasicMaterialModule, declarations: [MeshBasicMaterialDirective], exports: [MeshBasicMaterialDirective] });
ThreeMeshBasicMaterialModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeMeshBasicMaterialModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeMeshBasicMaterialModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [MeshBasicMaterialDirective],
                    exports: [MeshBasicMaterialDirective],
                }]
        }] });

// GENERATED

// GENERATED
class MeshMatcapMaterialDirective extends ThreeMaterial {
    constructor() {
        super(...arguments);
        this.materialType = MeshMatcapMaterial;
    }
}
MeshMatcapMaterialDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: MeshMatcapMaterialDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
MeshMatcapMaterialDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: MeshMatcapMaterialDirective, selector: "ngt-mesh-matcap-material", providers: [
        { provide: ThreeMaterial, useExisting: MeshMatcapMaterialDirective },
    ], exportAs: ["ngtMeshMatcapMaterial"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: MeshMatcapMaterialDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-mesh-matcap-material',
                    exportAs: 'ngtMeshMatcapMaterial',
                    providers: [
                        { provide: ThreeMaterial, useExisting: MeshMatcapMaterialDirective },
                    ],
                }]
        }] });

// GENERATED
class ThreeMeshMatcapMaterialModule {
}
ThreeMeshMatcapMaterialModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeMeshMatcapMaterialModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ThreeMeshMatcapMaterialModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeMeshMatcapMaterialModule, declarations: [MeshMatcapMaterialDirective], exports: [MeshMatcapMaterialDirective] });
ThreeMeshMatcapMaterialModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeMeshMatcapMaterialModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeMeshMatcapMaterialModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [MeshMatcapMaterialDirective],
                    exports: [MeshMatcapMaterialDirective],
                }]
        }] });

// GENERATED

// GENERATED
class LineDashedMaterialDirective extends ThreeMaterial {
    constructor() {
        super(...arguments);
        this.materialType = LineDashedMaterial;
    }
}
LineDashedMaterialDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: LineDashedMaterialDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
LineDashedMaterialDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: LineDashedMaterialDirective, selector: "ngt-line-dashed-material", providers: [
        { provide: ThreeMaterial, useExisting: LineDashedMaterialDirective },
    ], exportAs: ["ngtLineDashedMaterial"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: LineDashedMaterialDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-line-dashed-material',
                    exportAs: 'ngtLineDashedMaterial',
                    providers: [
                        { provide: ThreeMaterial, useExisting: LineDashedMaterialDirective },
                    ],
                }]
        }] });

// GENERATED
class ThreeLineDashedMaterialModule {
}
ThreeLineDashedMaterialModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeLineDashedMaterialModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ThreeLineDashedMaterialModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeLineDashedMaterialModule, declarations: [LineDashedMaterialDirective], exports: [LineDashedMaterialDirective] });
ThreeLineDashedMaterialModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeLineDashedMaterialModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeLineDashedMaterialModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [LineDashedMaterialDirective],
                    exports: [LineDashedMaterialDirective],
                }]
        }] });

// GENERATED

// GENERATED
class LineBasicMaterialDirective extends ThreeMaterial {
    constructor() {
        super(...arguments);
        this.materialType = LineBasicMaterial;
    }
}
LineBasicMaterialDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: LineBasicMaterialDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
LineBasicMaterialDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: LineBasicMaterialDirective, selector: "ngt-line-basic-material", providers: [
        { provide: ThreeMaterial, useExisting: LineBasicMaterialDirective },
    ], exportAs: ["ngtLineBasicMaterial"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: LineBasicMaterialDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-line-basic-material',
                    exportAs: 'ngtLineBasicMaterial',
                    providers: [
                        { provide: ThreeMaterial, useExisting: LineBasicMaterialDirective },
                    ],
                }]
        }] });

// GENERATED
class ThreeLineBasicMaterialModule {
}
ThreeLineBasicMaterialModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeLineBasicMaterialModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ThreeLineBasicMaterialModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeLineBasicMaterialModule, declarations: [LineBasicMaterialDirective], exports: [LineBasicMaterialDirective] });
ThreeLineBasicMaterialModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeLineBasicMaterialModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeLineBasicMaterialModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [LineBasicMaterialDirective],
                    exports: [LineBasicMaterialDirective],
                }]
        }] });

// GENERATED

// GENERATED

/**
 * Generated bundle index. Do not edit.
 */

export { LineBasicMaterialDirective, LineDashedMaterialDirective, MeshBasicMaterialDirective, MeshDepthMaterialDirective, MeshDistanceMaterialDirective, MeshLambertMaterialDirective, MeshMatcapMaterialDirective, MeshNormalMaterialDirective, MeshPhongMaterialDirective, MeshPhysicalMaterialDirective, MeshStandardMaterialDirective, MeshToonMaterialDirective, PointsMaterialDirective, RawShaderMaterialDirective, ShaderMaterialDirective, ShadowMaterialDirective, SpriteMaterialDirective, ThreeLineBasicMaterialModule, ThreeLineDashedMaterialModule, ThreeMeshBasicMaterialModule, ThreeMeshDepthMaterialModule, ThreeMeshDistanceMaterialModule, ThreeMeshLambertMaterialModule, ThreeMeshMatcapMaterialModule, ThreeMeshNormalMaterialModule, ThreeMeshPhongMaterialModule, ThreeMeshPhysicalMaterialModule, ThreeMeshStandardMaterialModule, ThreeMeshToonMaterialModule, ThreePointsMaterialModule, ThreeRawShaderMaterialModule, ThreeShaderMaterialModule, ThreeShadowMaterialModule, ThreeSpriteMaterialModule };
