// @angular-three/core/fesm2015/angular-three-core-lights.js
import { ThreeLight, ThreeObject3d, OBJECT_3D_CONTROLLER_PROVIDER } from '@angular-three/core';
import * as i0 from '@angular/core';
import { Directive, Input, NgModule } from '@angular/core';
import { LightProbe, AmbientLight, AmbientLightProbe, HemisphereLight, HemisphereLightProbe, DirectionalLight, PointLight, SpotLight, RectAreaLight } from 'three';

// GENERATED
class LightProbeDirective extends ThreeLight {
    constructor() {
        super(...arguments);
        this.lightType = LightProbe;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
LightProbeDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: LightProbeDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
LightProbeDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: LightProbeDirective, selector: "ngt-light-probe", inputs: { args: "args" }, providers: [
        {
            provide: ThreeObject3d,
            useExisting: LightProbeDirective,
        },
        OBJECT_3D_CONTROLLER_PROVIDER,
    ], exportAs: ["ngtLightProbe"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: LightProbeDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-light-probe',
                    exportAs: 'ngtLightProbe',
                    providers: [
                        {
                            provide: ThreeObject3d,
                            useExisting: LightProbeDirective,
                        },
                        OBJECT_3D_CONTROLLER_PROVIDER,
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });

// GENERATED
class ThreeLightProbeModule {
}
ThreeLightProbeModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeLightProbeModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ThreeLightProbeModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeLightProbeModule, declarations: [LightProbeDirective], exports: [LightProbeDirective] });
ThreeLightProbeModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeLightProbeModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeLightProbeModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [LightProbeDirective],
                    exports: [LightProbeDirective],
                }]
        }] });

// GENERATED

// GENERATED
class AmbientLightDirective extends ThreeLight {
    constructor() {
        super(...arguments);
        this.lightType = AmbientLight;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
AmbientLightDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: AmbientLightDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
AmbientLightDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: AmbientLightDirective, selector: "ngt-ambient-light", inputs: { args: "args" }, providers: [
        {
            provide: ThreeObject3d,
            useExisting: AmbientLightDirective,
        },
        OBJECT_3D_CONTROLLER_PROVIDER,
    ], exportAs: ["ngtAmbientLight"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: AmbientLightDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-ambient-light',
                    exportAs: 'ngtAmbientLight',
                    providers: [
                        {
                            provide: ThreeObject3d,
                            useExisting: AmbientLightDirective,
                        },
                        OBJECT_3D_CONTROLLER_PROVIDER,
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });

// GENERATED
class ThreeAmbientLightModule {
}
ThreeAmbientLightModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeAmbientLightModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ThreeAmbientLightModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeAmbientLightModule, declarations: [AmbientLightDirective], exports: [AmbientLightDirective] });
ThreeAmbientLightModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeAmbientLightModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeAmbientLightModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [AmbientLightDirective],
                    exports: [AmbientLightDirective],
                }]
        }] });

// GENERATED

// GENERATED
class AmbientLightProbeDirective extends ThreeLight {
    constructor() {
        super(...arguments);
        this.lightType = AmbientLightProbe;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
AmbientLightProbeDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: AmbientLightProbeDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
AmbientLightProbeDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: AmbientLightProbeDirective, selector: "ngt-ambient-light-probe", inputs: { args: "args" }, providers: [
        {
            provide: ThreeObject3d,
            useExisting: AmbientLightProbeDirective,
        },
        OBJECT_3D_CONTROLLER_PROVIDER,
    ], exportAs: ["ngtAmbientLightProbe"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: AmbientLightProbeDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-ambient-light-probe',
                    exportAs: 'ngtAmbientLightProbe',
                    providers: [
                        {
                            provide: ThreeObject3d,
                            useExisting: AmbientLightProbeDirective,
                        },
                        OBJECT_3D_CONTROLLER_PROVIDER,
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });

// GENERATED
class ThreeAmbientLightProbeModule {
}
ThreeAmbientLightProbeModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeAmbientLightProbeModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ThreeAmbientLightProbeModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeAmbientLightProbeModule, declarations: [AmbientLightProbeDirective], exports: [AmbientLightProbeDirective] });
ThreeAmbientLightProbeModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeAmbientLightProbeModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeAmbientLightProbeModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [AmbientLightProbeDirective],
                    exports: [AmbientLightProbeDirective],
                }]
        }] });

// GENERATED

// GENERATED
class HemisphereLightDirective extends ThreeLight {
    constructor() {
        super(...arguments);
        this.lightType = HemisphereLight;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
HemisphereLightDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: HemisphereLightDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
HemisphereLightDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: HemisphereLightDirective, selector: "ngt-hemisphere-light", inputs: { args: "args" }, providers: [
        {
            provide: ThreeObject3d,
            useExisting: HemisphereLightDirective,
        },
        OBJECT_3D_CONTROLLER_PROVIDER,
    ], exportAs: ["ngtHemisphereLight"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: HemisphereLightDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-hemisphere-light',
                    exportAs: 'ngtHemisphereLight',
                    providers: [
                        {
                            provide: ThreeObject3d,
                            useExisting: HemisphereLightDirective,
                        },
                        OBJECT_3D_CONTROLLER_PROVIDER,
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });

// GENERATED
class ThreeHemisphereLightModule {
}
ThreeHemisphereLightModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeHemisphereLightModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ThreeHemisphereLightModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeHemisphereLightModule, declarations: [HemisphereLightDirective], exports: [HemisphereLightDirective] });
ThreeHemisphereLightModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeHemisphereLightModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeHemisphereLightModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [HemisphereLightDirective],
                    exports: [HemisphereLightDirective],
                }]
        }] });

// GENERATED

// GENERATED
class HemisphereLightProbeDirective extends ThreeLight {
    constructor() {
        super(...arguments);
        this.lightType = HemisphereLightProbe;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
HemisphereLightProbeDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: HemisphereLightProbeDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
HemisphereLightProbeDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: HemisphereLightProbeDirective, selector: "ngt-hemisphere-light-probe", inputs: { args: "args" }, providers: [
        {
            provide: ThreeObject3d,
            useExisting: HemisphereLightProbeDirective,
        },
        OBJECT_3D_CONTROLLER_PROVIDER,
    ], exportAs: ["ngtHemisphereLightProbe"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: HemisphereLightProbeDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-hemisphere-light-probe',
                    exportAs: 'ngtHemisphereLightProbe',
                    providers: [
                        {
                            provide: ThreeObject3d,
                            useExisting: HemisphereLightProbeDirective,
                        },
                        OBJECT_3D_CONTROLLER_PROVIDER,
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });

// GENERATED
class ThreeHemisphereLightProbeModule {
}
ThreeHemisphereLightProbeModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeHemisphereLightProbeModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ThreeHemisphereLightProbeModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeHemisphereLightProbeModule, declarations: [HemisphereLightProbeDirective], exports: [HemisphereLightProbeDirective] });
ThreeHemisphereLightProbeModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeHemisphereLightProbeModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeHemisphereLightProbeModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [HemisphereLightProbeDirective],
                    exports: [HemisphereLightProbeDirective],
                }]
        }] });

// GENERATED

// GENERATED
class DirectionalLightDirective extends ThreeLight {
    constructor() {
        super(...arguments);
        this.lightType = DirectionalLight;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
DirectionalLightDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: DirectionalLightDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
DirectionalLightDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: DirectionalLightDirective, selector: "ngt-directional-light", inputs: { args: "args" }, providers: [
        {
            provide: ThreeObject3d,
            useExisting: DirectionalLightDirective,
        },
        OBJECT_3D_CONTROLLER_PROVIDER,
    ], exportAs: ["ngtDirectionalLight"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: DirectionalLightDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-directional-light',
                    exportAs: 'ngtDirectionalLight',
                    providers: [
                        {
                            provide: ThreeObject3d,
                            useExisting: DirectionalLightDirective,
                        },
                        OBJECT_3D_CONTROLLER_PROVIDER,
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });

// GENERATED
class ThreeDirectionalLightModule {
}
ThreeDirectionalLightModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeDirectionalLightModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ThreeDirectionalLightModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeDirectionalLightModule, declarations: [DirectionalLightDirective], exports: [DirectionalLightDirective] });
ThreeDirectionalLightModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeDirectionalLightModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeDirectionalLightModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [DirectionalLightDirective],
                    exports: [DirectionalLightDirective],
                }]
        }] });

// GENERATED

// GENERATED
class PointLightDirective extends ThreeLight {
    constructor() {
        super(...arguments);
        this.lightType = PointLight;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
PointLightDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: PointLightDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
PointLightDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: PointLightDirective, selector: "ngt-point-light", inputs: { args: "args" }, providers: [
        {
            provide: ThreeObject3d,
            useExisting: PointLightDirective,
        },
        OBJECT_3D_CONTROLLER_PROVIDER,
    ], exportAs: ["ngtPointLight"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: PointLightDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-point-light',
                    exportAs: 'ngtPointLight',
                    providers: [
                        {
                            provide: ThreeObject3d,
                            useExisting: PointLightDirective,
                        },
                        OBJECT_3D_CONTROLLER_PROVIDER,
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });

// GENERATED
class ThreePointLightModule {
}
ThreePointLightModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreePointLightModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ThreePointLightModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreePointLightModule, declarations: [PointLightDirective], exports: [PointLightDirective] });
ThreePointLightModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreePointLightModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreePointLightModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [PointLightDirective],
                    exports: [PointLightDirective],
                }]
        }] });

// GENERATED

// GENERATED
class SpotLightDirective extends ThreeLight {
    constructor() {
        super(...arguments);
        this.lightType = SpotLight;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
SpotLightDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: SpotLightDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
SpotLightDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: SpotLightDirective, selector: "ngt-spot-light", inputs: { args: "args" }, providers: [
        {
            provide: ThreeObject3d,
            useExisting: SpotLightDirective,
        },
        OBJECT_3D_CONTROLLER_PROVIDER,
    ], exportAs: ["ngtSpotLight"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: SpotLightDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-spot-light',
                    exportAs: 'ngtSpotLight',
                    providers: [
                        {
                            provide: ThreeObject3d,
                            useExisting: SpotLightDirective,
                        },
                        OBJECT_3D_CONTROLLER_PROVIDER,
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });

// GENERATED
class ThreeSpotLightModule {
}
ThreeSpotLightModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeSpotLightModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ThreeSpotLightModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeSpotLightModule, declarations: [SpotLightDirective], exports: [SpotLightDirective] });
ThreeSpotLightModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeSpotLightModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeSpotLightModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [SpotLightDirective],
                    exports: [SpotLightDirective],
                }]
        }] });

// GENERATED

// GENERATED
class RectAreaLightDirective extends ThreeLight {
    constructor() {
        super(...arguments);
        this.lightType = RectAreaLight;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
RectAreaLightDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: RectAreaLightDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
RectAreaLightDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: RectAreaLightDirective, selector: "ngt-rect-area-light", inputs: { args: "args" }, providers: [
        {
            provide: ThreeObject3d,
            useExisting: RectAreaLightDirective,
        },
        OBJECT_3D_CONTROLLER_PROVIDER,
    ], exportAs: ["ngtRectAreaLight"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: RectAreaLightDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-rect-area-light',
                    exportAs: 'ngtRectAreaLight',
                    providers: [
                        {
                            provide: ThreeObject3d,
                            useExisting: RectAreaLightDirective,
                        },
                        OBJECT_3D_CONTROLLER_PROVIDER,
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });

// GENERATED
class ThreeRectAreaLightModule {
}
ThreeRectAreaLightModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeRectAreaLightModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ThreeRectAreaLightModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeRectAreaLightModule, declarations: [RectAreaLightDirective], exports: [RectAreaLightDirective] });
ThreeRectAreaLightModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeRectAreaLightModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeRectAreaLightModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [RectAreaLightDirective],
                    exports: [RectAreaLightDirective],
                }]
        }] });

// GENERATED

// GENERATED

/**
 * Generated bundle index. Do not edit.
 */

export { AmbientLightDirective, AmbientLightProbeDirective, DirectionalLightDirective, HemisphereLightDirective, HemisphereLightProbeDirective, LightProbeDirective, PointLightDirective, RectAreaLightDirective, SpotLightDirective, ThreeAmbientLightModule, ThreeAmbientLightProbeModule, ThreeDirectionalLightModule, ThreeHemisphereLightModule, ThreeHemisphereLightProbeModule, ThreeLightProbeModule, ThreePointLightModule, ThreeRectAreaLightModule, ThreeSpotLightModule };
