// @angular-three/core/fesm2015/angular-three-core-stats.js
import * as i0 from '@angular/core';
import { Directive, Inject, NgModule } from '@angular/core';
import * as i1 from '@angular-three/core';
import { DOCUMENT } from '@angular/common';
import Stats from 'three/examples/jsm/libs/stats.module';

class StatsDirective {
    constructor(animationStore, document) {
        this.animationStore = animationStore;
        this.document = document;
    }
    ngOnInit() {
        this.stats = Stats();
        this.document.body.appendChild(this.stats.dom);
        this.animationStore.registerAnimation(() => {
            this.stats.update();
        });
    }
    ngOnDestroy() {
        this.stats.end();
    }
}
StatsDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: StatsDirective, deps: [{ token: i1.AnimationStore }, { token: DOCUMENT }], target: i0.ɵɵFactoryTarget.Directive });
StatsDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: StatsDirective, selector: "ngt-stats", exportAs: ["ngtStats"], ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: StatsDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-stats',
                    exportAs: 'ngtStats',
                }]
        }], ctorParameters: function () { return [{ type: i1.AnimationStore }, { type: Document, decorators: [{
                    type: Inject,
                    args: [DOCUMENT]
                }] }]; } });

class ThreeStatsModule {
}
ThreeStatsModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeStatsModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ThreeStatsModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeStatsModule, declarations: [StatsDirective], exports: [StatsDirective] });
ThreeStatsModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeStatsModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeStatsModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [StatsDirective],
                    exports: [StatsDirective],
                }]
        }] });

/**
 * Generated bundle index. Do not edit.
 */

export { StatsDirective, ThreeStatsModule };
