// @angular-three/core/fesm2015/angular-three-core-textures.js
import { ThreeTexture } from '@angular-three/core';
import * as i0 from '@angular/core';
import { Directive, Input, NgModule } from '@angular/core';
import { CanvasTexture, CompressedTexture, CubeTexture, DataTexture, DataTexture2DArray, DataTexture3D, DepthTexture, VideoTexture } from 'three';

// GENERATED
class CanvasTextureDirective extends ThreeTexture {
    constructor() {
        super(...arguments);
        this.textureType = CanvasTexture;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
CanvasTextureDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: CanvasTextureDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
CanvasTextureDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: CanvasTextureDirective, selector: "ngt-canvas-texture", inputs: { args: "args" }, providers: [
        {
            provide: ThreeTexture,
            useExisting: CanvasTextureDirective,
        },
    ], exportAs: ["ngtCanvasTexture"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: CanvasTextureDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-canvas-texture',
                    exportAs: 'ngtCanvasTexture',
                    providers: [
                        {
                            provide: ThreeTexture,
                            useExisting: CanvasTextureDirective,
                        },
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });

// GENERATED
class ThreeCanvasTextureModule {
}
ThreeCanvasTextureModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeCanvasTextureModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ThreeCanvasTextureModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeCanvasTextureModule, declarations: [CanvasTextureDirective], exports: [CanvasTextureDirective] });
ThreeCanvasTextureModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeCanvasTextureModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeCanvasTextureModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [CanvasTextureDirective],
                    exports: [CanvasTextureDirective],
                }]
        }] });

// GENERATED

// GENERATED
class CompressedTextureDirective extends ThreeTexture {
    constructor() {
        super(...arguments);
        this.textureType = CompressedTexture;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
CompressedTextureDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: CompressedTextureDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
CompressedTextureDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: CompressedTextureDirective, selector: "ngt-compressed-texture", inputs: { args: "args" }, providers: [
        {
            provide: ThreeTexture,
            useExisting: CompressedTextureDirective,
        },
    ], exportAs: ["ngtCompressedTexture"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: CompressedTextureDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-compressed-texture',
                    exportAs: 'ngtCompressedTexture',
                    providers: [
                        {
                            provide: ThreeTexture,
                            useExisting: CompressedTextureDirective,
                        },
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });

// GENERATED
class ThreeCompressedTextureModule {
}
ThreeCompressedTextureModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeCompressedTextureModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ThreeCompressedTextureModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeCompressedTextureModule, declarations: [CompressedTextureDirective], exports: [CompressedTextureDirective] });
ThreeCompressedTextureModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeCompressedTextureModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeCompressedTextureModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [CompressedTextureDirective],
                    exports: [CompressedTextureDirective],
                }]
        }] });

// GENERATED

// GENERATED
class CubeTextureDirective extends ThreeTexture {
    constructor() {
        super(...arguments);
        this.textureType = CubeTexture;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
CubeTextureDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: CubeTextureDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
CubeTextureDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: CubeTextureDirective, selector: "ngt-cube-texture", inputs: { args: "args" }, providers: [
        {
            provide: ThreeTexture,
            useExisting: CubeTextureDirective,
        },
    ], exportAs: ["ngtCubeTexture"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: CubeTextureDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-cube-texture',
                    exportAs: 'ngtCubeTexture',
                    providers: [
                        {
                            provide: ThreeTexture,
                            useExisting: CubeTextureDirective,
                        },
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });

// GENERATED
class ThreeCubeTextureModule {
}
ThreeCubeTextureModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeCubeTextureModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ThreeCubeTextureModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeCubeTextureModule, declarations: [CubeTextureDirective], exports: [CubeTextureDirective] });
ThreeCubeTextureModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeCubeTextureModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeCubeTextureModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [CubeTextureDirective],
                    exports: [CubeTextureDirective],
                }]
        }] });

// GENERATED

// GENERATED
class DataTextureDirective extends ThreeTexture {
    constructor() {
        super(...arguments);
        this.textureType = DataTexture;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
DataTextureDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: DataTextureDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
DataTextureDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: DataTextureDirective, selector: "ngt-data-texture", inputs: { args: "args" }, providers: [
        {
            provide: ThreeTexture,
            useExisting: DataTextureDirective,
        },
    ], exportAs: ["ngtDataTexture"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: DataTextureDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-data-texture',
                    exportAs: 'ngtDataTexture',
                    providers: [
                        {
                            provide: ThreeTexture,
                            useExisting: DataTextureDirective,
                        },
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });

// GENERATED
class ThreeDataTextureModule {
}
ThreeDataTextureModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeDataTextureModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ThreeDataTextureModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeDataTextureModule, declarations: [DataTextureDirective], exports: [DataTextureDirective] });
ThreeDataTextureModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeDataTextureModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeDataTextureModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [DataTextureDirective],
                    exports: [DataTextureDirective],
                }]
        }] });

// GENERATED

// GENERATED
class DataTexture2DArrayDirective extends ThreeTexture {
    constructor() {
        super(...arguments);
        this.textureType = DataTexture2DArray;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
DataTexture2DArrayDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: DataTexture2DArrayDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
DataTexture2DArrayDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: DataTexture2DArrayDirective, selector: "ngt-data-texture2-darray", inputs: { args: "args" }, providers: [
        {
            provide: ThreeTexture,
            useExisting: DataTexture2DArrayDirective,
        },
    ], exportAs: ["ngtDataTexture2DArray"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: DataTexture2DArrayDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-data-texture2-darray',
                    exportAs: 'ngtDataTexture2DArray',
                    providers: [
                        {
                            provide: ThreeTexture,
                            useExisting: DataTexture2DArrayDirective,
                        },
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });

// GENERATED
class ThreeDataTexture2DArrayModule {
}
ThreeDataTexture2DArrayModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeDataTexture2DArrayModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ThreeDataTexture2DArrayModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeDataTexture2DArrayModule, declarations: [DataTexture2DArrayDirective], exports: [DataTexture2DArrayDirective] });
ThreeDataTexture2DArrayModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeDataTexture2DArrayModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeDataTexture2DArrayModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [DataTexture2DArrayDirective],
                    exports: [DataTexture2DArrayDirective],
                }]
        }] });

// GENERATED

// GENERATED
class DataTexture3DDirective extends ThreeTexture {
    constructor() {
        super(...arguments);
        this.textureType = DataTexture3D;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
DataTexture3DDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: DataTexture3DDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
DataTexture3DDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: DataTexture3DDirective, selector: "ngt-data-texture3-d", inputs: { args: "args" }, providers: [
        {
            provide: ThreeTexture,
            useExisting: DataTexture3DDirective,
        },
    ], exportAs: ["ngtDataTexture3D"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: DataTexture3DDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-data-texture3-d',
                    exportAs: 'ngtDataTexture3D',
                    providers: [
                        {
                            provide: ThreeTexture,
                            useExisting: DataTexture3DDirective,
                        },
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });

// GENERATED
class ThreeDataTexture3DModule {
}
ThreeDataTexture3DModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeDataTexture3DModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ThreeDataTexture3DModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeDataTexture3DModule, declarations: [DataTexture3DDirective], exports: [DataTexture3DDirective] });
ThreeDataTexture3DModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeDataTexture3DModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeDataTexture3DModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [DataTexture3DDirective],
                    exports: [DataTexture3DDirective],
                }]
        }] });

// GENERATED

// GENERATED
class DepthTextureDirective extends ThreeTexture {
    constructor() {
        super(...arguments);
        this.textureType = DepthTexture;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
DepthTextureDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: DepthTextureDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
DepthTextureDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: DepthTextureDirective, selector: "ngt-depth-texture", inputs: { args: "args" }, providers: [
        {
            provide: ThreeTexture,
            useExisting: DepthTextureDirective,
        },
    ], exportAs: ["ngtDepthTexture"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: DepthTextureDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-depth-texture',
                    exportAs: 'ngtDepthTexture',
                    providers: [
                        {
                            provide: ThreeTexture,
                            useExisting: DepthTextureDirective,
                        },
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });

// GENERATED
class ThreeDepthTextureModule {
}
ThreeDepthTextureModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeDepthTextureModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ThreeDepthTextureModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeDepthTextureModule, declarations: [DepthTextureDirective], exports: [DepthTextureDirective] });
ThreeDepthTextureModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeDepthTextureModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeDepthTextureModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [DepthTextureDirective],
                    exports: [DepthTextureDirective],
                }]
        }] });

// GENERATED

// GENERATED
class VideoTextureDirective extends ThreeTexture {
    constructor() {
        super(...arguments);
        this.textureType = VideoTexture;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
VideoTextureDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: VideoTextureDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
VideoTextureDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: VideoTextureDirective, selector: "ngt-video-texture", inputs: { args: "args" }, providers: [
        {
            provide: ThreeTexture,
            useExisting: VideoTextureDirective,
        },
    ], exportAs: ["ngtVideoTexture"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: VideoTextureDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-video-texture',
                    exportAs: 'ngtVideoTexture',
                    providers: [
                        {
                            provide: ThreeTexture,
                            useExisting: VideoTextureDirective,
                        },
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });

// GENERATED
class ThreeVideoTextureModule {
}
ThreeVideoTextureModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeVideoTextureModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ThreeVideoTextureModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeVideoTextureModule, declarations: [VideoTextureDirective], exports: [VideoTextureDirective] });
ThreeVideoTextureModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeVideoTextureModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeVideoTextureModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [VideoTextureDirective],
                    exports: [VideoTextureDirective],
                }]
        }] });

// GENERATED

// GENERATED

/**
 * Generated bundle index. Do not edit.
 */

export { CanvasTextureDirective, CompressedTextureDirective, CubeTextureDirective, DataTexture2DArrayDirective, DataTexture3DDirective, DataTextureDirective, DepthTextureDirective, ThreeCanvasTextureModule, ThreeCompressedTextureModule, ThreeCubeTextureModule, ThreeDataTexture2DArrayModule, ThreeDataTexture3DModule, ThreeDataTextureModule, ThreeDepthTextureModule, ThreeVideoTextureModule, VideoTextureDirective };
