// @angular-three/core/fesm2015/angular-three-core-curves.js
import { ThreeCurve } from '@angular-three/core';
import * as i0 from '@angular/core';
import { Directive, Input, NgModule } from '@angular/core';
import { CatmullRomCurve3, CubicBezierCurve, CubicBezierCurve3, EllipseCurve, LineCurve, LineCurve3, QuadraticBezierCurve, QuadraticBezierCurve3, SplineCurve } from 'three';

// GENERATED
class CatmullRomCurve3Directive extends ThreeCurve {
    constructor() {
        super(...arguments);
        this.curveType = CatmullRomCurve3;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
CatmullRomCurve3Directive.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: CatmullRomCurve3Directive, deps: null, target: i0.ɵɵFactoryTarget.Directive });
CatmullRomCurve3Directive.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: CatmullRomCurve3Directive, selector: "ngt-catmull-rom-curve3", inputs: { args: "args" }, providers: [
        {
            provide: ThreeCurve,
            useExisting: CatmullRomCurve3Directive,
        },
    ], exportAs: ["ngtCatmullRomCurve3"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: CatmullRomCurve3Directive, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-catmull-rom-curve3',
                    exportAs: 'ngtCatmullRomCurve3',
                    providers: [
                        {
                            provide: ThreeCurve,
                            useExisting: CatmullRomCurve3Directive,
                        },
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });

// GENERATED
class ThreeCatmullRomCurve3Module {
}
ThreeCatmullRomCurve3Module.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeCatmullRomCurve3Module, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ThreeCatmullRomCurve3Module.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeCatmullRomCurve3Module, declarations: [CatmullRomCurve3Directive], exports: [CatmullRomCurve3Directive] });
ThreeCatmullRomCurve3Module.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeCatmullRomCurve3Module });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeCatmullRomCurve3Module, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [CatmullRomCurve3Directive],
                    exports: [CatmullRomCurve3Directive],
                }]
        }] });

// GENERATED

// GENERATED
class CubicBezierCurveDirective extends ThreeCurve {
    constructor() {
        super(...arguments);
        this.curveType = CubicBezierCurve;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
CubicBezierCurveDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: CubicBezierCurveDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
CubicBezierCurveDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: CubicBezierCurveDirective, selector: "ngt-cubic-bezier-curve", inputs: { args: "args" }, providers: [
        {
            provide: ThreeCurve,
            useExisting: CubicBezierCurveDirective,
        },
    ], exportAs: ["ngtCubicBezierCurve"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: CubicBezierCurveDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-cubic-bezier-curve',
                    exportAs: 'ngtCubicBezierCurve',
                    providers: [
                        {
                            provide: ThreeCurve,
                            useExisting: CubicBezierCurveDirective,
                        },
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });

// GENERATED
class ThreeCubicBezierCurveModule {
}
ThreeCubicBezierCurveModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeCubicBezierCurveModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ThreeCubicBezierCurveModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeCubicBezierCurveModule, declarations: [CubicBezierCurveDirective], exports: [CubicBezierCurveDirective] });
ThreeCubicBezierCurveModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeCubicBezierCurveModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeCubicBezierCurveModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [CubicBezierCurveDirective],
                    exports: [CubicBezierCurveDirective],
                }]
        }] });

// GENERATED

// GENERATED
class CubicBezierCurve3Directive extends ThreeCurve {
    constructor() {
        super(...arguments);
        this.curveType = CubicBezierCurve3;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
CubicBezierCurve3Directive.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: CubicBezierCurve3Directive, deps: null, target: i0.ɵɵFactoryTarget.Directive });
CubicBezierCurve3Directive.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: CubicBezierCurve3Directive, selector: "ngt-cubic-bezier-curve3", inputs: { args: "args" }, providers: [
        {
            provide: ThreeCurve,
            useExisting: CubicBezierCurve3Directive,
        },
    ], exportAs: ["ngtCubicBezierCurve3"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: CubicBezierCurve3Directive, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-cubic-bezier-curve3',
                    exportAs: 'ngtCubicBezierCurve3',
                    providers: [
                        {
                            provide: ThreeCurve,
                            useExisting: CubicBezierCurve3Directive,
                        },
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });

// GENERATED
class ThreeCubicBezierCurve3Module {
}
ThreeCubicBezierCurve3Module.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeCubicBezierCurve3Module, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ThreeCubicBezierCurve3Module.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeCubicBezierCurve3Module, declarations: [CubicBezierCurve3Directive], exports: [CubicBezierCurve3Directive] });
ThreeCubicBezierCurve3Module.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeCubicBezierCurve3Module });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeCubicBezierCurve3Module, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [CubicBezierCurve3Directive],
                    exports: [CubicBezierCurve3Directive],
                }]
        }] });

// GENERATED

// GENERATED
class EllipseCurveDirective extends ThreeCurve {
    constructor() {
        super(...arguments);
        this.curveType = EllipseCurve;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
EllipseCurveDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: EllipseCurveDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
EllipseCurveDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: EllipseCurveDirective, selector: "ngt-ellipse-curve", inputs: { args: "args" }, providers: [
        {
            provide: ThreeCurve,
            useExisting: EllipseCurveDirective,
        },
    ], exportAs: ["ngtEllipseCurve"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: EllipseCurveDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-ellipse-curve',
                    exportAs: 'ngtEllipseCurve',
                    providers: [
                        {
                            provide: ThreeCurve,
                            useExisting: EllipseCurveDirective,
                        },
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });

// GENERATED
class ThreeEllipseCurveModule {
}
ThreeEllipseCurveModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeEllipseCurveModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ThreeEllipseCurveModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeEllipseCurveModule, declarations: [EllipseCurveDirective], exports: [EllipseCurveDirective] });
ThreeEllipseCurveModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeEllipseCurveModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeEllipseCurveModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [EllipseCurveDirective],
                    exports: [EllipseCurveDirective],
                }]
        }] });

// GENERATED

// GENERATED
class LineCurveDirective extends ThreeCurve {
    constructor() {
        super(...arguments);
        this.curveType = LineCurve;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
LineCurveDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: LineCurveDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
LineCurveDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: LineCurveDirective, selector: "ngt-line-curve", inputs: { args: "args" }, providers: [
        {
            provide: ThreeCurve,
            useExisting: LineCurveDirective,
        },
    ], exportAs: ["ngtLineCurve"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: LineCurveDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-line-curve',
                    exportAs: 'ngtLineCurve',
                    providers: [
                        {
                            provide: ThreeCurve,
                            useExisting: LineCurveDirective,
                        },
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });

// GENERATED
class ThreeLineCurveModule {
}
ThreeLineCurveModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeLineCurveModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ThreeLineCurveModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeLineCurveModule, declarations: [LineCurveDirective], exports: [LineCurveDirective] });
ThreeLineCurveModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeLineCurveModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeLineCurveModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [LineCurveDirective],
                    exports: [LineCurveDirective],
                }]
        }] });

// GENERATED

// GENERATED
class LineCurve3Directive extends ThreeCurve {
    constructor() {
        super(...arguments);
        this.curveType = LineCurve3;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
LineCurve3Directive.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: LineCurve3Directive, deps: null, target: i0.ɵɵFactoryTarget.Directive });
LineCurve3Directive.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: LineCurve3Directive, selector: "ngt-line-curve3", inputs: { args: "args" }, providers: [
        {
            provide: ThreeCurve,
            useExisting: LineCurve3Directive,
        },
    ], exportAs: ["ngtLineCurve3"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: LineCurve3Directive, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-line-curve3',
                    exportAs: 'ngtLineCurve3',
                    providers: [
                        {
                            provide: ThreeCurve,
                            useExisting: LineCurve3Directive,
                        },
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });

// GENERATED
class ThreeLineCurve3Module {
}
ThreeLineCurve3Module.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeLineCurve3Module, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ThreeLineCurve3Module.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeLineCurve3Module, declarations: [LineCurve3Directive], exports: [LineCurve3Directive] });
ThreeLineCurve3Module.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeLineCurve3Module });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeLineCurve3Module, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [LineCurve3Directive],
                    exports: [LineCurve3Directive],
                }]
        }] });

// GENERATED

// GENERATED
class QuadraticBezierCurveDirective extends ThreeCurve {
    constructor() {
        super(...arguments);
        this.curveType = QuadraticBezierCurve;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
QuadraticBezierCurveDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: QuadraticBezierCurveDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
QuadraticBezierCurveDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: QuadraticBezierCurveDirective, selector: "ngt-quadratic-bezier-curve", inputs: { args: "args" }, providers: [
        {
            provide: ThreeCurve,
            useExisting: QuadraticBezierCurveDirective,
        },
    ], exportAs: ["ngtQuadraticBezierCurve"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: QuadraticBezierCurveDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-quadratic-bezier-curve',
                    exportAs: 'ngtQuadraticBezierCurve',
                    providers: [
                        {
                            provide: ThreeCurve,
                            useExisting: QuadraticBezierCurveDirective,
                        },
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });

// GENERATED
class ThreeQuadraticBezierCurveModule {
}
ThreeQuadraticBezierCurveModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeQuadraticBezierCurveModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ThreeQuadraticBezierCurveModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeQuadraticBezierCurveModule, declarations: [QuadraticBezierCurveDirective], exports: [QuadraticBezierCurveDirective] });
ThreeQuadraticBezierCurveModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeQuadraticBezierCurveModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeQuadraticBezierCurveModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [QuadraticBezierCurveDirective],
                    exports: [QuadraticBezierCurveDirective],
                }]
        }] });

// GENERATED

// GENERATED
class QuadraticBezierCurve3Directive extends ThreeCurve {
    constructor() {
        super(...arguments);
        this.curveType = QuadraticBezierCurve3;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
QuadraticBezierCurve3Directive.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: QuadraticBezierCurve3Directive, deps: null, target: i0.ɵɵFactoryTarget.Directive });
QuadraticBezierCurve3Directive.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: QuadraticBezierCurve3Directive, selector: "ngt-quadratic-bezier-curve3", inputs: { args: "args" }, providers: [
        {
            provide: ThreeCurve,
            useExisting: QuadraticBezierCurve3Directive,
        },
    ], exportAs: ["ngtQuadraticBezierCurve3"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: QuadraticBezierCurve3Directive, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-quadratic-bezier-curve3',
                    exportAs: 'ngtQuadraticBezierCurve3',
                    providers: [
                        {
                            provide: ThreeCurve,
                            useExisting: QuadraticBezierCurve3Directive,
                        },
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });

// GENERATED
class ThreeQuadraticBezierCurve3Module {
}
ThreeQuadraticBezierCurve3Module.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeQuadraticBezierCurve3Module, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ThreeQuadraticBezierCurve3Module.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeQuadraticBezierCurve3Module, declarations: [QuadraticBezierCurve3Directive], exports: [QuadraticBezierCurve3Directive] });
ThreeQuadraticBezierCurve3Module.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeQuadraticBezierCurve3Module });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeQuadraticBezierCurve3Module, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [QuadraticBezierCurve3Directive],
                    exports: [QuadraticBezierCurve3Directive],
                }]
        }] });

// GENERATED

// GENERATED
class SplineCurveDirective extends ThreeCurve {
    constructor() {
        super(...arguments);
        this.curveType = SplineCurve;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
SplineCurveDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: SplineCurveDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
SplineCurveDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: SplineCurveDirective, selector: "ngt-spline-curve", inputs: { args: "args" }, providers: [
        {
            provide: ThreeCurve,
            useExisting: SplineCurveDirective,
        },
    ], exportAs: ["ngtSplineCurve"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: SplineCurveDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-spline-curve',
                    exportAs: 'ngtSplineCurve',
                    providers: [
                        {
                            provide: ThreeCurve,
                            useExisting: SplineCurveDirective,
                        },
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });

// GENERATED
class ThreeSplineCurveModule {
}
ThreeSplineCurveModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeSplineCurveModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ThreeSplineCurveModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeSplineCurveModule, declarations: [SplineCurveDirective], exports: [SplineCurveDirective] });
ThreeSplineCurveModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeSplineCurveModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeSplineCurveModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [SplineCurveDirective],
                    exports: [SplineCurveDirective],
                }]
        }] });

// GENERATED

// GENERATED

/**
 * Generated bundle index. Do not edit.
 */

export { CatmullRomCurve3Directive, CubicBezierCurve3Directive, CubicBezierCurveDirective, EllipseCurveDirective, LineCurve3Directive, LineCurveDirective, QuadraticBezierCurve3Directive, QuadraticBezierCurveDirective, SplineCurveDirective, ThreeCatmullRomCurve3Module, ThreeCubicBezierCurve3Module, ThreeCubicBezierCurveModule, ThreeEllipseCurveModule, ThreeLineCurve3Module, ThreeLineCurveModule, ThreeQuadraticBezierCurve3Module, ThreeQuadraticBezierCurveModule, ThreeSplineCurveModule };
