// @angular-three/core/fesm2015/angular-three-core-sprites.js
import { ThreeSprite, ThreeObject3d, OBJECT_3D_CONTROLLER_PROVIDER } from '@angular-three/core';
import * as i0 from '@angular/core';
import { Directive, NgModule } from '@angular/core';
import { Sprite } from 'three';

// GENERATED
class SpriteDirective extends ThreeSprite {
    constructor() {
        super(...arguments);
        this.spriteType = Sprite;
    }
}
SpriteDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: SpriteDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
SpriteDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: SpriteDirective, selector: "ngt-sprite", providers: [
        {
            provide: ThreeObject3d,
            useExisting: SpriteDirective,
        },
        OBJECT_3D_CONTROLLER_PROVIDER,
    ], exportAs: ["ngtSprite"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: SpriteDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-sprite',
                    exportAs: 'ngtSprite',
                    providers: [
                        {
                            provide: ThreeObject3d,
                            useExisting: SpriteDirective,
                        },
                        OBJECT_3D_CONTROLLER_PROVIDER,
                    ],
                }]
        }] });

// GENERATED
class ThreeSpriteModule {
}
ThreeSpriteModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeSpriteModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ThreeSpriteModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeSpriteModule, declarations: [SpriteDirective], exports: [SpriteDirective] });
ThreeSpriteModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeSpriteModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeSpriteModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [SpriteDirective],
                    exports: [SpriteDirective],
                }]
        }] });

// GENERATED

// GENERATED

/**
 * Generated bundle index. Do not edit.
 */

export { SpriteDirective, ThreeSpriteModule };
