// @angular-three/core/fesm2015/angular-three-core-lines.js
import { ThreeLine, ThreeObject3d, OBJECT_3D_CONTROLLER_PROVIDER } from '@angular-three/core';
import * as i0 from '@angular/core';
import { Directive, NgModule } from '@angular/core';
import { Line, LineLoop, LineSegments } from 'three';

// GENERATED
class LineDirective extends ThreeLine {
    constructor() {
        super(...arguments);
        this.lineType = Line;
    }
}
LineDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: LineDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
LineDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: LineDirective, selector: "ngt-line", providers: [
        {
            provide: ThreeObject3d,
            useExisting: LineDirective,
        },
        OBJECT_3D_CONTROLLER_PROVIDER,
    ], exportAs: ["ngtLine"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: LineDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-line',
                    exportAs: 'ngtLine',
                    providers: [
                        {
                            provide: ThreeObject3d,
                            useExisting: LineDirective,
                        },
                        OBJECT_3D_CONTROLLER_PROVIDER,
                    ],
                }]
        }] });

// GENERATED
class ThreeLineModule {
}
ThreeLineModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeLineModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ThreeLineModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeLineModule, declarations: [LineDirective], exports: [LineDirective] });
ThreeLineModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeLineModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeLineModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [LineDirective],
                    exports: [LineDirective],
                }]
        }] });

// GENERATED

// GENERATED
class LineLoopDirective extends ThreeLine {
    constructor() {
        super(...arguments);
        this.lineType = LineLoop;
    }
}
LineLoopDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: LineLoopDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
LineLoopDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: LineLoopDirective, selector: "ngt-line-loop", providers: [
        {
            provide: ThreeObject3d,
            useExisting: LineLoopDirective,
        },
        OBJECT_3D_CONTROLLER_PROVIDER,
    ], exportAs: ["ngtLineLoop"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: LineLoopDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-line-loop',
                    exportAs: 'ngtLineLoop',
                    providers: [
                        {
                            provide: ThreeObject3d,
                            useExisting: LineLoopDirective,
                        },
                        OBJECT_3D_CONTROLLER_PROVIDER,
                    ],
                }]
        }] });

// GENERATED
class ThreeLineLoopModule {
}
ThreeLineLoopModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeLineLoopModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ThreeLineLoopModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeLineLoopModule, declarations: [LineLoopDirective], exports: [LineLoopDirective] });
ThreeLineLoopModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeLineLoopModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeLineLoopModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [LineLoopDirective],
                    exports: [LineLoopDirective],
                }]
        }] });

// GENERATED

// GENERATED
class LineSegmentsDirective extends ThreeLine {
    constructor() {
        super(...arguments);
        this.lineType = LineSegments;
    }
}
LineSegmentsDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: LineSegmentsDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
LineSegmentsDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: LineSegmentsDirective, selector: "ngt-line-segments", providers: [
        {
            provide: ThreeObject3d,
            useExisting: LineSegmentsDirective,
        },
        OBJECT_3D_CONTROLLER_PROVIDER,
    ], exportAs: ["ngtLineSegments"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: LineSegmentsDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-line-segments',
                    exportAs: 'ngtLineSegments',
                    providers: [
                        {
                            provide: ThreeObject3d,
                            useExisting: LineSegmentsDirective,
                        },
                        OBJECT_3D_CONTROLLER_PROVIDER,
                    ],
                }]
        }] });

// GENERATED
class ThreeLineSegmentsModule {
}
ThreeLineSegmentsModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeLineSegmentsModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ThreeLineSegmentsModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeLineSegmentsModule, declarations: [LineSegmentsDirective], exports: [LineSegmentsDirective] });
ThreeLineSegmentsModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeLineSegmentsModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeLineSegmentsModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [LineSegmentsDirective],
                    exports: [LineSegmentsDirective],
                }]
        }] });

// GENERATED

// GENERATED

/**
 * Generated bundle index. Do not edit.
 */

export { LineDirective, LineLoopDirective, LineSegmentsDirective, ThreeLineLoopModule, ThreeLineModule, ThreeLineSegmentsModule };
