// @angular-three/core/fesm2015/angular-three-core-helpers.js
import { ThreeHelper, ThreeObject3d, OBJECT_3D_CONTROLLER_PROVIDER } from '@angular-three/core';
import * as i0 from '@angular/core';
import { Directive, Input, NgModule } from '@angular/core';
import { ArrowHelper, AxesHelper, BoxHelper, Box3Helper, GridHelper, CameraHelper, DirectionalLightHelper, HemisphereLightHelper, PlaneHelper, PointLightHelper, PolarGridHelper, SkeletonHelper, SpotLightHelper } from 'three';

// GENERATED
class ArrowHelperDirective extends ThreeHelper {
    constructor() {
        super(...arguments);
        this.helperType = ArrowHelper;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
ArrowHelperDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ArrowHelperDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
ArrowHelperDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: ArrowHelperDirective, selector: "ngt-arrow-helper", inputs: { args: "args" }, providers: [
        {
            provide: ThreeObject3d,
            useExisting: ArrowHelperDirective,
        },
        OBJECT_3D_CONTROLLER_PROVIDER,
    ], exportAs: ["ngtArrowHelper"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ArrowHelperDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-arrow-helper',
                    exportAs: 'ngtArrowHelper',
                    providers: [
                        {
                            provide: ThreeObject3d,
                            useExisting: ArrowHelperDirective,
                        },
                        OBJECT_3D_CONTROLLER_PROVIDER,
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });

// GENERATED
class ThreeArrowHelperModule {
}
ThreeArrowHelperModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeArrowHelperModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ThreeArrowHelperModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeArrowHelperModule, declarations: [ArrowHelperDirective], exports: [ArrowHelperDirective] });
ThreeArrowHelperModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeArrowHelperModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeArrowHelperModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [ArrowHelperDirective],
                    exports: [ArrowHelperDirective],
                }]
        }] });

// GENERATED

// GENERATED
class AxesHelperDirective extends ThreeHelper {
    constructor() {
        super(...arguments);
        this.helperType = AxesHelper;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
AxesHelperDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: AxesHelperDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
AxesHelperDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: AxesHelperDirective, selector: "ngt-axes-helper", inputs: { args: "args" }, providers: [
        {
            provide: ThreeObject3d,
            useExisting: AxesHelperDirective,
        },
        OBJECT_3D_CONTROLLER_PROVIDER,
    ], exportAs: ["ngtAxesHelper"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: AxesHelperDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-axes-helper',
                    exportAs: 'ngtAxesHelper',
                    providers: [
                        {
                            provide: ThreeObject3d,
                            useExisting: AxesHelperDirective,
                        },
                        OBJECT_3D_CONTROLLER_PROVIDER,
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });

// GENERATED
class ThreeAxesHelperModule {
}
ThreeAxesHelperModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeAxesHelperModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ThreeAxesHelperModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeAxesHelperModule, declarations: [AxesHelperDirective], exports: [AxesHelperDirective] });
ThreeAxesHelperModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeAxesHelperModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeAxesHelperModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [AxesHelperDirective],
                    exports: [AxesHelperDirective],
                }]
        }] });

// GENERATED

// GENERATED
class BoxHelperDirective extends ThreeHelper {
    constructor() {
        super(...arguments);
        this.helperType = BoxHelper;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
BoxHelperDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: BoxHelperDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
BoxHelperDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: BoxHelperDirective, selector: "ngt-box-helper", inputs: { args: "args" }, providers: [
        {
            provide: ThreeObject3d,
            useExisting: BoxHelperDirective,
        },
        OBJECT_3D_CONTROLLER_PROVIDER,
    ], exportAs: ["ngtBoxHelper"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: BoxHelperDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-box-helper',
                    exportAs: 'ngtBoxHelper',
                    providers: [
                        {
                            provide: ThreeObject3d,
                            useExisting: BoxHelperDirective,
                        },
                        OBJECT_3D_CONTROLLER_PROVIDER,
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });

// GENERATED
class ThreeBoxHelperModule {
}
ThreeBoxHelperModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeBoxHelperModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ThreeBoxHelperModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeBoxHelperModule, declarations: [BoxHelperDirective], exports: [BoxHelperDirective] });
ThreeBoxHelperModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeBoxHelperModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeBoxHelperModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [BoxHelperDirective],
                    exports: [BoxHelperDirective],
                }]
        }] });

// GENERATED

// GENERATED
class Box3HelperDirective extends ThreeHelper {
    constructor() {
        super(...arguments);
        this.helperType = Box3Helper;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
Box3HelperDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: Box3HelperDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
Box3HelperDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: Box3HelperDirective, selector: "ngt-box3-helper", inputs: { args: "args" }, providers: [
        {
            provide: ThreeObject3d,
            useExisting: Box3HelperDirective,
        },
        OBJECT_3D_CONTROLLER_PROVIDER,
    ], exportAs: ["ngtBox3Helper"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: Box3HelperDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-box3-helper',
                    exportAs: 'ngtBox3Helper',
                    providers: [
                        {
                            provide: ThreeObject3d,
                            useExisting: Box3HelperDirective,
                        },
                        OBJECT_3D_CONTROLLER_PROVIDER,
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });

// GENERATED
class ThreeBox3HelperModule {
}
ThreeBox3HelperModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeBox3HelperModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ThreeBox3HelperModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeBox3HelperModule, declarations: [Box3HelperDirective], exports: [Box3HelperDirective] });
ThreeBox3HelperModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeBox3HelperModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeBox3HelperModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [Box3HelperDirective],
                    exports: [Box3HelperDirective],
                }]
        }] });

// GENERATED

// GENERATED
class GridHelperDirective extends ThreeHelper {
    constructor() {
        super(...arguments);
        this.helperType = GridHelper;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
GridHelperDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: GridHelperDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
GridHelperDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: GridHelperDirective, selector: "ngt-grid-helper", inputs: { args: "args" }, providers: [
        {
            provide: ThreeObject3d,
            useExisting: GridHelperDirective,
        },
        OBJECT_3D_CONTROLLER_PROVIDER,
    ], exportAs: ["ngtGridHelper"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: GridHelperDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-grid-helper',
                    exportAs: 'ngtGridHelper',
                    providers: [
                        {
                            provide: ThreeObject3d,
                            useExisting: GridHelperDirective,
                        },
                        OBJECT_3D_CONTROLLER_PROVIDER,
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });

// GENERATED
class ThreeGridHelperModule {
}
ThreeGridHelperModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeGridHelperModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ThreeGridHelperModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeGridHelperModule, declarations: [GridHelperDirective], exports: [GridHelperDirective] });
ThreeGridHelperModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeGridHelperModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeGridHelperModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [GridHelperDirective],
                    exports: [GridHelperDirective],
                }]
        }] });

// GENERATED

// GENERATED
class CameraHelperDirective extends ThreeHelper {
    constructor() {
        super(...arguments);
        this.helperType = CameraHelper;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
CameraHelperDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: CameraHelperDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
CameraHelperDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: CameraHelperDirective, selector: "ngt-camera-helper", inputs: { args: "args" }, providers: [
        {
            provide: ThreeObject3d,
            useExisting: CameraHelperDirective,
        },
        OBJECT_3D_CONTROLLER_PROVIDER,
    ], exportAs: ["ngtCameraHelper"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: CameraHelperDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-camera-helper',
                    exportAs: 'ngtCameraHelper',
                    providers: [
                        {
                            provide: ThreeObject3d,
                            useExisting: CameraHelperDirective,
                        },
                        OBJECT_3D_CONTROLLER_PROVIDER,
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });

// GENERATED
class ThreeCameraHelperModule {
}
ThreeCameraHelperModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeCameraHelperModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ThreeCameraHelperModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeCameraHelperModule, declarations: [CameraHelperDirective], exports: [CameraHelperDirective] });
ThreeCameraHelperModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeCameraHelperModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeCameraHelperModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [CameraHelperDirective],
                    exports: [CameraHelperDirective],
                }]
        }] });

// GENERATED

// GENERATED
class DirectionalLightHelperDirective extends ThreeHelper {
    constructor() {
        super(...arguments);
        this.helperType = DirectionalLightHelper;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
DirectionalLightHelperDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: DirectionalLightHelperDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
DirectionalLightHelperDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: DirectionalLightHelperDirective, selector: "ngt-directional-light-helper", inputs: { args: "args" }, providers: [
        {
            provide: ThreeObject3d,
            useExisting: DirectionalLightHelperDirective,
        },
        OBJECT_3D_CONTROLLER_PROVIDER,
    ], exportAs: ["ngtDirectionalLightHelper"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: DirectionalLightHelperDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-directional-light-helper',
                    exportAs: 'ngtDirectionalLightHelper',
                    providers: [
                        {
                            provide: ThreeObject3d,
                            useExisting: DirectionalLightHelperDirective,
                        },
                        OBJECT_3D_CONTROLLER_PROVIDER,
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });

// GENERATED
class ThreeDirectionalLightHelperModule {
}
ThreeDirectionalLightHelperModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeDirectionalLightHelperModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ThreeDirectionalLightHelperModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeDirectionalLightHelperModule, declarations: [DirectionalLightHelperDirective], exports: [DirectionalLightHelperDirective] });
ThreeDirectionalLightHelperModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeDirectionalLightHelperModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeDirectionalLightHelperModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [DirectionalLightHelperDirective],
                    exports: [DirectionalLightHelperDirective],
                }]
        }] });

// GENERATED

// GENERATED
class HemisphereLightHelperDirective extends ThreeHelper {
    constructor() {
        super(...arguments);
        this.helperType = HemisphereLightHelper;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
HemisphereLightHelperDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: HemisphereLightHelperDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
HemisphereLightHelperDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: HemisphereLightHelperDirective, selector: "ngt-hemisphere-light-helper", inputs: { args: "args" }, providers: [
        {
            provide: ThreeObject3d,
            useExisting: HemisphereLightHelperDirective,
        },
        OBJECT_3D_CONTROLLER_PROVIDER,
    ], exportAs: ["ngtHemisphereLightHelper"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: HemisphereLightHelperDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-hemisphere-light-helper',
                    exportAs: 'ngtHemisphereLightHelper',
                    providers: [
                        {
                            provide: ThreeObject3d,
                            useExisting: HemisphereLightHelperDirective,
                        },
                        OBJECT_3D_CONTROLLER_PROVIDER,
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });

// GENERATED
class ThreeHemisphereLightHelperModule {
}
ThreeHemisphereLightHelperModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeHemisphereLightHelperModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ThreeHemisphereLightHelperModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeHemisphereLightHelperModule, declarations: [HemisphereLightHelperDirective], exports: [HemisphereLightHelperDirective] });
ThreeHemisphereLightHelperModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeHemisphereLightHelperModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeHemisphereLightHelperModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [HemisphereLightHelperDirective],
                    exports: [HemisphereLightHelperDirective],
                }]
        }] });

// GENERATED

// GENERATED
class PlaneHelperDirective extends ThreeHelper {
    constructor() {
        super(...arguments);
        this.helperType = PlaneHelper;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
PlaneHelperDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: PlaneHelperDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
PlaneHelperDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: PlaneHelperDirective, selector: "ngt-plane-helper", inputs: { args: "args" }, providers: [
        {
            provide: ThreeObject3d,
            useExisting: PlaneHelperDirective,
        },
        OBJECT_3D_CONTROLLER_PROVIDER,
    ], exportAs: ["ngtPlaneHelper"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: PlaneHelperDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-plane-helper',
                    exportAs: 'ngtPlaneHelper',
                    providers: [
                        {
                            provide: ThreeObject3d,
                            useExisting: PlaneHelperDirective,
                        },
                        OBJECT_3D_CONTROLLER_PROVIDER,
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });

// GENERATED
class ThreePlaneHelperModule {
}
ThreePlaneHelperModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreePlaneHelperModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ThreePlaneHelperModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreePlaneHelperModule, declarations: [PlaneHelperDirective], exports: [PlaneHelperDirective] });
ThreePlaneHelperModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreePlaneHelperModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreePlaneHelperModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [PlaneHelperDirective],
                    exports: [PlaneHelperDirective],
                }]
        }] });

// GENERATED

// GENERATED
class PointLightHelperDirective extends ThreeHelper {
    constructor() {
        super(...arguments);
        this.helperType = PointLightHelper;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
PointLightHelperDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: PointLightHelperDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
PointLightHelperDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: PointLightHelperDirective, selector: "ngt-point-light-helper", inputs: { args: "args" }, providers: [
        {
            provide: ThreeObject3d,
            useExisting: PointLightHelperDirective,
        },
        OBJECT_3D_CONTROLLER_PROVIDER,
    ], exportAs: ["ngtPointLightHelper"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: PointLightHelperDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-point-light-helper',
                    exportAs: 'ngtPointLightHelper',
                    providers: [
                        {
                            provide: ThreeObject3d,
                            useExisting: PointLightHelperDirective,
                        },
                        OBJECT_3D_CONTROLLER_PROVIDER,
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });

// GENERATED
class ThreePointLightHelperModule {
}
ThreePointLightHelperModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreePointLightHelperModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ThreePointLightHelperModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreePointLightHelperModule, declarations: [PointLightHelperDirective], exports: [PointLightHelperDirective] });
ThreePointLightHelperModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreePointLightHelperModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreePointLightHelperModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [PointLightHelperDirective],
                    exports: [PointLightHelperDirective],
                }]
        }] });

// GENERATED

// GENERATED
class PolarGridHelperDirective extends ThreeHelper {
    constructor() {
        super(...arguments);
        this.helperType = PolarGridHelper;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
PolarGridHelperDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: PolarGridHelperDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
PolarGridHelperDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: PolarGridHelperDirective, selector: "ngt-polar-grid-helper", inputs: { args: "args" }, providers: [
        {
            provide: ThreeObject3d,
            useExisting: PolarGridHelperDirective,
        },
        OBJECT_3D_CONTROLLER_PROVIDER,
    ], exportAs: ["ngtPolarGridHelper"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: PolarGridHelperDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-polar-grid-helper',
                    exportAs: 'ngtPolarGridHelper',
                    providers: [
                        {
                            provide: ThreeObject3d,
                            useExisting: PolarGridHelperDirective,
                        },
                        OBJECT_3D_CONTROLLER_PROVIDER,
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });

// GENERATED
class ThreePolarGridHelperModule {
}
ThreePolarGridHelperModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreePolarGridHelperModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ThreePolarGridHelperModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreePolarGridHelperModule, declarations: [PolarGridHelperDirective], exports: [PolarGridHelperDirective] });
ThreePolarGridHelperModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreePolarGridHelperModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreePolarGridHelperModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [PolarGridHelperDirective],
                    exports: [PolarGridHelperDirective],
                }]
        }] });

// GENERATED

// GENERATED
class SkeletonHelperDirective extends ThreeHelper {
    constructor() {
        super(...arguments);
        this.helperType = SkeletonHelper;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
SkeletonHelperDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: SkeletonHelperDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
SkeletonHelperDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: SkeletonHelperDirective, selector: "ngt-skeleton-helper", inputs: { args: "args" }, providers: [
        {
            provide: ThreeObject3d,
            useExisting: SkeletonHelperDirective,
        },
        OBJECT_3D_CONTROLLER_PROVIDER,
    ], exportAs: ["ngtSkeletonHelper"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: SkeletonHelperDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-skeleton-helper',
                    exportAs: 'ngtSkeletonHelper',
                    providers: [
                        {
                            provide: ThreeObject3d,
                            useExisting: SkeletonHelperDirective,
                        },
                        OBJECT_3D_CONTROLLER_PROVIDER,
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });

// GENERATED
class ThreeSkeletonHelperModule {
}
ThreeSkeletonHelperModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeSkeletonHelperModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ThreeSkeletonHelperModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeSkeletonHelperModule, declarations: [SkeletonHelperDirective], exports: [SkeletonHelperDirective] });
ThreeSkeletonHelperModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeSkeletonHelperModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeSkeletonHelperModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [SkeletonHelperDirective],
                    exports: [SkeletonHelperDirective],
                }]
        }] });

// GENERATED

// GENERATED
class SpotLightHelperDirective extends ThreeHelper {
    constructor() {
        super(...arguments);
        this.helperType = SpotLightHelper;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
SpotLightHelperDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: SpotLightHelperDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
SpotLightHelperDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: SpotLightHelperDirective, selector: "ngt-spot-light-helper", inputs: { args: "args" }, providers: [
        {
            provide: ThreeObject3d,
            useExisting: SpotLightHelperDirective,
        },
        OBJECT_3D_CONTROLLER_PROVIDER,
    ], exportAs: ["ngtSpotLightHelper"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: SpotLightHelperDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-spot-light-helper',
                    exportAs: 'ngtSpotLightHelper',
                    providers: [
                        {
                            provide: ThreeObject3d,
                            useExisting: SpotLightHelperDirective,
                        },
                        OBJECT_3D_CONTROLLER_PROVIDER,
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });

// GENERATED
class ThreeSpotLightHelperModule {
}
ThreeSpotLightHelperModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeSpotLightHelperModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ThreeSpotLightHelperModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeSpotLightHelperModule, declarations: [SpotLightHelperDirective], exports: [SpotLightHelperDirective] });
ThreeSpotLightHelperModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeSpotLightHelperModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeSpotLightHelperModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [SpotLightHelperDirective],
                    exports: [SpotLightHelperDirective],
                }]
        }] });

// GENERATED

// GENERATED

/**
 * Generated bundle index. Do not edit.
 */

export { ArrowHelperDirective, AxesHelperDirective, Box3HelperDirective, BoxHelperDirective, CameraHelperDirective, DirectionalLightHelperDirective, GridHelperDirective, HemisphereLightHelperDirective, PlaneHelperDirective, PointLightHelperDirective, PolarGridHelperDirective, SkeletonHelperDirective, SpotLightHelperDirective, ThreeArrowHelperModule, ThreeAxesHelperModule, ThreeBox3HelperModule, ThreeBoxHelperModule, ThreeCameraHelperModule, ThreeDirectionalLightHelperModule, ThreeGridHelperModule, ThreeHemisphereLightHelperModule, ThreePlaneHelperModule, ThreePointLightHelperModule, ThreePolarGridHelperModule, ThreeSkeletonHelperModule, ThreeSpotLightHelperModule };
