// @angular-three/core/fesm2015/angular-three-core-group.js
import * as i0 from '@angular/core';
import { Directive, NgModule } from '@angular/core';
import { ThreeObject3d, OBJECT_3D_CONTROLLER_PROVIDER } from '@angular-three/core';
import { Group } from 'three';

class GroupDirective extends ThreeObject3d {
    ngOnInit() {
        this.init();
    }
    initObject() {
        this._group = new Group();
    }
    get object3d() {
        return this._group;
    }
}
GroupDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: GroupDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
GroupDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: GroupDirective, selector: "ngt-group", providers: [
        { provide: ThreeObject3d, useExisting: GroupDirective },
        OBJECT_3D_CONTROLLER_PROVIDER,
    ], exportAs: ["ngtGroup"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: GroupDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-group',
                    exportAs: 'ngtGroup',
                    providers: [
                        { provide: ThreeObject3d, useExisting: GroupDirective },
                        OBJECT_3D_CONTROLLER_PROVIDER,
                    ],
                }]
        }] });

class ThreeGroupModule {
}
ThreeGroupModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeGroupModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ThreeGroupModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeGroupModule, declarations: [GroupDirective], exports: [GroupDirective] });
ThreeGroupModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeGroupModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeGroupModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [GroupDirective],
                    exports: [GroupDirective],
                }]
        }] });

/**
 * Generated bundle index. Do not edit.
 */

export { GroupDirective, ThreeGroupModule };
