// @angular-three/core/fesm2015/angular-three-core-attributes.js
import * as i1 from '@angular-three/core';
import { ThreeAttribute } from '@angular-three/core';
import * as i0 from '@angular/core';
import { Directive, Optional, Input, NgModule } from '@angular/core';
import { InterleavedBufferAttribute, BufferAttribute, InstancedBufferAttribute, Float16BufferAttribute, Float32BufferAttribute, Float64BufferAttribute, Int8BufferAttribute, Int16BufferAttribute, Int32BufferAttribute, Uint8BufferAttribute, Uint16BufferAttribute, Uint32BufferAttribute, Uint8ClampedBufferAttribute } from 'three';

class InterleavedBufferAttributeDirective {
    constructor(ngZone, geometryDirective) {
        this.ngZone = ngZone;
        this.geometryDirective = geometryDirective;
    }
    ngOnChanges() {
        this.ngZone.runOutsideAngular(() => {
            if (this.attribute) {
                this.attribute.needsUpdate = true;
            }
        });
    }
    ngOnInit() {
        this.ngZone.runOutsideAngular(() => {
            if (this.geometryDirective && this.attach) {
                this._attribute = new InterleavedBufferAttribute(...this.args);
                if (this.attribute) {
                    this.geometryDirective.bufferGeometry.setAttribute(this.attach, this.attribute);
                }
            }
        });
    }
    ngOnDestroy() {
        this.ngZone.runOutsideAngular(() => {
            if (this.geometryDirective && this.attach) {
                this.geometryDirective.bufferGeometry.deleteAttribute(this.attach);
            }
        });
    }
    get attribute() {
        return this._attribute;
    }
}
InterleavedBufferAttributeDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: InterleavedBufferAttributeDirective, deps: [{ token: i0.NgZone }, { token: i1.ThreeBufferGeometry, optional: true }], target: i0.ɵɵFactoryTarget.Directive });
InterleavedBufferAttributeDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: InterleavedBufferAttributeDirective, selector: "ngt-interleaved-buffer-attribute", inputs: { args: "args", attach: "attach" }, exportAs: ["ngtInterleavedBufferAttribute"], usesOnChanges: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: InterleavedBufferAttributeDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-interleaved-buffer-attribute',
                    exportAs: 'ngtInterleavedBufferAttribute',
                }]
        }], ctorParameters: function () { return [{ type: i0.NgZone }, { type: i1.ThreeBufferGeometry, decorators: [{
                    type: Optional
                }] }]; }, propDecorators: { args: [{
                type: Input
            }], attach: [{
                type: Input
            }] } });

class InterleavedBufferAttributeModule {
}
InterleavedBufferAttributeModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: InterleavedBufferAttributeModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
InterleavedBufferAttributeModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: InterleavedBufferAttributeModule, declarations: [InterleavedBufferAttributeDirective], exports: [InterleavedBufferAttributeDirective] });
InterleavedBufferAttributeModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: InterleavedBufferAttributeModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: InterleavedBufferAttributeModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [InterleavedBufferAttributeDirective],
                    exports: [InterleavedBufferAttributeDirective],
                }]
        }] });

// GENERATED
class BufferAttributeDirective extends ThreeAttribute {
    constructor() {
        super(...arguments);
        this.attributeType = BufferAttribute;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
BufferAttributeDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: BufferAttributeDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
BufferAttributeDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: BufferAttributeDirective, selector: "ngt-buffer-attribute", inputs: { args: "args" }, providers: [
        {
            provide: ThreeAttribute,
            useExisting: BufferAttributeDirective,
        },
    ], exportAs: ["ngtBufferAttribute"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: BufferAttributeDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-buffer-attribute',
                    exportAs: 'ngtBufferAttribute',
                    providers: [
                        {
                            provide: ThreeAttribute,
                            useExisting: BufferAttributeDirective,
                        },
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });

// GENERATED
class ThreeBufferAttributeModule {
}
ThreeBufferAttributeModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeBufferAttributeModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ThreeBufferAttributeModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeBufferAttributeModule, declarations: [BufferAttributeDirective], exports: [BufferAttributeDirective] });
ThreeBufferAttributeModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeBufferAttributeModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeBufferAttributeModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [BufferAttributeDirective],
                    exports: [BufferAttributeDirective],
                }]
        }] });

// GENERATED

// GENERATED
class InstancedBufferAttributeDirective extends ThreeAttribute {
    constructor() {
        super(...arguments);
        this.attributeType = InstancedBufferAttribute;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
InstancedBufferAttributeDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: InstancedBufferAttributeDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
InstancedBufferAttributeDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: InstancedBufferAttributeDirective, selector: "ngt-instanced-buffer-attribute", inputs: { args: "args" }, providers: [
        {
            provide: ThreeAttribute,
            useExisting: InstancedBufferAttributeDirective,
        },
    ], exportAs: ["ngtInstancedBufferAttribute"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: InstancedBufferAttributeDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-instanced-buffer-attribute',
                    exportAs: 'ngtInstancedBufferAttribute',
                    providers: [
                        {
                            provide: ThreeAttribute,
                            useExisting: InstancedBufferAttributeDirective,
                        },
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });

// GENERATED
class ThreeInstancedBufferAttributeModule {
}
ThreeInstancedBufferAttributeModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeInstancedBufferAttributeModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ThreeInstancedBufferAttributeModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeInstancedBufferAttributeModule, declarations: [InstancedBufferAttributeDirective], exports: [InstancedBufferAttributeDirective] });
ThreeInstancedBufferAttributeModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeInstancedBufferAttributeModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeInstancedBufferAttributeModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [InstancedBufferAttributeDirective],
                    exports: [InstancedBufferAttributeDirective],
                }]
        }] });

// GENERATED

// GENERATED
class Float16BufferAttributeDirective extends ThreeAttribute {
    constructor() {
        super(...arguments);
        this.attributeType = Float16BufferAttribute;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
Float16BufferAttributeDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: Float16BufferAttributeDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
Float16BufferAttributeDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: Float16BufferAttributeDirective, selector: "ngt-float16-buffer-attribute", inputs: { args: "args" }, providers: [
        {
            provide: ThreeAttribute,
            useExisting: Float16BufferAttributeDirective,
        },
    ], exportAs: ["ngtFloat16BufferAttribute"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: Float16BufferAttributeDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-float16-buffer-attribute',
                    exportAs: 'ngtFloat16BufferAttribute',
                    providers: [
                        {
                            provide: ThreeAttribute,
                            useExisting: Float16BufferAttributeDirective,
                        },
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });

// GENERATED
class ThreeFloat16BufferAttributeModule {
}
ThreeFloat16BufferAttributeModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeFloat16BufferAttributeModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ThreeFloat16BufferAttributeModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeFloat16BufferAttributeModule, declarations: [Float16BufferAttributeDirective], exports: [Float16BufferAttributeDirective] });
ThreeFloat16BufferAttributeModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeFloat16BufferAttributeModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeFloat16BufferAttributeModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [Float16BufferAttributeDirective],
                    exports: [Float16BufferAttributeDirective],
                }]
        }] });

// GENERATED

// GENERATED
class Float32BufferAttributeDirective extends ThreeAttribute {
    constructor() {
        super(...arguments);
        this.attributeType = Float32BufferAttribute;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
Float32BufferAttributeDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: Float32BufferAttributeDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
Float32BufferAttributeDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: Float32BufferAttributeDirective, selector: "ngt-float32-buffer-attribute", inputs: { args: "args" }, providers: [
        {
            provide: ThreeAttribute,
            useExisting: Float32BufferAttributeDirective,
        },
    ], exportAs: ["ngtFloat32BufferAttribute"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: Float32BufferAttributeDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-float32-buffer-attribute',
                    exportAs: 'ngtFloat32BufferAttribute',
                    providers: [
                        {
                            provide: ThreeAttribute,
                            useExisting: Float32BufferAttributeDirective,
                        },
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });

// GENERATED
class ThreeFloat32BufferAttributeModule {
}
ThreeFloat32BufferAttributeModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeFloat32BufferAttributeModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ThreeFloat32BufferAttributeModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeFloat32BufferAttributeModule, declarations: [Float32BufferAttributeDirective], exports: [Float32BufferAttributeDirective] });
ThreeFloat32BufferAttributeModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeFloat32BufferAttributeModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeFloat32BufferAttributeModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [Float32BufferAttributeDirective],
                    exports: [Float32BufferAttributeDirective],
                }]
        }] });

// GENERATED

// GENERATED
class Float64BufferAttributeDirective extends ThreeAttribute {
    constructor() {
        super(...arguments);
        this.attributeType = Float64BufferAttribute;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
Float64BufferAttributeDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: Float64BufferAttributeDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
Float64BufferAttributeDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: Float64BufferAttributeDirective, selector: "ngt-float64-buffer-attribute", inputs: { args: "args" }, providers: [
        {
            provide: ThreeAttribute,
            useExisting: Float64BufferAttributeDirective,
        },
    ], exportAs: ["ngtFloat64BufferAttribute"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: Float64BufferAttributeDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-float64-buffer-attribute',
                    exportAs: 'ngtFloat64BufferAttribute',
                    providers: [
                        {
                            provide: ThreeAttribute,
                            useExisting: Float64BufferAttributeDirective,
                        },
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });

// GENERATED
class ThreeFloat64BufferAttributeModule {
}
ThreeFloat64BufferAttributeModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeFloat64BufferAttributeModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ThreeFloat64BufferAttributeModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeFloat64BufferAttributeModule, declarations: [Float64BufferAttributeDirective], exports: [Float64BufferAttributeDirective] });
ThreeFloat64BufferAttributeModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeFloat64BufferAttributeModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeFloat64BufferAttributeModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [Float64BufferAttributeDirective],
                    exports: [Float64BufferAttributeDirective],
                }]
        }] });

// GENERATED

// GENERATED
class Int8BufferAttributeDirective extends ThreeAttribute {
    constructor() {
        super(...arguments);
        this.attributeType = Int8BufferAttribute;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
Int8BufferAttributeDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: Int8BufferAttributeDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
Int8BufferAttributeDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: Int8BufferAttributeDirective, selector: "ngt-int8-buffer-attribute", inputs: { args: "args" }, providers: [
        {
            provide: ThreeAttribute,
            useExisting: Int8BufferAttributeDirective,
        },
    ], exportAs: ["ngtInt8BufferAttribute"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: Int8BufferAttributeDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-int8-buffer-attribute',
                    exportAs: 'ngtInt8BufferAttribute',
                    providers: [
                        {
                            provide: ThreeAttribute,
                            useExisting: Int8BufferAttributeDirective,
                        },
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });

// GENERATED
class ThreeInt8BufferAttributeModule {
}
ThreeInt8BufferAttributeModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeInt8BufferAttributeModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ThreeInt8BufferAttributeModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeInt8BufferAttributeModule, declarations: [Int8BufferAttributeDirective], exports: [Int8BufferAttributeDirective] });
ThreeInt8BufferAttributeModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeInt8BufferAttributeModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeInt8BufferAttributeModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [Int8BufferAttributeDirective],
                    exports: [Int8BufferAttributeDirective],
                }]
        }] });

// GENERATED

// GENERATED
class Int16BufferAttributeDirective extends ThreeAttribute {
    constructor() {
        super(...arguments);
        this.attributeType = Int16BufferAttribute;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
Int16BufferAttributeDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: Int16BufferAttributeDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
Int16BufferAttributeDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: Int16BufferAttributeDirective, selector: "ngt-int16-buffer-attribute", inputs: { args: "args" }, providers: [
        {
            provide: ThreeAttribute,
            useExisting: Int16BufferAttributeDirective,
        },
    ], exportAs: ["ngtInt16BufferAttribute"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: Int16BufferAttributeDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-int16-buffer-attribute',
                    exportAs: 'ngtInt16BufferAttribute',
                    providers: [
                        {
                            provide: ThreeAttribute,
                            useExisting: Int16BufferAttributeDirective,
                        },
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });

// GENERATED
class ThreeInt16BufferAttributeModule {
}
ThreeInt16BufferAttributeModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeInt16BufferAttributeModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ThreeInt16BufferAttributeModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeInt16BufferAttributeModule, declarations: [Int16BufferAttributeDirective], exports: [Int16BufferAttributeDirective] });
ThreeInt16BufferAttributeModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeInt16BufferAttributeModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeInt16BufferAttributeModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [Int16BufferAttributeDirective],
                    exports: [Int16BufferAttributeDirective],
                }]
        }] });

// GENERATED

// GENERATED
class Int32BufferAttributeDirective extends ThreeAttribute {
    constructor() {
        super(...arguments);
        this.attributeType = Int32BufferAttribute;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
Int32BufferAttributeDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: Int32BufferAttributeDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
Int32BufferAttributeDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: Int32BufferAttributeDirective, selector: "ngt-int32-buffer-attribute", inputs: { args: "args" }, providers: [
        {
            provide: ThreeAttribute,
            useExisting: Int32BufferAttributeDirective,
        },
    ], exportAs: ["ngtInt32BufferAttribute"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: Int32BufferAttributeDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-int32-buffer-attribute',
                    exportAs: 'ngtInt32BufferAttribute',
                    providers: [
                        {
                            provide: ThreeAttribute,
                            useExisting: Int32BufferAttributeDirective,
                        },
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });

// GENERATED
class ThreeInt32BufferAttributeModule {
}
ThreeInt32BufferAttributeModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeInt32BufferAttributeModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ThreeInt32BufferAttributeModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeInt32BufferAttributeModule, declarations: [Int32BufferAttributeDirective], exports: [Int32BufferAttributeDirective] });
ThreeInt32BufferAttributeModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeInt32BufferAttributeModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeInt32BufferAttributeModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [Int32BufferAttributeDirective],
                    exports: [Int32BufferAttributeDirective],
                }]
        }] });

// GENERATED

// GENERATED
class Uint8BufferAttributeDirective extends ThreeAttribute {
    constructor() {
        super(...arguments);
        this.attributeType = Uint8BufferAttribute;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
Uint8BufferAttributeDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: Uint8BufferAttributeDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
Uint8BufferAttributeDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: Uint8BufferAttributeDirective, selector: "ngt-uint8-buffer-attribute", inputs: { args: "args" }, providers: [
        {
            provide: ThreeAttribute,
            useExisting: Uint8BufferAttributeDirective,
        },
    ], exportAs: ["ngtUint8BufferAttribute"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: Uint8BufferAttributeDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-uint8-buffer-attribute',
                    exportAs: 'ngtUint8BufferAttribute',
                    providers: [
                        {
                            provide: ThreeAttribute,
                            useExisting: Uint8BufferAttributeDirective,
                        },
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });

// GENERATED
class ThreeUint8BufferAttributeModule {
}
ThreeUint8BufferAttributeModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeUint8BufferAttributeModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ThreeUint8BufferAttributeModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeUint8BufferAttributeModule, declarations: [Uint8BufferAttributeDirective], exports: [Uint8BufferAttributeDirective] });
ThreeUint8BufferAttributeModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeUint8BufferAttributeModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeUint8BufferAttributeModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [Uint8BufferAttributeDirective],
                    exports: [Uint8BufferAttributeDirective],
                }]
        }] });

// GENERATED

// GENERATED
class Uint16BufferAttributeDirective extends ThreeAttribute {
    constructor() {
        super(...arguments);
        this.attributeType = Uint16BufferAttribute;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
Uint16BufferAttributeDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: Uint16BufferAttributeDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
Uint16BufferAttributeDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: Uint16BufferAttributeDirective, selector: "ngt-uint16-buffer-attribute", inputs: { args: "args" }, providers: [
        {
            provide: ThreeAttribute,
            useExisting: Uint16BufferAttributeDirective,
        },
    ], exportAs: ["ngtUint16BufferAttribute"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: Uint16BufferAttributeDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-uint16-buffer-attribute',
                    exportAs: 'ngtUint16BufferAttribute',
                    providers: [
                        {
                            provide: ThreeAttribute,
                            useExisting: Uint16BufferAttributeDirective,
                        },
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });

// GENERATED
class ThreeUint16BufferAttributeModule {
}
ThreeUint16BufferAttributeModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeUint16BufferAttributeModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ThreeUint16BufferAttributeModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeUint16BufferAttributeModule, declarations: [Uint16BufferAttributeDirective], exports: [Uint16BufferAttributeDirective] });
ThreeUint16BufferAttributeModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeUint16BufferAttributeModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeUint16BufferAttributeModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [Uint16BufferAttributeDirective],
                    exports: [Uint16BufferAttributeDirective],
                }]
        }] });

// GENERATED

// GENERATED
class Uint32BufferAttributeDirective extends ThreeAttribute {
    constructor() {
        super(...arguments);
        this.attributeType = Uint32BufferAttribute;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
Uint32BufferAttributeDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: Uint32BufferAttributeDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
Uint32BufferAttributeDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: Uint32BufferAttributeDirective, selector: "ngt-uint32-buffer-attribute", inputs: { args: "args" }, providers: [
        {
            provide: ThreeAttribute,
            useExisting: Uint32BufferAttributeDirective,
        },
    ], exportAs: ["ngtUint32BufferAttribute"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: Uint32BufferAttributeDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-uint32-buffer-attribute',
                    exportAs: 'ngtUint32BufferAttribute',
                    providers: [
                        {
                            provide: ThreeAttribute,
                            useExisting: Uint32BufferAttributeDirective,
                        },
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });

// GENERATED
class ThreeUint32BufferAttributeModule {
}
ThreeUint32BufferAttributeModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeUint32BufferAttributeModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ThreeUint32BufferAttributeModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeUint32BufferAttributeModule, declarations: [Uint32BufferAttributeDirective], exports: [Uint32BufferAttributeDirective] });
ThreeUint32BufferAttributeModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeUint32BufferAttributeModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeUint32BufferAttributeModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [Uint32BufferAttributeDirective],
                    exports: [Uint32BufferAttributeDirective],
                }]
        }] });

// GENERATED

// GENERATED
class Uint8ClampedBufferAttributeDirective extends ThreeAttribute {
    constructor() {
        super(...arguments);
        this.attributeType = Uint8ClampedBufferAttribute;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
Uint8ClampedBufferAttributeDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: Uint8ClampedBufferAttributeDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
Uint8ClampedBufferAttributeDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: Uint8ClampedBufferAttributeDirective, selector: "ngt-uint8-clamped-buffer-attribute", inputs: { args: "args" }, providers: [
        {
            provide: ThreeAttribute,
            useExisting: Uint8ClampedBufferAttributeDirective,
        },
    ], exportAs: ["ngtUint8ClampedBufferAttribute"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: Uint8ClampedBufferAttributeDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-uint8-clamped-buffer-attribute',
                    exportAs: 'ngtUint8ClampedBufferAttribute',
                    providers: [
                        {
                            provide: ThreeAttribute,
                            useExisting: Uint8ClampedBufferAttributeDirective,
                        },
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });

// GENERATED
class ThreeUint8ClampedBufferAttributeModule {
}
ThreeUint8ClampedBufferAttributeModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeUint8ClampedBufferAttributeModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ThreeUint8ClampedBufferAttributeModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeUint8ClampedBufferAttributeModule, declarations: [Uint8ClampedBufferAttributeDirective], exports: [Uint8ClampedBufferAttributeDirective] });
ThreeUint8ClampedBufferAttributeModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeUint8ClampedBufferAttributeModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeUint8ClampedBufferAttributeModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [Uint8ClampedBufferAttributeDirective],
                    exports: [Uint8ClampedBufferAttributeDirective],
                }]
        }] });

// GENERATED

// GENERATED

/**
 * Generated bundle index. Do not edit.
 */

export { BufferAttributeDirective, Float16BufferAttributeDirective, Float32BufferAttributeDirective, Float64BufferAttributeDirective, InstancedBufferAttributeDirective, Int16BufferAttributeDirective, Int32BufferAttributeDirective, Int8BufferAttributeDirective, InterleavedBufferAttributeDirective, InterleavedBufferAttributeModule, ThreeBufferAttributeModule, ThreeFloat16BufferAttributeModule, ThreeFloat32BufferAttributeModule, ThreeFloat64BufferAttributeModule, ThreeInstancedBufferAttributeModule, ThreeInt16BufferAttributeModule, ThreeInt32BufferAttributeModule, ThreeInt8BufferAttributeModule, ThreeUint16BufferAttributeModule, ThreeUint32BufferAttributeModule, ThreeUint8BufferAttributeModule, ThreeUint8ClampedBufferAttributeModule, Uint16BufferAttributeDirective, Uint32BufferAttributeDirective, Uint8BufferAttributeDirective, Uint8ClampedBufferAttributeDirective };
