// @angular-three/core/fesm2015/angular-three-core-scene.js
import * as i0 from '@angular/core';
import { Directive, NgModule } from '@angular/core';
import { ThreeObject3d, OBJECT_3D_CONTROLLER_PROVIDER } from '@angular-three/core';
import { Scene } from 'three';

class SceneDirective extends ThreeObject3d {
    ngOnInit() {
        this.init();
    }
    initObject() {
        this._scene = new Scene();
    }
    get object3d() {
        return this._scene;
    }
}
SceneDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: SceneDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
SceneDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: SceneDirective, selector: "ngt-scene", providers: [
        { provide: ThreeObject3d, useExisting: SceneDirective },
        OBJECT_3D_CONTROLLER_PROVIDER,
    ], exportAs: ["ngtScene"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: SceneDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-scene',
                    exportAs: 'ngtScene',
                    providers: [
                        { provide: ThreeObject3d, useExisting: SceneDirective },
                        OBJECT_3D_CONTROLLER_PROVIDER,
                    ],
                }]
        }] });

class ThreeSceneModule {
}
ThreeSceneModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeSceneModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ThreeSceneModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeSceneModule, declarations: [SceneDirective], exports: [SceneDirective] });
ThreeSceneModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeSceneModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeSceneModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [SceneDirective],
                    exports: [SceneDirective],
                }]
        }] });

/**
 * Generated bundle index. Do not edit.
 */

export { SceneDirective, ThreeSceneModule };
