// @angular-three/core/fesm2015/angular-three-core-lod.js
import { ThreeObject3d, OBJECT_3D_CONTROLLER_PROVIDER } from '@angular-three/core';
import * as i0 from '@angular/core';
import { Directive, NgModule } from '@angular/core';
import { LOD } from 'three';

class LodDirective extends ThreeObject3d {
    ngOnInit() {
        this.init();
    }
    initObject() {
        this._lod = new LOD();
    }
    get object3d() {
        return this._lod;
    }
}
LodDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: LodDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
LodDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: LodDirective, selector: "ngt-lod", providers: [
        { provide: ThreeObject3d, useExisting: LodDirective },
        OBJECT_3D_CONTROLLER_PROVIDER,
    ], exportAs: ["ngtLod"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: LodDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-lod',
                    exportAs: 'ngtLod',
                    providers: [
                        { provide: ThreeObject3d, useExisting: LodDirective },
                        OBJECT_3D_CONTROLLER_PROVIDER,
                    ],
                }]
        }] });

class ThreeLodModule {
}
ThreeLodModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeLodModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ThreeLodModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeLodModule, declarations: [LodDirective], exports: [LodDirective] });
ThreeLodModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeLodModule });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ThreeLodModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [LodDirective],
                    exports: [LodDirective],
                }]
        }] });

/**
 * Generated bundle index. Do not edit.
 */

export { LodDirective, ThreeLodModule };
