// @angular-three/core/esm2015/geometries/lib/circle-buffer-geometry/circle-buffer-geometry.directive.js
// GENERATED
import { ThreeBufferGeometry } from '@angular-three/core';
import { Directive, Input } from '@angular/core';
import { CircleBufferGeometry } from 'three';
import * as i0 from "@angular/core";
export class CircleBufferGeometryDirective extends ThreeBufferGeometry {
    constructor() {
        super(...arguments);
        this.geometryType = CircleBufferGeometry;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
CircleBufferGeometryDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: CircleBufferGeometryDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
CircleBufferGeometryDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: CircleBufferGeometryDirective, selector: "ngt-circle-buffer-geometry,ngt-circle-geometry", inputs: { args: "args" }, providers: [
        {
            provide: ThreeBufferGeometry,
            useExisting: CircleBufferGeometryDirective,
        },
    ], exportAs: ["ngtCircleBufferGeometry"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: CircleBufferGeometryDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-circle-buffer-geometry,ngt-circle-geometry',
                    exportAs: 'ngtCircleBufferGeometry',
                    providers: [
                        {
                            provide: ThreeBufferGeometry,
                            useExisting: CircleBufferGeometryDirective,
                        },
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });
