// @angular-three/core/esm2015/materials/lib/line-dashed-material/line-dashed-material.directive.js
// GENERATED
import { ThreeMaterial } from '@angular-three/core';
import { Directive } from '@angular/core';
import { LineDashedMaterial } from 'three';
import * as i0 from "@angular/core";
export class LineDashedMaterialDirective extends ThreeMaterial {
    constructor() {
        super(...arguments);
        this.materialType = LineDashedMaterial;
    }
}
LineDashedMaterialDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: LineDashedMaterialDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
LineDashedMaterialDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: LineDashedMaterialDirective, selector: "ngt-line-dashed-material", providers: [
        { provide: ThreeMaterial, useExisting: LineDashedMaterialDirective },
    ], exportAs: ["ngtLineDashedMaterial"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: LineDashedMaterialDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-line-dashed-material',
                    exportAs: 'ngtLineDashedMaterial',
                    providers: [
                        { provide: ThreeMaterial, useExisting: LineDashedMaterialDirective },
                    ],
                }]
        }] });
