// @angular-three/core/esm2015/lights/lib/hemisphere-light-probe/hemisphere-light-probe.directive.js
// GENERATED
import { ThreeLight, ThreeObject3d, OBJECT_3D_CONTROLLER_PROVIDER, } from '@angular-three/core';
import { Directive, Input } from '@angular/core';
import { HemisphereLightProbe } from 'three';
import * as i0 from "@angular/core";
export class HemisphereLightProbeDirective extends ThreeLight {
    constructor() {
        super(...arguments);
        this.lightType = HemisphereLightProbe;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
HemisphereLightProbeDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: HemisphereLightProbeDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
HemisphereLightProbeDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: HemisphereLightProbeDirective, selector: "ngt-hemisphere-light-probe", inputs: { args: "args" }, providers: [
        {
            provide: ThreeObject3d,
            useExisting: HemisphereLightProbeDirective,
        },
        OBJECT_3D_CONTROLLER_PROVIDER,
    ], exportAs: ["ngtHemisphereLightProbe"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: HemisphereLightProbeDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-hemisphere-light-probe',
                    exportAs: 'ngtHemisphereLightProbe',
                    providers: [
                        {
                            provide: ThreeObject3d,
                            useExisting: HemisphereLightProbeDirective,
                        },
                        OBJECT_3D_CONTROLLER_PROVIDER,
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });
