// @angular-three/core/esm2015/meshes/lib/mesh/mesh.directive.js
import { OBJECT_3D_CONTROLLER_PROVIDER, ThreeMesh, ThreeObject3d, } from '@angular-three/core';
import { Directive } from '@angular/core';
import { Mesh } from 'three';
import * as i0 from "@angular/core";
export class MeshDirective extends ThreeMesh {
    constructor() {
        super(...arguments);
        this.meshType = Mesh;
    }
}
MeshDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: MeshDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
MeshDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: MeshDirective, selector: "ngt-mesh", providers: [
        { provide: ThreeObject3d, useExisting: MeshDirective },
        OBJECT_3D_CONTROLLER_PROVIDER,
    ], exportAs: ["ngtMesh"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: MeshDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-mesh',
                    exportAs: 'ngtMesh',
                    providers: [
                        { provide: ThreeObject3d, useExisting: MeshDirective },
                        OBJECT_3D_CONTROLLER_PROVIDER,
                    ],
                }]
        }] });
