// @angular-three/core/esm2015/materials/lib/points-material/points-material.directive.js
// GENERATED
import { ThreeMaterial } from '@angular-three/core';
import { Directive } from '@angular/core';
import { PointsMaterial } from 'three';
import * as i0 from "@angular/core";
export class PointsMaterialDirective extends ThreeMaterial {
    constructor() {
        super(...arguments);
        this.materialType = PointsMaterial;
    }
}
PointsMaterialDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: PointsMaterialDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
PointsMaterialDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: PointsMaterialDirective, selector: "ngt-points-material", providers: [{ provide: ThreeMaterial, useExisting: PointsMaterialDirective }], exportAs: ["ngtPointsMaterial"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: PointsMaterialDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-points-material',
                    exportAs: 'ngtPointsMaterial',
                    providers: [{ provide: ThreeMaterial, useExisting: PointsMaterialDirective }],
                }]
        }] });
