// @angular-three/core/esm2015/geometries/lib/cone-buffer-geometry/cone-buffer-geometry.directive.js
// GENERATED
import { ThreeBufferGeometry } from '@angular-three/core';
import { Directive, Input } from '@angular/core';
import { ConeBufferGeometry } from 'three';
import * as i0 from "@angular/core";
export class ConeBufferGeometryDirective extends ThreeBufferGeometry {
    constructor() {
        super(...arguments);
        this.geometryType = ConeBufferGeometry;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
ConeBufferGeometryDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ConeBufferGeometryDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
ConeBufferGeometryDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: ConeBufferGeometryDirective, selector: "ngt-cone-buffer-geometry,ngt-cone-geometry", inputs: { args: "args" }, providers: [
        {
            provide: ThreeBufferGeometry,
            useExisting: ConeBufferGeometryDirective,
        },
    ], exportAs: ["ngtConeBufferGeometry"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ConeBufferGeometryDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-cone-buffer-geometry,ngt-cone-geometry',
                    exportAs: 'ngtConeBufferGeometry',
                    providers: [
                        {
                            provide: ThreeBufferGeometry,
                            useExisting: ConeBufferGeometryDirective,
                        },
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });
