// @angular-three/core/esm2015/geometries/lib/teapot-geometry/teapot-geometry.directive.js
// GENERATED
import { ThreeBufferGeometry } from '@angular-three/core';
import { Directive, Input } from '@angular/core';
import { TeapotGeometry } from 'three/examples/jsm/geometries/TeapotGeometry';
import * as i0 from "@angular/core";
export class TeapotGeometryDirective extends ThreeBufferGeometry {
    constructor() {
        super(...arguments);
        this.geometryType = TeapotGeometry;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
TeapotGeometryDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: TeapotGeometryDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
TeapotGeometryDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: TeapotGeometryDirective, selector: "ngt-teapot-geometry", inputs: { args: "args" }, providers: [
        {
            provide: ThreeBufferGeometry,
            useExisting: TeapotGeometryDirective,
        },
    ], exportAs: ["ngtTeapotGeometry"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: TeapotGeometryDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-teapot-geometry',
                    exportAs: 'ngtTeapotGeometry',
                    providers: [
                        {
                            provide: ThreeBufferGeometry,
                            useExisting: TeapotGeometryDirective,
                        },
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });
