// @angular-three/core/esm2015/geometries/lib/instanced-buffer-geometry/instanced-buffer-geometry.directive.js
// GENERATED
import { ThreeBufferGeometry } from '@angular-three/core';
import { Directive, Input } from '@angular/core';
import { InstancedBufferGeometry } from 'three';
import * as i0 from "@angular/core";
export class InstancedBufferGeometryDirective extends ThreeBufferGeometry {
    constructor() {
        super(...arguments);
        this.geometryType = InstancedBufferGeometry;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
InstancedBufferGeometryDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: InstancedBufferGeometryDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
InstancedBufferGeometryDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: InstancedBufferGeometryDirective, selector: "ngt-instanced-buffer-geometry,ngt-instanced-geometry", inputs: { args: "args" }, providers: [
        {
            provide: ThreeBufferGeometry,
            useExisting: InstancedBufferGeometryDirective,
        },
    ], exportAs: ["ngtInstancedBufferGeometry"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: InstancedBufferGeometryDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-instanced-buffer-geometry,ngt-instanced-geometry',
                    exportAs: 'ngtInstancedBufferGeometry',
                    providers: [
                        {
                            provide: ThreeBufferGeometry,
                            useExisting: InstancedBufferGeometryDirective,
                        },
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });
