// @angular-three/core/esm2015/attributes/lib/float64-buffer-attribute/float64-buffer-attribute.directive.js
// GENERATED
import { ThreeAttribute } from '@angular-three/core';
import { Directive, Input } from '@angular/core';
import { Float64BufferAttribute } from 'three';
import * as i0 from "@angular/core";
export class Float64BufferAttributeDirective extends ThreeAttribute {
    constructor() {
        super(...arguments);
        this.attributeType = Float64BufferAttribute;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
Float64BufferAttributeDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: Float64BufferAttributeDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
Float64BufferAttributeDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: Float64BufferAttributeDirective, selector: "ngt-float64-buffer-attribute", inputs: { args: "args" }, providers: [
        {
            provide: ThreeAttribute,
            useExisting: Float64BufferAttributeDirective,
        },
    ], exportAs: ["ngtFloat64BufferAttribute"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: Float64BufferAttributeDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-float64-buffer-attribute',
                    exportAs: 'ngtFloat64BufferAttribute',
                    providers: [
                        {
                            provide: ThreeAttribute,
                            useExisting: Float64BufferAttributeDirective,
                        },
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });
