// @angular-three/core/esm2015/cameras/lib/stereo-camera/stereo-camera.directive.js
// GENERATED
import { ThreeCamera, ThreeObject3d, OBJECT_3D_CONTROLLER_PROVIDER, } from '@angular-three/core';
import { Directive, Input } from '@angular/core';
import { StereoCamera } from 'three';
import * as i0 from "@angular/core";
export class StereoCameraDirective extends ThreeCamera {
    constructor() {
        super(...arguments);
        this.cameraType = StereoCamera;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
StereoCameraDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: StereoCameraDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
StereoCameraDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: StereoCameraDirective, selector: "ngt-stereo-camera", inputs: { args: "args" }, providers: [
        {
            provide: ThreeObject3d,
            useExisting: StereoCameraDirective,
        },
        OBJECT_3D_CONTROLLER_PROVIDER,
    ], exportAs: ["ngtStereoCamera"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: StereoCameraDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-stereo-camera',
                    exportAs: 'ngtStereoCamera',
                    providers: [
                        {
                            provide: ThreeObject3d,
                            useExisting: StereoCameraDirective,
                        },
                        OBJECT_3D_CONTROLLER_PROVIDER,
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });
