// @angular-three/core/esm2015/stats/lib/stats.directive.js
import { AnimationStore } from '@angular-three/core';
import { DOCUMENT } from '@angular/common';
import { Directive, Inject } from '@angular/core';
import Stats from 'three/examples/jsm/libs/stats.module';
import * as i0 from "@angular/core";
import * as i1 from "@angular-three/core";
export class StatsDirective {
    constructor(animationStore, document) {
        this.animationStore = animationStore;
        this.document = document;
    }
    ngOnInit() {
        this.stats = Stats();
        this.document.body.appendChild(this.stats.dom);
        this.animationStore.registerAnimation(() => {
            this.stats.update();
        });
    }
    ngOnDestroy() {
        this.stats.end();
    }
}
StatsDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: StatsDirective, deps: [{ token: i1.AnimationStore }, { token: DOCUMENT }], target: i0.ɵɵFactoryTarget.Directive });
StatsDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: StatsDirective, selector: "ngt-stats", exportAs: ["ngtStats"], ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: StatsDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-stats',
                    exportAs: 'ngtStats',
                }]
        }], ctorParameters: function () { return [{ type: i1.AnimationStore }, { type: Document, decorators: [{
                    type: Inject,
                    args: [DOCUMENT]
                }] }]; } });
