// @angular-three/core/esm2015/materials/lib/raw-shader-material/raw-shader-material.directive.js
// GENERATED
import { ThreeMaterial } from '@angular-three/core';
import { Directive } from '@angular/core';
import { RawShaderMaterial } from 'three';
import * as i0 from "@angular/core";
export class RawShaderMaterialDirective extends ThreeMaterial {
    constructor() {
        super(...arguments);
        this.materialType = RawShaderMaterial;
    }
}
RawShaderMaterialDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: RawShaderMaterialDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
RawShaderMaterialDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: RawShaderMaterialDirective, selector: "ngt-raw-shader-material", providers: [
        { provide: ThreeMaterial, useExisting: RawShaderMaterialDirective },
    ], exportAs: ["ngtRawShaderMaterial"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: RawShaderMaterialDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-raw-shader-material',
                    exportAs: 'ngtRawShaderMaterial',
                    providers: [
                        { provide: ThreeMaterial, useExisting: RawShaderMaterialDirective },
                    ],
                }]
        }] });
