// @angular-three/core/esm2015/helpers/lib/point-light-helper/point-light-helper.directive.js
// GENERATED
import { ThreeHelper, ThreeObject3d, OBJECT_3D_CONTROLLER_PROVIDER, } from '@angular-three/core';
import { Directive, Input } from '@angular/core';
import { PointLightHelper } from 'three';
import * as i0 from "@angular/core";
export class PointLightHelperDirective extends ThreeHelper {
    constructor() {
        super(...arguments);
        this.helperType = PointLightHelper;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
PointLightHelperDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: PointLightHelperDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
PointLightHelperDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: PointLightHelperDirective, selector: "ngt-point-light-helper", inputs: { args: "args" }, providers: [
        {
            provide: ThreeObject3d,
            useExisting: PointLightHelperDirective,
        },
        OBJECT_3D_CONTROLLER_PROVIDER,
    ], exportAs: ["ngtPointLightHelper"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: PointLightHelperDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-point-light-helper',
                    exportAs: 'ngtPointLightHelper',
                    providers: [
                        {
                            provide: ThreeObject3d,
                            useExisting: PointLightHelperDirective,
                        },
                        OBJECT_3D_CONTROLLER_PROVIDER,
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });
