// @angular-three/core/esm2015/geometries/lib/polyhedron-buffer-geometry/polyhedron-buffer-geometry.directive.js
// GENERATED
import { ThreeBufferGeometry } from '@angular-three/core';
import { Directive, Input } from '@angular/core';
import { PolyhedronBufferGeometry } from 'three';
import * as i0 from "@angular/core";
export class PolyhedronBufferGeometryDirective extends ThreeBufferGeometry {
    constructor() {
        super(...arguments);
        this.geometryType = PolyhedronBufferGeometry;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
PolyhedronBufferGeometryDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: PolyhedronBufferGeometryDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
PolyhedronBufferGeometryDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: PolyhedronBufferGeometryDirective, selector: "ngt-polyhedron-buffer-geometry,ngt-polyhedron-geometry", inputs: { args: "args" }, providers: [
        {
            provide: ThreeBufferGeometry,
            useExisting: PolyhedronBufferGeometryDirective,
        },
    ], exportAs: ["ngtPolyhedronBufferGeometry"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: PolyhedronBufferGeometryDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-polyhedron-buffer-geometry,ngt-polyhedron-geometry',
                    exportAs: 'ngtPolyhedronBufferGeometry',
                    providers: [
                        {
                            provide: ThreeBufferGeometry,
                            useExisting: PolyhedronBufferGeometryDirective,
                        },
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });
