// @angular-three/core/esm2015/helpers/lib/box3-helper/box3-helper.directive.js
// GENERATED
import { ThreeHelper, ThreeObject3d, OBJECT_3D_CONTROLLER_PROVIDER, } from '@angular-three/core';
import { Directive, Input } from '@angular/core';
import { Box3Helper } from 'three';
import * as i0 from "@angular/core";
export class Box3HelperDirective extends ThreeHelper {
    constructor() {
        super(...arguments);
        this.helperType = Box3Helper;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
Box3HelperDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: Box3HelperDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
Box3HelperDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: Box3HelperDirective, selector: "ngt-box3-helper", inputs: { args: "args" }, providers: [
        {
            provide: ThreeObject3d,
            useExisting: Box3HelperDirective,
        },
        OBJECT_3D_CONTROLLER_PROVIDER,
    ], exportAs: ["ngtBox3Helper"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: Box3HelperDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-box3-helper',
                    exportAs: 'ngtBox3Helper',
                    providers: [
                        {
                            provide: ThreeObject3d,
                            useExisting: Box3HelperDirective,
                        },
                        OBJECT_3D_CONTROLLER_PROVIDER,
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });
