// @angular-three/core/esm2015/textures/lib/depth-texture/depth-texture.directive.js
// GENERATED
import { ThreeTexture } from '@angular-three/core';
import { Directive, Input } from '@angular/core';
import { DepthTexture } from 'three';
import * as i0 from "@angular/core";
export class DepthTextureDirective extends ThreeTexture {
    constructor() {
        super(...arguments);
        this.textureType = DepthTexture;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
DepthTextureDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: DepthTextureDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
DepthTextureDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: DepthTextureDirective, selector: "ngt-depth-texture", inputs: { args: "args" }, providers: [
        {
            provide: ThreeTexture,
            useExisting: DepthTextureDirective,
        },
    ], exportAs: ["ngtDepthTexture"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: DepthTextureDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-depth-texture',
                    exportAs: 'ngtDepthTexture',
                    providers: [
                        {
                            provide: ThreeTexture,
                            useExisting: DepthTextureDirective,
                        },
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });
