// @angular-three/core/esm2015/geometries/lib/tetrahedron-buffer-geometry/tetrahedron-buffer-geometry.directive.js
// GENERATED
import { ThreeBufferGeometry } from '@angular-three/core';
import { Directive, Input } from '@angular/core';
import { TetrahedronBufferGeometry } from 'three';
import * as i0 from "@angular/core";
export class TetrahedronBufferGeometryDirective extends ThreeBufferGeometry {
    constructor() {
        super(...arguments);
        this.geometryType = TetrahedronBufferGeometry;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
TetrahedronBufferGeometryDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: TetrahedronBufferGeometryDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
TetrahedronBufferGeometryDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: TetrahedronBufferGeometryDirective, selector: "ngt-tetrahedron-buffer-geometry,ngt-tetrahedron-geometry", inputs: { args: "args" }, providers: [
        {
            provide: ThreeBufferGeometry,
            useExisting: TetrahedronBufferGeometryDirective,
        },
    ], exportAs: ["ngtTetrahedronBufferGeometry"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: TetrahedronBufferGeometryDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-tetrahedron-buffer-geometry,ngt-tetrahedron-geometry',
                    exportAs: 'ngtTetrahedronBufferGeometry',
                    providers: [
                        {
                            provide: ThreeBufferGeometry,
                            useExisting: TetrahedronBufferGeometryDirective,
                        },
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });
