// @angular-three/core/esm2015/materials/lib/mesh-lambert-material/mesh-lambert-material.directive.js
// GENERATED
import { ThreeMaterial } from '@angular-three/core';
import { Directive } from '@angular/core';
import { MeshLambertMaterial } from 'three';
import * as i0 from "@angular/core";
export class MeshLambertMaterialDirective extends ThreeMaterial {
    constructor() {
        super(...arguments);
        this.materialType = MeshLambertMaterial;
    }
}
MeshLambertMaterialDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: MeshLambertMaterialDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
MeshLambertMaterialDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: MeshLambertMaterialDirective, selector: "ngt-mesh-lambert-material", providers: [
        { provide: ThreeMaterial, useExisting: MeshLambertMaterialDirective },
    ], exportAs: ["ngtMeshLambertMaterial"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: MeshLambertMaterialDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-mesh-lambert-material',
                    exportAs: 'ngtMeshLambertMaterial',
                    providers: [
                        { provide: ThreeMaterial, useExisting: MeshLambertMaterialDirective },
                    ],
                }]
        }] });
