// @angular-three/core/esm2015/textures/lib/data-texture2-darray/data-texture2-darray.directive.js
// GENERATED
import { ThreeTexture } from '@angular-three/core';
import { Directive, Input } from '@angular/core';
import { DataTexture2DArray } from 'three';
import * as i0 from "@angular/core";
export class DataTexture2DArrayDirective extends ThreeTexture {
    constructor() {
        super(...arguments);
        this.textureType = DataTexture2DArray;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
DataTexture2DArrayDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: DataTexture2DArrayDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
DataTexture2DArrayDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: DataTexture2DArrayDirective, selector: "ngt-data-texture2-darray", inputs: { args: "args" }, providers: [
        {
            provide: ThreeTexture,
            useExisting: DataTexture2DArrayDirective,
        },
    ], exportAs: ["ngtDataTexture2DArray"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: DataTexture2DArrayDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-data-texture2-darray',
                    exportAs: 'ngtDataTexture2DArray',
                    providers: [
                        {
                            provide: ThreeTexture,
                            useExisting: DataTexture2DArrayDirective,
                        },
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });
