// @angular-three/core/esm2015/geometries/lib/tube-buffer-geometry/tube-buffer-geometry.directive.js
// GENERATED
import { ThreeBufferGeometry } from '@angular-three/core';
import { Directive, Input } from '@angular/core';
import { TubeBufferGeometry } from 'three';
import * as i0 from "@angular/core";
export class TubeBufferGeometryDirective extends ThreeBufferGeometry {
    constructor() {
        super(...arguments);
        this.geometryType = TubeBufferGeometry;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
TubeBufferGeometryDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: TubeBufferGeometryDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
TubeBufferGeometryDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: TubeBufferGeometryDirective, selector: "ngt-tube-buffer-geometry,ngt-tube-geometry", inputs: { args: "args" }, providers: [
        {
            provide: ThreeBufferGeometry,
            useExisting: TubeBufferGeometryDirective,
        },
    ], exportAs: ["ngtTubeBufferGeometry"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: TubeBufferGeometryDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-tube-buffer-geometry,ngt-tube-geometry',
                    exportAs: 'ngtTubeBufferGeometry',
                    providers: [
                        {
                            provide: ThreeBufferGeometry,
                            useExisting: TubeBufferGeometryDirective,
                        },
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });
