// @angular-three/core/esm2015/lights/lib/rect-area-light/rect-area-light.directive.js
// GENERATED
import { ThreeLight, ThreeObject3d, OBJECT_3D_CONTROLLER_PROVIDER, } from '@angular-three/core';
import { Directive, Input } from '@angular/core';
import { RectAreaLight } from 'three';
import * as i0 from "@angular/core";
export class RectAreaLightDirective extends ThreeLight {
    constructor() {
        super(...arguments);
        this.lightType = RectAreaLight;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
RectAreaLightDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: RectAreaLightDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
RectAreaLightDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: RectAreaLightDirective, selector: "ngt-rect-area-light", inputs: { args: "args" }, providers: [
        {
            provide: ThreeObject3d,
            useExisting: RectAreaLightDirective,
        },
        OBJECT_3D_CONTROLLER_PROVIDER,
    ], exportAs: ["ngtRectAreaLight"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: RectAreaLightDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-rect-area-light',
                    exportAs: 'ngtRectAreaLight',
                    providers: [
                        {
                            provide: ThreeObject3d,
                            useExisting: RectAreaLightDirective,
                        },
                        OBJECT_3D_CONTROLLER_PROVIDER,
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });
