// @angular-three/core/esm2015/helpers/lib/grid-helper/grid-helper.directive.js
// GENERATED
import { ThreeHelper, ThreeObject3d, OBJECT_3D_CONTROLLER_PROVIDER, } from '@angular-three/core';
import { Directive, Input } from '@angular/core';
import { GridHelper } from 'three';
import * as i0 from "@angular/core";
export class GridHelperDirective extends ThreeHelper {
    constructor() {
        super(...arguments);
        this.helperType = GridHelper;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
GridHelperDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: GridHelperDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
GridHelperDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: GridHelperDirective, selector: "ngt-grid-helper", inputs: { args: "args" }, providers: [
        {
            provide: ThreeObject3d,
            useExisting: GridHelperDirective,
        },
        OBJECT_3D_CONTROLLER_PROVIDER,
    ], exportAs: ["ngtGridHelper"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: GridHelperDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-grid-helper',
                    exportAs: 'ngtGridHelper',
                    providers: [
                        {
                            provide: ThreeObject3d,
                            useExisting: GridHelperDirective,
                        },
                        OBJECT_3D_CONTROLLER_PROVIDER,
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });
