// @angular-three/core/esm2015/curves/lib/ellipse-curve/ellipse-curve.directive.js
// GENERATED
import { ThreeCurve } from '@angular-three/core';
import { Directive, Input } from '@angular/core';
import { EllipseCurve } from 'three';
import * as i0 from "@angular/core";
export class EllipseCurveDirective extends ThreeCurve {
    constructor() {
        super(...arguments);
        this.curveType = EllipseCurve;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
EllipseCurveDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: EllipseCurveDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
EllipseCurveDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: EllipseCurveDirective, selector: "ngt-ellipse-curve", inputs: { args: "args" }, providers: [
        {
            provide: ThreeCurve,
            useExisting: EllipseCurveDirective,
        },
    ], exportAs: ["ngtEllipseCurve"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: EllipseCurveDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-ellipse-curve',
                    exportAs: 'ngtEllipseCurve',
                    providers: [
                        {
                            provide: ThreeCurve,
                            useExisting: EllipseCurveDirective,
                        },
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });
