// @angular-three/core/esm2015/curves/lib/spline-curve/spline-curve.directive.js
// GENERATED
import { ThreeCurve } from '@angular-three/core';
import { Directive, Input } from '@angular/core';
import { SplineCurve } from 'three';
import * as i0 from "@angular/core";
export class SplineCurveDirective extends ThreeCurve {
    constructor() {
        super(...arguments);
        this.curveType = SplineCurve;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
SplineCurveDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: SplineCurveDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
SplineCurveDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: SplineCurveDirective, selector: "ngt-spline-curve", inputs: { args: "args" }, providers: [
        {
            provide: ThreeCurve,
            useExisting: SplineCurveDirective,
        },
    ], exportAs: ["ngtSplineCurve"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: SplineCurveDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-spline-curve',
                    exportAs: 'ngtSplineCurve',
                    providers: [
                        {
                            provide: ThreeCurve,
                            useExisting: SplineCurveDirective,
                        },
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });
