// @angular-three/core/esm2015/lines/lib/line-loop/line-loop.directive.js
// GENERATED
import { ThreeLine, ThreeObject3d, OBJECT_3D_CONTROLLER_PROVIDER, } from '@angular-three/core';
import { Directive } from '@angular/core';
import { LineLoop } from 'three';
import * as i0 from "@angular/core";
export class LineLoopDirective extends ThreeLine {
    constructor() {
        super(...arguments);
        this.lineType = LineLoop;
    }
}
LineLoopDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: LineLoopDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
LineLoopDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: LineLoopDirective, selector: "ngt-line-loop", providers: [
        {
            provide: ThreeObject3d,
            useExisting: LineLoopDirective,
        },
        OBJECT_3D_CONTROLLER_PROVIDER,
    ], exportAs: ["ngtLineLoop"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: LineLoopDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-line-loop',
                    exportAs: 'ngtLineLoop',
                    providers: [
                        {
                            provide: ThreeObject3d,
                            useExisting: LineLoopDirective,
                        },
                        OBJECT_3D_CONTROLLER_PROVIDER,
                    ],
                }]
        }] });
