// @angular-three/core/esm2015/cameras/lib/array-camera/array-camera.directive.js
// GENERATED
import { ThreeCamera, ThreeObject3d, OBJECT_3D_CONTROLLER_PROVIDER, } from '@angular-three/core';
import { Directive, Input } from '@angular/core';
import { ArrayCamera } from 'three';
import * as i0 from "@angular/core";
export class ArrayCameraDirective extends ThreeCamera {
    constructor() {
        super(...arguments);
        this.cameraType = ArrayCamera;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
ArrayCameraDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ArrayCameraDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
ArrayCameraDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: ArrayCameraDirective, selector: "ngt-array-camera", inputs: { args: "args" }, providers: [
        {
            provide: ThreeObject3d,
            useExisting: ArrayCameraDirective,
        },
        OBJECT_3D_CONTROLLER_PROVIDER,
    ], exportAs: ["ngtArrayCamera"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ArrayCameraDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-array-camera',
                    exportAs: 'ngtArrayCamera',
                    providers: [
                        {
                            provide: ThreeObject3d,
                            useExisting: ArrayCameraDirective,
                        },
                        OBJECT_3D_CONTROLLER_PROVIDER,
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });
