// @angular-three/core/esm2015/helpers/lib/polar-grid-helper/polar-grid-helper.directive.js
// GENERATED
import { ThreeHelper, ThreeObject3d, OBJECT_3D_CONTROLLER_PROVIDER, } from '@angular-three/core';
import { Directive, Input } from '@angular/core';
import { PolarGridHelper } from 'three';
import * as i0 from "@angular/core";
export class PolarGridHelperDirective extends ThreeHelper {
    constructor() {
        super(...arguments);
        this.helperType = PolarGridHelper;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
PolarGridHelperDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: PolarGridHelperDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
PolarGridHelperDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: PolarGridHelperDirective, selector: "ngt-polar-grid-helper", inputs: { args: "args" }, providers: [
        {
            provide: ThreeObject3d,
            useExisting: PolarGridHelperDirective,
        },
        OBJECT_3D_CONTROLLER_PROVIDER,
    ], exportAs: ["ngtPolarGridHelper"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: PolarGridHelperDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-polar-grid-helper',
                    exportAs: 'ngtPolarGridHelper',
                    providers: [
                        {
                            provide: ThreeObject3d,
                            useExisting: PolarGridHelperDirective,
                        },
                        OBJECT_3D_CONTROLLER_PROVIDER,
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });
