// @angular-three/core/esm2015/attributes/lib/uint8-clamped-buffer-attribute/uint8-clamped-buffer-attribute.directive.js
// GENERATED
import { ThreeAttribute } from '@angular-three/core';
import { Directive, Input } from '@angular/core';
import { Uint8ClampedBufferAttribute } from 'three';
import * as i0 from "@angular/core";
export class Uint8ClampedBufferAttributeDirective extends ThreeAttribute {
    constructor() {
        super(...arguments);
        this.attributeType = Uint8ClampedBufferAttribute;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
Uint8ClampedBufferAttributeDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: Uint8ClampedBufferAttributeDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
Uint8ClampedBufferAttributeDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: Uint8ClampedBufferAttributeDirective, selector: "ngt-uint8-clamped-buffer-attribute", inputs: { args: "args" }, providers: [
        {
            provide: ThreeAttribute,
            useExisting: Uint8ClampedBufferAttributeDirective,
        },
    ], exportAs: ["ngtUint8ClampedBufferAttribute"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: Uint8ClampedBufferAttributeDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-uint8-clamped-buffer-attribute',
                    exportAs: 'ngtUint8ClampedBufferAttribute',
                    providers: [
                        {
                            provide: ThreeAttribute,
                            useExisting: Uint8ClampedBufferAttributeDirective,
                        },
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });
