// @angular-three/core/esm2015/lights/lib/light-probe/light-probe.directive.js
// GENERATED
import { ThreeLight, ThreeObject3d, OBJECT_3D_CONTROLLER_PROVIDER, } from '@angular-three/core';
import { Directive, Input } from '@angular/core';
import { LightProbe } from 'three';
import * as i0 from "@angular/core";
export class LightProbeDirective extends ThreeLight {
    constructor() {
        super(...arguments);
        this.lightType = LightProbe;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
LightProbeDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: LightProbeDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
LightProbeDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: LightProbeDirective, selector: "ngt-light-probe", inputs: { args: "args" }, providers: [
        {
            provide: ThreeObject3d,
            useExisting: LightProbeDirective,
        },
        OBJECT_3D_CONTROLLER_PROVIDER,
    ], exportAs: ["ngtLightProbe"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: LightProbeDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-light-probe',
                    exportAs: 'ngtLightProbe',
                    providers: [
                        {
                            provide: ThreeObject3d,
                            useExisting: LightProbeDirective,
                        },
                        OBJECT_3D_CONTROLLER_PROVIDER,
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });
