// @angular-three/core/esm2015/materials/lib/line-basic-material/line-basic-material.directive.js
// GENERATED
import { ThreeMaterial } from '@angular-three/core';
import { Directive } from '@angular/core';
import { LineBasicMaterial } from 'three';
import * as i0 from "@angular/core";
export class LineBasicMaterialDirective extends ThreeMaterial {
    constructor() {
        super(...arguments);
        this.materialType = LineBasicMaterial;
    }
}
LineBasicMaterialDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: LineBasicMaterialDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
LineBasicMaterialDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: LineBasicMaterialDirective, selector: "ngt-line-basic-material", providers: [
        { provide: ThreeMaterial, useExisting: LineBasicMaterialDirective },
    ], exportAs: ["ngtLineBasicMaterial"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: LineBasicMaterialDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-line-basic-material',
                    exportAs: 'ngtLineBasicMaterial',
                    providers: [
                        { provide: ThreeMaterial, useExisting: LineBasicMaterialDirective },
                    ],
                }]
        }] });
