// @angular-three/core/esm2015/lights/lib/point-light/point-light.directive.js
// GENERATED
import { ThreeLight, ThreeObject3d, OBJECT_3D_CONTROLLER_PROVIDER, } from '@angular-three/core';
import { Directive, Input } from '@angular/core';
import { PointLight } from 'three';
import * as i0 from "@angular/core";
export class PointLightDirective extends ThreeLight {
    constructor() {
        super(...arguments);
        this.lightType = PointLight;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
PointLightDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: PointLightDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
PointLightDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: PointLightDirective, selector: "ngt-point-light", inputs: { args: "args" }, providers: [
        {
            provide: ThreeObject3d,
            useExisting: PointLightDirective,
        },
        OBJECT_3D_CONTROLLER_PROVIDER,
    ], exportAs: ["ngtPointLight"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: PointLightDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-point-light',
                    exportAs: 'ngtPointLight',
                    providers: [
                        {
                            provide: ThreeObject3d,
                            useExisting: PointLightDirective,
                        },
                        OBJECT_3D_CONTROLLER_PROVIDER,
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });
