// @angular-three/core/esm2015/lights/lib/spot-light/spot-light.directive.js
// GENERATED
import { ThreeLight, ThreeObject3d, OBJECT_3D_CONTROLLER_PROVIDER, } from '@angular-three/core';
import { Directive, Input } from '@angular/core';
import { SpotLight } from 'three';
import * as i0 from "@angular/core";
export class SpotLightDirective extends ThreeLight {
    constructor() {
        super(...arguments);
        this.lightType = SpotLight;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
SpotLightDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: SpotLightDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
SpotLightDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: SpotLightDirective, selector: "ngt-spot-light", inputs: { args: "args" }, providers: [
        {
            provide: ThreeObject3d,
            useExisting: SpotLightDirective,
        },
        OBJECT_3D_CONTROLLER_PROVIDER,
    ], exportAs: ["ngtSpotLight"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: SpotLightDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-spot-light',
                    exportAs: 'ngtSpotLight',
                    providers: [
                        {
                            provide: ThreeObject3d,
                            useExisting: SpotLightDirective,
                        },
                        OBJECT_3D_CONTROLLER_PROVIDER,
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });
