// @angular-three/core/esm2015/audios/lib/positional-audio/positional-audio.directive.js
import { Directive } from '@angular/core';
import { PositionalAudio } from 'three';
import { ThreeAudio } from '../abstracts';
import * as i0 from "@angular/core";
export class PositionalAudioDirective extends ThreeAudio {
    constructor() {
        super(...arguments);
        this.audioType = PositionalAudio;
    }
}
PositionalAudioDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: PositionalAudioDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
PositionalAudioDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: PositionalAudioDirective, selector: "ngt-positional-audio", exportAs: ["ngtPositionalAudio"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: PositionalAudioDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-positional-audio',
                    exportAs: 'ngtPositionalAudio',
                }]
        }] });
