// @angular-three/core/esm2015/helpers/lib/hemisphere-light-helper/hemisphere-light-helper.directive.js
// GENERATED
import { ThreeHelper, ThreeObject3d, OBJECT_3D_CONTROLLER_PROVIDER, } from '@angular-three/core';
import { Directive, Input } from '@angular/core';
import { HemisphereLightHelper } from 'three';
import * as i0 from "@angular/core";
export class HemisphereLightHelperDirective extends ThreeHelper {
    constructor() {
        super(...arguments);
        this.helperType = HemisphereLightHelper;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
HemisphereLightHelperDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: HemisphereLightHelperDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
HemisphereLightHelperDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: HemisphereLightHelperDirective, selector: "ngt-hemisphere-light-helper", inputs: { args: "args" }, providers: [
        {
            provide: ThreeObject3d,
            useExisting: HemisphereLightHelperDirective,
        },
        OBJECT_3D_CONTROLLER_PROVIDER,
    ], exportAs: ["ngtHemisphereLightHelper"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: HemisphereLightHelperDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-hemisphere-light-helper',
                    exportAs: 'ngtHemisphereLightHelper',
                    providers: [
                        {
                            provide: ThreeObject3d,
                            useExisting: HemisphereLightHelperDirective,
                        },
                        OBJECT_3D_CONTROLLER_PROVIDER,
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });
