// @angular-three/core/esm2015/cameras/lib/camera/camera.directive.js
// GENERATED
import { ThreeCamera, ThreeObject3d, OBJECT_3D_CONTROLLER_PROVIDER, } from '@angular-three/core';
import { Directive, Input } from '@angular/core';
import { Camera } from 'three';
import * as i0 from "@angular/core";
export class CameraDirective extends ThreeCamera {
    constructor() {
        super(...arguments);
        this.cameraType = Camera;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
CameraDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: CameraDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
CameraDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: CameraDirective, selector: "ngt-camera", inputs: { args: "args" }, providers: [
        {
            provide: ThreeObject3d,
            useExisting: CameraDirective,
        },
        OBJECT_3D_CONTROLLER_PROVIDER,
    ], exportAs: ["ngtCamera"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: CameraDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-camera',
                    exportAs: 'ngtCamera',
                    providers: [
                        {
                            provide: ThreeObject3d,
                            useExisting: CameraDirective,
                        },
                        OBJECT_3D_CONTROLLER_PROVIDER,
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });
