// @angular-three/core/esm2015/attributes/lib/float16-buffer-attribute/float16-buffer-attribute.directive.js
// GENERATED
import { ThreeAttribute } from '@angular-three/core';
import { Directive, Input } from '@angular/core';
import { Float16BufferAttribute } from 'three';
import * as i0 from "@angular/core";
export class Float16BufferAttributeDirective extends ThreeAttribute {
    constructor() {
        super(...arguments);
        this.attributeType = Float16BufferAttribute;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
Float16BufferAttributeDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: Float16BufferAttributeDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
Float16BufferAttributeDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: Float16BufferAttributeDirective, selector: "ngt-float16-buffer-attribute", inputs: { args: "args" }, providers: [
        {
            provide: ThreeAttribute,
            useExisting: Float16BufferAttributeDirective,
        },
    ], exportAs: ["ngtFloat16BufferAttribute"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: Float16BufferAttributeDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-float16-buffer-attribute',
                    exportAs: 'ngtFloat16BufferAttribute',
                    providers: [
                        {
                            provide: ThreeAttribute,
                            useExisting: Float16BufferAttributeDirective,
                        },
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });
