// @angular-three/core/esm2015/textures/lib/video-texture/video-texture.directive.js
// GENERATED
import { ThreeTexture } from '@angular-three/core';
import { Directive, Input } from '@angular/core';
import { VideoTexture } from 'three';
import * as i0 from "@angular/core";
export class VideoTextureDirective extends ThreeTexture {
    constructor() {
        super(...arguments);
        this.textureType = VideoTexture;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
VideoTextureDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: VideoTextureDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
VideoTextureDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: VideoTextureDirective, selector: "ngt-video-texture", inputs: { args: "args" }, providers: [
        {
            provide: ThreeTexture,
            useExisting: VideoTextureDirective,
        },
    ], exportAs: ["ngtVideoTexture"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: VideoTextureDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-video-texture',
                    exportAs: 'ngtVideoTexture',
                    providers: [
                        {
                            provide: ThreeTexture,
                            useExisting: VideoTextureDirective,
                        },
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });
