// @angular-three/core/esm2015/attributes/lib/instanced-buffer-attribute/instanced-buffer-attribute.directive.js
// GENERATED
import { ThreeAttribute } from '@angular-three/core';
import { Directive, Input } from '@angular/core';
import { InstancedBufferAttribute } from 'three';
import * as i0 from "@angular/core";
export class InstancedBufferAttributeDirective extends ThreeAttribute {
    constructor() {
        super(...arguments);
        this.attributeType = InstancedBufferAttribute;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
InstancedBufferAttributeDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: InstancedBufferAttributeDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
InstancedBufferAttributeDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: InstancedBufferAttributeDirective, selector: "ngt-instanced-buffer-attribute", inputs: { args: "args" }, providers: [
        {
            provide: ThreeAttribute,
            useExisting: InstancedBufferAttributeDirective,
        },
    ], exportAs: ["ngtInstancedBufferAttribute"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: InstancedBufferAttributeDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-instanced-buffer-attribute',
                    exportAs: 'ngtInstancedBufferAttribute',
                    providers: [
                        {
                            provide: ThreeAttribute,
                            useExisting: InstancedBufferAttributeDirective,
                        },
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });
