// @angular-three/core/esm2015/geometries/lib/box-buffer-geometry/box-buffer-geometry.directive.js
// GENERATED
import { ThreeBufferGeometry } from '@angular-three/core';
import { Directive, Input } from '@angular/core';
import { BoxBufferGeometry } from 'three';
import * as i0 from "@angular/core";
export class BoxBufferGeometryDirective extends ThreeBufferGeometry {
    constructor() {
        super(...arguments);
        this.geometryType = BoxBufferGeometry;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
BoxBufferGeometryDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: BoxBufferGeometryDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
BoxBufferGeometryDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: BoxBufferGeometryDirective, selector: "ngt-box-buffer-geometry,ngt-box-geometry", inputs: { args: "args" }, providers: [
        {
            provide: ThreeBufferGeometry,
            useExisting: BoxBufferGeometryDirective,
        },
    ], exportAs: ["ngtBoxBufferGeometry"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: BoxBufferGeometryDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-box-buffer-geometry,ngt-box-geometry',
                    exportAs: 'ngtBoxBufferGeometry',
                    providers: [
                        {
                            provide: ThreeBufferGeometry,
                            useExisting: BoxBufferGeometryDirective,
                        },
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });
