// @angular-three/core/esm2015/sprites/lib/sprite/sprite.directive.js
// GENERATED
import { ThreeSprite, ThreeObject3d, OBJECT_3D_CONTROLLER_PROVIDER, } from '@angular-three/core';
import { Directive } from '@angular/core';
import { Sprite } from 'three';
import * as i0 from "@angular/core";
export class SpriteDirective extends ThreeSprite {
    constructor() {
        super(...arguments);
        this.spriteType = Sprite;
    }
}
SpriteDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: SpriteDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
SpriteDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: SpriteDirective, selector: "ngt-sprite", providers: [
        {
            provide: ThreeObject3d,
            useExisting: SpriteDirective,
        },
        OBJECT_3D_CONTROLLER_PROVIDER,
    ], exportAs: ["ngtSprite"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: SpriteDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-sprite',
                    exportAs: 'ngtSprite',
                    providers: [
                        {
                            provide: ThreeObject3d,
                            useExisting: SpriteDirective,
                        },
                        OBJECT_3D_CONTROLLER_PROVIDER,
                    ],
                }]
        }] });
