// @angular-three/core/esm2015/textures/lib/cube-texture/cube-texture.directive.js
// GENERATED
import { ThreeTexture } from '@angular-three/core';
import { Directive, Input } from '@angular/core';
import { CubeTexture } from 'three';
import * as i0 from "@angular/core";
export class CubeTextureDirective extends ThreeTexture {
    constructor() {
        super(...arguments);
        this.textureType = CubeTexture;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
CubeTextureDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: CubeTextureDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
CubeTextureDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: CubeTextureDirective, selector: "ngt-cube-texture", inputs: { args: "args" }, providers: [
        {
            provide: ThreeTexture,
            useExisting: CubeTextureDirective,
        },
    ], exportAs: ["ngtCubeTexture"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: CubeTextureDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-cube-texture',
                    exportAs: 'ngtCubeTexture',
                    providers: [
                        {
                            provide: ThreeTexture,
                            useExisting: CubeTextureDirective,
                        },
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });
