// @angular-three/core/esm2015/meshes/lib/skinned-mesh/bone.directive.js
import { OBJECT_3D_CONTROLLER_PROVIDER, ThreeObject3d, } from '@angular-three/core';
import { Directive } from '@angular/core';
import { Bone } from 'three';
import * as i0 from "@angular/core";
export class BoneDirective extends ThreeObject3d {
    ngOnInit() {
        this.init();
    }
    initObject() {
        this._bone = new Bone();
    }
    get object3d() {
        return this._bone;
    }
}
BoneDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: BoneDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
BoneDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: BoneDirective, selector: "ngt-bone", providers: [
        { provide: ThreeObject3d, useExisting: BoneDirective },
        OBJECT_3D_CONTROLLER_PROVIDER,
    ], exportAs: ["ngtBone"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: BoneDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-bone',
                    exportAs: 'ngtBone',
                    providers: [
                        { provide: ThreeObject3d, useExisting: BoneDirective },
                        OBJECT_3D_CONTROLLER_PROVIDER,
                    ],
                }]
        }] });
