// @angular-three/core/esm2015/group/lib/group.directive.js
import { OBJECT_3D_CONTROLLER_PROVIDER, ThreeObject3d, } from '@angular-three/core';
import { Directive } from '@angular/core';
import { Group } from 'three';
import * as i0 from "@angular/core";
export class GroupDirective extends ThreeObject3d {
    ngOnInit() {
        this.init();
    }
    initObject() {
        this._group = new Group();
    }
    get object3d() {
        return this._group;
    }
}
GroupDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: GroupDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
GroupDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: GroupDirective, selector: "ngt-group", providers: [
        { provide: ThreeObject3d, useExisting: GroupDirective },
        OBJECT_3D_CONTROLLER_PROVIDER,
    ], exportAs: ["ngtGroup"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: GroupDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-group',
                    exportAs: 'ngtGroup',
                    providers: [
                        { provide: ThreeObject3d, useExisting: GroupDirective },
                        OBJECT_3D_CONTROLLER_PROVIDER,
                    ],
                }]
        }] });
