// @angular-three/core/esm2015/helpers/lib/plane-helper/plane-helper.directive.js
// GENERATED
import { ThreeHelper, ThreeObject3d, OBJECT_3D_CONTROLLER_PROVIDER, } from '@angular-three/core';
import { Directive, Input } from '@angular/core';
import { PlaneHelper } from 'three';
import * as i0 from "@angular/core";
export class PlaneHelperDirective extends ThreeHelper {
    constructor() {
        super(...arguments);
        this.helperType = PlaneHelper;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
PlaneHelperDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: PlaneHelperDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
PlaneHelperDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: PlaneHelperDirective, selector: "ngt-plane-helper", inputs: { args: "args" }, providers: [
        {
            provide: ThreeObject3d,
            useExisting: PlaneHelperDirective,
        },
        OBJECT_3D_CONTROLLER_PROVIDER,
    ], exportAs: ["ngtPlaneHelper"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: PlaneHelperDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-plane-helper',
                    exportAs: 'ngtPlaneHelper',
                    providers: [
                        {
                            provide: ThreeObject3d,
                            useExisting: PlaneHelperDirective,
                        },
                        OBJECT_3D_CONTROLLER_PROVIDER,
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });
