// @angular-three/core/esm2015/curves/lib/quadratic-bezier-curve/quadratic-bezier-curve.directive.js
// GENERATED
import { ThreeCurve } from '@angular-three/core';
import { Directive, Input } from '@angular/core';
import { QuadraticBezierCurve } from 'three';
import * as i0 from "@angular/core";
export class QuadraticBezierCurveDirective extends ThreeCurve {
    constructor() {
        super(...arguments);
        this.curveType = QuadraticBezierCurve;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
QuadraticBezierCurveDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: QuadraticBezierCurveDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
QuadraticBezierCurveDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: QuadraticBezierCurveDirective, selector: "ngt-quadratic-bezier-curve", inputs: { args: "args" }, providers: [
        {
            provide: ThreeCurve,
            useExisting: QuadraticBezierCurveDirective,
        },
    ], exportAs: ["ngtQuadraticBezierCurve"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: QuadraticBezierCurveDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-quadratic-bezier-curve',
                    exportAs: 'ngtQuadraticBezierCurve',
                    providers: [
                        {
                            provide: ThreeCurve,
                            useExisting: QuadraticBezierCurveDirective,
                        },
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });
