// @angular-three/core/esm2015/geometries/lib/text-geometry/text-geometry.directive.js
// GENERATED
import { ThreeBufferGeometry } from '@angular-three/core';
import { Directive, Input } from '@angular/core';
import { TextGeometry } from 'three/examples/jsm/geometries/TextGeometry';
import * as i0 from "@angular/core";
export class TextGeometryDirective extends ThreeBufferGeometry {
    constructor() {
        super(...arguments);
        this.geometryType = TextGeometry;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
TextGeometryDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: TextGeometryDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
TextGeometryDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: TextGeometryDirective, selector: "ngt-text-geometry", inputs: { args: "args" }, providers: [
        {
            provide: ThreeBufferGeometry,
            useExisting: TextGeometryDirective,
        },
    ], exportAs: ["ngtTextGeometry"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: TextGeometryDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-text-geometry',
                    exportAs: 'ngtTextGeometry',
                    providers: [
                        {
                            provide: ThreeBufferGeometry,
                            useExisting: TextGeometryDirective,
                        },
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });
