// @angular-three/core/esm2015/textures/lib/canvas-texture/canvas-texture.directive.js
// GENERATED
import { ThreeTexture } from '@angular-three/core';
import { Directive, Input } from '@angular/core';
import { CanvasTexture } from 'three';
import * as i0 from "@angular/core";
export class CanvasTextureDirective extends ThreeTexture {
    constructor() {
        super(...arguments);
        this.textureType = CanvasTexture;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
CanvasTextureDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: CanvasTextureDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
CanvasTextureDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: CanvasTextureDirective, selector: "ngt-canvas-texture", inputs: { args: "args" }, providers: [
        {
            provide: ThreeTexture,
            useExisting: CanvasTextureDirective,
        },
    ], exportAs: ["ngtCanvasTexture"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: CanvasTextureDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-canvas-texture',
                    exportAs: 'ngtCanvasTexture',
                    providers: [
                        {
                            provide: ThreeTexture,
                            useExisting: CanvasTextureDirective,
                        },
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });
