// @angular-three/core/esm2015/curves/lib/catmull-rom-curve3/catmull-rom-curve3.directive.js
// GENERATED
import { ThreeCurve } from '@angular-three/core';
import { Directive, Input } from '@angular/core';
import { CatmullRomCurve3 } from 'three';
import * as i0 from "@angular/core";
export class CatmullRomCurve3Directive extends ThreeCurve {
    constructor() {
        super(...arguments);
        this.curveType = CatmullRomCurve3;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
CatmullRomCurve3Directive.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: CatmullRomCurve3Directive, deps: null, target: i0.ɵɵFactoryTarget.Directive });
CatmullRomCurve3Directive.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: CatmullRomCurve3Directive, selector: "ngt-catmull-rom-curve3", inputs: { args: "args" }, providers: [
        {
            provide: ThreeCurve,
            useExisting: CatmullRomCurve3Directive,
        },
    ], exportAs: ["ngtCatmullRomCurve3"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: CatmullRomCurve3Directive, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-catmull-rom-curve3',
                    exportAs: 'ngtCatmullRomCurve3',
                    providers: [
                        {
                            provide: ThreeCurve,
                            useExisting: CatmullRomCurve3Directive,
                        },
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });
