// @angular-three/core/esm2015/attributes/lib/float32-buffer-attribute/float32-buffer-attribute.directive.js
// GENERATED
import { ThreeAttribute } from '@angular-three/core';
import { Directive, Input } from '@angular/core';
import { Float32BufferAttribute } from 'three';
import * as i0 from "@angular/core";
export class Float32BufferAttributeDirective extends ThreeAttribute {
    constructor() {
        super(...arguments);
        this.attributeType = Float32BufferAttribute;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
Float32BufferAttributeDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: Float32BufferAttributeDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
Float32BufferAttributeDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: Float32BufferAttributeDirective, selector: "ngt-float32-buffer-attribute", inputs: { args: "args" }, providers: [
        {
            provide: ThreeAttribute,
            useExisting: Float32BufferAttributeDirective,
        },
    ], exportAs: ["ngtFloat32BufferAttribute"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: Float32BufferAttributeDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-float32-buffer-attribute',
                    exportAs: 'ngtFloat32BufferAttribute',
                    providers: [
                        {
                            provide: ThreeAttribute,
                            useExisting: Float32BufferAttributeDirective,
                        },
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });
