// @angular-three/core/esm2015/attributes/lib/int32-buffer-attribute/int32-buffer-attribute.directive.js
// GENERATED
import { ThreeAttribute } from '@angular-three/core';
import { Directive, Input } from '@angular/core';
import { Int32BufferAttribute } from 'three';
import * as i0 from "@angular/core";
export class Int32BufferAttributeDirective extends ThreeAttribute {
    constructor() {
        super(...arguments);
        this.attributeType = Int32BufferAttribute;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
Int32BufferAttributeDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: Int32BufferAttributeDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
Int32BufferAttributeDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: Int32BufferAttributeDirective, selector: "ngt-int32-buffer-attribute", inputs: { args: "args" }, providers: [
        {
            provide: ThreeAttribute,
            useExisting: Int32BufferAttributeDirective,
        },
    ], exportAs: ["ngtInt32BufferAttribute"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: Int32BufferAttributeDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-int32-buffer-attribute',
                    exportAs: 'ngtInt32BufferAttribute',
                    providers: [
                        {
                            provide: ThreeAttribute,
                            useExisting: Int32BufferAttributeDirective,
                        },
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });
