// @angular-three/core/esm2015/materials/lib/mesh-phong-material/mesh-phong-material.directive.js
// GENERATED
import { ThreeMaterial } from '@angular-three/core';
import { Directive } from '@angular/core';
import { MeshPhongMaterial } from 'three';
import * as i0 from "@angular/core";
export class MeshPhongMaterialDirective extends ThreeMaterial {
    constructor() {
        super(...arguments);
        this.materialType = MeshPhongMaterial;
    }
}
MeshPhongMaterialDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: MeshPhongMaterialDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
MeshPhongMaterialDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: MeshPhongMaterialDirective, selector: "ngt-mesh-phong-material", providers: [
        { provide: ThreeMaterial, useExisting: MeshPhongMaterialDirective },
    ], exportAs: ["ngtMeshPhongMaterial"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: MeshPhongMaterialDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-mesh-phong-material',
                    exportAs: 'ngtMeshPhongMaterial',
                    providers: [
                        { provide: ThreeMaterial, useExisting: MeshPhongMaterialDirective },
                    ],
                }]
        }] });
