// @angular-three/core/esm2015/helpers/lib/arrow-helper/arrow-helper.directive.js
// GENERATED
import { ThreeHelper, ThreeObject3d, OBJECT_3D_CONTROLLER_PROVIDER, } from '@angular-three/core';
import { Directive, Input } from '@angular/core';
import { ArrowHelper } from 'three';
import * as i0 from "@angular/core";
export class ArrowHelperDirective extends ThreeHelper {
    constructor() {
        super(...arguments);
        this.helperType = ArrowHelper;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
ArrowHelperDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ArrowHelperDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
ArrowHelperDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: ArrowHelperDirective, selector: "ngt-arrow-helper", inputs: { args: "args" }, providers: [
        {
            provide: ThreeObject3d,
            useExisting: ArrowHelperDirective,
        },
        OBJECT_3D_CONTROLLER_PROVIDER,
    ], exportAs: ["ngtArrowHelper"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ArrowHelperDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-arrow-helper',
                    exportAs: 'ngtArrowHelper',
                    providers: [
                        {
                            provide: ThreeObject3d,
                            useExisting: ArrowHelperDirective,
                        },
                        OBJECT_3D_CONTROLLER_PROVIDER,
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });
