// @angular-three/core/esm2015/geometries/lib/torus-buffer-geometry/torus-buffer-geometry.directive.js
// GENERATED
import { ThreeBufferGeometry } from '@angular-three/core';
import { Directive, Input } from '@angular/core';
import { TorusBufferGeometry } from 'three';
import * as i0 from "@angular/core";
export class TorusBufferGeometryDirective extends ThreeBufferGeometry {
    constructor() {
        super(...arguments);
        this.geometryType = TorusBufferGeometry;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
TorusBufferGeometryDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: TorusBufferGeometryDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
TorusBufferGeometryDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: TorusBufferGeometryDirective, selector: "ngt-torus-buffer-geometry,ngt-torus-geometry", inputs: { args: "args" }, providers: [
        {
            provide: ThreeBufferGeometry,
            useExisting: TorusBufferGeometryDirective,
        },
    ], exportAs: ["ngtTorusBufferGeometry"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: TorusBufferGeometryDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-torus-buffer-geometry,ngt-torus-geometry',
                    exportAs: 'ngtTorusBufferGeometry',
                    providers: [
                        {
                            provide: ThreeBufferGeometry,
                            useExisting: TorusBufferGeometryDirective,
                        },
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });
