// @angular-three/core/esm2015/curves/lib/cubic-bezier-curve/cubic-bezier-curve.directive.js
// GENERATED
import { ThreeCurve } from '@angular-three/core';
import { Directive, Input } from '@angular/core';
import { CubicBezierCurve } from 'three';
import * as i0 from "@angular/core";
export class CubicBezierCurveDirective extends ThreeCurve {
    constructor() {
        super(...arguments);
        this.curveType = CubicBezierCurve;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
CubicBezierCurveDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: CubicBezierCurveDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
CubicBezierCurveDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: CubicBezierCurveDirective, selector: "ngt-cubic-bezier-curve", inputs: { args: "args" }, providers: [
        {
            provide: ThreeCurve,
            useExisting: CubicBezierCurveDirective,
        },
    ], exportAs: ["ngtCubicBezierCurve"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: CubicBezierCurveDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-cubic-bezier-curve',
                    exportAs: 'ngtCubicBezierCurve',
                    providers: [
                        {
                            provide: ThreeCurve,
                            useExisting: CubicBezierCurveDirective,
                        },
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });
