// @angular-three/core/esm2015/lines/lib/line/line.directive.js
// GENERATED
import { ThreeLine, ThreeObject3d, OBJECT_3D_CONTROLLER_PROVIDER, } from '@angular-three/core';
import { Directive } from '@angular/core';
import { Line } from 'three';
import * as i0 from "@angular/core";
export class LineDirective extends ThreeLine {
    constructor() {
        super(...arguments);
        this.lineType = Line;
    }
}
LineDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: LineDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
LineDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: LineDirective, selector: "ngt-line", providers: [
        {
            provide: ThreeObject3d,
            useExisting: LineDirective,
        },
        OBJECT_3D_CONTROLLER_PROVIDER,
    ], exportAs: ["ngtLine"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: LineDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-line',
                    exportAs: 'ngtLine',
                    providers: [
                        {
                            provide: ThreeObject3d,
                            useExisting: LineDirective,
                        },
                        OBJECT_3D_CONTROLLER_PROVIDER,
                    ],
                }]
        }] });
