// @angular-three/core/esm2015/materials/lib/mesh-normal-material/mesh-normal-material.directive.js
// GENERATED
import { ThreeMaterial } from '@angular-three/core';
import { Directive } from '@angular/core';
import { MeshNormalMaterial } from 'three';
import * as i0 from "@angular/core";
export class MeshNormalMaterialDirective extends ThreeMaterial {
    constructor() {
        super(...arguments);
        this.materialType = MeshNormalMaterial;
    }
}
MeshNormalMaterialDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: MeshNormalMaterialDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
MeshNormalMaterialDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: MeshNormalMaterialDirective, selector: "ngt-mesh-normal-material", providers: [
        { provide: ThreeMaterial, useExisting: MeshNormalMaterialDirective },
    ], exportAs: ["ngtMeshNormalMaterial"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: MeshNormalMaterialDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-mesh-normal-material',
                    exportAs: 'ngtMeshNormalMaterial',
                    providers: [
                        { provide: ThreeMaterial, useExisting: MeshNormalMaterialDirective },
                    ],
                }]
        }] });
