// @angular-three/core/esm2015/geometries/lib/shape-buffer-geometry/shape-buffer-geometry.directive.js
// GENERATED
import { ThreeBufferGeometry } from '@angular-three/core';
import { Directive, Input } from '@angular/core';
import { ShapeBufferGeometry } from 'three';
import * as i0 from "@angular/core";
export class ShapeBufferGeometryDirective extends ThreeBufferGeometry {
    constructor() {
        super(...arguments);
        this.geometryType = ShapeBufferGeometry;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
ShapeBufferGeometryDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ShapeBufferGeometryDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
ShapeBufferGeometryDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: ShapeBufferGeometryDirective, selector: "ngt-shape-buffer-geometry,ngt-shape-geometry", inputs: { args: "args" }, providers: [
        {
            provide: ThreeBufferGeometry,
            useExisting: ShapeBufferGeometryDirective,
        },
    ], exportAs: ["ngtShapeBufferGeometry"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ShapeBufferGeometryDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-shape-buffer-geometry,ngt-shape-geometry',
                    exportAs: 'ngtShapeBufferGeometry',
                    providers: [
                        {
                            provide: ThreeBufferGeometry,
                            useExisting: ShapeBufferGeometryDirective,
                        },
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });
