// @angular-three/core/esm2015/materials/lib/mesh-physical-material/mesh-physical-material.directive.js
// GENERATED
import { ThreeMaterial } from '@angular-three/core';
import { Directive } from '@angular/core';
import { MeshPhysicalMaterial } from 'three';
import * as i0 from "@angular/core";
export class MeshPhysicalMaterialDirective extends ThreeMaterial {
    constructor() {
        super(...arguments);
        this.materialType = MeshPhysicalMaterial;
    }
}
MeshPhysicalMaterialDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: MeshPhysicalMaterialDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
MeshPhysicalMaterialDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: MeshPhysicalMaterialDirective, selector: "ngt-mesh-physical-material", providers: [
        { provide: ThreeMaterial, useExisting: MeshPhysicalMaterialDirective },
    ], exportAs: ["ngtMeshPhysicalMaterial"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: MeshPhysicalMaterialDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-mesh-physical-material',
                    exportAs: 'ngtMeshPhysicalMaterial',
                    providers: [
                        { provide: ThreeMaterial, useExisting: MeshPhysicalMaterialDirective },
                    ],
                }]
        }] });
