// @angular-three/core/esm2015/helpers/lib/directional-light-helper/directional-light-helper.directive.js
// GENERATED
import { ThreeHelper, ThreeObject3d, OBJECT_3D_CONTROLLER_PROVIDER, } from '@angular-three/core';
import { Directive, Input } from '@angular/core';
import { DirectionalLightHelper } from 'three';
import * as i0 from "@angular/core";
export class DirectionalLightHelperDirective extends ThreeHelper {
    constructor() {
        super(...arguments);
        this.helperType = DirectionalLightHelper;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
DirectionalLightHelperDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: DirectionalLightHelperDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
DirectionalLightHelperDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: DirectionalLightHelperDirective, selector: "ngt-directional-light-helper", inputs: { args: "args" }, providers: [
        {
            provide: ThreeObject3d,
            useExisting: DirectionalLightHelperDirective,
        },
        OBJECT_3D_CONTROLLER_PROVIDER,
    ], exportAs: ["ngtDirectionalLightHelper"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: DirectionalLightHelperDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-directional-light-helper',
                    exportAs: 'ngtDirectionalLightHelper',
                    providers: [
                        {
                            provide: ThreeObject3d,
                            useExisting: DirectionalLightHelperDirective,
                        },
                        OBJECT_3D_CONTROLLER_PROVIDER,
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });
