// @angular-three/core/esm2015/geometries/lib/icosahedron-buffer-geometry/icosahedron-buffer-geometry.directive.js
// GENERATED
import { ThreeBufferGeometry } from '@angular-three/core';
import { Directive, Input } from '@angular/core';
import { IcosahedronBufferGeometry } from 'three';
import * as i0 from "@angular/core";
export class IcosahedronBufferGeometryDirective extends ThreeBufferGeometry {
    constructor() {
        super(...arguments);
        this.geometryType = IcosahedronBufferGeometry;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
IcosahedronBufferGeometryDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: IcosahedronBufferGeometryDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
IcosahedronBufferGeometryDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: IcosahedronBufferGeometryDirective, selector: "ngt-icosahedron-buffer-geometry,ngt-icosahedron-geometry", inputs: { args: "args" }, providers: [
        {
            provide: ThreeBufferGeometry,
            useExisting: IcosahedronBufferGeometryDirective,
        },
    ], exportAs: ["ngtIcosahedronBufferGeometry"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: IcosahedronBufferGeometryDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-icosahedron-buffer-geometry,ngt-icosahedron-geometry',
                    exportAs: 'ngtIcosahedronBufferGeometry',
                    providers: [
                        {
                            provide: ThreeBufferGeometry,
                            useExisting: IcosahedronBufferGeometryDirective,
                        },
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });
