// @angular-three/core/esm2015/lights/lib/ambient-light-probe/ambient-light-probe.directive.js
// GENERATED
import { ThreeLight, ThreeObject3d, OBJECT_3D_CONTROLLER_PROVIDER, } from '@angular-three/core';
import { Directive, Input } from '@angular/core';
import { AmbientLightProbe } from 'three';
import * as i0 from "@angular/core";
export class AmbientLightProbeDirective extends ThreeLight {
    constructor() {
        super(...arguments);
        this.lightType = AmbientLightProbe;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
AmbientLightProbeDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: AmbientLightProbeDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
AmbientLightProbeDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: AmbientLightProbeDirective, selector: "ngt-ambient-light-probe", inputs: { args: "args" }, providers: [
        {
            provide: ThreeObject3d,
            useExisting: AmbientLightProbeDirective,
        },
        OBJECT_3D_CONTROLLER_PROVIDER,
    ], exportAs: ["ngtAmbientLightProbe"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: AmbientLightProbeDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-ambient-light-probe',
                    exportAs: 'ngtAmbientLightProbe',
                    providers: [
                        {
                            provide: ThreeObject3d,
                            useExisting: AmbientLightProbeDirective,
                        },
                        OBJECT_3D_CONTROLLER_PROVIDER,
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });
