// @angular-three/core/esm2015/geometries/lib/wireframe-geometry/wireframe-geometry.directive.js
// GENERATED
import { ThreeBufferGeometry } from '@angular-three/core';
import { Directive, Input } from '@angular/core';
import { WireframeGeometry } from 'three';
import * as i0 from "@angular/core";
export class WireframeGeometryDirective extends ThreeBufferGeometry {
    constructor() {
        super(...arguments);
        this.geometryType = WireframeGeometry;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
WireframeGeometryDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: WireframeGeometryDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
WireframeGeometryDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: WireframeGeometryDirective, selector: "ngt-wireframe-geometry", inputs: { args: "args" }, providers: [
        {
            provide: ThreeBufferGeometry,
            useExisting: WireframeGeometryDirective,
        },
    ], exportAs: ["ngtWireframeGeometry"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: WireframeGeometryDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-wireframe-geometry',
                    exportAs: 'ngtWireframeGeometry',
                    providers: [
                        {
                            provide: ThreeBufferGeometry,
                            useExisting: WireframeGeometryDirective,
                        },
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });
