// @angular-three/core/esm2015/geometries/lib/decal-geometry/decal-geometry.directive.js
// GENERATED
import { ThreeBufferGeometry } from '@angular-three/core';
import { Directive, Input } from '@angular/core';
import { DecalGeometry } from 'three/examples/jsm/geometries/DecalGeometry';
import * as i0 from "@angular/core";
export class DecalGeometryDirective extends ThreeBufferGeometry {
    constructor() {
        super(...arguments);
        this.geometryType = DecalGeometry;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
DecalGeometryDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: DecalGeometryDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
DecalGeometryDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: DecalGeometryDirective, selector: "ngt-decal-geometry", inputs: { args: "args" }, providers: [
        {
            provide: ThreeBufferGeometry,
            useExisting: DecalGeometryDirective,
        },
    ], exportAs: ["ngtDecalGeometry"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: DecalGeometryDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-decal-geometry',
                    exportAs: 'ngtDecalGeometry',
                    providers: [
                        {
                            provide: ThreeBufferGeometry,
                            useExisting: DecalGeometryDirective,
                        },
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });
