// @angular-three/core/esm2015/materials/lib/mesh-toon-material/mesh-toon-material.directive.js
// GENERATED
import { ThreeMaterial } from '@angular-three/core';
import { Directive } from '@angular/core';
import { MeshToonMaterial } from 'three';
import * as i0 from "@angular/core";
export class MeshToonMaterialDirective extends ThreeMaterial {
    constructor() {
        super(...arguments);
        this.materialType = MeshToonMaterial;
    }
}
MeshToonMaterialDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: MeshToonMaterialDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
MeshToonMaterialDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: MeshToonMaterialDirective, selector: "ngt-mesh-toon-material", providers: [
        { provide: ThreeMaterial, useExisting: MeshToonMaterialDirective },
    ], exportAs: ["ngtMeshToonMaterial"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: MeshToonMaterialDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-mesh-toon-material',
                    exportAs: 'ngtMeshToonMaterial',
                    providers: [
                        { provide: ThreeMaterial, useExisting: MeshToonMaterialDirective },
                    ],
                }]
        }] });
