// @angular-three/core/esm2015/materials/lib/shadow-material/shadow-material.directive.js
// GENERATED
import { ThreeMaterial } from '@angular-three/core';
import { Directive } from '@angular/core';
import { ShadowMaterial } from 'three';
import * as i0 from "@angular/core";
export class ShadowMaterialDirective extends ThreeMaterial {
    constructor() {
        super(...arguments);
        this.materialType = ShadowMaterial;
    }
}
ShadowMaterialDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ShadowMaterialDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
ShadowMaterialDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: ShadowMaterialDirective, selector: "ngt-shadow-material", providers: [{ provide: ThreeMaterial, useExisting: ShadowMaterialDirective }], exportAs: ["ngtShadowMaterial"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: ShadowMaterialDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-shadow-material',
                    exportAs: 'ngtShadowMaterial',
                    providers: [{ provide: ThreeMaterial, useExisting: ShadowMaterialDirective }],
                }]
        }] });
