// @angular-three/core/esm2015/geometries/lib/lathe-buffer-geometry/lathe-buffer-geometry.directive.js
// GENERATED
import { ThreeBufferGeometry } from '@angular-three/core';
import { Directive, Input } from '@angular/core';
import { LatheBufferGeometry } from 'three';
import * as i0 from "@angular/core";
export class LatheBufferGeometryDirective extends ThreeBufferGeometry {
    constructor() {
        super(...arguments);
        this.geometryType = LatheBufferGeometry;
    }
    set args(v) {
        this.extraArgs = v;
    }
}
LatheBufferGeometryDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: LatheBufferGeometryDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive });
LatheBufferGeometryDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.10", type: LatheBufferGeometryDirective, selector: "ngt-lathe-buffer-geometry,ngt-lathe-geometry", inputs: { args: "args" }, providers: [
        {
            provide: ThreeBufferGeometry,
            useExisting: LatheBufferGeometryDirective,
        },
    ], exportAs: ["ngtLatheBufferGeometry"], usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.10", ngImport: i0, type: LatheBufferGeometryDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: 'ngt-lathe-buffer-geometry,ngt-lathe-geometry',
                    exportAs: 'ngtLatheBufferGeometry',
                    providers: [
                        {
                            provide: ThreeBufferGeometry,
                            useExisting: LatheBufferGeometryDirective,
                        },
                    ],
                }]
        }], propDecorators: { args: [{
                type: Input
            }] } });
